<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="cytoscape-automation-in-r-using-rcy3.html">
<link rel="next" href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-everyone"><i class="fa fa-check"></i><b>1.1</b> For Everyone</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#for-workshop-authors"><i class="fa fa-check"></i><b>1.2</b> For Workshop Authors</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#description"><i class="fa fa-check"></i><b>1.2.1</b> DESCRIPTION</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#classifying-your-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Classifying your workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#bookdown"><i class="fa fa-check"></i><b>1.2.3</b> Bookdown</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#deadlines-for-bioc2018"><i class="fa fa-check"></i><b>1.3</b> Deadlines for Bioc2018</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html"><i class="fa fa-check"></i><b>2</b> <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#description-1"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="2.1.7" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-goals"><i class="fa fa-check"></i><b>2.1.7</b> Learning goals</a></li>
<li class="chapter" data-level="2.1.8" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-objectives"><i class="fa fa-check"></i><b>2.1.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-r"><i class="fa fa-check"></i><b>2.2</b> Introduction to <em>R</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#rstudio-orientation"><i class="fa fa-check"></i><b>2.2.1</b> <em>RStudio</em> orientation</a></li>
<li class="chapter" data-level="2.2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#very-basics"><i class="fa fa-check"></i><b>2.2.2</b> Very basics</a></li>
<li class="chapter" data-level="2.2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#data-input-and-manipulation"><i class="fa fa-check"></i><b>2.2.3</b> Data input and manipulation</a></li>
<li class="chapter" data-level="2.2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#writing-r-scripts"><i class="fa fa-check"></i><b>2.2.4</b> Writing <em>R</em> scripts</a></li>
<li class="chapter" data-level="2.2.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#using-packages"><i class="fa fa-check"></i><b>2.2.5</b> Using packages</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.3</b> Introduction to <em>Bioconductor</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#discovering-installing-and-learning-how-to-use-bioconductor-packages"><i class="fa fa-check"></i><b>2.3.1</b> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-genomic-ranges"><i class="fa fa-check"></i><b>2.3.2</b> Working with Genomic Ranges</a></li>
<li class="chapter" data-level="2.3.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-summarized-experimental-data"><i class="fa fa-check"></i><b>2.3.3</b> Working with summarized experimental data</a></li>
<li class="chapter" data-level="2.3.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#down-stream-analysis"><i class="fa fa-check"></i><b>2.3.4</b> Down-stream analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>3</b> RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a><ul>
<li class="chapter" data-level="3.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-2"><i class="fa fa-check"></i><b>3.1.1</b> Description</a></li>
<li class="chapter" data-level="3.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-1"><i class="fa fa-check"></i><b>3.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="3.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation-1"><i class="fa fa-check"></i><b>3.1.3</b> Participation</a></li>
<li class="chapter" data-level="3.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="3.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-1"><i class="fa fa-check"></i><b>3.1.5</b> Time outline</a></li>
<li class="chapter" data-level="3.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>3.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="3.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>3.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="3.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>3.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="3.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>3.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="3.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>3.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>3.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="3.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>3.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="3.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>3.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="3.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>3.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>3.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="3.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>3.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="3.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>3.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="3.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>3.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>3.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="3.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>3.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="3.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>3.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub"><i class="fa fa-check"></i><b>3.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="3.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub"><i class="fa fa-check"></i><b>3.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="3.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>3.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>3.7</b> Building reports</a><ul>
<li class="chapter" data-level="3.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>3.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="3.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>3.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>3.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="3.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>3.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="3.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>3.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="3.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>3.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="3.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>3.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="3.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>3.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="3.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>3.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>4</b> Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-2"><i class="fa fa-check"></i><b>4.1</b> Overview</a><ul>
<li class="chapter" data-level="4.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-3"><i class="fa fa-check"></i><b>4.1.1</b> Description</a></li>
<li class="chapter" data-level="4.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-2"><i class="fa fa-check"></i><b>4.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-2"><i class="fa fa-check"></i><b>4.1.3</b> Participation</a></li>
<li class="chapter" data-level="4.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-2"><i class="fa fa-check"></i><b>4.1.5</b> Time outline</a></li>
<li class="chapter" data-level="4.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started"><i class="fa fa-check"></i><b>4.2</b> Getting started</a><ul>
<li class="chapter" data-level="4.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>4.2.1</b> The data</a></li>
<li class="chapter" data-level="4.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>4.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>4.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="4.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>4.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="4.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>4.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>4.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="4.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>4.5.1</b> Normalization</a></li>
<li class="chapter" data-level="4.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>4.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>4.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="4.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>4.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="4.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>4.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>5</b> Cytoscape Automation in R using Rcy3</a><ul>
<li class="chapter" data-level="5.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-3"><i class="fa fa-check"></i><b>5.1</b> Overview</a><ul>
<li class="chapter" data-level="5.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>5.1.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="5.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description"><i class="fa fa-check"></i><b>5.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="5.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>5.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="5.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>5.1.4</b> Background:</a></li>
<li class="chapter" data-level="5.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation"><i class="fa fa-check"></i><b>5.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>5.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="5.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-3"><i class="fa fa-check"></i><b>5.1.7</b> Time outline</a></li>
<li class="chapter" data-level="5.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>5.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>5.2</b> Background</a><ul>
<li class="chapter" data-level="5.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>5.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="5.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>5.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="5.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>5.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>5.3</b> Translating biological data into Cytoscape using RCy3</a></li>
<li class="chapter" data-level="5.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>5.4</b> Set up</a><ul>
<li class="chapter" data-level="5.4.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started-1"><i class="fa fa-check"></i><b>5.4.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>5.5</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="5.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>5.6</b> Example Data Set</a></li>
<li class="chapter" data-level="5.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>5.7</b> Finding Network Data</a></li>
<li class="chapter" data-level="5.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>5.8</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="5.9" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>5.9</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="5.10" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>5.10</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
<li class="chapter" data-level="5.11" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#references"><i class="fa fa-check"></i><b>5.11</b> References</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>6</b> Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="6.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-4"><i class="fa fa-check"></i><b>6.1</b> Overview</a><ul>
<li class="chapter" data-level="6.1.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#description-4"><i class="fa fa-check"></i><b>6.1.1</b> Description</a></li>
<li class="chapter" data-level="6.1.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.1.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#participation-3"><i class="fa fa-check"></i><b>6.1.3</b> Participation</a></li>
<li class="chapter" data-level="6.1.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>6.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.1.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-4"><i class="fa fa-check"></i><b>6.1.5</b> Time outline</a></li>
<li class="chapter" data-level="6.1.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>6.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#introductory-material"><i class="fa fa-check"></i><b>6.2</b> Introductory material</a><ul>
<li class="chapter" data-level="6.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#discussion"><i class="fa fa-check"></i><b>6.2.1</b> Discussion</a></li>
<li class="chapter" data-level="6.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#why-learn-delayedarray"><i class="fa fa-check"></i><b>6.2.2</b> Why learn DelayedArray?</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-of-delayedarray-framework"><i class="fa fa-check"></i><b>6.3</b> Overview of DelayedArray framework</a><ul>
<li class="chapter" data-level="6.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-package"><i class="fa fa-check"></i><b>6.3.1</b> The <strong>DelayedArray</strong> package</a></li>
<li class="chapter" data-level="6.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#packages-that-extend-delayedarray"><i class="fa fa-check"></i><b>6.3.2</b> Packages that extend <strong>DelayedArray</strong></a></li>
<li class="chapter" data-level="6.3.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#packages-that-use-delayedarray"><i class="fa fa-check"></i><b>6.3.3</b> Packages that use <strong>DelayedArray</strong></a></li>
<li class="chapter" data-level="6.3.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-class"><i class="fa fa-check"></i><b>6.3.4</b> The <em>DelayedArray</em> class</a></li>
<li class="chapter" data-level="6.3.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fundamental-concepts"><i class="fa fa-check"></i><b>6.3.5</b> Fundamental concepts</a></li>
<li class="chapter" data-level="6.3.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#whats-out-there-already"><i class="fa fa-check"></i><b>6.3.6</b> What’s out there already?</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#incoporating-delayedarray-into-a-package"><i class="fa fa-check"></i><b>6.4</b> Incoporating DelayedArray into a package</a><ul>
<li class="chapter" data-level="6.4.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#writing-algorithms-to-process-delayedarray-instances"><i class="fa fa-check"></i><b>6.4.1</b> Writing algorithms to process <em>DelayedArray</em> instances</a></li>
</ul></li>
<li class="chapter" data-level="6.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#questions-and-discussion"><i class="fa fa-check"></i><b>6.5</b> Questions and discussion</a></li>
<li class="chapter" data-level="6.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#todos"><i class="fa fa-check"></i><b>6.6</b> TODOs</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><i class="fa fa-check"></i><b>7</b> Variant Functional Annotation using StatePaintR, FunciVar &amp; MotifBreakR</a><ul>
<li class="chapter" data-level="7.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>7.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="7.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-description-1"><i class="fa fa-check"></i><b>7.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="7.2.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.2.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-participation-1"><i class="fa fa-check"></i><b>7.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="7.2.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>7.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.2.4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#time-outline-5"><i class="fa fa-check"></i><b>7.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>7.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="7.3.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#key-concepts"><i class="fa fa-check"></i><b>7.3.1</b> Key Concepts</a></li>
<li class="chapter" data-level="7.3.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#learning-objectives-4"><i class="fa fa-check"></i><b>7.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>8</b> Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="8.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>8.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="8.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-2"><i class="fa fa-check"></i><b>8.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="8.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-2"><i class="fa fa-check"></i><b>8.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="8.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>8.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-6"><i class="fa fa-check"></i><b>8.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>8.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="8.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop"><i class="fa fa-check"></i><b>8.4</b> Workshop</a></li>
<li class="chapter" data-level="8.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>8.5</b> Where does it all come from?</a></li>
<li class="chapter" data-level="8.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>8.6</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="8.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>8.7</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="8.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>8.8</b> Data types</a></li>
<li class="chapter" data-level="8.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>8.9</b> Differential expression analysis</a></li>
<li class="chapter" data-level="8.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>8.10</b> Gene sets</a></li>
<li class="chapter" data-level="8.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>8.11</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="8.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>8.12</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="8.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>8.13</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="8.14" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>8.14</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>9</b> RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="9.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>9.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-7"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>9.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="9.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-4"><i class="fa fa-check"></i><b>9.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="9.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-5"><i class="fa fa-check"></i><b>9.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="9.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-1"><i class="fa fa-check"></i><b>9.4</b> Introduction</a></li>
<li class="chapter" data-level="9.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>9.5</b> Data packaging</a><ul>
<li class="chapter" data-level="9.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>9.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="9.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>9.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="9.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>9.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="9.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>9.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="9.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>9.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="9.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>9.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="9.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>9.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="9.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>9.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="9.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>9.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="9.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>9.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="9.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>9.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="9.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>9.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="9.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>9.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="9.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>9.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="9.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>9.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>10</b> Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="10.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information-3"><i class="fa fa-check"></i><b>10.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-4"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-4"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>10.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-8"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-5"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-6"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>10.4</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="10.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>10.5</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>10.6</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>10.7</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>10.8</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="10.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#grangeslist-groups-of-genomic-ranges"><i class="fa fa-check"></i><b>10.9</b> <em>GRangesList</em>: Groups of Genomic Ranges</a></li>
<li class="chapter" data-level="10.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-grangeslist-accessors"><i class="fa fa-check"></i><b>10.10</b> Basic <em>GRangesList</em> accessors</a></li>
<li class="chapter" data-level="10.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#combining-grangeslist-objects"><i class="fa fa-check"></i><b>10.11</b> Combining <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-grangeslist-objects"><i class="fa fa-check"></i><b>10.12</b> Basic interval operations for <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-grangeslist-objects"><i class="fa fa-check"></i><b>10.13</b> Subsetting <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#looping-over-grangeslist-objects"><i class="fa fa-check"></i><b>10.14</b> Looping over <em>GRangesList</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>11</b> Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="11.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructors-names-and-contact-information-4"><i class="fa fa-check"></i><b>11.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="11.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-5"><i class="fa fa-check"></i><b>11.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="11.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-8"><i class="fa fa-check"></i><b>11.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="11.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-5"><i class="fa fa-check"></i><b>11.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>11.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="11.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-9"><i class="fa fa-check"></i><b>11.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="11.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-6"><i class="fa fa-check"></i><b>11.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="11.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-7"><i class="fa fa-check"></i><b>11.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-1"><i class="fa fa-check"></i><b>11.4</b> Workshop</a></li>
<li class="chapter" data-level="11.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-2"><i class="fa fa-check"></i><b>11.5</b> Introduction</a><ul>
<li class="chapter" data-level="11.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>11.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="11.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>11.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="11.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>11.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="11.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup"><i class="fa fa-check"></i><b>11.6</b> Setup</a></li>
<li class="chapter" data-level="11.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>11.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="11.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>11.8</b> The Grammar</a><ul>
<li class="chapter" data-level="11.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>11.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="11.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>11.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="11.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises"><i class="fa fa-check"></i><b>11.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="11.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>11.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="11.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>11.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="11.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>11.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="11.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>11.10</b> What’s next?</a></li>
<li class="chapter" data-level="11.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>11.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>12</b> Workflow for Multi-omics Analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="12.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>12.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="12.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-6"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-9"><i class="fa fa-check"></i><b>12.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-6"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>12.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="12.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-10"><i class="fa fa-check"></i><b>12.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-7"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-8"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>12.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="12.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>12.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="12.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>12.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="12.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>12.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>12.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="12.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>12.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="12.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>12.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="12.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>12.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>12.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="12.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>12.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="12.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>12.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="12.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>12.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="12.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>12.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="12.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>12.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="12.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>12.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>12.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="12.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>12.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="12.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>12.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="12.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>12.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>12.9</b> Complete cases</a></li>
<li class="chapter" data-level="12.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>12.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="12.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>12.11</b> Extraction</a><ul>
<li class="chapter" data-level="12.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>12.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="12.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>12.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="12.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>12.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="12.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>12.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="12.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>12.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="12.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>12.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="12.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>12.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="12.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>12.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="12.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>12.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="12.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>12.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="12.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>12.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="12.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>12.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="12.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>12.16.1</b> “Simplification” of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="12.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>12.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="12.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>12.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="12.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>12.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="12.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>12.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="12.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>12.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="12.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>12.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="12.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>12.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="12.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>12.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="12.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>12.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="12.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>12.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="12.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>12.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>13</b> Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="13.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors"><i class="fa fa-check"></i><b>13.1</b> Instructors:</a></li>
<li class="chapter" data-level="13.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-7"><i class="fa fa-check"></i><b>13.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="13.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-10"><i class="fa fa-check"></i><b>13.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-7"><i class="fa fa-check"></i><b>13.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>13.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="13.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-11"><i class="fa fa-check"></i><b>13.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-8"><i class="fa fa-check"></i><b>13.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-9"><i class="fa fa-check"></i><b>13.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>14</b> Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="14.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>14.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="14.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-8"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-11"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-4"><i class="fa fa-check"></i><b>14.2.2</b> Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>14.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="14.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-4"><i class="fa fa-check"></i><b>14.3</b> Introduction</a></li>
<li class="chapter" data-level="14.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>14.4</b> Why maintain your package</a></li>
<li class="chapter" data-level="14.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>14.5</b> Infrastructure used for maintenance</a></li>
<li class="chapter" data-level="14.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>14.6</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="14.6.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>14.6.1</b> Build report</a></li>
<li class="chapter" data-level="14.6.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>14.6.2</b> Landing page badges</a></li>
<li class="chapter" data-level="14.6.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>14.6.3</b> Download statistics</a></li>
<li class="chapter" data-level="14.6.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>14.6.4</b> RSS feeds</a></li>
<li class="chapter" data-level="14.6.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>14.6.5</b> Support site, Bioc-devel mailing list</a></li>
<li class="chapter" data-level="14.6.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github"><i class="fa fa-check"></i><b>14.6.6</b> GitHub</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>14.7</b> Round up of resources available</a></li>
<li class="chapter" data-level="14.8" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#acknowledgements"><i class="fa fa-check"></i><b>14.8</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>15</b> Public Data Resources and Bioconductor</a><ul>
<li class="chapter" data-level="15.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>15.1</b> Instructor name(s) and contact information</a></li>
<li class="chapter" data-level="15.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>15.2</b> Syllabus</a><ul>
<li class="chapter" data-level="15.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-9"><i class="fa fa-check"></i><b>15.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="15.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-12"><i class="fa fa-check"></i><b>15.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="15.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-8"><i class="fa fa-check"></i><b>15.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="15.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>15.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="15.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-12"><i class="fa fa-check"></i><b>15.2.5</b> Time outline</a></li>
<li class="chapter" data-level="15.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.2.6</b> workshop goals and objectives</a></li>
<li class="chapter" data-level="15.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-9"><i class="fa fa-check"></i><b>15.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="15.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-10"><i class="fa fa-check"></i><b>15.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-5"><i class="fa fa-check"></i><b>15.3</b> Overview</a></li>
<li class="chapter" data-level="15.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>15.4</b> GEOquery</a><ul>
<li class="chapter" data-level="15.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>15.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="15.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>15.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="15.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>15.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="15.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>15.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="15.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>15.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="15.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>15.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="15.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>15.5.2</b> Basic design</a></li>
<li class="chapter" data-level="15.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>15.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>15.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="15.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>15.6.1</b> Authentication</a></li>
<li class="chapter" data-level="15.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>15.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="15.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>15.7</b> Sequence Read Archive</a></li>
<li class="chapter" data-level="15.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>15.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="15.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>15.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="15.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>15.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="15.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>15.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="15.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>15.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="15.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>15.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="15.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>15.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="15.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>15.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="15.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>15.12</b> Pharmacogenomics</a></li>
<li class="chapter" data-level="15.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>15.13</b> Bibliography</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets" class="section level1">
<h1><span class="header-section-number">6</span> Effectively using the DelayedArray framework to support the analysis of large datasets</h1>
<p>Authors: Peter Francis Hickey<a href="#fn9" class="footnoteRef" id="fnref9"><sup>9</sup></a>, <br/> Last modified: 12 June, 2018.</p>
<div id="overview-4" class="section level2">
<h2><span class="header-section-number">6.1</span> Overview</h2>
<div id="description-4" class="section level3">
<h3><span class="header-section-number">6.1.1</span> Description</h3>
<p>This workshop will teach the fundamental concepts underlying the DelayedArray framework and related infrastructure. It is intended for package developers who want to learn how to use the DelayedArray framework to support the analysis of large datasets, particularly through the use of on-disk data storage.</p>
<p>The first part of the workshop will provide an overview of the DelayedArray infrastructure and introduce computing on DelayedArray objects using delayed operations and block-processing. The second part of the workshop will present strategies for adding support for DelayedArray to an existing package and extending the DelayedArray framework.</p>
<p>Students can expect a mixture of lecture and question-and-answer session to teach the fundamental concepts. There will be plenty of examples to illustrate common design patterns for writing performant code, although we will not be writing much code during the workshop.</p>
</div>
<div id="pre-requisites-3" class="section level3">
<h3><span class="header-section-number">6.1.2</span> Pre-requisites</h3>
<ul>
<li>Solid understanding of R</li>
<li>Familiarity with common operations on arrays (e.g., <code>colSums()</code> and those available in the <strong>matrixStats</strong> package)</li>
<li>Familiarity with object oriented programming, particularly S4, will be useful but is not essential</li>
<li>No familiarity required of technical details of particular data storage backends (e.g., HDF5, sparse matrices)</li>
<li>No familiarity required of particular biological techniques (e.g., single-cell RNA-seq)</li>
</ul>
</div>
<div id="participation-3" class="section level3">
<h3><span class="header-section-number">6.1.3</span> Participation</h3>
<p>Questions and discussion are encouraged! This will be especially important to guide the second half of the worshop which focuses on integrating DelayedArray into an existing or new Bioconductor package. Students will be expected to be able to follow and reason about example R code.</p>
</div>
<div id="r-bioconductor-packages-used-4" class="section level3">
<h3><span class="header-section-number">6.1.4</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>DelayedArray</li>
<li>HDF5Array</li>
<li>SummarizedExperiment</li>
<li>DelayedMatrixStats</li>
<li>beachmat</li>
</ul>
</div>
<div id="time-outline-4" class="section level3">
<h3><span class="header-section-number">6.1.5</span> Time outline</h3>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Introductory slides</td>
<td>15m</td>
</tr>
<tr class="even">
<td>Part 1: Overview of DelayedArray framework</td>
<td>45m</td>
</tr>
<tr class="odd">
<td>Part 2: Incoporating DelayedArray into a package</td>
<td>45m</td>
</tr>
<tr class="even">
<td>Questions and discussion</td>
<td>15m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives-4" class="section level3">
<h3><span class="header-section-number">6.1.6</span> Workshop goals and objectives</h3>
<div id="learning-goals-1" class="section level4">
<h4><span class="header-section-number">6.1.6.1</span> Learning goals</h4>
<ul>
<li>Identify when it is useful to use a <em>DelayedArray</em> instead of an ordinary array or other array-like data structure.</li>
<li>Become familiar with the fundamental concepts of delayed operations, block-processing, and realization.</li>
<li>Learn of existing functions and packages for constructing and computing on DelayedArray objects, avoiding the need to re-invent the wheel.</li>
<li>Learn common design patterns for writing performant code that operates on a DelayedArray.</li>
<li>Evaluate whether an existing function that operates on an ordinary array can be readily adapted to work on a DelayedArray.</li>
<li>Reason about potential bottlenecks in algorithms operating on DelayedArray objects.</li>
</ul>
</div>
<div id="learning-objectives-1" class="section level4">
<h4><span class="header-section-number">6.1.6.2</span> Learning objectives</h4>
<ul>
<li>Understand the differences between a <em>DelayedArray</em> instance and an instance of a subclass (e.g., <em>HDF5Array</em>, <em>RleArray</em>).</li>
<li>Know what types of operations ‘degrade’ an instance of a <em>DelayedArray</em> subclass to a <em>DelayedArray</em>, as well as when and why this matters.</li>
<li>Construct a <em>DelayedArray</em>:</li>
<li>From an in-memory array-like object.</li>
<li>From an on-disk data store (e.g., HDF5).</li>
<li>From scratch by writing data to a <em>RealizationSink</em>.</li>
<li>Take a function that operates on rows or columns of a matrix and apply it to a DelayedMatrix.</li>
<li>Use block-processing on a <em>DelayedArray</em> to compute:</li>
<li>A univariate (scalar) summary statistic (e.g., <code>max()</code>).</li>
<li>A multivariate (vector) summary statistic (e.g., <code>colSum()</code> or <code>rowMean()</code>).</li>
<li>A multivariate (array-like) summary statistic (e.g., <code>rowRanks()</code>).</li>
<li>Design an algorithm that imports data into a DelayedArray.</li>
</ul>
</div>
</div>
</div>
<div id="introductory-material" class="section level2">
<h2><span class="header-section-number">6.2</span> Introductory material</h2>
<p>Data from a high-throughput biological assay, such as single-cell RNA-sequencing (scRNA-seq), will often be summarised as a matrix of counts, where rows correspond to features and columns to samples<a href="#fn10" class="footnoteRef" id="fnref10"><sup>10</sup></a>. Within <strong>Bioconductor</strong>, the <em>SummarizedExperiment</em> class is the recommended container for such data, offering a rich interface that tightly links assay measurements to data on the features and the samples.</p>
<div class="figure">
<img src="https://docs.google.com/feeds/download/drawings/Export?id=1kiC8Qlo1mhSnLDqkGiRNPSo6GWn3C2duBszCFbJCB-g&amp;exportFormat=svg" alt="The SummarizedExperiment class is used to store rectangular arrays of experimental results (assays). Although each assay is here drawn as a matrix, higher-dimensional arrays are also supported." />
<p class="caption">The <em>SummarizedExperiment</em> class is used to store rectangular arrays of experimental results (<em>assays</em>). Although each <em>assay</em> is here drawn as a matrix, higher-dimensional arrays are also supported.</p>
</div>
<p>Traditionally, the assay data are stored in-memory as an ordinary <em>array</em> object<a href="#fn11" class="footnoteRef" id="fnref11"><sup>11</sup></a>. Storing the data in-memory becomes a real pain with the ever-growing size of ’omics datasets; it is now not uncommon to collect <span class="math inline">\(10,000-100,000,000\)</span> measurements on <span class="math inline">\(100 - 1,000,000\)</span> samples, which would occupy <span class="math inline">\(10-1,000\)</span> gigabytes (GB) if stored in-memory as ordinary R arrays!</p>
<p>Let’s take as an example some single-cell RNA-seq data on 1.3 million brain cells from embryonic mice, generated by 10X Genomics<a href="#fn12" class="footnoteRef" id="fnref12"><sup>12</sup></a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(TENxBrainData)
<span class="co"># </span><span class="al">NOTE</span><span class="co">: This will download the data and may take a little while on the first </span>
<span class="co">#       run. The result will be cached, however, so subsequent runs are near </span>
<span class="co">#       instantaneous.</span>
tenx &lt;-<span class="st"> </span><span class="kw">TENxBrainData</span>()
<span class="co"># The data are stored in a SingleCellExperiment, an extension of the </span>
<span class="co"># SummarizedExperiment class.</span>
<span class="kw">class</span>(tenx)
<span class="co">#&gt; [1] &quot;SingleCellExperiment&quot;</span>
<span class="co">#&gt; attr(,&quot;package&quot;)</span>
<span class="co">#&gt; [1] &quot;SingleCellExperiment&quot;</span>
<span class="kw">dim</span>(tenx)
<span class="co">#&gt; [1]   27998 1306127</span>

<span class="co"># How big much memory do the counts data use?</span>
counts &lt;-<span class="st"> </span><span class="kw">assay</span>(tenx, <span class="st">&quot;counts&quot;</span>, <span class="dt">withDimnames =</span> <span class="ot">FALSE</span>)
<span class="kw">print</span>(<span class="kw">object.size</span>(counts))
<span class="co">#&gt; 2144 bytes</span></code></pre></div>
<p>The counts data only require a tiny amount of RAM despite it having 27998 rows and 1306127 columns. And the <code>counts</code> object still “feels” like an ordinary R matrix:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Oooh, pretty-printing.</span>
counts
<span class="co">#&gt; &lt;27998 x 1306127&gt; HDF5Matrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;                [,1]       [,2]       [,3]       [,4] ... [,1306124]</span>
<span class="co">#&gt;     [1,]          0          0          0          0   .          0</span>
<span class="co">#&gt;     [2,]          0          0          0          0   .          0</span>
<span class="co">#&gt;     [3,]          0          0          0          0   .          0</span>
<span class="co">#&gt;     [4,]          0          0          0          0   .          0</span>
<span class="co">#&gt;     [5,]          0          0          0          0   .          0</span>
<span class="co">#&gt;      ...          .          .          .          .   .          .</span>
<span class="co">#&gt; [27994,]          0          0          0          0   .          0</span>
<span class="co">#&gt; [27995,]          1          0          0          2   .          0</span>
<span class="co">#&gt; [27996,]          0          0          0          0   .          0</span>
<span class="co">#&gt; [27997,]          0          0          0          0   .          0</span>
<span class="co">#&gt; [27998,]          0          0          0          0   .          0</span>
<span class="co">#&gt;          [,1306125] [,1306126] [,1306127]</span>
<span class="co">#&gt;     [1,]          0          0          0</span>
<span class="co">#&gt;     [2,]          0          0          0</span>
<span class="co">#&gt;     [3,]          0          0          0</span>
<span class="co">#&gt;     [4,]          0          0          0</span>
<span class="co">#&gt;     [5,]          0          0          0</span>
<span class="co">#&gt;      ...          .          .          .</span>
<span class="co">#&gt; [27994,]          0          0          0</span>
<span class="co">#&gt; [27995,]          1          0          0</span>
<span class="co">#&gt; [27996,]          0          0          0</span>
<span class="co">#&gt; [27997,]          0          0          0</span>
<span class="co">#&gt; [27998,]          0          0          0</span>

<span class="co"># Let&#39;s take a subset of the data</span>
counts[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>]
<span class="co">#&gt; &lt;10 x 10&gt; DelayedMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1]  [,2]  [,3]  [,4] ...  [,7]  [,8]  [,9] [,10]</span>
<span class="co">#&gt;  [1,]     0     0     0     0   .     0     0     0     0</span>
<span class="co">#&gt;  [2,]     0     0     0     0   .     0     0     0     0</span>
<span class="co">#&gt;  [3,]     0     0     0     0   .     0     0     0     0</span>
<span class="co">#&gt;  [4,]     0     0     0     0   .     0     0     0     0</span>
<span class="co">#&gt;  [5,]     0     0     0     0   .     0     0     0     0</span>
<span class="co">#&gt;  [6,]     0     0     0     0   .     0     0     0     0</span>
<span class="co">#&gt;  [7,]     0     0     0     0   .     0     0     0     0</span>
<span class="co">#&gt;  [8,]     0     0     0     0   .     2     0     1     1</span>
<span class="co">#&gt;  [9,]     0     0     1     0   .     0     0     0     0</span>
<span class="co">#&gt; [10,]     0     0     0     0   .     0     0     0     0</span>

<span class="co"># Let&#39;s compute column sums (crude library sizes) for the first 100 samples.</span>
<span class="co"># </span><span class="al">TODO</span><span class="co">: DelayedArray:: prefix shouldn&#39;t be necessary</span>
DelayedArray<span class="op">::</span><span class="kw">colSums</span>(counts[, <span class="dv">1</span><span class="op">:</span><span class="dv">100</span>])
<span class="co">#&gt;   [1]  4046  2087  4654  3193  8444 11178  2375  3672  3115  4592  7899</span>
<span class="co">#&gt;  [12]  5474  2894  5887  8349  7310  3340 23410  7326  3446  4601  5921</span>
<span class="co">#&gt;  [23]  4746  3063  3879  3582  2854  4053  6987  9155  3747  1534  6632</span>
<span class="co">#&gt;  [34]  4099  2846  5025  6688  3742  2982  1998  1808  3121 10561  3874</span>
<span class="co">#&gt;  [45]  4143  1500  2280  3060  4325  3161  2522  1979  6033  3721  2546</span>
<span class="co">#&gt;  [56]  6317  2756  3896  4475  5580  1879  8746  4873  2202  4517  2815</span>
<span class="co">#&gt;  [67]  3809  2580  4655  3523  4717  6436  2434  5704  2962 11654  4848</span>
<span class="co">#&gt;  [78]  5288  6689  5761 11539 15745  2986  2736  3666  2476  2251  3052</span>
<span class="co">#&gt;  [89]  5480  1721  4166  4451  1893  5606  2551  2810  1555  1840  2972</span>
<span class="co">#&gt; [100]  2404</span></code></pre></div>
<p>The reason for the small memory footprint and matrix-like “feel” of the <code>counts</code> object is because the counts data are in fact stored on-disk in a Hierarchical Data Format (<strong>HDF5</strong>) file and we are interacting with the data via the DelayedArray framework.</p>
<div id="discussion" class="section level3">
<h3><span class="header-section-number">6.2.1</span> Discussion</h3>
<p><strong>TODO:</strong> Make these ‘discussions/challenges’ into a boxes that visually break out</p>
<ul>
<li>Talk with your neighbour about the sorts of ‘big data’ you are analysing, the challenges you’ve faced, and strategies you’re using to tackle this.</li>
<li><strong>TODO:</strong> (keep this?) Play around with the <code>tenx</code> and <code>counts</code> data. Could use this to demonstrate the upcoming challenges of chunking (e.g., compute row-wise summary).</li>
</ul>
<p><strong>TABLE</strong> gives some examples of contemporary experiments and the memory requirements if these data are to be stored in-memory as ordinary R arrays.</p>
<table>
<thead>
<tr class="header">
<th align="left">Assay</th>
<th align="left">Size</th>
<th align="left"><code>nrow</code></th>
<th align="left"><code>ncol</code></th>
<th align="right">Number of assays</th>
<th align="left">Type</th>
<th align="left">Reference</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">WGBS (mCG)</td>
<td align="left">67.992 GB</td>
<td align="left">29,307,073</td>
<td align="left">145</td>
<td align="right">3</td>
<td align="left">2 x dense integer, 1 x dense double</td>
<td align="left">eGTEx (unpublished)</td>
</tr>
<tr class="even">
<td align="left">scRNA-seq</td>
<td align="left">146.276 GB</td>
<td align="left">27,998</td>
<td align="left">1,306,127</td>
<td align="right">1</td>
<td align="left">sparse integer</td>
<td align="left"><a href="https://bioconductor.org/packages/TENxBrainData/" class="uri">https://bioconductor.org/packages/TENxBrainData/</a></td>
</tr>
</tbody>
</table>
<p>There are various strategies for handling such large data in R. For example, we could use:</p>
<ol style="list-style-type: decimal">
<li>Sparse matrices for single-cell RNA-seq data.</li>
<li>Run length encoded vectors for ChIP-seq coverage profiles.</li>
<li>Storing data on disk, and only bringing into memory as required, for whole genome methylation data<a href="#fn13" class="footnoteRef" id="fnref13"><sup>13</sup></a>.</li>
</ol>
<p>Each of these approaches has its strengths, as well as weaknesses and idiosyncrasies. For example,</p>
<ul>
<li><strong>Matrix</strong> doesn’t support long vectors</li>
<li>Sparsity is readily lost (e.g., normalization often destroys it)</li>
<li>etc.</li>
</ul>
<p><strong>TODO:</strong> Lead discussion of limitations</p>
</div>
<div id="why-learn-delayedarray" class="section level3">
<h3><span class="header-section-number">6.2.2</span> Why learn DelayedArray?</h3>
<p>The high-level goals of the DelayedArray framework are:</p>
<ul>
<li>Provide a common R interface to array-like data, where the data may be in-memory or on-disk.</li>
<li>Support delayed operations, which avoid doing any computation until the result is required.</li>
<li>Support block-processing of the data, which enables bounded-memory and parallel computations.</li>
</ul>
<p>From the <strong>DelayedArray</strong> <code>DESCRIPTION</code>:</p>
<p><strong>TODO:</strong> Use <strong>desc</strong> to extract Description field?</p>
<blockquote>
<p>Wrapping an array-like object (typically an on-disk object) in a DelayedArray object allows one to perform common array operations on it without loading the object in memory. In order to reduce memory usage and optimize performance, operations on the object are either delayed or executed using a block processing mechanism. Note that this also works on in-memory array-like objects like DataFrame objects (typically with Rle columns), Matrix objects, and ordinary arrays and data frames. (<a href="https://bioconductor.org/packages/release/bioc/html/DelayedArray.html" class="uri">https://bioconductor.org/packages/release/bioc/html/DelayedArray.html</a>)</p>
</blockquote>
<p>These goals are similar to that of the <strong>tibble</strong> and <strong>dplyr</strong> packages:</p>
<blockquote>
<p>A tibble, or <code>tbl_df</code>, is a modern reimagining of the data.frame, keeping what time has proven to be effective, and throwing out what is not. Tibbles are data.frames that are lazy and surly (<a href="http://tibble.tidyverse.org/#overview" class="uri">http://tibble.tidyverse.org/#overview</a>)</p>
</blockquote>
<blockquote>
<p>dplyr is designed to abstract over how the data is stored. That means as well as working with local data frames, you can also work with remote database tables, using exactly the same R code. (<a href="https://dplyr.tidyverse.org/#overview" class="uri">https://dplyr.tidyverse.org/#overview</a>)</p>
</blockquote>
<p>An important feature of the DelayedArray framework is that it supports all these strategies, and more, with a common interface that aims to feel like the interface to ordinary arrays, which provides familiarity to R users.</p>
<div id="learning-goal" class="section level4">
<h4><span class="header-section-number">6.2.2.1</span> Learning goal</h4>
<ul>
<li>Identify when it is useful to use a <em>DelayedArray</em> instead of an ordinary array or other array-like data structure.</li>
</ul>
</div>
</div>
</div>
<div id="overview-of-delayedarray-framework" class="section level2">
<h2><span class="header-section-number">6.3</span> Overview of DelayedArray framework</h2>
<p>The core of DelayedArray framework is implemented in the <strong>DelayedArray</strong> Bioconductor package. Other packages extend the framework or simply use it as is to enable analyses of large datasets.</p>
<div id="the-delayedarray-package" class="section level3">
<h3><span class="header-section-number">6.3.1</span> The <strong>DelayedArray</strong> package</h3>
<p>The <strong>DelayedArray</strong> package defines the key classes, generics, and methods<a href="#fn14" class="footnoteRef" id="fnref14"><sup>14</sup></a>, as well as miscellaneous helper functions, that implement the DelayedArray framework.</p>
<p>The reverse dependencies of <strong>DelayedArray</strong> are shown below:</p>
<p><img src="500_Effectively_Using_the_DelayedArray_Framework_files/figure-html/unnamed-chunk-5-1.png" width="672" /></p>
<p>The above figures includes packages that extend the DelayedArray framework in various ways, as well as those that simply use the DelayedArray framework to analyse specific types of ’omics data. We briefly discuss some of these:</p>
</div>
<div id="packages-that-extend-delayedarray" class="section level3">
<h3><span class="header-section-number">6.3.2</span> Packages that extend <strong>DelayedArray</strong></h3>
<p>There are two ways a package may extend the DelayedArray framework. The first kind of package adds support for a new <em>realization backend</em> (<strong>TODO</strong> Define?). Examples of this are:</p>
<ul>
<li>The <strong>HDF5Array</strong> package adds the <em>HDF5Array</em> realization backend for accessing and creating data stored on-disk in an HDF5 file. This is typically used for on-disk representation of multidimensional (numeric) arrays.</li>
<li>The <strong>GDSArray</strong> package adds the <em>GDSArray</em> backend for accessing and creating data stored on-disk in a GDS file. This is typically used for on-disk representation of genotyping or sequence data.</li>
<li>The <strong>rhdf5client</strong> packages adds the <em>H5S_Array</em> realization backend for accessing and creating data stored on a HDF Server, a Python-based web service that can be used to send and receive HDF5 data using an HTTP-based REST interface. This is typically used for on-disk representation of multidimensional (numeric) arrays that need to be shared with multiple users from a central location.</li>
</ul>
<p>The second kind of package adds methods for computing on <em>DelayedArray</em> instances. Examples of this are:</p>
<ul>
<li>The <strong>DelayedMatrixStats</strong> package provides methods for computing commonly used row- and column-wise summaries of a 2-dimensional <em>DelayedArray</em>.</li>
<li>The <strong>beachmat</strong> package provides a consistent C++ class interface for a variety of commonly used matrix types, including ordinary R arrays, sparse matrices, and <em>DelayedArray</em> with various backends.</li>
<li>The <strong>kmknn</strong> package provides methods for performing k-means for k-nearest neighbours of data stored in a <em>DelayedArray</em>.</li>
<li>Packages for performing matrix factorizations, (generalized) linear regression, , etc. (work in progress)</li>
</ul>
</div>
<div id="packages-that-use-delayedarray" class="section level3">
<h3><span class="header-section-number">6.3.3</span> Packages that use <strong>DelayedArray</strong></h3>
<ul>
<li>The <strong>bsseq</strong> package uses the DelayedArray framework to support the analysis of large whole-genome bisulfite methylation sequencing experiments.</li>
<li><strong>TODO</strong>: minfi, scater, scran, others</li>
</ul>
</div>
<div id="the-delayedarray-class" class="section level3">
<h3><span class="header-section-number">6.3.4</span> The <em>DelayedArray</em> class</h3>
<p>The <em>DelayedArray</em> class is the key data structure that end users of the DelayedArray framework will interact with. A <em>DelayedMatrix</em> is the same thing as a two-dimensional <em>DelayedArray</em>, just as a <em>matrix</em> is the same thing as a two-dimensional <em>array</em>.</p>
<p>The <em>DelayedArray</em> class has a single slot called the <em>seed</em>. This name is evocative, it is the core of the object.</p>
<p>A package developer may create a subclass of <em>DelayedArray</em>; we will make extensive use of the <em>HDF5Array</em> class, for example.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># </span><span class="al">TODO</span><span class="co">: Graphical representation of this network?</span>
<span class="co"># </span><span class="al">TODO</span><span class="co">: Illustrate How classes depend on one another</span>
<span class="co">#       a. with DelayedArray as the root</span>
<span class="co">#       b. With DelayedArray as the leaf</span>
<span class="kw">showClass</span>(<span class="kw">getClass</span>(<span class="st">&quot;DelayedArray&quot;</span>, <span class="dt">where =</span> <span class="st">&quot;DelayedArray&quot;</span>))
<span class="co">#&gt; Class &quot;DelayedArray&quot; [package &quot;DelayedArray&quot;]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slots:</span>
<span class="co">#&gt;            </span>
<span class="co">#&gt; Name:  seed</span>
<span class="co">#&gt; Class:  ANY</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Extends: </span>
<span class="co">#&gt; Class &quot;DelayedUnaryOp&quot;, directly</span>
<span class="co">#&gt; Class &quot;DelayedOp&quot;, by class &quot;DelayedUnaryOp&quot;, distance 2</span>
<span class="co">#&gt; Class &quot;Array&quot;, by class &quot;DelayedUnaryOp&quot;, distance 3</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Known Subclasses: </span>
<span class="co">#&gt; Class &quot;DelayedMatrix&quot;, directly, with explicit coerce</span>
<span class="co">#&gt; Class &quot;DelayedArray1&quot;, directly</span>
<span class="co">#&gt; Class &quot;RleArray&quot;, directly</span>
<span class="co">#&gt; Class &quot;RleMatrix&quot;, by class &quot;DelayedMatrix&quot;, distance 2</span>
<span class="kw">showClass</span>(<span class="kw">getClass</span>(<span class="st">&quot;HDF5Array&quot;</span>, <span class="dt">where =</span> <span class="st">&quot;HDF5Array&quot;</span>))
<span class="co">#&gt; Class &quot;HDF5Array&quot; [package &quot;HDF5Array&quot;]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slots:</span>
<span class="co">#&gt;            </span>
<span class="co">#&gt; Name:  seed</span>
<span class="co">#&gt; Class:  ANY</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Extends: </span>
<span class="co">#&gt; Class &quot;DelayedArray&quot;, directly</span>
<span class="co">#&gt; Class &quot;DelayedUnaryOp&quot;, by class &quot;DelayedArray&quot;, distance 2</span>
<span class="co">#&gt; Class &quot;DelayedOp&quot;, by class &quot;DelayedArray&quot;, distance 3</span>
<span class="co">#&gt; Class &quot;Array&quot;, by class &quot;DelayedArray&quot;, distance 4</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Known Subclasses: </span>
<span class="co">#&gt; Class &quot;HDF5Matrix&quot;, directly, with explicit coerce</span></code></pre></div>
<div id="learning-objectives-2" class="section level4">
<h4><span class="header-section-number">6.3.4.1</span> Learning objectives</h4>
<p>In this section, we’ll learn how an end user can construct a <em>DelayedArray</em> from:</p>
<ul>
<li>An in-memory array-like object.</li>
<li>A local, on-disk data store, such as an HDF5 file.</li>
<li>A remote data store, such as from a HDF Server.</li>
</ul>
<p>We’ll also learn:</p>
<ul>
<li>What defines the “seed contract”</li>
<li>What types of operations ‘degrade’ an instance of a <em>DelayedArray</em> subclass to a <em>DelayedArray</em>, as well as when and why this matters (<strong>TODO</strong> Save the when and why it matters to later?).</li>
</ul>
</div>
<div id="the-seed-of-a-delayedarray" class="section level4">
<h4><span class="header-section-number">6.3.4.2</span> The seed of a <em>DelayedArray</em></h4>
<p>From an end user’s perspective, there are a two broad categories of seed:</p>
<ol style="list-style-type: decimal">
<li>In-memory seeds</li>
<li>Out-of-memory seeds
<ol style="list-style-type: lower-alpha">
<li>Local, on-disk seeds</li>
<li>Remote (in-memory or on-disk) seeds</li>
</ol></li>
</ol>
<p><strong>TODO:</strong> Box this out as an aside</p>
<blockquote>
<p>For a developer’s perspective, there’s a third category of seed that is defined by the <em>DelayedOp</em> class. Instances of the <em>DelayedOp</em> class are not intended to be manipulated directly by the end user, but they are central to the concept of delayed operations, which we will learn about later (<strong>TODO:</strong> Link to section).</p>
</blockquote>
<p>A seed must implement the “seed contract”<a href="#fn15" class="footnoteRef" id="fnref15"><sup>15</sup></a>:</p>
<ul>
<li><code>dim(x)</code>: Return the dimensions of the seed.</li>
<li><code>dimnames(x)</code>: Return the (possibly <code>NULL</code>) dimension names of the seed.</li>
<li><code>extract_array(x, index)</code>: Return a slice, specified by <code>index</code>, of the seed as an ordinary R <em>array</em>.</li>
</ul>
</div>
<div id="in-memory-seeds" class="section level4">
<h4><span class="header-section-number">6.3.4.3</span> In-memory seeds</h4>
<p>To begin, we’ll consider a <em>DelayedArray</em> instance with the simplest <em>seed</em>, an in-memory array:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(DelayedArray)

mat &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">rep</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">20</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">20</span>), <span class="dt">ncol =</span> <span class="dv">2</span>)
da_mat &lt;-<span class="st"> </span><span class="kw">DelayedArray</span>(<span class="dt">seed =</span> mat) 
da_mat
<span class="co">#&gt; &lt;105 x 2&gt; DelayedMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    1   15</span>
<span class="co">#&gt;   [2,]    2   15</span>
<span class="co">#&gt;   [3,]    2   15</span>
<span class="co">#&gt;   [4,]    3   15</span>
<span class="co">#&gt;   [5,]    3   15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   14   20</span>
<span class="co">#&gt; [102,]   14   20</span>
<span class="co">#&gt; [103,]   14   20</span>
<span class="co">#&gt; [104,]   14   20</span>
<span class="co">#&gt; [105,]   14   20</span>
<span class="kw">str</span>(da_mat)
<span class="co">#&gt; Formal class &#39;DelayedMatrix&#39; [package &quot;DelayedArray&quot;] with 1 slot</span>
<span class="co">#&gt;   ..@ seed: int [1:105, 1:2] 1 2 2 3 3 3 4 4 4 4 ...</span></code></pre></div>
<p>We can use other, more complex, array-like objects as the <em>seed</em>, such as <em>Matrix</em> objects from the <strong>Matrix</strong> package:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(Matrix)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;Matrix&#39;</span>
<span class="co">#&gt; The following object is masked from &#39;package:S4Vectors&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     expand</span>
Mat &lt;-<span class="st"> </span><span class="kw">Matrix</span>(mat)
da_Mat &lt;-<span class="st"> </span><span class="kw">DelayedArray</span>(<span class="dt">seed =</span> Mat)
<span class="co"># </span><span class="al">NOTE</span><span class="co">: The type is &quot;double&quot; because of how the Matrix package stores the data.</span>
da_Mat
<span class="co">#&gt; &lt;105 x 2&gt; DelayedMatrix object of type &quot;double&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    1   15</span>
<span class="co">#&gt;   [2,]    2   15</span>
<span class="co">#&gt;   [3,]    2   15</span>
<span class="co">#&gt;   [4,]    3   15</span>
<span class="co">#&gt;   [5,]    3   15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   14   20</span>
<span class="co">#&gt; [102,]   14   20</span>
<span class="co">#&gt; [103,]   14   20</span>
<span class="co">#&gt; [104,]   14   20</span>
<span class="co">#&gt; [105,]   14   20</span>
<span class="kw">str</span>(da_Mat)
<span class="co">#&gt; Formal class &#39;DelayedMatrix&#39; [package &quot;DelayedArray&quot;] with 1 slot</span>
<span class="co">#&gt;   ..@ seed:Formal class &#39;dgeMatrix&#39; [package &quot;Matrix&quot;] with 4 slots</span>
<span class="co">#&gt;   .. .. ..@ x       : num [1:210] 1 2 2 3 3 3 4 4 4 4 ...</span>
<span class="co">#&gt;   .. .. ..@ Dim     : int [1:2] 105 2</span>
<span class="co">#&gt;   .. .. ..@ Dimnames:List of 2</span>
<span class="co">#&gt;   .. .. .. ..$ : NULL</span>
<span class="co">#&gt;   .. .. .. ..$ : NULL</span>
<span class="co">#&gt;   .. .. ..@ factors : list()</span></code></pre></div>
<p>We can even use data frames as the <em>seed</em> of a two-dimensional <em>DelayedArray</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span><span class="kw">as.data.frame</span>(mat)
da_df &lt;-<span class="st"> </span><span class="kw">DelayedArray</span>(<span class="dt">seed =</span> df)
<span class="co"># </span><span class="al">NOTE</span><span class="co">: This inherits the (default) column names of the data.frame.</span>
da_df
<span class="co">#&gt; &lt;105 x 2&gt; DelayedMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;     V1 V2</span>
<span class="co">#&gt;   1  1 15</span>
<span class="co">#&gt;   2  2 15</span>
<span class="co">#&gt;   3  2 15</span>
<span class="co">#&gt;   4  3 15</span>
<span class="co">#&gt;   5  3 15</span>
<span class="co">#&gt; ...  .  .</span>
<span class="co">#&gt; 101 14 20</span>
<span class="co">#&gt; 102 14 20</span>
<span class="co">#&gt; 103 14 20</span>
<span class="co">#&gt; 104 14 20</span>
<span class="co">#&gt; 105 14 20</span>
<span class="kw">str</span>(da_df)
<span class="co">#&gt; Formal class &#39;DelayedMatrix&#39; [package &quot;DelayedArray&quot;] with 1 slot</span>
<span class="co">#&gt;   ..@ seed:&#39;data.frame&#39;: 105 obs. of  2 variables:</span>
<span class="co">#&gt;   .. ..$ V1: int [1:105] 1 2 2 3 3 3 4 4 4 4 ...</span>
<span class="co">#&gt;   .. ..$ V2: int [1:105] 15 15 15 15 15 15 15 15 15 15 ...</span>

<span class="kw">library</span>(tibble)
tbl &lt;-<span class="st"> </span><span class="kw">as_tibble</span>(mat)
da_tbl &lt;-<span class="st"> </span><span class="kw">DelayedArray</span>(<span class="dt">seed =</span> tbl)
<span class="co"># </span><span class="al">NOTE</span><span class="co">: This inherits the (default) column names of the tibble.</span>
da_tbl
<span class="co">#&gt; &lt;105 x 2&gt; DelayedMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;     V1 V2</span>
<span class="co">#&gt;   1  1 15</span>
<span class="co">#&gt;   2  2 15</span>
<span class="co">#&gt;   3  2 15</span>
<span class="co">#&gt;   4  3 15</span>
<span class="co">#&gt;   5  3 15</span>
<span class="co">#&gt; ...  .  .</span>
<span class="co">#&gt; 101 14 20</span>
<span class="co">#&gt; 102 14 20</span>
<span class="co">#&gt; 103 14 20</span>
<span class="co">#&gt; 104 14 20</span>
<span class="co">#&gt; 105 14 20</span>
<span class="kw">str</span>(da_tbl)
<span class="co">#&gt; Formal class &#39;DelayedMatrix&#39; [package &quot;DelayedArray&quot;] with 1 slot</span>
<span class="co">#&gt;   ..@ seed:Classes &#39;tbl_df&#39;, &#39;tbl&#39; and &#39;data.frame&#39;: 105 obs. of  2 variables:</span>
<span class="co">#&gt;   .. ..$ V1: int [1:105] 1 2 2 3 3 3 4 4 4 4 ...</span>
<span class="co">#&gt;   .. ..$ V2: int [1:105] 15 15 15 15 15 15 15 15 15 15 ...</span>

DF &lt;-<span class="st"> </span><span class="kw">as</span>(mat, <span class="st">&quot;DataFrame&quot;</span>)
da_DF &lt;-<span class="st"> </span><span class="kw">DelayedArray</span>(<span class="dt">seed =</span> DF)
<span class="co"># </span><span class="al">NOTE</span><span class="co">: This inherits the (default) column names of the DataFrame.</span>
da_DF
<span class="co">#&gt; &lt;105 x 2&gt; DelayedMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        V1 V2</span>
<span class="co">#&gt;   [1,]  1 15</span>
<span class="co">#&gt;   [2,]  2 15</span>
<span class="co">#&gt;   [3,]  2 15</span>
<span class="co">#&gt;   [4,]  3 15</span>
<span class="co">#&gt;   [5,]  3 15</span>
<span class="co">#&gt;    ...  .  .</span>
<span class="co">#&gt; [101,] 14 20</span>
<span class="co">#&gt; [102,] 14 20</span>
<span class="co">#&gt; [103,] 14 20</span>
<span class="co">#&gt; [104,] 14 20</span>
<span class="co">#&gt; [105,] 14 20</span>
<span class="kw">str</span>(da_DF)
<span class="co">#&gt; Formal class &#39;DelayedMatrix&#39; [package &quot;DelayedArray&quot;] with 1 slot</span>
<span class="co">#&gt;   ..@ seed:Formal class &#39;DataFrame&#39; [package &quot;S4Vectors&quot;] with 6 slots</span>
<span class="co">#&gt;   .. .. ..@ rownames       : NULL</span>
<span class="co">#&gt;   .. .. ..@ nrows          : int 105</span>
<span class="co">#&gt;   .. .. ..@ listData       :List of 2</span>
<span class="co">#&gt;   .. .. .. ..$ V1: int [1:105] 1 2 2 3 3 3 4 4 4 4 ...</span>
<span class="co">#&gt;   .. .. .. ..$ V2: int [1:105] 15 15 15 15 15 15 15 15 15 15 ...</span>
<span class="co">#&gt;   .. .. ..@ elementType    : chr &quot;ANY&quot;</span>
<span class="co">#&gt;   .. .. ..@ elementMetadata: NULL</span>
<span class="co">#&gt;   .. .. ..@ metadata       : list()</span></code></pre></div>
<p>A package developer can also implement a novel in-memory seed. For example, the <strong>DelayedArray</strong> package defines the <em>RleArraySeed</em> class<a href="#fn16" class="footnoteRef" id="fnref16"><sup>16</sup></a>. This can be used as the seed of an <em>RleArray</em>, a <em>DelayedArray</em> subclass for storing run-length encoded data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># </span><span class="al">NOTE</span><span class="co">: The DelayedArray package does not expose the RleArraySeed() constructor.</span>
<span class="co">#       Instead, we directly call the RleArray() constructor on the run-length </span>
<span class="co">#       encoded data.</span>
da_Rle &lt;-<span class="st"> </span><span class="kw">RleArray</span>(<span class="dt">rle =</span> <span class="kw">Rle</span>(mat), <span class="dt">dim =</span> <span class="kw">dim</span>(mat))
da_Rle
<span class="co">#&gt; &lt;105 x 2&gt; RleMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    1   15</span>
<span class="co">#&gt;   [2,]    2   15</span>
<span class="co">#&gt;   [3,]    2   15</span>
<span class="co">#&gt;   [4,]    3   15</span>
<span class="co">#&gt;   [5,]    3   15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   14   20</span>
<span class="co">#&gt; [102,]   14   20</span>
<span class="co">#&gt; [103,]   14   20</span>
<span class="co">#&gt; [104,]   14   20</span>
<span class="co">#&gt; [105,]   14   20</span>
<span class="kw">str</span>(da_Rle)
<span class="co">#&gt; Formal class &#39;RleMatrix&#39; [package &quot;DelayedArray&quot;] with 1 slot</span>
<span class="co">#&gt;   ..@ seed:Formal class &#39;SolidRleArraySeed&#39; [package &quot;DelayedArray&quot;] with 3 slots</span>
<span class="co">#&gt;   .. .. ..@ rle     :Formal class &#39;Rle&#39; [package &quot;S4Vectors&quot;] with 4 slots</span>
<span class="co">#&gt;   .. .. .. .. ..@ values         : int [1:20] 1 2 3 4 5 6 7 8 9 10 ...</span>
<span class="co">#&gt;   .. .. .. .. ..@ lengths        : int [1:20] 1 2 3 4 5 6 7 8 9 10 ...</span>
<span class="co">#&gt;   .. .. .. .. ..@ elementMetadata: NULL</span>
<span class="co">#&gt;   .. .. .. .. ..@ metadata       : list()</span>
<span class="co">#&gt;   .. .. ..@ DIM     : int [1:2] 105 2</span>
<span class="co">#&gt;   .. .. ..@ DIMNAMES:List of 2</span>
<span class="co">#&gt;   .. .. .. ..$ : NULL</span>
<span class="co">#&gt;   .. .. .. ..$ : NULL</span></code></pre></div>
<p>The <em>RleArray</em> examples illustrates some important concepts in the DelayedArray class hierarchy that warrants reiteration and expansion.</p>
</div>
<div id="degrading-delayedarray-subclasses" class="section level4">
<h4><span class="header-section-number">6.3.4.4</span> Degrading DelayedArray subclasses</h4>
<p>The <code>da_Rle</code> object is an <em>RleMatrix</em>, a direct subclass of <em>RleArray</em> and a direct subclass of <em>DelayedMatrix</em>. Both <em>RleArray</em> and <em>DelayedMatrix</em> are direct subclasses of a <em>DelayedArray</em>. As such, in accordance with properties of S4 class inheritance, <strong>an <em>RleMatrix</em> is a <em>DelayedArray</em></strong>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">da_Rle
<span class="co">#&gt; &lt;105 x 2&gt; RleMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    1   15</span>
<span class="co">#&gt;   [2,]    2   15</span>
<span class="co">#&gt;   [3,]    2   15</span>
<span class="co">#&gt;   [4,]    3   15</span>
<span class="co">#&gt;   [5,]    3   15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   14   20</span>
<span class="co">#&gt; [102,]   14   20</span>
<span class="co">#&gt; [103,]   14   20</span>
<span class="co">#&gt; [104,]   14   20</span>
<span class="co">#&gt; [105,]   14   20</span>
<span class="kw">is</span>(da_Rle, <span class="st">&quot;DelayedArray&quot;</span>)
<span class="co">#&gt; [1] TRUE</span>
<span class="kw">showClass</span>(<span class="kw">getClass</span>(<span class="st">&quot;RleMatrix&quot;</span>, <span class="dt">where =</span> <span class="st">&quot;DelayedArray&quot;</span>))
<span class="co">#&gt; Class &quot;RleMatrix&quot; [package &quot;DelayedArray&quot;]</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Slots:</span>
<span class="co">#&gt;            </span>
<span class="co">#&gt; Name:  seed</span>
<span class="co">#&gt; Class:  ANY</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Extends: </span>
<span class="co">#&gt; Class &quot;DelayedMatrix&quot;, directly</span>
<span class="co">#&gt; Class &quot;RleArray&quot;, directly</span>
<span class="co">#&gt; Class &quot;DelayedArray&quot;, by class &quot;DelayedMatrix&quot;, distance 2</span>
<span class="co">#&gt; Class &quot;DelayedUnaryOp&quot;, by class &quot;DelayedMatrix&quot;, distance 2</span>
<span class="co">#&gt; Class &quot;DelayedOp&quot;, by class &quot;DelayedMatrix&quot;, distance 2</span>
<span class="co">#&gt; Class &quot;Array&quot;, by class &quot;DelayedMatrix&quot;, distance 2</span>
<span class="co">#&gt; Class &quot;DataTable&quot;, by class &quot;DelayedMatrix&quot;, distance 2</span>
<span class="co">#&gt; Class &quot;DataTable_OR_NULL&quot;, by class &quot;DelayedMatrix&quot;, distance 3</span></code></pre></div>
<p>However, if we do (almost) anything to the <em>RleMatrix</em>, the result is ‘degraded’ to a <em>DelayedMatrix</em>. This ‘degradation’ isn’t an issue in and of itself, but it can be surprising and can complicate writing functions that expect a certain type of input (e.g., S4 methods).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># </span><span class="al">NOTE</span><span class="co">: Adding one to each element will &#39;degrade&#39; the result.</span>
da_Rle <span class="op">+</span><span class="st"> </span><span class="dv">1</span>
<span class="co">#&gt; &lt;105 x 2&gt; DelayedMatrix object of type &quot;double&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    2   16</span>
<span class="co">#&gt;   [2,]    3   16</span>
<span class="co">#&gt;   [3,]    3   16</span>
<span class="co">#&gt;   [4,]    4   16</span>
<span class="co">#&gt;   [5,]    4   16</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   15   21</span>
<span class="co">#&gt; [102,]   15   21</span>
<span class="co">#&gt; [103,]   15   21</span>
<span class="co">#&gt; [104,]   15   21</span>
<span class="co">#&gt; [105,]   15   21</span>
<span class="kw">is</span>(da_Rle <span class="op">+</span><span class="st"> </span><span class="dv">1</span>, <span class="st">&quot;RleMatrix&quot;</span>)
<span class="co">#&gt; [1] FALSE</span>

<span class="co"># </span><span class="al">NOTE</span><span class="co">: Subsetting will &#39;degrade&#39; the result.</span>
da_Rle[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, ]
<span class="co">#&gt; &lt;10 x 2&gt; DelayedMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;       [,1] [,2]</span>
<span class="co">#&gt;  [1,]    1   15</span>
<span class="co">#&gt;  [2,]    2   15</span>
<span class="co">#&gt;  [3,]    2   15</span>
<span class="co">#&gt;  [4,]    3   15</span>
<span class="co">#&gt;  [5,]    3   15</span>
<span class="co">#&gt;  [6,]    3   15</span>
<span class="co">#&gt;  [7,]    4   15</span>
<span class="co">#&gt;  [8,]    4   15</span>
<span class="co">#&gt;  [9,]    4   15</span>
<span class="co">#&gt; [10,]    4   15</span>
<span class="kw">is</span>(da_Rle[<span class="dv">1</span><span class="op">:</span><span class="dv">10</span>, ], <span class="st">&quot;RleMatrix&quot;</span>)
<span class="co">#&gt; [1] FALSE</span>

<span class="co"># </span><span class="al">NOTE</span><span class="co">: Changing the dimnames will &#39;degrade&#39; the result.</span>
da_Rle_with_dimnames &lt;-<span class="st"> </span>da_Rle
<span class="kw">colnames</span>(da_Rle_with_dimnames) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>)
da_Rle_with_dimnames
<span class="co">#&gt; &lt;105 x 2&gt; DelayedMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;         A  B</span>
<span class="co">#&gt;   [1,]  1 15</span>
<span class="co">#&gt;   [2,]  2 15</span>
<span class="co">#&gt;   [3,]  2 15</span>
<span class="co">#&gt;   [4,]  3 15</span>
<span class="co">#&gt;   [5,]  3 15</span>
<span class="co">#&gt;    ...  .  .</span>
<span class="co">#&gt; [101,] 14 20</span>
<span class="co">#&gt; [102,] 14 20</span>
<span class="co">#&gt; [103,] 14 20</span>
<span class="co">#&gt; [104,] 14 20</span>
<span class="co">#&gt; [105,] 14 20</span>
<span class="kw">is</span>(da_Rle_with_dimnames, <span class="st">&quot;RleMatrix&quot;</span>)
<span class="co">#&gt; [1] FALSE</span>

<span class="co"># </span><span class="al">NOTE</span><span class="co">: Transposing will &#39;degrade&#39; the result.</span>
<span class="kw">t</span>(da_Rle)
<span class="co">#&gt; &lt;2 x 105&gt; DelayedMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1]   [,2]   [,3]   [,4] ... [,102] [,103] [,104] [,105]</span>
<span class="co">#&gt; [1,]      1      2      2      3   .     14     14     14     14</span>
<span class="co">#&gt; [2,]     15     15     15     15   .     20     20     20     20</span>
<span class="kw">is</span>(<span class="kw">t</span>(da_Rle), <span class="st">&quot;RleMatrix&quot;</span>)
<span class="co">#&gt; [1] FALSE</span>

<span class="co"># </span><span class="al">NOTE</span><span class="co">: Even adding zero (conceptually a no-op) will &#39;degrade&#39; the result.</span>
da_Rle <span class="op">+</span><span class="st"> </span>0L
<span class="co">#&gt; &lt;105 x 2&gt; DelayedMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    1   15</span>
<span class="co">#&gt;   [2,]    2   15</span>
<span class="co">#&gt;   [3,]    2   15</span>
<span class="co">#&gt;   [4,]    3   15</span>
<span class="co">#&gt;   [5,]    3   15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   14   20</span>
<span class="co">#&gt; [102,]   14   20</span>
<span class="co">#&gt; [103,]   14   20</span>
<span class="co">#&gt; [104,]   14   20</span>
<span class="co">#&gt; [105,]   14   20</span></code></pre></div>
<p>There are some exceptions to this rule, when the DelayedArray framework can recognise/guarantee that the operation is a no-op that will leave the object in its current state:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># </span><span class="al">NOTE</span><span class="co">: Subsetting to select the entire object does not &#39;degrade&#39; the result.</span>
da_Rle[<span class="kw">seq_len</span>(<span class="kw">nrow</span>(da_Rle)), <span class="kw">seq_len</span>(<span class="kw">ncol</span>(da_Rle))]
<span class="co">#&gt; &lt;105 x 2&gt; RleMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    1   15</span>
<span class="co">#&gt;   [2,]    2   15</span>
<span class="co">#&gt;   [3,]    2   15</span>
<span class="co">#&gt;   [4,]    3   15</span>
<span class="co">#&gt;   [5,]    3   15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   14   20</span>
<span class="co">#&gt; [102,]   14   20</span>
<span class="co">#&gt; [103,]   14   20</span>
<span class="co">#&gt; [104,]   14   20</span>
<span class="co">#&gt; [105,]   14   20</span>
<span class="kw">is</span>(da_Rle[<span class="kw">seq_len</span>(<span class="kw">nrow</span>(da_Rle)), <span class="kw">seq_len</span>(<span class="kw">ncol</span>(da_Rle))], <span class="st">&quot;RleMatrix&quot;</span>)
<span class="co">#&gt; [1] TRUE</span>

<span class="co"># </span><span class="al">NOTE</span><span class="co">: Transposing and transposing back does not &#39;degrade&#39; the result.</span>
<span class="kw">t</span>(<span class="kw">t</span>(da_Rle))
<span class="co">#&gt; &lt;105 x 2&gt; RleMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    1   15</span>
<span class="co">#&gt;   [2,]    2   15</span>
<span class="co">#&gt;   [3,]    2   15</span>
<span class="co">#&gt;   [4,]    3   15</span>
<span class="co">#&gt;   [5,]    3   15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   14   20</span>
<span class="co">#&gt; [102,]   14   20</span>
<span class="co">#&gt; [103,]   14   20</span>
<span class="co">#&gt; [104,]   14   20</span>
<span class="co">#&gt; [105,]   14   20</span>
<span class="kw">is</span>(<span class="kw">t</span>(<span class="kw">t</span>(da_Rle)), <span class="st">&quot;RleMatrix&quot;</span>)
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p>A particularly important example of this ‘degrading’ to be aware of is when accessing the assays of a <em>SummarizedExperiment</em> via the <code>assay()</code> and <code>assays()</code> getters. Each of these getters has an argument <code>withDimnames</code> with a default value of <code>TRUE</code>; this copies the dimnames of the <em>SummarizedExperiment</em> to the returned assay. Consequently, this may ‘degrade’ the object(s) returned by <code>assay()</code> and <code>assays()</code> to <em>DelayedArray</em>. To avoid this, use <code>withDimnames = FALSE</code> in the call to <code>assay()</code> and <code>assays()</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(SummarizedExperiment)
<span class="co"># Construct a SummarizedExperiment with column names &#39;A&#39; and &#39;B&#39; and da_Rle as </span>
<span class="co"># the assay data.</span>
se &lt;-<span class="st"> </span><span class="kw">SummarizedExperiment</span>(da_Rle, <span class="dt">colData =</span> <span class="kw">DataFrame</span>(<span class="dt">row.names =</span> <span class="kw">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>)))
se
<span class="co">#&gt; class: SummarizedExperiment </span>
<span class="co">#&gt; dim: 105 2 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): &#39;&#39;</span>
<span class="co">#&gt; rownames: NULL</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(2): A B</span>
<span class="co">#&gt; colData names(0):</span>

<span class="co"># </span><span class="al">NOTE</span><span class="co">: dimnames are copied, so the result is &#39;degraded&#39;.</span>
<span class="kw">assay</span>(se)
<span class="co">#&gt; &lt;105 x 2&gt; DelayedMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;         A  B</span>
<span class="co">#&gt;   [1,]  1 15</span>
<span class="co">#&gt;   [2,]  2 15</span>
<span class="co">#&gt;   [3,]  2 15</span>
<span class="co">#&gt;   [4,]  3 15</span>
<span class="co">#&gt;   [5,]  3 15</span>
<span class="co">#&gt;    ...  .  .</span>
<span class="co">#&gt; [101,] 14 20</span>
<span class="co">#&gt; [102,] 14 20</span>
<span class="co">#&gt; [103,] 14 20</span>
<span class="co">#&gt; [104,] 14 20</span>
<span class="co">#&gt; [105,] 14 20</span>
<span class="kw">is</span>(<span class="kw">assay</span>(se), <span class="st">&quot;RleMatrix&quot;</span>)
<span class="co">#&gt; [1] FALSE</span>

<span class="co"># </span><span class="al">NOTE</span><span class="co">: dimnames are not copied, so the result is not &#39;degraded&#39;.</span>
<span class="kw">assay</span>(se, <span class="dt">withDimnames =</span> <span class="ot">FALSE</span>)
<span class="co">#&gt; &lt;105 x 2&gt; RleMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    1   15</span>
<span class="co">#&gt;   [2,]    2   15</span>
<span class="co">#&gt;   [3,]    2   15</span>
<span class="co">#&gt;   [4,]    3   15</span>
<span class="co">#&gt;   [5,]    3   15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   14   20</span>
<span class="co">#&gt; [102,]   14   20</span>
<span class="co">#&gt; [103,]   14   20</span>
<span class="co">#&gt; [104,]   14   20</span>
<span class="co">#&gt; [105,]   14   20</span>
<span class="kw">is</span>(<span class="kw">assay</span>(se, <span class="dt">withDimnames =</span> <span class="ot">FALSE</span>), <span class="st">&quot;RleMatrix&quot;</span>)
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p>As noted up front, the degradation isn’t in and of itself a problem. However, if you are passing an object to a function that expects an <em>RleMatrix</em>, for example, some care needs to be taken to ensure that the object isn’t accidentally ‘degraded’ along the way to a <em>DelayedArray</em>.</p>
<p>To summarise, the lessons here are:</p>
<ul>
<li>Modifying an instance of a <em>DelayedArray</em> subclass will almost always ‘degrade’ it to a <em>DelayedArray</em>.</li>
<li>It is very easy to accidentally trigger this ‘degradation’.</li>
<li>This degradation can complicate method dispatch.</li>
</ul>
</div>
<div id="out-of-memory-seeds" class="section level4">
<h4><span class="header-section-number">6.3.4.5</span> Out-of-memory seeds</h4>
<p>The DelayedArray framework really shines when working with out-of-memory seeds. It can give the user the “feel” of interacting with an ordinary R array but allows for the data to be stored on a local disk or even on a remote server, thus reducing (local) memory usage.</p>
<div id="local-on-disk-seeds" class="section level5">
<h5><span class="header-section-number">6.3.4.5.1</span> Local on-disk seeds</h5>
<p>The <strong>HDF5Array</strong> package defines the <em>HDF5Array</em> class, a <em>DelayedArray</em> subclass for data stored on disk in a HDF5 file. The <em>seed</em> of an <em>HDF5Array</em> is a <em>HDF5ArraySeed</em>. It is important to note that creating a <em>HDF5Array</em> does not read the data into memory! The data remain on disk until requested (we’ll see how to do this later on in the workshop).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(HDF5Array)

hdf5_file &lt;-<span class="st"> </span><span class="kw">file.path</span>(
    <span class="st">&quot;500_Effectively_Using_the_DelayedArray_Framework&quot;</span>, 
    <span class="st">&quot;hdf5_mat.h5&quot;</span>)
<span class="co"># </span><span class="al">NOTE</span><span class="co">: We can use rhdf5::h5ls() to take a look what is in the HDF5 file.</span>
<span class="co">#       This is very useful when working interactively!</span>
rhdf5<span class="op">::</span><span class="kw">h5ls</span>(hdf5_file)
<span class="co">#&gt;   group     name       otype  dclass  dim</span>
<span class="co">#&gt; 0     / hdf5_mat H5I_DATASET INTEGER  x 2</span>

<span class="co"># We can create the HDF5Array by first creating a HDF5ArraySeed and then </span>
<span class="co"># creating the HDF5Array.</span>
hdf5_seed &lt;-<span class="st"> </span><span class="kw">HDF5ArraySeed</span>(<span class="dt">filepath =</span> hdf5_file, <span class="dt">name =</span> <span class="st">&quot;hdf5_mat&quot;</span>)
da_hdf5 &lt;-<span class="st"> </span><span class="kw">DelayedArray</span>(<span class="dt">seed =</span> hdf5_seed)
da_hdf5
<span class="co">#&gt; &lt;105 x 2&gt; HDF5Matrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    1   15</span>
<span class="co">#&gt;   [2,]    2   15</span>
<span class="co">#&gt;   [3,]    2   15</span>
<span class="co">#&gt;   [4,]    3   15</span>
<span class="co">#&gt;   [5,]    3   15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   14   20</span>
<span class="co">#&gt; [102,]   14   20</span>
<span class="co">#&gt; [103,]   14   20</span>
<span class="co">#&gt; [104,]   14   20</span>
<span class="co">#&gt; [105,]   14   20</span>

<span class="co"># Alternatively, we can create this in one go using the HDF5Array() constructor.    </span>
da_hdf5 &lt;-<span class="st"> </span><span class="kw">HDF5Array</span>(<span class="dt">filepath =</span> hdf5_file, <span class="dt">name =</span> <span class="st">&quot;hdf5_mat&quot;</span>)
da_hdf5
<span class="co">#&gt; &lt;105 x 2&gt; HDF5Matrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    1   15</span>
<span class="co">#&gt;   [2,]    2   15</span>
<span class="co">#&gt;   [3,]    2   15</span>
<span class="co">#&gt;   [4,]    3   15</span>
<span class="co">#&gt;   [5,]    3   15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   14   20</span>
<span class="co">#&gt; [102,]   14   20</span>
<span class="co">#&gt; [103,]   14   20</span>
<span class="co">#&gt; [104,]   14   20</span>
<span class="co">#&gt; [105,]   14   20</span>
<span class="kw">str</span>(da_hdf5)
<span class="co">#&gt; Formal class &#39;HDF5Matrix&#39; [package &quot;HDF5Array&quot;] with 1 slot</span>
<span class="co">#&gt;   ..@ seed:Formal class &#39;HDF5ArraySeed&#39; [package &quot;HDF5Array&quot;] with 5 slots</span>
<span class="co">#&gt;   .. .. ..@ filepath : chr &quot;/home/mramos/Source/github/BiocWorkshops/500_Effectively_Using_the_DelayedArray_Framework/hdf5_mat.h5&quot;</span>
<span class="co">#&gt;   .. .. ..@ name     : chr &quot;hdf5_mat&quot;</span>
<span class="co">#&gt;   .. .. ..@ dim      : int [1:2] 105 2</span>
<span class="co">#&gt;   .. .. ..@ first_val: int 1</span>
<span class="co">#&gt;   .. .. ..@ chunkdim : int [1:2] 105 2</span></code></pre></div>
<p>Other on-disk seeds are possible such as <strong>fst</strong>, <strong>bigmemory</strong>, <strong>ff</strong>, or <strong>matter</strong>.</p>
</div>
<div id="remote-seeds" class="section level5">
<h5><span class="header-section-number">6.3.4.5.2</span> Remote seeds</h5>
<p>The <strong>rhdf5client</strong> packages defines the <em>H5S_Array</em> class, a <em>DelayedArray</em> subclass for data stored on a remote HDF Server. The <em>seed</em> of an <em>H5S_Array</em> is a <em>H5S_ArraySeed</em>. It is important to note that creating a <em>H5S_Array</em> does not read the data into memory! The data remain on the server until requested.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(rhdf5client)
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;rhdf5client&#39;</span>
<span class="co">#&gt; The following object is masked from &#39;package:tidygraph&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     groups</span>

<span class="co"># </span><span class="al">TODO</span><span class="co">: Ask Vince to host the file?</span></code></pre></div>
</div>
</div>
<div id="so-what-seed-should-i-use" class="section level4">
<h4><span class="header-section-number">6.3.4.6</span> So what seed should I use?</h4>
<p>Notably, <code>da_mat</code>, <code>da_Mat</code>, <code>da_tbl</code>, <code>da_df</code>, <code>da_Rle</code>, <code>da_hdf5</code>, and <code>da_h5s</code> all “look” and “feel” much the same. The <em>DelayedArray</em> is a very light wrapper around the <em>seed</em> that formalises this consistent “look” and “feel”.</p>
<p>If your data always fit in memory, and it’s still comfortable to work with it interactively, then you should probably stick with using ordinary <em>matrix</em> and <em>array</em> objects. If, however, your data sometimes don’t fit in memory or it becomes painful to work with them when they are in-memory, you should consider a disk-backed <em>DelayedArray</em>. If you need to share a large dataset with a number of uses, and most users only need slices of it, you may want to consider a remote server-backed <em>DelayedArray</em>.</p>
<p>Not suprisingly, working with in-memory seeds will typically be faster than disk-backed or remote seeds; you are trading off memory-usage and data access times to obtain faster performance.</p>
<p>Below are some further opinionated recommendations.</p>
<p>Although the <em>DelayedArray</em> is a light wrapper, it does introduce some overhead. For example, operations on a <em>DelayedArray</em> with an <em>array</em> or <em>Matrix</em> seed will be slower than operating on the <em>array</em> or <em>Matrix</em> directly. For some operations this is almost non-existant (e.g., seed-aware operations in the <strong>DelayedMatrixStats</strong> package), but for others this overhead may be noticeable. Conversely, a potential upside to wrapping a <em>Matrix</em> in a <em>DelayedArray</em> is that it can now leverage the block-processing strategy of the DelayedArray framework.</p>
<p>If you need a disk-backed <em>DelayedArray</em>, I would recommend using <em>HDF5Array</em> as the backend at this time. HDF5 is a well-established scientific data format and the <strong>HDF5Array</strong> package is developed by the core Bioconductor team. Furthermore, several bioinformatics tools natively export HDF5 files (e.g., <strong>kallisto</strong>, <strong>CellRanger</strong> from 10x Genomics). One potential downside of HDF5 is the recent split into “Community” and “Enterprise Support” editions. It’s not yet clear how this will affect the HDF5 library or its community. Other on-disk backends may not yet have well-established formats or libraries (e.g., <strong>TileDB</strong>). A topic of on-going research in the Bioconductor community is to compare the performance of HDF5 to other disk-backed data stores. Another topic is how to best choose the layout of the data on disk, often referred to as the “chunking” of the data. For example, if the data is stored in column-major order then it might be expected that row-wise data access suffers.</p>
<p>One final comment. If you are trying to support both in-memory and on-disk data, it may be worth considering using <em>DelayedArray</em> for everything rather than trying to support both <em>DelayedArray</em> and ordinary arrays. This is particularly true for internal, non-user facing code. Otherwise your software may need separate code paths for <em>array</em> and <em>DelayedArray</em> inputs, although the need for separate branches is reducing. For example, when I initially added support for the DelayedArray framework in <strong>bsseq</strong> (~2 years ago), I opted to remove support for ordinary arrays. However, more recently I added support for the DelayedArray framework in <strong>minfi</strong>, and there I opted to maintain support for ordinary arrays. I’ve found that as the DelayedArray framework matures and expands, it is increasingly common that the same code “just works” for both <em>array</em> and <em>DelayedArray</em> objects. Maintaining support for ordinary arrays can also be critical for popular Bioconductor packages, although this can be handled by having functions that accept an <em>array</em> as input simply wrapping them in a <em>DelayedArray</em> for internal computations.</p>
</div>
</div>
<div id="fundamental-concepts" class="section level3">
<h3><span class="header-section-number">6.3.5</span> Fundamental concepts</h3>
<p><strong>UP TO HERE</strong></p>
<ul>
<li>delayed ops</li>
<li>realization</li>
<li>block-processing
<ul>
<li>array grid</li>
<li>array viewport</li>
</ul></li>
</ul>
<div id="learning-goals-2" class="section level4">
<h4><span class="header-section-number">6.3.5.1</span> Learning goals</h4>
<ul>
<li>Become familiar with the fundamental concepts of delayed operations, block-processing, and realization.</li>
<li>Construct a <em>DelayedArray</em>:</li>
<li>From scratch by writing data to a <em>RealizationSink</em>.</li>
</ul>
</div>
<div id="delayed-operations" class="section level4">
<h4><span class="header-section-number">6.3.5.2</span> Delayed operations</h4>
<div id="the-delayedop-class" class="section level5">
<h5><span class="header-section-number">6.3.5.2.1</span> The <em>DelayedOp</em> class</h5>
<p>The final type of seed is a <em>DelayedOp</em>, used to ‘record’ delayed operations. <em>DelayedOp</em> objects are not intended to be manipulated directly by the user but are used inside <em>DelayedArray</em> objects. A ‘delayed operation’ is one that is not actually performed until the result is required. Here’s a simple example of a delayed operations: taking the negative of every element of a <em>HDF5Matrix</em>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">da_hdf5
<span class="co">#&gt; &lt;105 x 2&gt; HDF5Matrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]    1   15</span>
<span class="co">#&gt;   [2,]    2   15</span>
<span class="co">#&gt;   [3,]    2   15</span>
<span class="co">#&gt;   [4,]    3   15</span>
<span class="co">#&gt;   [5,]    3   15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]   14   20</span>
<span class="co">#&gt; [102,]   14   20</span>
<span class="co">#&gt; [103,]   14   20</span>
<span class="co">#&gt; [104,]   14   20</span>
<span class="co">#&gt; [105,]   14   20</span>
<span class="op">-</span>da_hdf5
<span class="co">#&gt; &lt;105 x 2&gt; DelayedMatrix object of type &quot;integer&quot;:</span>
<span class="co">#&gt;        [,1] [,2]</span>
<span class="co">#&gt;   [1,]   -1  -15</span>
<span class="co">#&gt;   [2,]   -2  -15</span>
<span class="co">#&gt;   [3,]   -2  -15</span>
<span class="co">#&gt;   [4,]   -3  -15</span>
<span class="co">#&gt;   [5,]   -3  -15</span>
<span class="co">#&gt;    ...    .    .</span>
<span class="co">#&gt; [101,]  -14  -20</span>
<span class="co">#&gt; [102,]  -14  -20</span>
<span class="co">#&gt; [103,]  -14  -20</span>
<span class="co">#&gt; [104,]  -14  -20</span>
<span class="co">#&gt; [105,]  -14  -20</span></code></pre></div>
<p>It may look like constructing <code>-da_hdf5</code> has taken the negative of every value. However, we can use the <code>showtree()</code> function to inspect the internal state of these objects and see this is not the case<a href="#fn17" class="footnoteRef" id="fnref17"><sup>17</sup></a>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">showtree</span>(da_hdf5)
<span class="co">#&gt; 105x2 integer: HDF5Matrix object</span>
<span class="co">#&gt; └─ 105x2 integer: [seed] HDF5ArraySeed object</span>

<span class="co"># </span><span class="al">NOTE</span><span class="co">: See the &quot;Unary iso op&quot; line, below.</span>
<span class="kw">showtree</span>(<span class="op">-</span>da_hdf5)
<span class="co">#&gt; 105x2 integer: DelayedMatrix object</span>
<span class="co">#&gt; └─ 105x2 integer: Unary iso op</span>
<span class="co">#&gt;    └─ 105x2 integer: [seed] HDF5ArraySeed object</span></code></pre></div>
<p>Rather than modifying the data stored in the HDF5 file, which can be costly for large datasets, we’ve recorded the ‘idea’ of this operation as a <em>DelayedOp</em>.</p>
<p>In a <em>DelayedArray</em> object the delayed operations are stored as a tree of <em>DelayedOp</em> objects. Each node in the tree is represented by a <em>DelayedOp</em> object, of which there are 6 concrete subclasses:</p>
<table>
<thead>
<tr class="header">
<th>Node type</th>
<th>Out-degree</th>
<th>Operation</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><em>DelayedSubset</em></td>
<td>1</td>
<td>Multi-dimensional single bracket subsetting</td>
</tr>
<tr class="even">
<td><em>DelayedAperm</em></td>
<td>1</td>
<td>Extended <code>aperm()</code> (can drop dimensions)</td>
</tr>
<tr class="odd">
<td><em>DelayedUnaryIsoOp</em></td>
<td>1</td>
<td>Unary op that preserves the geometry (e.g., <code>-</code>, <code>log()</code>)</td>
</tr>
<tr class="even">
<td><em>DelayedDimnames</em></td>
<td>1</td>
<td>Set dimnames</td>
</tr>
<tr class="odd">
<td><em>DelayedNaryIsoOp</em></td>
<td>N</td>
<td>N-ary op that preserves the geometry</td>
</tr>
<tr class="even">
<td><em>DelayedAbind</em></td>
<td>N</td>
<td><code>abind()</code></td>
</tr>
</tbody>
</table>
<p>We’ll discuss delayed operations more, shortly.</p>
</div>
</div>
</div>
<div id="whats-out-there-already" class="section level3">
<h3><span class="header-section-number">6.3.6</span> What’s out there already?</h3>
<div id="learning-goal-1" class="section level4">
<h4><span class="header-section-number">6.3.6.1</span> Learning goal</h4>
<ul>
<li>Learn of existing functions and packages for constructing and computing on DelayedArray objects, avoiding the need to re-invent the wheel.</li>
</ul>
</div>
</div>
</div>
<div id="incoporating-delayedarray-into-a-package" class="section level2">
<h2><span class="header-section-number">6.4</span> Incoporating DelayedArray into a package</h2>
<div id="writing-algorithms-to-process-delayedarray-instances" class="section level3">
<h3><span class="header-section-number">6.4.1</span> Writing algorithms to process <em>DelayedArray</em> instances</h3>
<div id="learning-goals-3" class="section level4">
<h4><span class="header-section-number">6.4.1.1</span> Learning goals</h4>
<ul>
<li>Learn common design patterns for writing performant code that operates on a DelayedArray.</li>
<li>Evaluate whether an existing function that operates on an ordinary array can be readily adapted to work on a DelayedArray.</li>
<li>Reason about potential bottlenecks in algorithms operating on DelayedArray objects.</li>
</ul>
</div>
<div id="learning-objectives-3" class="section level4">
<h4><span class="header-section-number">6.4.1.2</span> Learning objectives</h4>
<ul>
<li>Take a function that operates on rows or columns of a matrix and apply it to a DelayedMatrix.</li>
<li>Use block-processing on a <em>DelayedArray</em> to compute:</li>
<li>A univariate (scalar) summary statistic (e.g., <code>max()</code>).</li>
<li>A multivariate (vector) summary statistic (e.g., <code>colSum()</code> or <code>rowMean()</code>).</li>
<li>A multivariate (array-like) summary statistic (e.g., <code>rowRanks()</code>).</li>
<li>Design an algorithm that imports data into a DelayedArray.</li>
</ul>
</div>
</div>
</div>
<div id="questions-and-discussion" class="section level2">
<h2><span class="header-section-number">6.5</span> Questions and discussion</h2>
<p>This section will be updated to address questions and to summarise the discussion from the presentation of this workshop at BioC2018.</p>
</div>
<div id="todos" class="section level2">
<h2><span class="header-section-number">6.6</span> TODOs</h2>
<ul>
<li>Use <strong>BiocStyle</strong>?</li>
<li>Show packages depend on one another, with HDF5Array as the root (i.e. explain the HDF5 stack)</li>
<li>Use <code>suppressPackageStartupMessages()</code> or equivalent.</li>
<li>Note that we’ll be focusing on numerical array-like data, i.e. no real discussion of <strong>GDSArray</strong>.</li>
<li>Remove <strong>memuse</strong> dependency</li>
</ul>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="9">
<li id="fn9"><p>Department of Biostatistics, Johns Hopkins University<a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fnref9">↩</a></p></li>
<li id="fn10"><p>Higher-dimensional arrays may be appropriate for some types of assays.<a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fnref10">↩</a></p></li>
<li id="fn11"><p>In R, a <em>matrix</em> is just a 2-dimensional <em>array</em><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fnref11">↩</a></p></li>
<li id="fn12"><p>These data are available in the <strong>TENxBrainData</strong> Bioconductor package<a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fnref12">↩</a></p></li>
<li id="fn13"><p>This strategy can also be effectively applied to sparse and repetitive data by using on-disk compression of the data.<a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fnref13">↩</a></p></li>
<li id="fn14"><p>The <strong>DelayedArray</strong> package, like all core Bioconductor packages, uses the <a href="http://adv-r.had.co.nz/OO-essentials.html#s4">S4 object oriented programming system.</a><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fnref14">↩</a></p></li>
<li id="fn15"><p>For further details, see the vignette in the <strong>DelayedArray</strong> package (available via <code>vignette(&quot;02-Implementing_a_backend&quot;, &quot;DelayedArray&quot;)</code>)<a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fnref15">↩</a></p></li>
<li id="fn16"><p>In fact, the <em>RleArraySeed</em> class is a virtual class, with concrete subclasses <em>SolidRleArraySeed</em> and <em>ChunkedRleArraySeed</em><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fnref16">↩</a></p></li>
<li id="fn17"><p>You may like to use the <code>str()</code> function for more detailed and verbose output<a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#fnref17">↩</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="cytoscape-automation-in-r-using-rcy3.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
