<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7.13 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html">
<link rel="next" href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; position: absolute; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; }
pre.numberSource a.sourceLine:empty
  { position: absolute; }
pre.numberSource a.sourceLine::before
  { content: attr(data-line-number);
    position: absolute; left: -5em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-everyone"><i class="fa fa-check"></i><b>1.1</b> For Everyone</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#for-workshop-authors"><i class="fa fa-check"></i><b>1.2</b> For Workshop Authors</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#deadlines-for-bioc2018"><i class="fa fa-check"></i><b>1.3</b> Deadlines for Bioc2018</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><i class="fa fa-check"></i><b>2</b> Variant Functional Annotation using StatePaintR, FunciVar &amp; MotifBreakR</a><ul>
<li class="chapter" data-level="2.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#authors"><i class="fa fa-check"></i><b>2.1</b> Authors:</a></li>
<li class="chapter" data-level="2.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-description"><i class="fa fa-check"></i><b>2.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="2.2.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-participation"><i class="fa fa-check"></i><b>2.2.1</b> Workshop Participation</a></li>
<li class="chapter" data-level="2.2.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.2.2</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.2.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#time-outline"><i class="fa fa-check"></i><b>2.2.3</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="2.3.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#key-concepts"><i class="fa fa-check"></i><b>2.3.1</b> Key Concepts</a></li>
<li class="chapter" data-level="2.3.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#learning-objectives"><i class="fa fa-check"></i><b>2.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html"><i class="fa fa-check"></i><b>3</b> Working with Genomic Data in R with the DECIPHER package</a><ul>
<li class="chapter" data-level="3.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#authors-1"><i class="fa fa-check"></i><b>3.1</b> Authors:</a></li>
<li class="chapter" data-level="3.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-description-1"><i class="fa fa-check"></i><b>3.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="3.2.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#pre-requisites"><i class="fa fa-check"></i><b>3.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="3.2.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-participation-1"><i class="fa fa-check"></i><b>3.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="3.2.3" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="3.2.4" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#time-outline-1"><i class="fa fa-check"></i><b>3.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="3.3.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-goals"><i class="fa fa-check"></i><b>3.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="3.3.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-objectives-1"><i class="fa fa-check"></i><b>3.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>4</b> Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>4.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-description-2"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-1"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-participation-2"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-2"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="4.3.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#learning-goals-1"><i class="fa fa-check"></i><b>4.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="4.3.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#learning-objectives-2"><i class="fa fa-check"></i><b>4.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>5</b> Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="5.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>5.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-3"><i class="fa fa-check"></i><b>5.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="5.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-2"><i class="fa fa-check"></i><b>5.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="5.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-3"><i class="fa fa-check"></i><b>5.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>5.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="5.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-3"><i class="fa fa-check"></i><b>5.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>5.3</b> Goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>6</b> Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="6.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>6.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="6.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-description-4"><i class="fa fa-check"></i><b>6.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="6.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-participation-4"><i class="fa fa-check"></i><b>6.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="6.2.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>6.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.2.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-4"><i class="fa fa-check"></i><b>6.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>6.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="6.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#learning-goals-2"><i class="fa fa-check"></i><b>6.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="6.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#learning-objectives-3"><i class="fa fa-check"></i><b>6.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>7</b> Cytoscape Automation in R using Rcy3</a><ul>
<li class="chapter" data-level="7.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructors-names-and-contact-information-3"><i class="fa fa-check"></i><b>7.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="7.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description-5"><i class="fa fa-check"></i><b>7.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="7.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation-5"><i class="fa fa-check"></i><b>7.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="7.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>7.2.3</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="7.2.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-5"><i class="fa fa-check"></i><b>7.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>7.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="7.3.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#learning-goals-3"><i class="fa fa-check"></i><b>7.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="7.3.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#learning-objectives-4"><i class="fa fa-check"></i><b>7.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>8</b> RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="8.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#author"><i class="fa fa-check"></i><b>8.1</b> Author:</a></li>
<li class="chapter" data-level="8.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-6"><i class="fa fa-check"></i><b>8.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="8.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-6"><i class="fa fa-check"></i><b>8.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="8.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>8.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-6"><i class="fa fa-check"></i><b>8.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>8.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="8.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-4"><i class="fa fa-check"></i><b>8.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="8.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-5"><i class="fa fa-check"></i><b>8.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>9</b> Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="9.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information-4"><i class="fa fa-check"></i><b>9.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-7"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-7"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-7"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>9.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="9.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-5"><i class="fa fa-check"></i><b>9.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="9.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-6"><i class="fa fa-check"></i><b>9.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>10</b> Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="10.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructors-names-and-contact-information-5"><i class="fa fa-check"></i><b>10.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-8"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-8"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>10.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-8"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-6"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-7"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>11</b> RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="11.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#instructors-names-and-contact-information-6"><i class="fa fa-check"></i><b>11.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="11.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-description-9"><i class="fa fa-check"></i><b>11.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="11.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-8"><i class="fa fa-check"></i><b>11.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="11.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-participation-9"><i class="fa fa-check"></i><b>11.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>11.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="11.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-9"><i class="fa fa-check"></i><b>11.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.3</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>12</b> Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="12.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>12.1</b> Instructors</a></li>
<li class="chapter" data-level="12.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description-10"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>12.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation-10"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>12.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-7"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-8"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>13</b> Workflow for Multi-omics Analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>13.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-11"><i class="fa fa-check"></i><b>13.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="13.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-11"><i class="fa fa-check"></i><b>13.2.2</b> workshop Participation</a></li>
<li class="chapter" data-level="13.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>13.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="13.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-10"><i class="fa fa-check"></i><b>13.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-8"><i class="fa fa-check"></i><b>13.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-9"><i class="fa fa-check"></i><b>13.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>14</b> Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="14.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>14.1</b> Instructors:</a></li>
<li class="chapter" data-level="14.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-12"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-10"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-12"><i class="fa fa-check"></i><b>14.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-11"><i class="fa fa-check"></i><b>14.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="14.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-11"><i class="fa fa-check"></i><b>14.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-9"><i class="fa fa-check"></i><b>14.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-10"><i class="fa fa-check"></i><b>14.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>15</b> Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="15.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#authors-2"><i class="fa fa-check"></i><b>15.1</b> Authors:</a></li>
<li class="chapter" data-level="15.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-13"><i class="fa fa-check"></i><b>15.2</b> Workshop Description:</a></li>
<li class="chapter" data-level="15.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-11"><i class="fa fa-check"></i><b>15.3</b> Pre-requisites:</a></li>
<li class="chapter" data-level="15.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation"><i class="fa fa-check"></i><b>15.4</b> Participation:</a></li>
<li class="chapter" data-level="15.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>15.5</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="15.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.6</b> Workshop goals and objectives:</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>16</b> Public Data Resources and Bioconductor</a><ul>
<li class="chapter" data-level="16.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructors-2"><i class="fa fa-check"></i><b>16.1</b> Instructors</a></li>
<li class="chapter" data-level="16.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-14"><i class="fa fa-check"></i><b>16.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="16.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-12"><i class="fa fa-check"></i><b>16.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="16.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-13"><i class="fa fa-check"></i><b>16.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="16.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>16.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="16.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-12"><i class="fa fa-check"></i><b>16.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-13"><i class="fa fa-check"></i><b>16.3</b> workshop goals and objectives</a><ul>
<li class="chapter" data-level="16.3.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-10"><i class="fa fa-check"></i><b>16.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="16.3.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-11"><i class="fa fa-check"></i><b>16.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="functional-enrichment-analysis-of-high-throughput-omics-data" class="section level1">
<h1><span class="header-section-number">5</span> Functional enrichment analysis of high-throughput omics data</h1>
<div id="instructors-names-and-contact-information-1" class="section level2">
<h2><span class="header-section-number">5.1</span> Instructor(s) name(s) and contact information</h2>
<ul>
<li>Ludwig Geistlinger (<a href="mailto:Ludwig.Geistlinger@sph.cuny.edu">Ludwig.Geistlinger@sph.cuny.edu</a>)</li>
<li>Levi Waldron</li>
</ul>
<p>CUNY School of Public Health
55 W 125th St, New York, NY 10027</p>
</div>
<div id="workshop-description-3" class="section level2">
<h2><span class="header-section-number">5.2</span> Workshop Description</h2>
<p>This workshop gives an in-depth overview of existing methods for enrichment
analysis of gene expression data with regard to functional gene sets, pathways,
and networks.
The workshop will help participants understand the distinctions between
assumptions and hypotheses of existing methods as well as the differences in
objectives and interpretation of results.
It will provide code and hands-on practice of all necessary steps for differential
expression analysis, gene set- and network-based enrichment analysis, and
identification of enriched genomic regions and regulatory elements, along with
visualization and exploration of results.
A previous well-attended (&gt;50 participants) version of this workshop was presented at Bioc2017
(<a href="https://www.bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/OmicsData/doc/enrichOmics.html" class="uri">https://www.bioconductor.org/help/course-materials/2017/BioC2017/Day1/Workshops/OmicsData/doc/enrichOmics.html</a>).</p>
<div id="pre-requisites-2" class="section level3">
<h3><span class="header-section-number">5.2.1</span> Pre-requisites</h3>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Familiarity with the SummarizedExperiment class</li>
<li><p>Familiarity with the GenomicRanges class</p></li>
<li>Familiarity with high-throughput gene expression data as obtained with
microarrays and RNA-seq</li>
<li><p>Familiarity with the concept of differential expression analysis
(with e.g. limma, edgeR, DESeq2)</p></li>
</ul>
</div>
<div id="workshop-participation-3" class="section level3">
<h3><span class="header-section-number">5.2.2</span> Workshop Participation</h3>
<p>Execution of example code and hands-on practice</p>
</div>
<div id="r-bioconductor-packages-used-3" class="section level3">
<h3><span class="header-section-number">5.2.3</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>EnrichmentBrowser</li>
<li>regioneR</li>
</ul>
</div>
<div id="time-outline-3" class="section level3">
<h3><span class="header-section-number">5.2.4</span> Time outline</h3>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Background</td>
<td>30m</td>
</tr>
<tr class="even">
<td>Differential expression analysis</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Gene set analysis</td>
<td>30m</td>
</tr>
<tr class="even">
<td>Gene network analysis</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Genomic region analysis</td>
<td>15m</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="goals-and-objectives" class="section level2">
<h2><span class="header-section-number">5.3</span> Goals and objectives</h2>
<p>Theory
* Gene sets, pathways &amp; regulatory networks
* Resources
* Gene set analysis vs. gene set enrichment analysis
* Underlying null: competitive vs. self-contained
* Generations: ora, fcs &amp; topology-based</p>
<p>Practice:
* Data types: microarray vs. RNA-seq
* Differential expression analysis
* Defining gene sets according to GO and KEGG
* GO/KEGG overrepresentation analysis
* Functional class scoring &amp; permutation testing
* Network-based enrichment analysis
* Genomic region enrichment analysis</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
