<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html">
<link rel="next" href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-122837239-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-122837239-1');
</script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-workshops"><i class="fa fa-check"></i><b>1.1</b> The Workshops</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#how-to-use-these-workshops"><i class="fa fa-check"></i><b>1.2</b> How to use these workshops</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#install-on-your-own-computer"><i class="fa fa-check"></i><b>1.2.1</b> Install on your own computer</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#use-the-exact-same-ami-as-the-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Use the exact same AMI as the workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#make-your-own-docker-image-and-ami"><i class="fa fa-check"></i><b>1.2.3</b> Make your own Docker image and AMI</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html"><i class="fa fa-check"></i><b>2</b> 100: <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#description"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="2.1.7" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-goals"><i class="fa fa-check"></i><b>2.1.7</b> Learning goals</a></li>
<li class="chapter" data-level="2.1.8" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-objectives"><i class="fa fa-check"></i><b>2.1.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-r"><i class="fa fa-check"></i><b>2.2</b> Introduction to <em>R</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#rstudio-orientation"><i class="fa fa-check"></i><b>2.2.1</b> <em>RStudio</em> orientation</a></li>
<li class="chapter" data-level="2.2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#very-basics"><i class="fa fa-check"></i><b>2.2.2</b> Very basics</a></li>
<li class="chapter" data-level="2.2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#data-input-and-manipulation"><i class="fa fa-check"></i><b>2.2.3</b> Data input and manipulation</a></li>
<li class="chapter" data-level="2.2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#writing-r-scripts"><i class="fa fa-check"></i><b>2.2.4</b> Writing <em>R</em> scripts</a></li>
<li class="chapter" data-level="2.2.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#using-packages"><i class="fa fa-check"></i><b>2.2.5</b> Using packages</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.3</b> Introduction to <em>Bioconductor</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#discovering-installing-and-learning-how-to-use-bioconductor-packages"><i class="fa fa-check"></i><b>2.3.1</b> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-genomic-ranges"><i class="fa fa-check"></i><b>2.3.2</b> Working with Genomic Ranges</a></li>
<li class="chapter" data-level="2.3.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-summarized-experimental-data"><i class="fa fa-check"></i><b>2.3.3</b> Working with summarized experimental data</a></li>
<li class="chapter" data-level="2.3.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#down-stream-analysis"><i class="fa fa-check"></i><b>2.3.4</b> Down-stream analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>3</b> 101: Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>3.1</b> Instructors</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description"><i class="fa fa-check"></i><b>3.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="3.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>3.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="3.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation"><i class="fa fa-check"></i><b>3.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="3.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="3.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-1"><i class="fa fa-check"></i><b>3.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="3.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-1"><i class="fa fa-check"></i><b>3.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>3.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="3.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>3.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="3.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>3.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="3.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>3.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="3.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>3.4.4</b> Data containers</a></li>
<li class="chapter" data-level="3.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>3.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="3.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>3.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>3.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>3.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="3.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>3.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="3.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>3.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="3.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>3.4.11</b> Simple example</a></li>
<li class="chapter" data-level="3.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>3.4.12</b> Questions!</a></li>
<li class="chapter" data-level="3.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>3.4.13</b> More questions!</a></li>
<li class="chapter" data-level="3.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>3.4.14</b> Another example</a></li>
<li class="chapter" data-level="3.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>3.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="3.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>3.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="3.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>3.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="3.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>3.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="3.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>3.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="3.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>3.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="3.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>3.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="3.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>3.4.22</b> GRanges</a></li>
<li class="chapter" data-level="3.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>3.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="3.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>3.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="3.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>3.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="3.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>3.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="3.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>3.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="3.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>3.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="3.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>3.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>3.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>3.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="3.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>3.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="3.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr"><i class="fa fa-check"></i><b>3.4.33</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-1"><i class="fa fa-check"></i><b>3.4.34</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-2"><i class="fa fa-check"></i><b>3.4.35</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>3.4.36</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="3.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>3.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>3.4.38</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>3.4.39</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>3.4.40</b> BSgenome exercises</a></li>
<li class="chapter" data-level="3.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub"><i class="fa fa-check"></i><b>3.4.41</b> AnnotationHub</a></li>
<li class="chapter" data-level="3.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub"><i class="fa fa-check"></i><b>3.4.42</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="3.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>3.4.43</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="3.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>3.4.44</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="3.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>3.4.45</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="3.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>3.4.46</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="3.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>3.4.47</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>3.4.48</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>3.4.49</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="3.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>3.4.50</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="3.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>3.4.51</b> biomaRt</a></li>
<li class="chapter" data-level="3.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>3.4.52</b> biomaRt data sets</a></li>
<li class="chapter" data-level="3.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>3.4.53</b> biomaRt queries</a></li>
<li class="chapter" data-level="3.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>3.4.54</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="3.4.55" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>3.4.55</b> biomaRt query</a></li>
<li class="chapter" data-level="3.4.56" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>3.4.56</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>4</b> 102: Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>4.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-1"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-1"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-1"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-1"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="4.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-2"><i class="fa fa-check"></i><b>4.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="4.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-2"><i class="fa fa-check"></i><b>4.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#introduction-1"><i class="fa fa-check"></i><b>4.4</b> Introduction</a><ul>
<li class="chapter" data-level="4.4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#what-is-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.1</b> What is the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#why-use-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.2</b> Why use the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>4.4.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#setup"><i class="fa fa-check"></i><b>4.5</b> Setup</a></li>
<li class="chapter" data-level="4.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>4.6</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="4.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#constructing-a-granges-object-from-data.frame"><i class="fa fa-check"></i><b>4.7</b> Constructing a <em>GRanges</em> object from data.frame</a></li>
<li class="chapter" data-level="4.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#loading-a-granges-object-from-a-standard-file-format"><i class="fa fa-check"></i><b>4.8</b> Loading a <em>GRanges</em> object from a standard file format</a></li>
<li class="chapter" data-level="4.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-manipulation-of-granges-objects"><i class="fa fa-check"></i><b>4.9</b> Basic manipulation of <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>4.10</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>4.11</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#aggregating-granges-objects"><i class="fa fa-check"></i><b>4.12</b> Aggregating <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.13</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.14</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.15" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#finding-overlaps-between-granges-objects"><i class="fa fa-check"></i><b>4.15</b> Finding overlaps between <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.16" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises"><i class="fa fa-check"></i><b>4.16</b> Exercises</a></li>
<li class="chapter" data-level="4.17" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>4.17</b> Example: exploring BigWig files from AnnotationHub</a><ul>
<li class="chapter" data-level="4.17.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#extracting-data-from-annotationhub"><i class="fa fa-check"></i><b>4.17.1</b> Extracting data from AnnotationHub</a></li>
<li class="chapter" data-level="4.17.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#reading-bigwig-files"><i class="fa fa-check"></i><b>4.17.2</b> Reading BigWig files</a></li>
<li class="chapter" data-level="4.17.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-1"><i class="fa fa-check"></i><b>4.17.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.18" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>4.18</b> Worked example: coverage analysis of BAM files</a><ul>
<li class="chapter" data-level="4.18.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-2"><i class="fa fa-check"></i><b>4.18.1</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.19" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#conclusions"><i class="fa fa-check"></i><b>4.19</b> Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>5</b> 103: Public data resources and Bioconductor</a><ul>
<li class="chapter" data-level="5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>5.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>5.2</b> Syllabus</a><ul>
<li class="chapter" data-level="5.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-2"><i class="fa fa-check"></i><b>5.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="5.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-2"><i class="fa fa-check"></i><b>5.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="5.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-2"><i class="fa fa-check"></i><b>5.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>5.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="5.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-2"><i class="fa fa-check"></i><b>5.2.5</b> Time outline</a></li>
<li class="chapter" data-level="5.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>5.2.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="5.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-3"><i class="fa fa-check"></i><b>5.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="5.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-3"><i class="fa fa-check"></i><b>5.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-1"><i class="fa fa-check"></i><b>5.3</b> Overview</a></li>
<li class="chapter" data-level="5.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>5.4</b> GEOquery</a><ul>
<li class="chapter" data-level="5.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>5.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="5.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>5.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="5.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>5.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="5.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>5.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>5.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="5.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>5.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="5.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>5.5.2</b> Basic design</a></li>
<li class="chapter" data-level="5.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>5.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>5.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="5.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>5.6.1</b> Authentication</a></li>
<li class="chapter" data-level="5.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>5.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>5.7</b> Sequence Read Archive</a><ul>
<li class="chapter" data-level="5.7.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage-1"><i class="fa fa-check"></i><b>5.7.1</b> Usage</a></li>
<li class="chapter" data-level="5.7.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#query-syntax"><i class="fa fa-check"></i><b>5.7.2</b> Query syntax</a></li>
<li class="chapter" data-level="5.7.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#using-the-raw-api-without-rbioconductor"><i class="fa fa-check"></i><b>5.7.3</b> Using the raw API without R/Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>5.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="5.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>5.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="5.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>5.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="5.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>5.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="5.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>5.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="5.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>5.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="5.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>5.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="5.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>5.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>5.12</b> Pharmacogenomics</a><ul>
<li class="chapter" data-level="5.12.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started"><i class="fa fa-check"></i><b>5.12.1</b> Getting started</a></li>
<li class="chapter" data-level="5.12.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-pharmacogx-datasets-pharmacosets"><i class="fa fa-check"></i><b>5.12.2</b> Overview of PharmacoGx datasets (PharmacoSets)</a></li>
<li class="chapter" data-level="5.12.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#exploring-drug-sensitivity-datasets"><i class="fa fa-check"></i><b>5.12.3</b> Exploring drug sensitivity datasets</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>5.13</b> Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>6</b> 200: RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>6.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>6.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="6.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>6.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="6.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>6.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-3"><i class="fa fa-check"></i><b>6.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>6.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="6.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-4"><i class="fa fa-check"></i><b>6.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="6.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-4"><i class="fa fa-check"></i><b>6.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-2"><i class="fa fa-check"></i><b>6.4</b> Introduction</a></li>
<li class="chapter" data-level="6.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>6.5</b> Data packaging</a><ul>
<li class="chapter" data-level="6.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>6.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="6.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>6.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="6.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>6.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>6.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="6.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>6.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="6.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>6.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="6.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>6.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="6.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>6.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis"><i class="fa fa-check"></i><b>6.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="6.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>6.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="6.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>6.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="6.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>6.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="6.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>6.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="6.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>6.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="6.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>6.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>7</b> 201: RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview-2"><i class="fa fa-check"></i><b>7.1</b> Overview</a><ul>
<li class="chapter" data-level="7.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-1"><i class="fa fa-check"></i><b>7.1.1</b> Description</a></li>
<li class="chapter" data-level="7.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation-1"><i class="fa fa-check"></i><b>7.1.3</b> Participation</a></li>
<li class="chapter" data-level="7.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>7.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-4"><i class="fa fa-check"></i><b>7.1.5</b> Time outline</a></li>
<li class="chapter" data-level="7.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>7.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>7.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="7.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>7.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="7.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>7.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="7.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>7.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="7.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>7.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>7.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="7.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>7.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="7.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>7.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="7.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>7.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>7.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="7.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>7.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="7.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>7.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="7.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>7.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>7.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="7.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>7.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="7.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>7.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub-1"><i class="fa fa-check"></i><b>7.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="7.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>7.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="7.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>7.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>7.7</b> Building reports</a><ul>
<li class="chapter" data-level="7.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>7.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="7.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>7.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>7.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="7.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>7.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>7.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>7.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="7.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>7.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="7.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>7.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="7.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>7.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>8</b> 202: Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="8.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-3"><i class="fa fa-check"></i><b>8.1</b> Overview</a><ul>
<li class="chapter" data-level="8.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-2"><i class="fa fa-check"></i><b>8.1.1</b> Description</a></li>
<li class="chapter" data-level="8.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-2"><i class="fa fa-check"></i><b>8.1.3</b> Participation</a></li>
<li class="chapter" data-level="8.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>8.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-5"><i class="fa fa-check"></i><b>8.1.5</b> Time outline</a></li>
<li class="chapter" data-level="8.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>8.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started-1"><i class="fa fa-check"></i><b>8.2</b> Getting started</a><ul>
<li class="chapter" data-level="8.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>8.2.1</b> The data</a></li>
<li class="chapter" data-level="8.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>8.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>8.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="8.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>8.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="8.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>8.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>8.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="8.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>8.5.1</b> Normalization</a></li>
<li class="chapter" data-level="8.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>8.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>8.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="8.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>8.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="8.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>8.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>9</b> 210: Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="9.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>9.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-4"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-4"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-6"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>9.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="9.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>9.4</b> Where does it all come from?</a></li>
<li class="chapter" data-level="9.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.5</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="9.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>9.6</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="9.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>9.7</b> Data types</a></li>
<li class="chapter" data-level="9.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>9.8</b> Differential expression analysis</a></li>
<li class="chapter" data-level="9.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>9.9</b> Gene sets</a></li>
<li class="chapter" data-level="9.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>9.10</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="9.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>9.11</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="9.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.12</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="9.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>9.13</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>10</b> 220: Workflow for multi-omics analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="10.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information-2"><i class="fa fa-check"></i><b>10.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-5"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-5"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>10.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-7"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-5"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-5"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>10.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="10.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="10.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>10.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="10.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>10.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>10.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="10.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>10.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="10.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>10.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="10.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>10.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>10.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="10.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>10.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="10.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>10.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="10.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>10.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="10.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>10.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="10.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>10.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>10.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="10.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>10.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="10.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>10.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="10.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>10.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>10.9</b> Complete cases</a></li>
<li class="chapter" data-level="10.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>10.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="10.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>10.11</b> Extraction</a><ul>
<li class="chapter" data-level="10.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>10.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="10.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>10.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="10.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>10.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="10.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>10.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="10.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>10.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="10.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>10.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="10.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>10.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="10.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>10.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="10.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>10.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>10.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="10.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>10.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="10.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>10.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="10.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>10.16.1</b> “Simplification” of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="10.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>10.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="10.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>10.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="10.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>10.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="10.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>10.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="10.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>10.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="10.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>10.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="10.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>10.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="10.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>10.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="10.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>10.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="10.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>10.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="10.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>10.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>11</b> 230: Cytoscape automation in R using Rcy3</a><ul>
<li class="chapter" data-level="11.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-4"><i class="fa fa-check"></i><b>11.1</b> Overview</a><ul>
<li class="chapter" data-level="11.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructor-names-and-contact-information-3"><i class="fa fa-check"></i><b>11.1.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="11.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description-6"><i class="fa fa-check"></i><b>11.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="11.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>11.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="11.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>11.1.4</b> Background:</a></li>
<li class="chapter" data-level="11.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation-6"><i class="fa fa-check"></i><b>11.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>11.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="11.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-8"><i class="fa fa-check"></i><b>11.1.7</b> Time outline</a></li>
<li class="chapter" data-level="11.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>11.2</b> Background</a><ul>
<li class="chapter" data-level="11.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>11.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="11.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>11.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="11.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>11.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>11.3</b> Translating biological data into Cytoscape using RCy3</a></li>
<li class="chapter" data-level="11.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>11.4</b> Set up</a><ul>
<li class="chapter" data-level="11.4.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started-2"><i class="fa fa-check"></i><b>11.4.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>11.5</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="11.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>11.6</b> Example Data Set</a></li>
<li class="chapter" data-level="11.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>11.7</b> Finding Network Data</a></li>
<li class="chapter" data-level="11.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>11.8</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="11.9" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>11.9</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="11.10" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>11.10</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>12</b> 240: Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="12.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructor-names-and-contact-information-4"><i class="fa fa-check"></i><b>12.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="12.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-7"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-8"><i class="fa fa-check"></i><b>12.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-7"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>12.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="12.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-9"><i class="fa fa-check"></i><b>12.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-6"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-6"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop"><i class="fa fa-check"></i><b>12.4</b> Workshop</a></li>
<li class="chapter" data-level="12.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-3"><i class="fa fa-check"></i><b>12.5</b> Introduction</a><ul>
<li class="chapter" data-level="12.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>12.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="12.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>12.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="12.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for-1"><i class="fa fa-check"></i><b>12.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup-1"><i class="fa fa-check"></i><b>12.6</b> Setup</a></li>
<li class="chapter" data-level="12.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>12.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="12.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>12.8</b> The Grammar</a><ul>
<li class="chapter" data-level="12.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>12.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="12.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>12.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="12.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises-3"><i class="fa fa-check"></i><b>12.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>12.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="12.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>12.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="12.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files-1"><i class="fa fa-check"></i><b>12.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="12.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>12.10</b> What’s next?</a></li>
<li class="chapter" data-level="12.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>12.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html"><i class="fa fa-check"></i><b>13</b> 250: Working with genomic data in R with the DECIPHER package</a><ul>
<li class="chapter" data-level="13.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#overview-5"><i class="fa fa-check"></i><b>13.1</b> Overview</a><ul>
<li class="chapter" data-level="13.1.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-description-8"><i class="fa fa-check"></i><b>13.1.1</b> Workshop Description</a></li>
<li class="chapter" data-level="13.1.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.1.3" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-participation-8"><i class="fa fa-check"></i><b>13.1.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.1.4" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>13.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="13.1.5" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#time-outline-10"><i class="fa fa-check"></i><b>13.1.5</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.2</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.2.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-goals-7"><i class="fa fa-check"></i><b>13.2.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.2.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-objectives-7"><i class="fa fa-check"></i><b>13.2.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>14</b> 260: Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="14.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>14.1</b> Instructors:</a></li>
<li class="chapter" data-level="14.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-9"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-10"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-9"><i class="fa fa-check"></i><b>14.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>14.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="14.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-11"><i class="fa fa-check"></i><b>14.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-8"><i class="fa fa-check"></i><b>14.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-8"><i class="fa fa-check"></i><b>14.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#abstract"><i class="fa fa-check"></i><b>14.4</b> Abstract</a></li>
<li class="chapter" data-level="14.5" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#introduction-5"><i class="fa fa-check"></i><b>14.5</b> Introduction</a><ul>
<li class="chapter" data-level="14.5.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#downloading-pharmacoset-objects"><i class="fa fa-check"></i><b>14.5.1</b> Downloading PharmacoSet objects</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#reproducibility"><i class="fa fa-check"></i><b>14.6</b> Reproducibility</a><ul>
<li class="chapter" data-level="14.6.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#plotting-drug-dose-response-data"><i class="fa fa-check"></i><b>14.6.1</b> Plotting Drug-Dose Response Data</a></li>
<li class="chapter" data-level="14.6.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pharmacological-profiles"><i class="fa fa-check"></i><b>14.6.2</b> Pharmacological profiles</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#replication"><i class="fa fa-check"></i><b>14.7</b> Replication</a><ul>
<li class="chapter" data-level="14.7.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-of-pharmacological-profiles"><i class="fa fa-check"></i><b>14.7.1</b> Consistency of pharmacological profiles</a></li>
<li class="chapter" data-level="14.7.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-assessment-improved-by-modified-concordance-index"><i class="fa fa-check"></i><b>14.7.2</b> consistency assessment improved by Modified Concordance Index</a></li>
<li class="chapter" data-level="14.7.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#known-biomarkers"><i class="fa fa-check"></i><b>14.7.3</b> Known Biomarkers</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#machine-learning-and-biomarker-discovery"><i class="fa fa-check"></i><b>14.8</b> Machine Learning and Biomarker Discovery</a><ul>
<li class="chapter" data-level="14.8.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#bonus-using-the-connectivity-map-for-drug-repurposing"><i class="fa fa-check"></i><b>14.8.1</b> Bonus: Using the Connectivity Map for drug repurposing</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#session-info"><i class="fa fa-check"></i><b>14.9</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>15</b> 500: Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="15.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-6"><i class="fa fa-check"></i><b>15.1</b> Overview</a><ul>
<li class="chapter" data-level="15.1.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#description-3"><i class="fa fa-check"></i><b>15.1.1</b> Description</a></li>
<li class="chapter" data-level="15.1.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-11"><i class="fa fa-check"></i><b>15.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="15.1.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#participation-3"><i class="fa fa-check"></i><b>15.1.3</b> Participation</a></li>
<li class="chapter" data-level="15.1.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-11"><i class="fa fa-check"></i><b>15.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="15.1.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-12"><i class="fa fa-check"></i><b>15.1.5</b> Time outline</a></li>
<li class="chapter" data-level="15.1.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#introductory-material"><i class="fa fa-check"></i><b>15.2</b> Introductory material</a><ul>
<li class="chapter" data-level="15.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#discussion"><i class="fa fa-check"></i><b>15.2.1</b> Discussion</a></li>
<li class="chapter" data-level="15.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#why-learn-delayedarray"><i class="fa fa-check"></i><b>15.2.2</b> Why learn DelayedArray?</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-of-delayedarray-framework"><i class="fa fa-check"></i><b>15.3</b> Overview of DelayedArray framework</a><ul>
<li class="chapter" data-level="15.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-package"><i class="fa fa-check"></i><b>15.3.1</b> The <strong>DelayedArray</strong> package</a></li>
<li class="chapter" data-level="15.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-class"><i class="fa fa-check"></i><b>15.3.2</b> The <em>DelayedArray</em> class</a></li>
<li class="chapter" data-level="15.3.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#operating-on-delayedarray-objects"><i class="fa fa-check"></i><b>15.3.3</b> Operating on <em>DelayedArray</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#whats-out-there-already"><i class="fa fa-check"></i><b>15.4</b> What’s out there already?</a></li>
<li class="chapter" data-level="15.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#incoporating-delayedarray-into-a-package"><i class="fa fa-check"></i><b>15.5</b> Incoporating DelayedArray into a package</a><ul>
<li class="chapter" data-level="15.5.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#writing-algorithms-to-process-delayedarray-instances"><i class="fa fa-check"></i><b>15.5.1</b> Writing algorithms to process <em>DelayedArray</em> instances</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#questions-and-discussion"><i class="fa fa-check"></i><b>15.6</b> Questions and discussion</a></li>
<li class="chapter" data-level="15.7" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#todos"><i class="fa fa-check"></i><b>15.7</b> TODOs</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>16</b> 510: Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="16.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-2"><i class="fa fa-check"></i><b>16.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="16.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-10"><i class="fa fa-check"></i><b>16.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="16.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-12"><i class="fa fa-check"></i><b>16.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="16.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-4"><i class="fa fa-check"></i><b>16.2.2</b> Participation</a></li>
<li class="chapter" data-level="16.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>16.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="16.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-13"><i class="fa fa-check"></i><b>16.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-6"><i class="fa fa-check"></i><b>16.3</b> Introduction</a><ul>
<li class="chapter" data-level="16.3.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>16.3.1</b> Why maintain your package</a></li>
<li class="chapter" data-level="16.3.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>16.3.2</b> Infrastructure used for maintenance</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>16.4</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="16.4.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>16.4.1</b> Build report</a></li>
<li class="chapter" data-level="16.4.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#version-numbering"><i class="fa fa-check"></i><b>16.4.2</b> Version Numbering</a></li>
<li class="chapter" data-level="16.4.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>16.4.3</b> Landing page badges</a></li>
<li class="chapter" data-level="16.4.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>16.4.4</b> Download statistics</a></li>
<li class="chapter" data-level="16.4.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>16.4.5</b> RSS feeds</a></li>
<li class="chapter" data-level="16.4.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>16.4.6</b> Support site, Bioc-devel mailing list</a></li>
<li class="chapter" data-level="16.4.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github"><i class="fa fa-check"></i><b>16.4.7</b> GitHub</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>16.5</b> Round up of resources available</a></li>
<li class="chapter" data-level="16.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#acknowledgements"><i class="fa fa-check"></i><b>16.6</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="bibliography-1.html"><a href="bibliography-1.html"><i class="fa fa-check"></i><b>17</b> Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="functional-enrichment-analysis-of-high-throughput-omics-data" class="section level1">
<h1><span class="header-section-number">9</span> 210: Functional enrichment analysis of high-throughput omics data</h1>
<div id="instructor-names-and-contact-information-1" class="section level2">
<h2><span class="header-section-number">9.1</span> Instructor names and contact information</h2>
<ul>
<li>Ludwig Geistlinger (<a href="mailto:Ludwig.Geistlinger@sph.cuny.edu">Ludwig.Geistlinger@sph.cuny.edu</a>)</li>
<li>Levi Waldron</li>
</ul>
<p>CUNY School of Public Health 55 W 125th St, New York, NY 10027</p>
</div>
<div id="workshop-description-4" class="section level2">
<h2><span class="header-section-number">9.2</span> Workshop Description</h2>
<p>This workshop gives an in-depth overview of existing methods for enrichment analysis of gene expression data with regard to functional gene sets, pathways, and networks. The workshop will help participants understand the distinctions between assumptions and hypotheses of existing methods as well as the differences in objectives and interpretation of results. It will provide code and hands-on practice of all necessary steps for differential expression analysis, gene set- and network-based enrichment analysis, and identification of enriched genomic regions and regulatory elements, along with visualization and exploration of results.</p>
<div id="pre-requisites-6" class="section level3">
<h3><span class="header-section-number">9.2.1</span> Pre-requisites</h3>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Familiarity with the SummarizedExperiment class</li>
<li><p>Familiarity with the GenomicRanges class</p></li>
<li>Familiarity with high-throughput gene expression data as obtained with microarrays and RNA-seq</li>
<li><p>Familiarity with the concept of differential expression analysis (with e.g. limma, edgeR, DESeq2)</p></li>
</ul>
</div>
<div id="workshop-participation-4" class="section level3">
<h3><span class="header-section-number">9.2.2</span> Workshop Participation</h3>
<p>Execution of example code and hands-on practice</p>
</div>
<div id="r-bioconductor-packages-used-6" class="section level3">
<h3><span class="header-section-number">9.2.3</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li><a href="http://bioconductor.org/packages/EnrichmentBrowser">EnrichmentBrowser</a></li>
<li><a href="http://bioconductor.org/packages/regioneR">regioneR</a></li>
<li><a href="http://bioconductor.org/packages/airway">airway</a></li>
<li><a href="http://bioconductor.org/packages/ALL">ALL</a></li>
<li><a href="http://bioconductor.org/packages/hgu95av2.db">hgu95av2.db</a></li>
<li><a href="http://bioconductor.org/packages/BSgenome.Hsapiens.UCSC.hg19.masked">BSgenome.Hsapiens.UCSC.hg19.masked</a></li>
</ul>
</div>
<div id="time-outline-6" class="section level3">
<h3><span class="header-section-number">9.2.4</span> Time outline</h3>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Background</td>
<td>30m</td>
</tr>
<tr class="even">
<td>Differential expression analysis</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Gene set analysis</td>
<td>30m</td>
</tr>
<tr class="even">
<td>Gene network analysis</td>
<td>15m</td>
</tr>
<tr class="odd">
<td>Genomic region analysis</td>
<td>15m</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="goals-and-objectives" class="section level2">
<h2><span class="header-section-number">9.3</span> Goals and objectives</h2>
<p>Theory:</p>
<ul>
<li>Gene sets, pathways &amp; regulatory networks</li>
<li>Resources</li>
<li>Gene set analysis vs. gene set enrichment analysis</li>
<li>Underlying null: competitive vs. self-contained</li>
<li>Generations: ora, fcs &amp; topology-based</li>
</ul>
<p>Practice:</p>
<ul>
<li>Data types: microarray vs. RNA-seq</li>
<li>Differential expression analysis</li>
<li>Defining gene sets according to GO and KEGG</li>
<li>GO/KEGG overrepresentation analysis</li>
<li>Functional class scoring &amp; permutation testing</li>
<li>Network-based enrichment analysis</li>
<li>Genomic region enrichment analysis</li>
</ul>
</div>
<div id="where-does-it-all-come-from" class="section level2">
<h2><span class="header-section-number">9.4</span> Where does it all come from?</h2>
<p>Test whether known biological functions or processes are over-represented (= enriched) in an experimentally-derived gene list, e.g. a list of differentially expressed (DE) genes. See <a href="https://doi.org/10.1093/bioinformatics/btm051">Goeman and Buehlmann, 2007</a> for a critical review.</p>
<p>Example: Transcriptomic study, in which 12,671 genes have been tested for differential expression between two sample conditions and 529 genes were found DE.</p>
<p>Among the DE genes, 28 are annotated to a specific functional gene set, which contains in total 170 genes. This setup corresponds to a 2x2 contingency table,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">deTable &lt;-
<span class="st">     </span><span class="kw">matrix</span>(<span class="kw">c</span>(<span class="dv">28</span>, <span class="dv">142</span>, <span class="dv">501</span>, <span class="dv">12000</span>),
            <span class="dt">nrow =</span> <span class="dv">2</span>,
            <span class="dt">dimnames =</span> <span class="kw">list</span>(<span class="kw">c</span>(<span class="st">&quot;DE&quot;</span>, <span class="st">&quot;Not.DE&quot;</span>),
                            <span class="kw">c</span>(<span class="st">&quot;In.gene.set&quot;</span>, <span class="st">&quot;Not.in.gene.set&quot;</span>)))
deTable
<span class="co">#&gt;        In.gene.set Not.in.gene.set</span>
<span class="co">#&gt; DE              28             501</span>
<span class="co">#&gt; Not.DE         142           12000</span></code></pre></div>
<p>where the overlap of 28 genes can be assessed based on the hypergeometric distribution. This corresponds to a one-sided version of Fisher’s exact test, yielding here a significant enrichment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">fisher.test</span>(deTable, <span class="dt">alternative =</span> <span class="st">&quot;greater&quot;</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt;  Fisher&#39;s Exact Test for Count Data</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; data:  deTable</span>
<span class="co">#&gt; p-value = 4.088e-10</span>
<span class="co">#&gt; alternative hypothesis: true odds ratio is greater than 1</span>
<span class="co">#&gt; 95 percent confidence interval:</span>
<span class="co">#&gt;  3.226736      Inf</span>
<span class="co">#&gt; sample estimates:</span>
<span class="co">#&gt; odds ratio </span>
<span class="co">#&gt;   4.721744</span></code></pre></div>
<p>This basic principle is at the foundation of major public and commercial enrichment tools such as <a href="https://david.ncifcrf.gov/">DAVID</a> and <a href="https://www.pathwaystudio.com">Pathway Studio</a>.</p>
<p>Although gene set enrichment methods have been primarily developed and applied on transcriptomic data, they have recently been modified, extended and applied also in other fields of genomic and biomedical research. This includes novel approaches for functional enrichment analysis of proteomic and metabolomic data as well as genomic regions and disease phenotypes, <a href="https://doi.org/10.1002/0471250953.bi1328s53">Lavallee and Yates, 2016</a>, <a href="https://doi.org/10.1007/978-1-4939-3572-7_20">Chagoyen et al., 2016</a>, <a href="https://doi.org/10.1038/nbt.1630">McLean et al., 2010</a>, <a href="https://doi.org/10.1002/gepi.21617">Ried et al., 2012</a>.</p>
</div>
<div id="gene-expression-based-enrichment-analysis" class="section level2">
<h2><span class="header-section-number">9.5</span> Gene expression-based enrichment analysis</h2>
<p>The first part of the workshop is largely based on the <a href="http://bioconductor.org/packages/EnrichmentBrowser">EnrichmentBrowser</a> package, which implements an analysis pipeline for high-throughput gene expression data as measured with microarrays and RNA-seq. In a workflow-like manner, the package brings together a selection of established Bioconductor packages for gene expression data analysis. It integrates a wide range of gene set enrichment analysis methods and facilitates combination and exploration of results across methods.</p>
<p><img src="https://github.com/waldronlab/BrownCOBRE2018/blob/master/notebooks_day2/EnrichmentBrowserWorkflow.png?raw=true", alt="EnrichmentBrowserWorkflow image", style="width:650px"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(EnrichmentBrowser))</code></pre></div>
<p>Further information can be found in the <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/EnrichmentBrowser/inst/doc/EnrichmentBrowser.pdf">vignette</a> and <a href="https://doi.org/10.1186/s12859-016-0884-1">publication</a>.</p>
</div>
<div id="a-primer-on-terminology-existing-methods-statistical-theory" class="section level2">
<h2><span class="header-section-number">9.6</span> A primer on terminology, existing methods &amp; statistical theory</h2>
<p><strong>Gene sets, pathways &amp; regulatory networks</strong></p>
<p>Gene sets are simple lists of usually functionally related genes without further specification of relationships between genes.</p>
<p>Pathways can be interpreted as specific gene sets, typically representing a group of genes that work together in a biological process. Pathways are commonly divided in metabolic and signaling pathways. Metabolic pathways such as glycolysis represent biochemical substrate conversions by specific enzymes. Signaling pathways such as the MAPK signaling pathway describe signal transduction cascades from receptor proteins to transcription factors, resulting in activation or inhibition of specific target genes.</p>
<p>Gene regulatory networks describe the interplay and effects of regulatory factors (such as transcription factors and microRNAs) on the expression of their target genes.</p>
<p><strong>Resources</strong></p>
<p><a href="http://www.geneontology.org">GO</a> and <a href="http://www.genome.jp/kegg">KEGG</a> annotations are most frequently used for the enrichment analysis of functional gene sets. Despite an increasing number of gene set and pathway databases, they are typically the first choice due to their long-standing curation and availability for a wide range of species.</p>
<p><em>GO</em>: The Gene Ontology (GO) consists of three major sub-ontologies that classify gene products according to molecular function (MF), biological process (BP) and cellular component (CC). Each ontology consists of GO terms that define MFs, BPs or CCs to which specific genes are annotated. The terms are organized in a directed acyclic graph, where edges between the terms represent relationships of different types. They relate the terms according to a parent-child scheme, i.e. parent terms denote more general entities, whereas child terms represent more specific entities.</p>
<p><em>KEGG</em>: The Kyoto Encyclopedia of Genes and Genomes (KEGG) is a collection of manually drawn pathway maps representing molecular interaction and reaction networks. These pathways cover a wide range of biochemical processes that can be divided in 7 broad categories: metabolism, genetic and environmental information processing, cellular processes, organismal systems, human diseases, and drug development. Metabolism and drug development pathways differ from pathways of the other 5 categories by illustrating reactions between chemical compounds. Pathways of the other 5 categories illustrate molecular interactions between genes and gene products.</p>
<p><strong>Gene set analysis vs. gene set enrichment analysis</strong></p>
<p>The two predominantly used enrichment methods are:</p>
<ul>
<li>Overrepresentation analysis (ORA), testing whether a gene set contains disproportional many genes of significant expression change, based on the procedure outlined in the first section</li>
<li>Gene set enrichment analysis (GSEA), testing whether genes of a gene set accumulate at the top or bottom of the full gene vector ordered by direction and magnitude of expression change <a href="https://doi.org/10.1073/pnas.0506580102">Subramanian et al., 2005</a></li>
</ul>
<p>However, the term <em>gene set enrichment analysis</em> nowadays subsumes a general strategy implemented by a wide range of methods <a href="https://doi.org/10.1093/nar/gkn923">Huang et al., 2009</a>. Those methods have in common the same goal, although approach and statistical model can vary substantially <a href="https://doi.org/10.1093/bioinformatics/btm051">Goeman and Buehlmann, 2007</a>, <a href="https://doi.org/10.1371/journal.pcbi.1002375">Khatri et al., 2012</a>.</p>
<p>To better distinguish from the specific method, some authors use the term <em>gene set analysis</em> to denote the general strategy. However, there is also a specific method from <a href="https://doi.org/10.1214/07-AOAS101">Efron and Tibshirani, 2007</a> of this name.</p>
<p><strong>Underlying null: competitive vs. self-contained</strong></p>
<p><a href="https://doi.org/10.1093/bioinformatics/btm051">Goeman and Buehlmann, 2007</a> classified existing enrichment methods into <em>competitive</em> and <em>self-contained</em> based on the underlying null hypothesis.</p>
<ul>
<li><em>Competitive</em> null hypothesis: the genes in the set of interest are at most as often DE as the genes not in the set,</li>
<li><em>Self-contained</em> null hypothesis: no genes in the set of interest are DE.</li>
</ul>
<p>Although the authors argue that a self-contained null is closer to the actual question of interest, the vast majority of enrichment methods is competitive.</p>
<p>Goeman and Buehlmann further raise several critical issues concerning the 2x2 ORA:</p>
<ul>
<li>rather arbitrary classification of genes in DE / not DE</li>
<li>based on gene sampling, although sampling of subjects is appropriate</li>
<li>unrealistic independence assumption between genes, resulting in highly anti-conservative <em>p</em>-values</li>
</ul>
<p>With regard to these statistical concerns, GSEA is considered superior:</p>
<ul>
<li>takes all measured genes into account</li>
<li>subject sampling via permutation of class labels<br />
</li>
<li>the incorporated permutation procedure implicitly accounts for correlations between genes</li>
</ul>
<p>However, the simplicity and general applicability of ORA is unmet by subsequent methods improving on these issues. For instance, GSEA requires the expression data as input, which is not available for gene lists derived from other experiment types. On the other hand, the involved sample permutation procedure has been proven inaccurate and time-consuming <a href="https://doi.org/10.1214/07-AOAS101">Efron and Tibshirani, 2007</a>, <a href="https://doi.org/10.2202/1544-6115.1585">Phipson and Smyth, 2010</a>, <a href="https://doi.org/10.1186/s12859-015-0571-7">Larson and Owen, 2015</a>.</p>
<p><strong>Generations: ora, fcs &amp; topology-based</strong></p>
<p><a href="https://doi.org/10.1371/journal.pcbi.1002375">Khatri et al., 2012</a> have taken a slightly different approach by classifying methods along the timeline of development into three generations:</p>
<ol style="list-style-type: decimal">
<li>Generation: ORA methods based on the 2x2 contingency table test,</li>
<li>Generation: functional class scoring (FCS) methods such as GSEA, which compute gene set (= functional class) scores by summarizing per-gene DE statistics,</li>
<li>Generation: topology-based methods, explicitly taking into account interactions between genes as defined in signaling pathways and gene regulatory networks (<a href="https://doi.org/10.1093/bioinformatics/btr228">Geistlinger et al., 2011</a> for an example).</li>
</ol>
<p>Although topology-based (also: network-based) methods appear to be most realistic, their straightforward application can be impaired by features that are not-detectable on the transcriptional level (such as protein-protein interactions) and insufficient network knowledge <a href="https://doi.org/10.1093/nar/gkt631">Geistlinger et al., 2013</a>, <a href="https://doi.org/10.1186/s12859-015-0751-5">Bayerlova et al., 2015</a>.</p>
<p>Given the individual benefits and limitations of existing methods, cautious interpretation of results is required to derive valid conclusions. Whereas no single method is best suited for all application scenarios, applying multiple methods can be beneficial. This has been shown to filter out spurious hits of individual methods, thereby reducing the outcome to gene sets accumulating evidence from different methods <a href="https://doi.org/10.1186/s12859-016-0884-1">Geistlinger et al., 2016</a>, <a href="https://doi.org/10.1093/bioinformatics/btw623">Alhamdoosh et al., 2017</a>.</p>
</div>
<div id="data-types" class="section level2">
<h2><span class="header-section-number">9.7</span> Data types</h2>
<p>Although RNA-seq (read count data) has become the <em>de facto</em> standard for transcriptomic profiling, it is important to know that many methods for differential expression and gene set enrichment analysis have been originally developed for microarray data (intensity measurements).</p>
<p>However, differences in data distribution assumptions (microarray: quasi-normal, RNA-seq: negative binomial) made adaptations in differential expression analysis and, to some extent, also in gene set enrichment analysis necessary.</p>
<p>Thus, we consider two example datasets - a microarray and a RNA-seq dataset, and discuss similarities and differences of the respective analysis steps.</p>
<p>For microarray data, we consider expression measurements of patients with acute lymphoblastic leukemia <a href="https://doi.org/10.1182/blood-2003-09-3243">Chiaretti et al., 2004</a>. A frequent chromosomal defect found among these patients is a translocation, in which parts of chromosome 9 and 22 swap places. This results in the oncogenic fusion gene BCR/ABL created by positioning the ABL1 gene on chromosome 9 to a part of the BCR gene on chromosome 22.</p>
<p>We load the <a href="http://bioconductor.org/packages/ALL">ALL</a> dataset</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(ALL)
<span class="kw">data</span>(ALL)</code></pre></div>
<p>and select B-cell ALL patients with and without the BCR/ABL fusion, as described previously <a href="https://www.bioconductor.org/help/publications/books/bioinformatics-and-computational-biology-solutions">Gentleman et al., 2005</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ind.bs &lt;-<span class="st"> </span><span class="kw">grep</span>(<span class="st">&quot;^B&quot;</span>, ALL<span class="op">$</span>BT)
ind.mut &lt;-<span class="st"> </span><span class="kw">which</span>(ALL<span class="op">$</span>mol.biol <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;BCR/ABL&quot;</span>, <span class="st">&quot;NEG&quot;</span>))
sset &lt;-<span class="st"> </span><span class="kw">intersect</span>(ind.bs, ind.mut)
all.eset &lt;-<span class="st"> </span>ALL[, sset]</code></pre></div>
<p>We can now access the expression values, which are intensity measurements on a log-scale for 12,625 probes (rows) across 79 patients (columns).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(all.eset)
<span class="co">#&gt; Features  Samples </span>
<span class="co">#&gt;    12625       79</span>
<span class="kw">exprs</span>(all.eset)[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]
<span class="co">#&gt;              01005    01010    03002    04007</span>
<span class="co">#&gt; 1000_at   7.597323 7.479445 7.567593 7.905312</span>
<span class="co">#&gt; 1001_at   5.046194 4.932537 4.799294 4.844565</span>
<span class="co">#&gt; 1002_f_at 3.900466 4.208155 3.886169 3.416923</span>
<span class="co">#&gt; 1003_s_at 5.903856 6.169024 5.860459 5.687997</span></code></pre></div>
<p>As we often have more than one probe per gene, we compute gene expression values as the average of the corresponding probe values.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">allSE &lt;-<span class="st"> </span><span class="kw">probe2gene</span>(all.eset) 
<span class="kw">head</span>(<span class="kw">names</span>(allSE))
<span class="co">#&gt; [1] &quot;5595&quot; &quot;7075&quot; &quot;1557&quot; &quot;643&quot;  &quot;1843&quot; &quot;4319&quot;</span></code></pre></div>
<p>For RNA-seq data, we consider transcriptome profiles of four primary human airway smooth muscle cell lines in two conditions: control and treatment with dexamethasone <a href="https://doi.org/10.1371/journal.pone.0099625">Himes et al., 2014</a>.</p>
<p>We load the <a href="http://bioconductor.org/packages/airway">airway</a> dataset</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(airway)
<span class="kw">data</span>(airway)</code></pre></div>
<p>For further analysis, we only keep genes that are annotated to an ENSEMBL gene ID.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">airSE &lt;-<span class="st"> </span>airway[<span class="kw">grep</span>(<span class="st">&quot;^ENSG&quot;</span>, <span class="kw">names</span>(airway)), ]
<span class="kw">dim</span>(airSE)
<span class="co">#&gt; [1] 63677     8</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">assay</span>(airSE)[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>]
<span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512 SRR1039513</span>
<span class="co">#&gt; ENSG00000000003        679        448        873        408</span>
<span class="co">#&gt; ENSG00000000005          0          0          0          0</span>
<span class="co">#&gt; ENSG00000000419        467        515        621        365</span>
<span class="co">#&gt; ENSG00000000457        260        211        263        164</span></code></pre></div>
</div>
<div id="differential-expression-analysis-2" class="section level2">
<h2><span class="header-section-number">9.8</span> Differential expression analysis</h2>
<p>Normalization of high-throughput expression data is essential to make results within and between experiments comparable. Microarray (intensity measurements) and RNA-seq (read counts) data typically show distinct features that need to be normalized for. As this is beyond the scope of this workshop, we refer to <a href="http://bioconductor.org/packages/limma">limma</a> for microarray normalization and <a href="http://bioconductor.org/packages/EDASeq">EDASeq</a> for RNA-seq normalization. See also <code>EnrichmentBrowser::normalize</code>, which wraps commonly used functionality for normalization.</p>
<p>The EnrichmentBrowser incorporates established functionality from the <a href="http://bioconductor.org/packages/limma">limma</a> package for differential expression analysis. This involves the <code>voom</code> transformation when applied to RNA-seq data. Alternatively, differential expression analysis for RNA-seq data can also be carried out based on the negative binomial distribution with <a href="http://bioconductor.org/packages/edgeR">edgeR</a> and <a href="http://bioconductor.org/packages/DESeq2">DESeq2</a>.</p>
<p>This can be performed using the function <code>EnrichmentBrowser::deAna</code> and assumes some standardized variable names:</p>
<ul>
<li><strong>GROUP</strong> defines the sample groups being contrasted,</li>
<li><strong>BLOCK</strong> defines paired samples or sample blocks, as e.g. for batch effects.</li>
</ul>
<p>For more information on experimental design, see the <a href="https://www.bioconductor.org/packages/devel/bioc/vignettes/limma/inst/doc/usersguide.pdf">limma user’s guide</a>, chapter 9.</p>
<p>For the ALL dataset, the <strong>GROUP</strong> variable indicates whether the BCR-ABL gene fusion is present (1) or not (0).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">allSE<span class="op">$</span>GROUP &lt;-<span class="st"> </span><span class="kw">ifelse</span>(allSE<span class="op">$</span>mol.biol <span class="op">==</span><span class="st"> &quot;BCR/ABL&quot;</span>, <span class="dv">1</span>, <span class="dv">0</span>)
<span class="kw">table</span>(allSE<span class="op">$</span>GROUP)
<span class="co">#&gt; </span>
<span class="co">#&gt;  0  1 </span>
<span class="co">#&gt; 42 37</span></code></pre></div>
<p>For the airway dataset, it indicates whether the cell lines have been treated with dexamethasone (1) or not (0).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">airSE<span class="op">$</span>GROUP &lt;-<span class="st"> </span><span class="kw">ifelse</span>(<span class="kw">colData</span>(airway)<span class="op">$</span>dex <span class="op">==</span><span class="st"> &quot;trt&quot;</span>, <span class="dv">1</span>, <span class="dv">0</span>)
<span class="kw">table</span>(airSE<span class="op">$</span>GROUP)
<span class="co">#&gt; </span>
<span class="co">#&gt; 0 1 </span>
<span class="co">#&gt; 4 4</span></code></pre></div>
<p>Paired samples, or in general sample batches/blocks, can be defined via a <code>BLOCK</code> column in the <code>colData</code> slot. For the airway dataset, the sample blocks correspond to the four different cell lines.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">airSE<span class="op">$</span>BLOCK &lt;-<span class="st"> </span>airway<span class="op">$</span>cell
<span class="kw">table</span>(airSE<span class="op">$</span>BLOCK)
<span class="co">#&gt; </span>
<span class="co">#&gt; N052611 N061011 N080611  N61311 </span>
<span class="co">#&gt;       2       2       2       2</span></code></pre></div>
<p>For microarray data, the <code>EnrichmentBrowser::deAna</code> function carries out differential expression analysis based on functionality from the <em>limma</em> package. Resulting log2 fold changes and <em>t</em>-test derived <em>p</em>-values for each gene are appended to the <code>rowData</code> slot.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">allSE &lt;-<span class="st"> </span><span class="kw">deAna</span>(allSE)
<span class="kw">rowData</span>(allSE, <span class="dt">use.names=</span><span class="ot">TRUE</span>)
<span class="co">#&gt; DataFrame with 9010 rows and 3 columns</span>
<span class="co">#&gt;                        FC          ADJ.PVAL         limma.STAT</span>
<span class="co">#&gt;                 &lt;numeric&gt;         &lt;numeric&gt;          &lt;numeric&gt;</span>
<span class="co">#&gt; 5595   0.0429698599842595 0.899246817310775  0.734679177472013</span>
<span class="co">#&gt; 7075   0.0320835027449625 0.949001013642671    0.4546910829318</span>
<span class="co">#&gt; 1557  -0.0439401425131442 0.818330132411339  -1.06578261967549</span>
<span class="co">#&gt; 643   -0.0277543539240438 0.929148567589577 -0.567394394416651</span>
<span class="co">#&gt; 1843   -0.427302534257363 0.566034751753148  -1.75050227190017</span>
<span class="co">#&gt; ...                   ...               ...                ...</span>
<span class="co">#&gt; 6300   -0.026651766164237 0.922828548631225 -0.608608859328046</span>
<span class="co">#&gt; 7297   -0.124257678078831 0.804578494190681  -1.11279493778184</span>
<span class="co">#&gt; 2246   0.0522428857778935 0.748021044717352   1.27408420746691</span>
<span class="co">#&gt; 7850 -0.00908229596065303 0.991826450687159 -0.102406339091096</span>
<span class="co">#&gt; 1593 -0.00747713820802068 0.989532971314233 -0.145650256847251</span></code></pre></div>
<p>Nominal <em>p</em>-values are already corrected for multiple testing (<code>ADJ.PVAL</code>) using the method from Benjamini and Hochberg implemented in <code>stats::p.adjust</code>.</p>
<p>For RNA-seq data, the <code>deAna</code> function can be used to carry out differential expression analysis between the two groups either based on functionality from <em>limma</em> (that includes the <code>voom</code> transformation), or alternatively, the frequently used <em>edgeR</em> or <em>DESeq2</em> package. Here, we use the analysis based on <em>edgeR</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">airSE &lt;-<span class="st"> </span><span class="kw">deAna</span>(airSE, <span class="dt">de.method=</span><span class="st">&quot;edgeR&quot;</span>)
<span class="co">#&gt; Excluding 50740 genes not satisfying min.cpm threshold</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rowData</span>(airSE, <span class="dt">use.names=</span><span class="ot">TRUE</span>)
<span class="co">#&gt; DataFrame with 12937 rows and 3 columns</span>
<span class="co">#&gt;                                  FC            ADJ.PVAL</span>
<span class="co">#&gt;                           &lt;numeric&gt;           &lt;numeric&gt;</span>
<span class="co">#&gt; ENSG00000000003  -0.404945626610932 0.00213458295385943</span>
<span class="co">#&gt; ENSG00000000419   0.182985434777532  0.0915691945172958</span>
<span class="co">#&gt; ENSG00000000457  0.0143477674070903   0.922279475399443</span>
<span class="co">#&gt; ENSG00000000460  -0.141173372957311   0.619013213521635</span>
<span class="co">#&gt; ENSG00000000971   0.402240426474172 0.00403820532305827</span>
<span class="co">#&gt; ...                             ...                 ...</span>
<span class="co">#&gt; ENSG00000273270  -0.129793853337261   0.495892935815041</span>
<span class="co">#&gt; ENSG00000273290   0.505580471641003 0.00639218387702899</span>
<span class="co">#&gt; ENSG00000273311 0.00161557580855148   0.996356136956657</span>
<span class="co">#&gt; ENSG00000273329  -0.222817127090519   0.388294594068803</span>
<span class="co">#&gt; ENSG00000273344  0.0151704005097403   0.962777106053257</span>
<span class="co">#&gt;                           edgeR.STAT</span>
<span class="co">#&gt;                            &lt;numeric&gt;</span>
<span class="co">#&gt; ENSG00000000003     35.8743710016552</span>
<span class="co">#&gt; ENSG00000000419     5.90960619951737</span>
<span class="co">#&gt; ENSG00000000457   0.0233923316990905</span>
<span class="co">#&gt; ENSG00000000460    0.492929955080604</span>
<span class="co">#&gt; ENSG00000000971     27.8509962017407</span>
<span class="co">#&gt; ...                              ...</span>
<span class="co">#&gt; ENSG00000273270    0.901598359265205</span>
<span class="co">#&gt; ENSG00000273290     23.0905678847871</span>
<span class="co">#&gt; ENSG00000273311 8.04821152029429e-05</span>
<span class="co">#&gt; ENSG00000273329     1.42723325850597</span>
<span class="co">#&gt; ENSG00000273344  0.00543503273765429</span></code></pre></div>
<p><em>Exercise:</em> Compare the number of differentially expressed genes as obtained on the <code>airSE</code> with <code>limma/voom</code>, <code>edgeR</code>, and <code>DESeq2</code>.</p>
</div>
<div id="gene-sets" class="section level2">
<h2><span class="header-section-number">9.9</span> Gene sets</h2>
<p>We are now interested in whether pre-defined sets of genes that are known to work together, e.g. as defined in the <a href="http://www.geneontology.org">Gene Ontology</a> or the <a href="http://www.genome.jp/kegg">KEGG</a> pathway annotation, are coordinately differentially expressed.</p>
<p>The function <code>getGenesets</code> can be used to download gene sets from databases such as GO and KEGG. Here, we use the function to download all KEGG pathways for a chosen organism (here: ) as gene sets.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">kegg.gs &lt;-<span class="st"> </span><span class="kw">getGenesets</span>(<span class="dt">org=</span><span class="st">&quot;hsa&quot;</span>, <span class="dt">db=</span><span class="st">&quot;kegg&quot;</span>)</code></pre></div>
<p>Analogously, the function <code>getGenesets</code> can be used to retrieve GO terms of a selected ontology (here: biological process, BP) as defined in the <em>GO.db</em> annotation package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">go.gs &lt;-<span class="st"> </span><span class="kw">getGenesets</span>(<span class="dt">org=</span><span class="st">&quot;hsa&quot;</span>, <span class="dt">db=</span><span class="st">&quot;go&quot;</span>, <span class="dt">go.onto=</span><span class="st">&quot;BP&quot;</span>, <span class="dt">go.mode=</span><span class="st">&quot;GO.db&quot;</span>)
<span class="co">#&gt; </span></code></pre></div>
<p>If provided a file, the function <code>getGenesets</code> parses user-defined gene sets from GMT file format. Here, we use this functionality for reading a list of already downloaded KEGG gene sets for <em>Homo sapiens</em> containing NCBI Entrez Gene IDs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">data.dir &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="dt">package=</span><span class="st">&quot;EnrichmentBrowser&quot;</span>)
gmt.file &lt;-<span class="st"> </span><span class="kw">file.path</span>(data.dir, <span class="st">&quot;hsa_kegg_gs.gmt&quot;</span>)
hsa.gs &lt;-<span class="st"> </span><span class="kw">getGenesets</span>(gmt.file)
hsa.gs[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]
<span class="co">#&gt; $hsa05416_Viral_myocarditis</span>
<span class="co">#&gt;  [1] &quot;100509457&quot; &quot;101060835&quot; &quot;1525&quot;      &quot;1604&quot;      &quot;1605&quot;     </span>
<span class="co">#&gt;  [6] &quot;1756&quot;      &quot;1981&quot;      &quot;1982&quot;      &quot;25&quot;        &quot;2534&quot;     </span>
<span class="co">#&gt; [11] &quot;27&quot;        &quot;3105&quot;      &quot;3106&quot;      &quot;3107&quot;      &quot;3108&quot;     </span>
<span class="co">#&gt; [16] &quot;3109&quot;      &quot;3111&quot;      &quot;3112&quot;      &quot;3113&quot;      &quot;3115&quot;     </span>
<span class="co">#&gt; [21] &quot;3117&quot;      &quot;3118&quot;      &quot;3119&quot;      &quot;3122&quot;      &quot;3123&quot;     </span>
<span class="co">#&gt; [26] &quot;3125&quot;      &quot;3126&quot;      &quot;3127&quot;      &quot;3133&quot;      &quot;3134&quot;     </span>
<span class="co">#&gt; [31] &quot;3135&quot;      &quot;3383&quot;      &quot;3683&quot;      &quot;3689&quot;      &quot;3908&quot;     </span>
<span class="co">#&gt; [36] &quot;4624&quot;      &quot;4625&quot;      &quot;54205&quot;     &quot;5551&quot;      &quot;5879&quot;     </span>
<span class="co">#&gt; [41] &quot;5880&quot;      &quot;5881&quot;      &quot;595&quot;       &quot;60&quot;        &quot;637&quot;      </span>
<span class="co">#&gt; [46] &quot;6442&quot;      &quot;6443&quot;      &quot;6444&quot;      &quot;6445&quot;      &quot;71&quot;       </span>
<span class="co">#&gt; [51] &quot;836&quot;       &quot;841&quot;       &quot;842&quot;       &quot;857&quot;       &quot;8672&quot;     </span>
<span class="co">#&gt; [56] &quot;940&quot;       &quot;941&quot;       &quot;942&quot;       &quot;958&quot;       &quot;959&quot;      </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`hsa04622_RIG-I-like_receptor_signaling_pathway`</span>
<span class="co">#&gt;  [1] &quot;10010&quot;  &quot;1147&quot;   &quot;1432&quot;   &quot;1540&quot;   &quot;1654&quot;   &quot;23586&quot;  &quot;26007&quot; </span>
<span class="co">#&gt;  [8] &quot;29110&quot;  &quot;338376&quot; &quot;340061&quot; &quot;3439&quot;   &quot;3440&quot;   &quot;3441&quot;   &quot;3442&quot;  </span>
<span class="co">#&gt; [15] &quot;3443&quot;   &quot;3444&quot;   &quot;3445&quot;   &quot;3446&quot;   &quot;3447&quot;   &quot;3448&quot;   &quot;3449&quot;  </span>
<span class="co">#&gt; [22] &quot;3451&quot;   &quot;3452&quot;   &quot;3456&quot;   &quot;3467&quot;   &quot;3551&quot;   &quot;3576&quot;   &quot;3592&quot;  </span>
<span class="co">#&gt; [29] &quot;3593&quot;   &quot;3627&quot;   &quot;3661&quot;   &quot;3665&quot;   &quot;4214&quot;   &quot;4790&quot;   &quot;4792&quot;  </span>
<span class="co">#&gt; [36] &quot;4793&quot;   &quot;5300&quot;   &quot;54941&quot;  &quot;55593&quot;  &quot;5599&quot;   &quot;5600&quot;   &quot;5601&quot;  </span>
<span class="co">#&gt; [43] &quot;5602&quot;   &quot;5603&quot;   &quot;56832&quot;  &quot;57506&quot;  &quot;5970&quot;   &quot;6300&quot;   &quot;64135&quot; </span>
<span class="co">#&gt; [50] &quot;64343&quot;  &quot;6885&quot;   &quot;7124&quot;   &quot;7186&quot;   &quot;7187&quot;   &quot;7189&quot;   &quot;7706&quot;  </span>
<span class="co">#&gt; [57] &quot;79132&quot;  &quot;79671&quot;  &quot;80143&quot;  &quot;841&quot;    &quot;843&quot;    &quot;8517&quot;   &quot;8717&quot;  </span>
<span class="co">#&gt; [64] &quot;8737&quot;   &quot;8772&quot;   &quot;9140&quot;   &quot;9474&quot;   &quot;9636&quot;   &quot;9641&quot;   &quot;9755&quot;</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hsa.gs[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]
<span class="co">#&gt; $hsa05416_Viral_myocarditis</span>
<span class="co">#&gt;  [1] &quot;100509457&quot; &quot;101060835&quot; &quot;1525&quot;      &quot;1604&quot;      &quot;1605&quot;     </span>
<span class="co">#&gt;  [6] &quot;1756&quot;      &quot;1981&quot;      &quot;1982&quot;      &quot;25&quot;        &quot;2534&quot;     </span>
<span class="co">#&gt; [11] &quot;27&quot;        &quot;3105&quot;      &quot;3106&quot;      &quot;3107&quot;      &quot;3108&quot;     </span>
<span class="co">#&gt; [16] &quot;3109&quot;      &quot;3111&quot;      &quot;3112&quot;      &quot;3113&quot;      &quot;3115&quot;     </span>
<span class="co">#&gt; [21] &quot;3117&quot;      &quot;3118&quot;      &quot;3119&quot;      &quot;3122&quot;      &quot;3123&quot;     </span>
<span class="co">#&gt; [26] &quot;3125&quot;      &quot;3126&quot;      &quot;3127&quot;      &quot;3133&quot;      &quot;3134&quot;     </span>
<span class="co">#&gt; [31] &quot;3135&quot;      &quot;3383&quot;      &quot;3683&quot;      &quot;3689&quot;      &quot;3908&quot;     </span>
<span class="co">#&gt; [36] &quot;4624&quot;      &quot;4625&quot;      &quot;54205&quot;     &quot;5551&quot;      &quot;5879&quot;     </span>
<span class="co">#&gt; [41] &quot;5880&quot;      &quot;5881&quot;      &quot;595&quot;       &quot;60&quot;        &quot;637&quot;      </span>
<span class="co">#&gt; [46] &quot;6442&quot;      &quot;6443&quot;      &quot;6444&quot;      &quot;6445&quot;      &quot;71&quot;       </span>
<span class="co">#&gt; [51] &quot;836&quot;       &quot;841&quot;       &quot;842&quot;       &quot;857&quot;       &quot;8672&quot;     </span>
<span class="co">#&gt; [56] &quot;940&quot;       &quot;941&quot;       &quot;942&quot;       &quot;958&quot;       &quot;959&quot;      </span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`hsa04622_RIG-I-like_receptor_signaling_pathway`</span>
<span class="co">#&gt;  [1] &quot;10010&quot;  &quot;1147&quot;   &quot;1432&quot;   &quot;1540&quot;   &quot;1654&quot;   &quot;23586&quot;  &quot;26007&quot; </span>
<span class="co">#&gt;  [8] &quot;29110&quot;  &quot;338376&quot; &quot;340061&quot; &quot;3439&quot;   &quot;3440&quot;   &quot;3441&quot;   &quot;3442&quot;  </span>
<span class="co">#&gt; [15] &quot;3443&quot;   &quot;3444&quot;   &quot;3445&quot;   &quot;3446&quot;   &quot;3447&quot;   &quot;3448&quot;   &quot;3449&quot;  </span>
<span class="co">#&gt; [22] &quot;3451&quot;   &quot;3452&quot;   &quot;3456&quot;   &quot;3467&quot;   &quot;3551&quot;   &quot;3576&quot;   &quot;3592&quot;  </span>
<span class="co">#&gt; [29] &quot;3593&quot;   &quot;3627&quot;   &quot;3661&quot;   &quot;3665&quot;   &quot;4214&quot;   &quot;4790&quot;   &quot;4792&quot;  </span>
<span class="co">#&gt; [36] &quot;4793&quot;   &quot;5300&quot;   &quot;54941&quot;  &quot;55593&quot;  &quot;5599&quot;   &quot;5600&quot;   &quot;5601&quot;  </span>
<span class="co">#&gt; [43] &quot;5602&quot;   &quot;5603&quot;   &quot;56832&quot;  &quot;57506&quot;  &quot;5970&quot;   &quot;6300&quot;   &quot;64135&quot; </span>
<span class="co">#&gt; [50] &quot;64343&quot;  &quot;6885&quot;   &quot;7124&quot;   &quot;7186&quot;   &quot;7187&quot;   &quot;7189&quot;   &quot;7706&quot;  </span>
<span class="co">#&gt; [57] &quot;79132&quot;  &quot;79671&quot;  &quot;80143&quot;  &quot;841&quot;    &quot;843&quot;    &quot;8517&quot;   &quot;8717&quot;  </span>
<span class="co">#&gt; [64] &quot;8737&quot;   &quot;8772&quot;   &quot;9140&quot;   &quot;9474&quot;   &quot;9636&quot;   &quot;9641&quot;   &quot;9755&quot;</span></code></pre></div>
<p>See also the <a href="http://software.broadinstitute.org/gsea/msigdb">MSigDb</a> for additional gene set collections.</p>
</div>
<div id="gokegg-overrepresentation-analysis" class="section level2">
<h2><span class="header-section-number">9.10</span> GO/KEGG overrepresentation analysis</h2>
<p>A variety of gene set analysis methods have been proposed <a href="https://doi.org/10.1371/journal.pcbi.1002375">Khatri et al., 2012</a>. The most basic, yet frequently used, method is the over-representation analysis (ORA) with gene sets defined according to GO or KEGG. As outlined in the first section, ORA tests the overlap between DE genes (typically DE <em>p</em>-value &lt; 0.05) and genes in a gene set based on the hypergeometric distribution. Here, we choose a significance level <span class="math inline">\(\alpha = 0.2\)</span> for demonstration.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ora.all &lt;-<span class="st"> </span><span class="kw">sbea</span>(<span class="dt">method=</span><span class="st">&quot;ora&quot;</span>, <span class="dt">se=</span>allSE, <span class="dt">gs=</span>hsa.gs, <span class="dt">perm=</span><span class="dv">0</span>, <span class="dt">alpha=</span><span class="fl">0.2</span>)
<span class="kw">gsRanking</span>(ora.all)
<span class="co">#&gt; DataFrame with 7 rows and 4 columns</span>
<span class="co">#&gt;                                                          GENE.SET</span>
<span class="co">#&gt;                                                       &lt;character&gt;</span>
<span class="co">#&gt; 1                hsa05202_Transcriptional_misregulation_in_cancer</span>
<span class="co">#&gt; 2 hsa05412_Arrhythmogenic_right_ventricular_cardiomyopathy_(ARVC)</span>
<span class="co">#&gt; 3                                                hsa05144_Malaria</span>
<span class="co">#&gt; 4                   hsa04670_Leukocyte_transendothelial_migration</span>
<span class="co">#&gt; 5                 hsa05100_Bacterial_invasion_of_epithelial_cells</span>
<span class="co">#&gt; 6                  hsa04622_RIG-I-like_receptor_signaling_pathway</span>
<span class="co">#&gt; 7                  hsa05130_Pathogenic_Escherichia_coli_infection</span>
<span class="co">#&gt;    NR.GENES NR.SIG.GENES   P.VALUE</span>
<span class="co">#&gt;   &lt;numeric&gt;    &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; 1       153           17    0.0351</span>
<span class="co">#&gt; 2        63            8    0.0717</span>
<span class="co">#&gt; 3        45            6    0.0932</span>
<span class="co">#&gt; 4        94           10     0.122</span>
<span class="co">#&gt; 5        64            7     0.162</span>
<span class="co">#&gt; 6        54            6     0.178</span>
<span class="co">#&gt; 7        43            5     0.184</span></code></pre></div>
<p>Such a ranked list is the standard output of most existing enrichment tools. Using the <code>eaBrowse</code> function creates a HTML summary from which each gene set can be inspected in more detail.</p>
<p><img src="https://github.com/waldronlab/BrownCOBRE2018/blob/master/notebooks_day2/EnrichmentBrowserNavigation.png?raw=true", alt="EnrichmentBrowserNavigation image", style="width:600px"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">eaBrowse</span>(ora.all)
<span class="co">#&gt; Creating gene report ...</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Creating set view ...</span>
<span class="co">#&gt; Creating kegg view ...</span></code></pre></div>
<p>The resulting summary page includes for each significant gene set</p>
<ul>
<li>a gene report, which lists all genes of a set along with fold change and DE <span class="math inline">\(p\)</span>-value (click on links in column <code>NR.GENES</code>),</li>
<li>interactive overview plots such as heatmap and volcano plot (column <code>SET.VIEW</code>, supports mouse-over and click-on),</li>
<li>for KEGG pathways: highlighting of differentially expressed genes on the pathway maps (column <code>PATH.VIEW</code>, supports mouse-over and click-on).</li>
</ul>
<p>As ORA works on the list of DE genes and not the actual expression values, it can be straightforward applied to RNA-seq data. However, as the gene sets here contain NCBI Entrez gene IDs and the airway dataset contains ENSEMBL gene ids, we first map the airway dataset to Entrez IDs.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">airSE &lt;-<span class="st"> </span><span class="kw">idMap</span>(airSE, <span class="dt">org=</span><span class="st">&quot;hsa&quot;</span>, <span class="dt">from=</span><span class="st">&quot;ENSEMBL&quot;</span>, <span class="dt">to=</span><span class="st">&quot;ENTREZID&quot;</span>)
<span class="co">#&gt; &#39;select()&#39; returned 1:many mapping between keys and columns</span>
<span class="co">#&gt; Excluded 1133 genes without a corresponding to.ID</span>
<span class="co">#&gt; Encountered 8 from.IDs with &gt;1 corresponding to.ID (a single to.ID was chosen for each of them)</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ora.air &lt;-<span class="st"> </span><span class="kw">sbea</span>(<span class="dt">method=</span><span class="st">&quot;ora&quot;</span>, <span class="dt">se=</span>airSE, <span class="dt">gs=</span>hsa.gs, <span class="dt">perm=</span><span class="dv">0</span>)
<span class="kw">gsRanking</span>(ora.air)
<span class="co">#&gt; DataFrame with 9 rows and 4 columns</span>
<span class="co">#&gt;                                                          GENE.SET</span>
<span class="co">#&gt;                                                       &lt;character&gt;</span>
<span class="co">#&gt; 1                                    hsa05206_MicroRNAs_in_cancer</span>
<span class="co">#&gt; 2                                               hsa05218_Melanoma</span>
<span class="co">#&gt; 3                                                 hsa05214_Glioma</span>
<span class="co">#&gt; 4                                            hsa05131_Shigellosis</span>
<span class="co">#&gt; 5                      hsa05410_Hypertrophic_cardiomyopathy_(HCM)</span>
<span class="co">#&gt; 6                   hsa04670_Leukocyte_transendothelial_migration</span>
<span class="co">#&gt; 7                 hsa05100_Bacterial_invasion_of_epithelial_cells</span>
<span class="co">#&gt; 8                         hsa04514_Cell_adhesion_molecules_(CAMs)</span>
<span class="co">#&gt; 9 hsa05412_Arrhythmogenic_right_ventricular_cardiomyopathy_(ARVC)</span>
<span class="co">#&gt;    NR.GENES NR.SIG.GENES   P.VALUE</span>
<span class="co">#&gt;   &lt;numeric&gt;    &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; 1       118           68  0.000508</span>
<span class="co">#&gt; 2        50           33  0.000662</span>
<span class="co">#&gt; 3        48           30   0.00419</span>
<span class="co">#&gt; 4        53           31    0.0142</span>
<span class="co">#&gt; 5        53           31    0.0142</span>
<span class="co">#&gt; 6        62           34    0.0353</span>
<span class="co">#&gt; 7        60           33     0.036</span>
<span class="co">#&gt; 8        60           33     0.036</span>
<span class="co">#&gt; 9        48           27    0.0402</span></code></pre></div>
<p>Note #1: <a href="https://doi.org/10.1186/gb-2010-11-2-r14">Young et al., 2010</a>, have reported biased results for ORA on RNA-seq data due to over-detection of differential expression for long and highly expressed transcripts. The <a href="http://bioconductor.org/packages/goseq">goseq</a> package and <code>limma::goana</code> implement possibilities to adjust ORA for gene length and abundance bias.</p>
<p>Note #2: Independent of the expression data type under investigation, overlap between gene sets can result in redundant findings. This is well-documented for GO (parent-child structure, <a href="https://doi.org/10.1038/nrg2363">Rhee et al., 2008</a>) and KEGG (pathway overlap/crosstalk, <a href="https://doi.org/10.1101/gr.153551.112">Donato et al., 2013</a>). The <a href="http://bioconductor.org/packages/topGO">topGO</a> package (explicitly designed for GO) and <a href="http://bioconductor.org/packages/mgsa">mgsa</a> (applicable to arbitrary gene set definitions) implement modifications of ORA to account for such redundancies.</p>
</div>
<div id="functional-class-scoring-permutation-testing" class="section level2">
<h2><span class="header-section-number">9.11</span> Functional class scoring &amp; permutation testing</h2>
<p>A major limitation of ORA is that it restricts analysis to DE genes, excluding genes not satisfying the chosen significance threshold (typically the vast majority).</p>
<p>This is resolved by gene set enrichment analysis (GSEA), which scores the tendency of gene set members to appear rather at the top or bottom of the ranked list of all measured genes <a href="https://doi.org/10.1073/pnas.0506580102">Subramanian et al., 2005</a>. The statistical significance of the enrichment score (ES) of a gene set is assessed via sample permutation, i.e. (1) sample labels (= group assignment) are shuffled, (2) per-gene DE statistics are recomputed, and (3) the enrichment score is recomputed. Repeating this procedure many times allows to determine the empirical distribution of the enrichment score and to compare the observed enrichment score against it. Here, we carry out GSEA with 1000 permutations.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gsea.all &lt;-<span class="st"> </span><span class="kw">sbea</span>(<span class="dt">method=</span><span class="st">&quot;gsea&quot;</span>, <span class="dt">se=</span>allSE, <span class="dt">gs=</span>hsa.gs, <span class="dt">perm=</span><span class="dv">1000</span>)  
<span class="co">#&gt; Permutations: 1 -- 100</span>
<span class="co">#&gt; Permutations: 101 -- 200</span>
<span class="co">#&gt; Permutations: 201 -- 300</span>
<span class="co">#&gt; Permutations: 301 -- 400</span>
<span class="co">#&gt; Permutations: 401 -- 500</span>
<span class="co">#&gt; Permutations: 501 -- 600</span>
<span class="co">#&gt; Permutations: 601 -- 700</span>
<span class="co">#&gt; Permutations: 701 -- 800</span>
<span class="co">#&gt; Permutations: 801 -- 900</span>
<span class="co">#&gt; Permutations: 901 -- 1000</span>
<span class="co">#&gt; Processing ...</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gsRanking</span>(gsea.all)
<span class="co">#&gt; DataFrame with 20 rows and 4 columns</span>
<span class="co">#&gt;                                                            GENE.SET</span>
<span class="co">#&gt;                                                         &lt;character&gt;</span>
<span class="co">#&gt; 1   hsa05412_Arrhythmogenic_right_ventricular_cardiomyopathy_(ARVC)</span>
<span class="co">#&gt; 2                     hsa04670_Leukocyte_transendothelial_migration</span>
<span class="co">#&gt; 3                                        hsa04520_Adherens_junction</span>
<span class="co">#&gt; 4                                  hsa04390_Hippo_signaling_pathway</span>
<span class="co">#&gt; 5                                     hsa05323_Rheumatoid_arthritis</span>
<span class="co">#&gt; ...                                                             ...</span>
<span class="co">#&gt; 16                                    hsa05217_Basal_cell_carcinoma</span>
<span class="co">#&gt; 17                                               hsa04210_Apoptosis</span>
<span class="co">#&gt; 18                   hsa05130_Pathogenic_Escherichia_coli_infection</span>
<span class="co">#&gt; 19                       hsa05410_Hypertrophic_cardiomyopathy_(HCM)</span>
<span class="co">#&gt; 20                                             hsa05131_Shigellosis</span>
<span class="co">#&gt;            ES       NES   P.VALUE</span>
<span class="co">#&gt;     &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; 1       0.511      1.92         0</span>
<span class="co">#&gt; 2       0.499      1.78         0</span>
<span class="co">#&gt; 3       0.488      1.74         0</span>
<span class="co">#&gt; 4       0.459      1.67         0</span>
<span class="co">#&gt; 5       0.574      1.66    0.0019</span>
<span class="co">#&gt; ...       ...       ...       ...</span>
<span class="co">#&gt; 16      0.559      1.64    0.0248</span>
<span class="co">#&gt; 17      0.424      1.44    0.0336</span>
<span class="co">#&gt; 18      0.486      1.54    0.0347</span>
<span class="co">#&gt; 19      0.386      1.45    0.0406</span>
<span class="co">#&gt; 20      0.479      1.49    0.0436</span></code></pre></div>
<p>As GSEA’s permutation procedure involves re-computation of per-gene DE statistics, adaptations are necessary for RNA-seq. The EnrichmentBrowser implements an accordingly adapted version of GSEA, which allows incorporation of limma/voom, edgeR, or DESeq2 for repeated DE re-computation within GSEA. However, this is computationally intensive (for limma/voom the least, for DESeq2 the most). Note the relatively long running times for only 100 permutations having used edgeR for DE analysis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gsea.air &lt;-<span class="st"> </span><span class="kw">sbea</span>(<span class="dt">method=</span><span class="st">&quot;gsea&quot;</span>, <span class="dt">se=</span>airSE, <span class="dt">gs=</span>hsa.gs, <span class="dt">perm=</span><span class="dv">100</span>)  
<span class="co">#&gt; 100 permutations completed</span></code></pre></div>
<p>While it might be in some cases necessary to apply permutation-based GSEA for RNA-seq data, there are also alternatives avoiding permutation. Among them is ROtAtion gene Set Testing (ROAST), which uses rotation instead of permutation <a href="https://doi.org/10.1093/bioinformatics/btq401">Wu et al., 2010</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">roast.air &lt;-<span class="st"> </span><span class="kw">sbea</span>(<span class="dt">method=</span><span class="st">&quot;roast&quot;</span>, <span class="dt">se=</span>airSE, <span class="dt">gs=</span>hsa.gs)
<span class="kw">gsRanking</span>(roast.air)  
<span class="co">#&gt; DataFrame with 27 rows and 4 columns</span>
<span class="co">#&gt;                                         GENE.SET  NR.GENES       DIR</span>
<span class="co">#&gt;                                      &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; 1     hsa05410_Hypertrophic_cardiomyopathy_(HCM)        53         1</span>
<span class="co">#&gt; 2                         hsa05134_Legionellosis        35         1</span>
<span class="co">#&gt; 3                     hsa05416_Viral_myocarditis        33         1</span>
<span class="co">#&gt; 4                   hsa00790_Folate_biosynthesis        11         1</span>
<span class="co">#&gt; 5                       hsa03030_DNA_replication        33        -1</span>
<span class="co">#&gt; ...                                          ...       ...       ...</span>
<span class="co">#&gt; 23               hsa04150_mTOR_signaling_pathway        50         1</span>
<span class="co">#&gt; 24           hsa04350_TGF-beta_signaling_pathway        63         1</span>
<span class="co">#&gt; 25              hsa00561_Glycerolipid_metabolism        39         1</span>
<span class="co">#&gt; 26  hsa04621_NOD-like_receptor_signaling_pathway        40        -1</span>
<span class="co">#&gt; 27       hsa04514_Cell_adhesion_molecules_(CAMs)        60        -1</span>
<span class="co">#&gt;       P.VALUE</span>
<span class="co">#&gt;     &lt;numeric&gt;</span>
<span class="co">#&gt; 1       0.001</span>
<span class="co">#&gt; 2       0.001</span>
<span class="co">#&gt; 3       0.001</span>
<span class="co">#&gt; 4       0.001</span>
<span class="co">#&gt; 5       0.001</span>
<span class="co">#&gt; ...       ...</span>
<span class="co">#&gt; 23      0.027</span>
<span class="co">#&gt; 24      0.029</span>
<span class="co">#&gt; 25      0.032</span>
<span class="co">#&gt; 26      0.033</span>
<span class="co">#&gt; 27      0.035</span></code></pre></div>
<p>A selection of additional methods is also available:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">sbeaMethods</span>()
<span class="co">#&gt;  [1] &quot;ora&quot;        &quot;safe&quot;       &quot;gsea&quot;       &quot;gsa&quot;        &quot;padog&quot;     </span>
<span class="co">#&gt;  [6] &quot;globaltest&quot; &quot;roast&quot;      &quot;camera&quot;     &quot;gsva&quot;       &quot;samgs&quot;     </span>
<span class="co">#&gt; [11] &quot;ebm&quot;        &quot;mgsa&quot;</span></code></pre></div>
<p><em>Exercise</em>: Carry out a GO overrepresentation analysis for the <code>allSE</code> and <code>airSE</code>. How many significant gene sets do you observe in each case?</p>
</div>
<div id="network-based-enrichment-analysis" class="section level2">
<h2><span class="header-section-number">9.12</span> Network-based enrichment analysis</h2>
<p>Having found gene sets that show enrichment for differential expression, we are now interested whether these findings can be supported by known regulatory interactions.</p>
<p>For example, we want to know whether transcription factors and their target genes are expressed in accordance to the connecting regulations (activation/inhibition). Such information is usually given in a gene regulatory network derived from specific experiments or compiled from the literature (<a href="https://doi.org/10.1093/nar/gkt631">Geistlinger et al., 2013</a> for an example).</p>
<p>There are well-studied processes and organisms for which comprehensive and well-annotated regulatory networks are available, e.g. the <a href="http://regulondb.ccg.unam.mx">RegulonDB</a> for <em>E. coli</em> and <a href="http://www.yeastract.com">Yeastract</a> for <em>S. cerevisiae</em>.</p>
<p>However, there are also cases where such a network is missing or at least incomplete. A basic workaround is to compile a network from regulations in pathway databases such as KEGG.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hsa.grn &lt;-<span class="st"> </span><span class="kw">compileGRN</span>(<span class="dt">org=</span><span class="st">&quot;hsa&quot;</span>, <span class="dt">db=</span><span class="st">&quot;kegg&quot;</span>)
<span class="kw">head</span>(hsa.grn)
<span class="co">#&gt;      FROM    TO          TYPE</span>
<span class="co">#&gt; [1,] &quot;10000&quot; &quot;100132074&quot; &quot;-&quot; </span>
<span class="co">#&gt; [2,] &quot;10000&quot; &quot;1026&quot;      &quot;-&quot; </span>
<span class="co">#&gt; [3,] &quot;10000&quot; &quot;1026&quot;      &quot;+&quot; </span>
<span class="co">#&gt; [4,] &quot;10000&quot; &quot;1027&quot;      &quot;-&quot; </span>
<span class="co">#&gt; [5,] &quot;10000&quot; &quot;10488&quot;     &quot;+&quot; </span>
<span class="co">#&gt; [6,] &quot;10000&quot; &quot;107&quot;       &quot;+&quot;</span></code></pre></div>
<p>Signaling pathway impact analysis (SPIA) is a network-based enrichment analysis method, which is explicitly designed for KEGG signaling pathways <a href="https://doi.org/%2010.1093/bioinformatics/btn577">Tarca et al., 2009</a>. The method evaluates whether expression changes are propagated across the pathway topology in combination with ORA.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">spia.all &lt;-<span class="st"> </span><span class="kw">nbea</span>(<span class="dt">method=</span><span class="st">&quot;spia&quot;</span>, <span class="dt">se=</span>allSE, <span class="dt">gs=</span>hsa.gs, <span class="dt">grn=</span>hsa.grn, <span class="dt">alpha=</span><span class="fl">0.2</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; Done pathway 1 : RNA transport..</span>
<span class="co">#&gt; Done pathway 2 : RNA degradation..</span>
<span class="co">#&gt; Done pathway 3 : PPAR signaling pathway..</span>
<span class="co">#&gt; Done pathway 4 : Fanconi anemia pathway..</span>
<span class="co">#&gt; Done pathway 5 : MAPK signaling pathway..</span>
<span class="co">#&gt; Done pathway 6 : ErbB signaling pathway..</span>
<span class="co">#&gt; Done pathway 7 : Calcium signaling pathway..</span>
<span class="co">#&gt; Done pathway 8 : Cytokine-cytokine receptor int..</span>
<span class="co">#&gt; Done pathway 9 : Chemokine signaling pathway..</span>
<span class="co">#&gt; Done pathway 10 : NF-kappa B signaling pathway..</span>
<span class="co">#&gt; Done pathway 11 : Phosphatidylinositol signaling..</span>
<span class="co">#&gt; Done pathway 12 : Neuroactive ligand-receptor in..</span>
<span class="co">#&gt; Done pathway 13 : Cell cycle..</span>
<span class="co">#&gt; Done pathway 14 : Oocyte meiosis..</span>
<span class="co">#&gt; Done pathway 15 : p53 signaling pathway..</span>
<span class="co">#&gt; Done pathway 16 : Sulfur relay system..</span>
<span class="co">#&gt; Done pathway 17 : SNARE interactions in vesicula..</span>
<span class="co">#&gt; Done pathway 18 : Regulation of autophagy..</span>
<span class="co">#&gt; Done pathway 19 : Protein processing in endoplas..</span>
<span class="co">#&gt; Done pathway 20 : Lysosome..</span>
<span class="co">#&gt; Done pathway 21 : mTOR signaling pathway..</span>
<span class="co">#&gt; Done pathway 22 : Apoptosis..</span>
<span class="co">#&gt; Done pathway 23 : Vascular smooth muscle contrac..</span>
<span class="co">#&gt; Done pathway 24 : Wnt signaling pathway..</span>
<span class="co">#&gt; Done pathway 25 : Dorso-ventral axis formation..</span>
<span class="co">#&gt; Done pathway 26 : Notch signaling pathway..</span>
<span class="co">#&gt; Done pathway 27 : Hedgehog signaling pathway..</span>
<span class="co">#&gt; Done pathway 28 : TGF-beta signaling pathway..</span>
<span class="co">#&gt; Done pathway 29 : Axon guidance..</span>
<span class="co">#&gt; Done pathway 30 : VEGF signaling pathway..</span>
<span class="co">#&gt; Done pathway 31 : Osteoclast differentiation..</span>
<span class="co">#&gt; Done pathway 32 : Focal adhesion..</span>
<span class="co">#&gt; Done pathway 33 : ECM-receptor interaction..</span>
<span class="co">#&gt; Done pathway 34 : Cell adhesion molecules (CAMs)..</span>
<span class="co">#&gt; Done pathway 35 : Adherens junction..</span>
<span class="co">#&gt; Done pathway 36 : Tight junction..</span>
<span class="co">#&gt; Done pathway 37 : Gap junction..</span>
<span class="co">#&gt; Done pathway 38 : Complement and coagulation cas..</span>
<span class="co">#&gt; Done pathway 39 : Antigen processing and present..</span>
<span class="co">#&gt; Done pathway 40 : Toll-like receptor signaling p..</span>
<span class="co">#&gt; Done pathway 41 : NOD-like receptor signaling pa..</span>
<span class="co">#&gt; Done pathway 42 : RIG-I-like receptor signaling ..</span>
<span class="co">#&gt; Done pathway 43 : Cytosolic DNA-sensing pathway..</span>
<span class="co">#&gt; Done pathway 44 : Jak-STAT signaling pathway..</span>
<span class="co">#&gt; Done pathway 45 : Natural killer cell mediated c..</span>
<span class="co">#&gt; Done pathway 46 : T cell receptor signaling path..</span>
<span class="co">#&gt; Done pathway 47 : B cell receptor signaling path..</span>
<span class="co">#&gt; Done pathway 48 : Fc epsilon RI signaling pathwa..</span>
<span class="co">#&gt; Done pathway 49 : Fc gamma R-mediated phagocytos..</span>
<span class="co">#&gt; Done pathway 50 : Leukocyte transendothelial mig..</span>
<span class="co">#&gt; Done pathway 51 : Intestinal immune network for ..</span>
<span class="co">#&gt; Done pathway 52 : Circadian rhythm - mammal..</span>
<span class="co">#&gt; Done pathway 53 : Long-term potentiation..</span>
<span class="co">#&gt; Done pathway 54 : Neurotrophin signaling pathway..</span>
<span class="co">#&gt; Done pathway 55 : Retrograde endocannabinoid sig..</span>
<span class="co">#&gt; Done pathway 56 : Glutamatergic synapse..</span>
<span class="co">#&gt; Done pathway 57 : Cholinergic synapse..</span>
<span class="co">#&gt; Done pathway 58 : Serotonergic synapse..</span>
<span class="co">#&gt; Done pathway 59 : GABAergic synapse..</span>
<span class="co">#&gt; Done pathway 60 : Dopaminergic synapse..</span>
<span class="co">#&gt; Done pathway 61 : Long-term depression..</span>
<span class="co">#&gt; Done pathway 62 : Olfactory transduction..</span>
<span class="co">#&gt; Done pathway 63 : Taste transduction..</span>
<span class="co">#&gt; Done pathway 64 : Phototransduction..</span>
<span class="co">#&gt; Done pathway 65 : Regulation of actin cytoskelet..</span>
<span class="co">#&gt; Done pathway 66 : Insulin signaling pathway..</span>
<span class="co">#&gt; Done pathway 67 : GnRH signaling pathway..</span>
<span class="co">#&gt; Done pathway 68 : Progesterone-mediated oocyte m..</span>
<span class="co">#&gt; Done pathway 69 : Melanogenesis..</span>
<span class="co">#&gt; Done pathway 70 : Adipocytokine signaling pathwa..</span>
<span class="co">#&gt; Done pathway 71 : Type II diabetes mellitus..</span>
<span class="co">#&gt; Done pathway 72 : Type I diabetes mellitus..</span>
<span class="co">#&gt; Done pathway 73 : Maturity onset diabetes of the..</span>
<span class="co">#&gt; Done pathway 74 : Aldosterone-regulated sodium r..</span>
<span class="co">#&gt; Done pathway 75 : Endocrine and other factor-reg..</span>
<span class="co">#&gt; Done pathway 76 : Vasopressin-regulated water re..</span>
<span class="co">#&gt; Done pathway 77 : Salivary secretion..</span>
<span class="co">#&gt; Done pathway 78 : Gastric acid secretion..</span>
<span class="co">#&gt; Done pathway 79 : Pancreatic secretion..</span>
<span class="co">#&gt; Done pathway 80 : Carbohydrate digestion and abs..</span>
<span class="co">#&gt; Done pathway 81 : Bile secretion..</span>
<span class="co">#&gt; Done pathway 82 : Mineral absorption..</span>
<span class="co">#&gt; Done pathway 83 : Alzheimer&#39;s disease..</span>
<span class="co">#&gt; Done pathway 84 : Parkinson&#39;s disease..</span>
<span class="co">#&gt; Done pathway 85 : Amyotrophic lateral sclerosis ..</span>
<span class="co">#&gt; Done pathway 86 : Huntington&#39;s disease..</span>
<span class="co">#&gt; Done pathway 87 : Prion diseases..</span>
<span class="co">#&gt; Done pathway 88 : Cocaine addiction..</span>
<span class="co">#&gt; Done pathway 89 : Amphetamine addiction..</span>
<span class="co">#&gt; Done pathway 90 : Morphine addiction..</span>
<span class="co">#&gt; Done pathway 91 : Alcoholism..</span>
<span class="co">#&gt; Done pathway 92 : Bacterial invasion of epitheli..</span>
<span class="co">#&gt; Done pathway 93 : Vibrio cholerae infection..</span>
<span class="co">#&gt; Done pathway 94 : Epithelial cell signaling in H..</span>
<span class="co">#&gt; Done pathway 95 : Pathogenic Escherichia coli in..</span>
<span class="co">#&gt; Done pathway 96 : Shigellosis..</span>
<span class="co">#&gt; Done pathway 97 : Salmonella infection..</span>
<span class="co">#&gt; Done pathway 98 : Pertussis..</span>
<span class="co">#&gt; Done pathway 99 : Legionellosis..</span>
<span class="co">#&gt; Done pathway 100 : Leishmaniasis..</span>
<span class="co">#&gt; Done pathway 101 : Chagas disease (American trypa..</span>
<span class="co">#&gt; Done pathway 102 : African trypanosomiasis..</span>
<span class="co">#&gt; Done pathway 103 : Malaria..</span>
<span class="co">#&gt; Done pathway 104 : Toxoplasmosis..</span>
<span class="co">#&gt; Done pathway 105 : Amoebiasis..</span>
<span class="co">#&gt; Done pathway 106 : Staphylococcus aureus infectio..</span>
<span class="co">#&gt; Done pathway 107 : Tuberculosis..</span>
<span class="co">#&gt; Done pathway 108 : Hepatitis C..</span>
<span class="co">#&gt; Done pathway 109 : Measles..</span>
<span class="co">#&gt; Done pathway 110 : Influenza A..</span>
<span class="co">#&gt; Done pathway 111 : HTLV-I infection..</span>
<span class="co">#&gt; Done pathway 112 : Herpes simplex infection..</span>
<span class="co">#&gt; Done pathway 113 : Epstein-Barr virus infection..</span>
<span class="co">#&gt; Done pathway 114 : Pathways in cancer..</span>
<span class="co">#&gt; Done pathway 115 : Transcriptional misregulation ..</span>
<span class="co">#&gt; Done pathway 116 : Viral carcinogenesis..</span>
<span class="co">#&gt; Done pathway 117 : Colorectal cancer..</span>
<span class="co">#&gt; Done pathway 118 : Renal cell carcinoma..</span>
<span class="co">#&gt; Done pathway 119 : Pancreatic cancer..</span>
<span class="co">#&gt; Done pathway 120 : Endometrial cancer..</span>
<span class="co">#&gt; Done pathway 121 : Glioma..</span>
<span class="co">#&gt; Done pathway 122 : Prostate cancer..</span>
<span class="co">#&gt; Done pathway 123 : Thyroid cancer..</span>
<span class="co">#&gt; Done pathway 124 : Basal cell carcinoma..</span>
<span class="co">#&gt; Done pathway 125 : Melanoma..</span>
<span class="co">#&gt; Done pathway 126 : Bladder cancer..</span>
<span class="co">#&gt; Done pathway 127 : Chronic myeloid leukemia..</span>
<span class="co">#&gt; Done pathway 128 : Acute myeloid leukemia..</span>
<span class="co">#&gt; Done pathway 129 : Small cell lung cancer..</span>
<span class="co">#&gt; Done pathway 130 : Non-small cell lung cancer..</span>
<span class="co">#&gt; Done pathway 131 : Asthma..</span>
<span class="co">#&gt; Done pathway 132 : Autoimmune thyroid disease..</span>
<span class="co">#&gt; Done pathway 133 : Systemic lupus erythematosus..</span>
<span class="co">#&gt; Done pathway 134 : Rheumatoid arthritis..</span>
<span class="co">#&gt; Done pathway 135 : Allograft rejection..</span>
<span class="co">#&gt; Done pathway 136 : Graft-versus-host disease..</span>
<span class="co">#&gt; Done pathway 137 : Arrhythmogenic right ventricul..</span>
<span class="co">#&gt; Done pathway 138 : Dilated cardiomyopathy..</span>
<span class="co">#&gt; Done pathway 139 : Viral myocarditis..</span>
<span class="co">#&gt; Finished SPIA analysis</span>
<span class="kw">gsRanking</span>(spia.all)
<span class="co">#&gt; DataFrame with 7 rows and 6 columns</span>
<span class="co">#&gt;                                           GENE.SET      SIZE       NDE</span>
<span class="co">#&gt;                                        &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; 1    hsa04620_Toll-like_receptor_signaling_pathway        74         6</span>
<span class="co">#&gt; 2 hsa05202_Transcriptional_misregulation_in_cancer        89         9</span>
<span class="co">#&gt; 3                       hsa05416_Viral_myocarditis        45         5</span>
<span class="co">#&gt; 4              hsa04630_Jak-STAT_signaling_pathway        72         8</span>
<span class="co">#&gt; 5               hsa04910_Insulin_signaling_pathway       115         5</span>
<span class="co">#&gt; 6                 hsa05143_African_trypanosomiasis        22         4</span>
<span class="co">#&gt; 7                      hsa04978_Mineral_absorption         3         1</span>
<span class="co">#&gt;       T.ACT    STATUS   P.VALUE</span>
<span class="co">#&gt;   &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; 1     -3.62        -1    0.0497</span>
<span class="co">#&gt; 2    -0.209        -1    0.0726</span>
<span class="co">#&gt; 3      2.98         1    0.0768</span>
<span class="co">#&gt; 4     -1.06        -1    0.0861</span>
<span class="co">#&gt; 5     -7.35        -1     0.153</span>
<span class="co">#&gt; 6     0.263         1     0.193</span>
<span class="co">#&gt; 7         0        -1     0.199</span></code></pre></div>
<p>More generally applicable is gene graph enrichment analysis (GGEA), which evaluates consistency of interactions in a given gene regulatory network with the observed expression data <a href="https://doi.org/10.1093/bioinformatics/btr228">Geistlinger et al., 2011</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">ggea.all &lt;-<span class="st"> </span><span class="kw">nbea</span>(<span class="dt">method=</span><span class="st">&quot;ggea&quot;</span>, <span class="dt">se=</span>allSE, <span class="dt">gs=</span>hsa.gs, <span class="dt">grn=</span>hsa.grn)
<span class="kw">gsRanking</span>(ggea.all)
<span class="co">#&gt; DataFrame with 9 rows and 5 columns</span>
<span class="co">#&gt;                                                          GENE.SET</span>
<span class="co">#&gt;                                                       &lt;character&gt;</span>
<span class="co">#&gt; 1                                hsa04390_Hippo_signaling_pathway</span>
<span class="co">#&gt; 2                                      hsa05416_Viral_myocarditis</span>
<span class="co">#&gt; 3                                   hsa05217_Basal_cell_carcinoma</span>
<span class="co">#&gt; 4                                      hsa04520_Adherens_junction</span>
<span class="co">#&gt; 5                             hsa04350_TGF-beta_signaling_pathway</span>
<span class="co">#&gt; 6 hsa05412_Arrhythmogenic_right_ventricular_cardiomyopathy_(ARVC)</span>
<span class="co">#&gt; 7                              hsa04910_Insulin_signaling_pathway</span>
<span class="co">#&gt; 8                  hsa04622_RIG-I-like_receptor_signaling_pathway</span>
<span class="co">#&gt; 9                                              hsa04210_Apoptosis</span>
<span class="co">#&gt;     NR.RELS RAW.SCORE NORM.SCORE   P.VALUE</span>
<span class="co">#&gt;   &lt;numeric&gt; &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; 1        62      22.8      0.367     0.002</span>
<span class="co">#&gt; 2         7       3.3      0.471     0.003</span>
<span class="co">#&gt; 3        18      6.92      0.385   0.00799</span>
<span class="co">#&gt; 4        11       4.5      0.409     0.014</span>
<span class="co">#&gt; 5        14      5.36      0.383     0.024</span>
<span class="co">#&gt; 6         4      1.75      0.437     0.032</span>
<span class="co">#&gt; 7        31      11.1      0.359     0.039</span>
<span class="co">#&gt; 8        35      12.5      0.356     0.041</span>
<span class="co">#&gt; 9        47      16.5       0.35     0.044</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">nbeaMethods</span>()
<span class="co">#&gt; [1] &quot;ggea&quot;        &quot;spia&quot;        &quot;pathnet&quot;     &quot;degraph&quot;     &quot;ganpa&quot;      </span>
<span class="co">#&gt; [6] &quot;cepa&quot;        &quot;topologygsa&quot; &quot;netgsa&quot;</span></code></pre></div>
<p>Note #1: As network-based enrichment methods typically do not involve sample permutation but rather network permutation, thus avoiding DE re-computation, they can likewise be applied to RNA-seq data.</p>
<p>Note #2: Given the various enrichment methods with individual benefits and limitations, combining multiple methods can be beneficial, e.g. combined application of a set-based and a network-based method. This has been shown to filter out spurious hits of individual methods and to reduce the outcome to gene sets accumulating evidence from different methods <a href="https://doi.org/10.1186/s12859-016-0884-1">Geistlinger et al., 2016</a>, <a href="https://doi.org/10.1093/bioinformatics/btw623">Alhamdoosh et al., 2017</a>.</p>
<p>The function <code>combResults</code> implements the straightforward combination of results, thereby facilitating seamless comparison of results across methods. For demonstration, we use the ORA and GSEA results for the ALL dataset from the previous section:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">res.list &lt;-<span class="st"> </span><span class="kw">list</span>(ora.all, gsea.all)
comb.res &lt;-<span class="st"> </span><span class="kw">combResults</span>(res.list)
<span class="kw">gsRanking</span>(comb.res)
<span class="co">#&gt; DataFrame with 20 rows and 6 columns</span>
<span class="co">#&gt;                                                            GENE.SET</span>
<span class="co">#&gt;                                                         &lt;character&gt;</span>
<span class="co">#&gt; 1   hsa05412_Arrhythmogenic_right_ventricular_cardiomyopathy_(ARVC)</span>
<span class="co">#&gt; 2                  hsa05202_Transcriptional_misregulation_in_cancer</span>
<span class="co">#&gt; 3                     hsa04670_Leukocyte_transendothelial_migration</span>
<span class="co">#&gt; 4                                        hsa04520_Adherens_junction</span>
<span class="co">#&gt; 5                                      hsa05206_MicroRNAs_in_cancer</span>
<span class="co">#&gt; ...                                                             ...</span>
<span class="co">#&gt; 16                       hsa05410_Hypertrophic_cardiomyopathy_(HCM)</span>
<span class="co">#&gt; 17                          hsa04514_Cell_adhesion_molecules_(CAMs)</span>
<span class="co">#&gt; 18                     hsa04621_NOD-like_receptor_signaling_pathway</span>
<span class="co">#&gt; 19                   hsa04622_RIG-I-like_receptor_signaling_pathway</span>
<span class="co">#&gt; 20                              hsa04350_TGF-beta_signaling_pathway</span>
<span class="co">#&gt;      ORA.RANK GSEA.RANK MEAN.RANK  ORA.PVAL GSEA.PVAL</span>
<span class="co">#&gt;     &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; 1         5.1      10.3       7.7    0.0717         0</span>
<span class="co">#&gt; 2         2.6      17.9      10.3    0.0351   0.00374</span>
<span class="co">#&gt; 3        10.3      10.3      10.3     0.122         0</span>
<span class="co">#&gt; 4        20.5      10.3      15.4     0.201         0</span>
<span class="co">#&gt; 5        23.1      25.6      24.4     0.225   0.00781</span>
<span class="co">#&gt; ...       ...       ...       ...       ...       ...</span>
<span class="co">#&gt; 16         41      48.7      44.9     0.403    0.0406</span>
<span class="co">#&gt; 17       69.2      23.1      46.2     0.649    0.0058</span>
<span class="co">#&gt; 18       61.5      33.3      47.4     0.631    0.0172</span>
<span class="co">#&gt; 19       15.4      84.6        50     0.178     0.354</span>
<span class="co">#&gt; 20       38.5      61.5        50      0.39     0.113</span></code></pre></div>
<p><em>Exercise:</em> Carry out <code>SPIA</code> and <code>GGEA</code> for the <code>airSE</code> and combine the results. How many gene sets are rendered significant by both methods?</p>
</div>
<div id="genomic-region-enrichment-analysis" class="section level2">
<h2><span class="header-section-number">9.13</span> Genomic region enrichment analysis</h2>
<p>Microarrays and next-generation sequencing are also widely applied for large-scale detection of variable and regulatory genomic regions, e.g. single nucleotide polymorphisms, copy number variations, and transcription factor binding sites.</p>
<p><img src="https://github.com/waldronlab/BrownCOBRE2018/blob/master/notebooks_day2/ENCODE.png?raw=true", alt="ENCODE image", style="width:600px"></p>
<p>Such experimentally-derived genomic region sets are raising similar questions regarding functional enrichment as in gene expression data analysis.</p>
<p>Of particular interest is thereby whether experimentally-derived regions overlap more (enrichment) or less (depletion) than expected by chance with regions representing known functional features such as genes or promoters.</p>
<p>The <a href="http://bioconductor.org/packages/regioneR">regioneR</a> package implements a general framework for testing overlaps of genomic regions based on permutation sampling. This allows to repeatedly sample random regions from the genome, matching size and chromosomal distribution of the region set under study. By recomputing the overlap with the functional features in each permutation, statistical significance of the observed overlap can be assessed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(regioneR))</code></pre></div>
<p>To demonstrate the basic functionality of the package, we consider the overlap of gene promoter regions and CpG islands in the human genome. We expect to find an enrichment as promoter regions are known to be GC-rich. Hence, is the overlap between CpG islands and promoters greater than expected by chance?</p>
<p>We use the collection of CpG islands described in <a href="https://doi.org/10.1093/biostatistics/kxq005">Wu et al., 2010</a> and restrict them to the set of canonical chromosomes 1-23, <em>X</em>, and <em>Y</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cpgHMM &lt;-<span class="st"> </span><span class="kw">toGRanges</span>(<span class="st">&quot;http://www.haowulab.org/software/makeCGI/model-based-cpg-islands-hg19.txt&quot;</span>)
cpgHMM &lt;-<span class="st"> </span><span class="kw">filterChromosomes</span>(cpgHMM, <span class="dt">chr.type=</span><span class="st">&quot;canonical&quot;</span>)
cpgHMM &lt;-<span class="st"> </span><span class="kw">sort</span>(cpgHMM)
cpgHMM
<span class="co">#&gt; GRanges object with 63705 ranges and 5 metadata columns:</span>
<span class="co">#&gt;           seqnames            ranges strand |    length  CpGcount</span>
<span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;       [1]     chr1       10497-11241      * |       745       110</span>
<span class="co">#&gt;       [2]     chr1       28705-29791      * |      1087       115</span>
<span class="co">#&gt;       [3]     chr1     135086-135805      * |       720        42</span>
<span class="co">#&gt;       [4]     chr1     136164-137362      * |      1199        71</span>
<span class="co">#&gt;       [5]     chr1     137665-138121      * |       457        22</span>
<span class="co">#&gt;       ...      ...               ...    ... .       ...       ...</span>
<span class="co">#&gt;   [63701]     chrY 59213702-59214290      * |       589        43</span>
<span class="co">#&gt;   [63702]     chrY 59240512-59241057      * |       546        40</span>
<span class="co">#&gt;   [63703]     chrY 59348047-59348370      * |       324        17</span>
<span class="co">#&gt;   [63704]     chrY 59349137-59349565      * |       429        31</span>
<span class="co">#&gt;   [63705]     chrY 59361489-59362401      * |       913       128</span>
<span class="co">#&gt;           GCcontent     pctGC    obsExp</span>
<span class="co">#&gt;           &lt;integer&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;       [1]       549     0.737     1.106</span>
<span class="co">#&gt;       [2]       792     0.729     0.818</span>
<span class="co">#&gt;       [3]       484     0.672     0.548</span>
<span class="co">#&gt;       [4]       832     0.694     0.524</span>
<span class="co">#&gt;       [5]       301     0.659     0.475</span>
<span class="co">#&gt;       ...       ...       ...       ...</span>
<span class="co">#&gt;   [63701]       366     0.621     0.765</span>
<span class="co">#&gt;   [63702]       369     0.676     0.643</span>
<span class="co">#&gt;   [63703]       193     0.596     0.593</span>
<span class="co">#&gt;   [63704]       276     0.643       0.7</span>
<span class="co">#&gt;   [63705]       650     0.712     1.108</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Analogously, we load promoter regions in the <em>hg19</em> human genome assembly as available from <a href="https://genome.ucsc.edu/">UCSC</a>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">promoters &lt;-<span class="st"> </span><span class="kw">toGRanges</span>(<span class="st">&quot;http://gattaca.imppc.org/regioner/data/UCSC.promoters.hg19.bed&quot;</span>)
promoters &lt;-<span class="st"> </span><span class="kw">filterChromosomes</span>(promoters, <span class="dt">chr.type=</span><span class="st">&quot;canonical&quot;</span>)
promoters &lt;-<span class="st"> </span><span class="kw">sort</span>(promoters)
promoters
<span class="co">#&gt; GRanges object with 49049 ranges and 3 metadata columns:</span>
<span class="co">#&gt;           seqnames            ranges strand |       V4       V5       V6</span>
<span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;</span>
<span class="co">#&gt;       [1]     chr1        9873-12073      * |      TSS        .        +</span>
<span class="co">#&gt;       [2]     chr1       16565-18765      * |      TSS        .        -</span>
<span class="co">#&gt;       [3]     chr1       17551-19751      * |      TSS        .        -</span>
<span class="co">#&gt;       [4]     chr1       17861-20061      * |      TSS        .        -</span>
<span class="co">#&gt;       [5]     chr1       19559-21759      * |      TSS        .        -</span>
<span class="co">#&gt;       ...      ...               ...    ... .      ...      ...      ...</span>
<span class="co">#&gt;   [49045]     chrY 59211948-59214148      * |      TSS        .        +</span>
<span class="co">#&gt;   [49046]     chrY 59328251-59330451      * |      TSS        .        +</span>
<span class="co">#&gt;   [49047]     chrY 59350972-59353172      * |      TSS        .        +</span>
<span class="co">#&gt;   [49048]     chrY 59352984-59355184      * |      TSS        .        +</span>
<span class="co">#&gt;   [49049]     chrY 59360654-59362854      * |      TSS        .        -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>To speed up the example, we restrict analysis to chromosomes 21 and 22. Note that this is done for demonstration only. To make an accurate claim, the complete region set should be used (which, however, runs considerably longer).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cpg &lt;-<span class="st"> </span>cpgHMM[<span class="kw">seqnames</span>(cpgHMM) <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;chr21&quot;</span>, <span class="st">&quot;chr22&quot;</span>)]
prom &lt;-<span class="st"> </span>promoters[<span class="kw">seqnames</span>(promoters) <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;chr21&quot;</span>, <span class="st">&quot;chr22&quot;</span>)]</code></pre></div>
<p>Now, we are applying an overlap permutation test with 100 permutations (<code>ntimes=100</code>), while maintaining chromosomal distribution of the CpG island region set (<code>per.chromosome=TRUE</code>). Furthermore, we use the option <code>count.once=TRUE</code> to count an overlapping CpG island only once, even if it overlaps with 2 or more promoters.</p>
<p>Note that we use 100 permutations for demonstration only. To draw robust conclusions a minimum of 1000 permutations should be carried out.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pt &lt;-<span class="st"> </span><span class="kw">overlapPermTest</span>(cpg, prom, <span class="dt">genome=</span><span class="st">&quot;hg19&quot;</span>, <span class="dt">ntimes=</span><span class="dv">100</span>, <span class="dt">per.chromosome=</span><span class="ot">TRUE</span>, <span class="dt">count.once=</span><span class="ot">TRUE</span>)
pt
<span class="co">#&gt; $numOverlaps</span>
<span class="co">#&gt; P-value: 0.0099009900990099</span>
<span class="co">#&gt; Z-score: 43.4132</span>
<span class="co">#&gt; Number of iterations: 100</span>
<span class="co">#&gt; Alternative: greater</span>
<span class="co">#&gt; Evaluation of the original region set: 719</span>
<span class="co">#&gt; Evaluation function: numOverlaps</span>
<span class="co">#&gt; Randomization function: randomizeRegions</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,&quot;class&quot;)</span>
<span class="co">#&gt; [1] &quot;permTestResultsList&quot;</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(pt[[<span class="dv">1</span>]]<span class="op">$</span>permuted)
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#&gt;   145.0   162.0   171.0   171.6   179.2   209.0</span></code></pre></div>
<p>The resulting permutation <em>p</em>-value indicates a significant enrichment. Out of the 2859 CpG islands, 719 overlap with at least one promoter. In contrast, when repeatedly drawing random regions matching the CpG islands in size and chromosomal distribution, the mean number of overlapping regions across permutations was 117.7 <span class="math inline">\(\pm\)</span> 11.8.</p>
<p>Note #1: The function <code>regioneR::permTest</code> allows to incorporate user-defined functions for randomizing regions and evaluating additional measures of overlap such as total genomic size in bp.</p>
<p>Note #2: The <a href="http://bioconductor.org/packages/LOLA">LOLA</a> package implements a genomic region ORA, which assesses genomic region overlap based on the hypergeometric distribution using a library of pre-defined functional region sets.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
