<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7.13 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="workshop.html">
<link rel="next" href="solving-common-bioinformatic-challenges-using-genomicranges.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-everyone"><i class="fa fa-check"></i><b>1.1</b> For Everyone</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#for-workshop-authors"><i class="fa fa-check"></i><b>1.2</b> For Workshop Authors</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#description"><i class="fa fa-check"></i><b>1.2.1</b> DESCRIPTION</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#classifying-your-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Classifying your workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#bookdown"><i class="fa fa-check"></i><b>1.2.3</b> Bookdown</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#deadlines-for-bioc2018"><i class="fa fa-check"></i><b>1.3</b> Deadlines for Bioc2018</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>2</b> RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-1"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>2.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>2.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="2.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>2.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="2.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>2.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="2.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>2.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>2.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>2.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="2.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>2.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="2.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>2.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>2.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="2.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>2.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="2.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>2.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="2.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>2.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="2.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>2.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="2.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>2.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="2.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>2.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="2.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub"><i class="fa fa-check"></i><b>2.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="2.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub"><i class="fa fa-check"></i><b>2.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="2.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>2.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="2.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>2.7</b> Building reports</a><ul>
<li class="chapter" data-level="2.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>2.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="2.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>2.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="2.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>2.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="2.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>2.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="2.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>2.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="2.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>2.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="2.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>2.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="2.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>2.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="2.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>2.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>3</b> Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="3.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a><ul>
<li class="chapter" data-level="3.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-2"><i class="fa fa-check"></i><b>3.1.1</b> Description</a></li>
<li class="chapter" data-level="3.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-1"><i class="fa fa-check"></i><b>3.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="3.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-1"><i class="fa fa-check"></i><b>3.1.3</b> Participation</a></li>
<li class="chapter" data-level="3.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="3.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-1"><i class="fa fa-check"></i><b>3.1.5</b> Time outline</a></li>
<li class="chapter" data-level="3.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started"><i class="fa fa-check"></i><b>3.2</b> Getting started</a><ul>
<li class="chapter" data-level="3.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>3.2.1</b> The data</a></li>
<li class="chapter" data-level="3.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>3.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>3.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="3.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>3.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="3.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>3.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>3.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="3.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>3.5.1</b> Normalization</a></li>
<li class="chapter" data-level="3.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>3.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>3.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="3.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>3.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="3.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>3.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><i class="fa fa-check"></i><b>4</b> Variant Functional Annotation using StatePaintR, FunciVar &amp; MotifBreakR</a><ul>
<li class="chapter" data-level="4.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>4.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-description"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#pre-requisites-2"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-participation"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#time-outline-2"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="4.3.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#key-concepts"><i class="fa fa-check"></i><b>4.3.1</b> Key Concepts</a></li>
<li class="chapter" data-level="4.3.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#learning-objectives"><i class="fa fa-check"></i><b>4.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>5</b> Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="5.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>5.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-1"><i class="fa fa-check"></i><b>5.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="5.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-3"><i class="fa fa-check"></i><b>5.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="5.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-1"><i class="fa fa-check"></i><b>5.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>5.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="5.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-3"><i class="fa fa-check"></i><b>5.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>5.3</b> Goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="workshop.html"><a href="workshop.html"><i class="fa fa-check"></i><b>6</b> Workshop</a><ul>
<li class="chapter" data-level="6.1" data-path="workshop.html"><a href="workshop.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>6.1</b> Where does it all come from?</a></li>
<li class="chapter" data-level="6.2" data-path="workshop.html"><a href="workshop.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>6.2</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="6.3" data-path="workshop.html"><a href="workshop.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>6.3</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="6.4" data-path="workshop.html"><a href="workshop.html#data-types"><i class="fa fa-check"></i><b>6.4</b> Data types</a></li>
<li class="chapter" data-level="6.5" data-path="workshop.html"><a href="workshop.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>6.5</b> Differential expression analysis</a></li>
<li class="chapter" data-level="6.6" data-path="workshop.html"><a href="workshop.html#gene-sets"><i class="fa fa-check"></i><b>6.6</b> Gene sets</a></li>
<li class="chapter" data-level="6.7" data-path="workshop.html"><a href="workshop.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>6.7</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="6.8" data-path="workshop.html"><a href="workshop.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>6.8</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="6.9" data-path="workshop.html"><a href="workshop.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>6.9</b> Network-based enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="genomic-region-enrichment-analysis.html"><a href="genomic-region-enrichment-analysis.html"><i class="fa fa-check"></i><b>7</b> Genomic region enrichment analysis</a></li>
<li class="chapter" data-level="8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>8</b> Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="8.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>8.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="8.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-2"><i class="fa fa-check"></i><b>8.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="8.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-4"><i class="fa fa-check"></i><b>8.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-2"><i class="fa fa-check"></i><b>8.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="8.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>8.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-4"><i class="fa fa-check"></i><b>8.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>8.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="8.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals"><i class="fa fa-check"></i><b>8.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="8.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-1"><i class="fa fa-check"></i><b>8.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="granges-genomic-ranges.html"><a href="granges-genomic-ranges.html"><i class="fa fa-check"></i><b>9</b> <em>GRanges</em>: Genomic Ranges</a><ul>
<li class="chapter" data-level="9.1" data-path="granges-genomic-ranges.html"><a href="granges-genomic-ranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>9.1</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="9.2" data-path="granges-genomic-ranges.html"><a href="granges-genomic-ranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>9.2</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="9.3" data-path="granges-genomic-ranges.html"><a href="granges-genomic-ranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>9.3</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="9.4" data-path="granges-genomic-ranges.html"><a href="granges-genomic-ranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>9.4</b> Interval set operations for <em>GRanges</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html"><i class="fa fa-check"></i><b>10</b> <em>GRangesList</em>: Groups of Genomic Ranges</a><ul>
<li class="chapter" data-level="10.1" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html#basic-grangeslist-accessors"><i class="fa fa-check"></i><b>10.1</b> Basic <em>GRangesList</em> accessors</a></li>
<li class="chapter" data-level="10.2" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html#combining-grangeslist-objects"><i class="fa fa-check"></i><b>10.2</b> Combining <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.3" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html#basic-interval-operations-for-grangeslist-objects"><i class="fa fa-check"></i><b>10.3</b> Basic interval operations for <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.4" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html#subsetting-grangeslist-objects"><i class="fa fa-check"></i><b>10.4</b> Subsetting <em>GRangesList</em> objects</a></li>
<li class="chapter" data-level="10.5" data-path="grangeslist-groups-of-genomic-ranges.html"><a href="grangeslist-groups-of-genomic-ranges.html#looping-over-grangeslist-objects"><i class="fa fa-check"></i><b>10.5</b> Looping over <em>GRangesList</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>11</b> Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="11.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>11.1</b> Instructors</a></li>
<li class="chapter" data-level="11.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description-3"><i class="fa fa-check"></i><b>11.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="11.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>11.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="11.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation-3"><i class="fa fa-check"></i><b>11.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>11.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>11.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="11.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-1"><i class="fa fa-check"></i><b>11.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="11.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-2"><i class="fa fa-check"></i><b>11.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>11.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="11.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>11.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="11.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>11.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="11.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>11.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="11.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>11.4.4</b> Data containers</a></li>
<li class="chapter" data-level="11.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>11.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="11.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>11.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="11.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>11.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="11.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>11.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="11.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>11.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="11.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>11.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="11.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>11.4.11</b> Simple example</a></li>
<li class="chapter" data-level="11.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>11.4.12</b> Questions!</a></li>
<li class="chapter" data-level="11.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>11.4.13</b> More questions!</a></li>
<li class="chapter" data-level="11.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>11.4.14</b> Another example</a></li>
<li class="chapter" data-level="11.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>11.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="11.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>11.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="11.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>11.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="11.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>11.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="11.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>11.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="11.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>11.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="11.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>11.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="11.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>11.4.22</b> GRanges</a></li>
<li class="chapter" data-level="11.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>11.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="11.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>11.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="11.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>11.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="11.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>11.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="11.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>11.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="11.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>11.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="11.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>11.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="11.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>11.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="11.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>11.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="11.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>11.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="11.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dply-example"><i class="fa fa-check"></i><b>11.4.33</b> Organism.dply example</a></li>
<li class="chapter" data-level="11.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>11.4.34</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="11.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>11.4.35</b> BSgenome packages</a></li>
<li class="chapter" data-level="11.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>11.4.36</b> BSgenome packages</a></li>
<li class="chapter" data-level="11.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>11.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="11.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>11.4.38</b> BSgenome exercises</a></li>
<li class="chapter" data-level="11.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-1"><i class="fa fa-check"></i><b>11.4.39</b> AnnotationHub</a></li>
<li class="chapter" data-level="11.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>11.4.40</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="11.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>11.4.41</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="11.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>11.4.42</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="11.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>11.4.43</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="11.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>11.4.44</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="11.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>11.4.45</b> AnnotationHub query</a></li>
<li class="chapter" data-level="11.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>11.4.46</b> AnnotationHub query</a></li>
<li class="chapter" data-level="11.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>11.4.47</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="11.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>11.4.48</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="11.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>11.4.49</b> biomaRt</a></li>
<li class="chapter" data-level="11.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>11.4.50</b> biomaRt data sets</a></li>
<li class="chapter" data-level="11.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>11.4.51</b> biomaRt queries</a></li>
<li class="chapter" data-level="11.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>11.4.52</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="11.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>11.4.53</b> biomaRt query</a></li>
<li class="chapter" data-level="11.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>11.4.54</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>12</b> Workflow for Multi-omics Analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="12.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>12.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="12.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-4"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-5"><i class="fa fa-check"></i><b>12.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-4"><i class="fa fa-check"></i><b>12.2.2</b> workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>12.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="12.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-5"><i class="fa fa-check"></i><b>12.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-2"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-3"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>12.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="12.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summarizedexperiment-raggedexperiment"><i class="fa fa-check"></i><b>12.4.1</b> <code>SummarizedExperiment</code> / <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="12.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>12.4.2</b> <code>MultiAssayExperiment</code></a></li>
<li class="chapter" data-level="12.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>12.4.3</b> <code>RaggedExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>12.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="12.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>12.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="12.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#using-granges-objects"><i class="fa fa-check"></i><b>12.5.2</b> Using <code>GRanges</code> objects</a></li>
<li class="chapter" data-level="12.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>12.5.3</b> *Assay functions</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#qreduceassay"><i class="fa fa-check"></i><b>12.6</b> qreduceAssay</a></li>
<li class="chapter" data-level="12.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>12.7</b> Working with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="12.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>12.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="12.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>12.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="12.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>12.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="12.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>12.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="12.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>12.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="12.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>12.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="12.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>12.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="12.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>12.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="12.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>12.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#patients-with-complete-data"><i class="fa fa-check"></i><b>12.9</b> Patients with complete data</a></li>
<li class="chapter" data-level="12.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>12.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="12.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>12.11</b> Extraction</a><ul>
<li class="chapter" data-level="12.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>12.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="12.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>12.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="12.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>12.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="12.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>12.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="12.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>12.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="12.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>12.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="12.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>12.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="12.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>12.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="12.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>12.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="12.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>12.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="12.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiments"><i class="fa fa-check"></i><b>12.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiments</a></li>
<li class="chapter" data-level="12.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>12.16</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="12.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>12.16.1</b> How many samples have data for each combination of assays?</a></li>
</ul></li>
<li class="chapter" data-level="12.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#tcgautils-package"><i class="fa fa-check"></i><b>12.17</b> <code>TCGAutils</code> package</a><ul>
<li class="chapter" data-level="12.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>12.17.1</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="12.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#is-there-subtype-data-available-in-the-multiassayexperiment-obtained-from"><i class="fa fa-check"></i><b>12.17.2</b> Is there subtype data available in the <code>MultiAssayExperiment</code> obtained from</a></li>
<li class="chapter" data-level="12.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>12.17.3</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="12.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>12.17.4</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="12.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>12.17.5</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="12.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>12.17.6</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="12.17.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>12.17.7</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>13</b> Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="13.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#authors"><i class="fa fa-check"></i><b>13.1</b> Authors:</a></li>
<li class="chapter" data-level="13.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-5"><i class="fa fa-check"></i><b>13.2</b> Workshop Description:</a></li>
<li class="chapter" data-level="13.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-6"><i class="fa fa-check"></i><b>13.3</b> Pre-requisites:</a></li>
<li class="chapter" data-level="13.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-2"><i class="fa fa-check"></i><b>13.4</b> Participation:</a></li>
<li class="chapter" data-level="13.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>13.5</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="13.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>13.6</b> Workshop goals and objectives:</a></li>
</ul></li>
<li class="chapter" data-level="14" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>14</b> Public Data Resources and Bioconductor</a><ul>
<li class="chapter" data-level="14.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-6"><i class="fa fa-check"></i><b>14.1</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.1.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-7"><i class="fa fa-check"></i><b>14.1.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.1.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-5"><i class="fa fa-check"></i><b>14.1.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.1.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>14.1.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="14.1.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-6"><i class="fa fa-check"></i><b>14.1.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>14.2</b> workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-3"><i class="fa fa-check"></i><b>14.2.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-4"><i class="fa fa-check"></i><b>14.2.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-2"><i class="fa fa-check"></i><b>14.3</b> Overview</a></li>
<li class="chapter" data-level="14.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>14.4</b> GEOquery</a></li>
<li class="chapter" data-level="14.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>14.5</b> GenomicDataCommons</a></li>
<li class="chapter" data-level="14.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>14.6</b> Sequence Read Archive</a></li>
<li class="chapter" data-level="14.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>14.7</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="14.7.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata"><i class="fa fa-check"></i><b>14.7.1</b> curatedTCGAData</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#gbm-dataset-example"><i class="fa fa-check"></i><b>14.8</b> GBM dataset example</a><ul>
<li class="chapter" data-level="14.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#subtype-information"><i class="fa fa-check"></i><b>14.8.1</b> Subtype information</a></li>
<li class="chapter" data-level="14.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>14.8.2</b> TCGABiolinks</a></li>
<li class="chapter" data-level="14.8.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons-1"><i class="fa fa-check"></i><b>14.8.3</b> GenomicDataCommons</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>14.9</b> Microbiome data</a></li>
<li class="chapter" data-level="14.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>14.10</b> Pharmacogenomics</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="genomic-region-enrichment-analysis" class="section level1">
<h1><span class="header-section-number">7</span> Genomic region enrichment analysis</h1>
<p>Microarrays and next-generation sequencing are also widely applied for large-scale detection of variable and regulatory genomic regions, e.g. single nucleotide polymorphisms, copy number variations, and transcription factor binding sites.</p>
<p><img src="https://github.com/waldronlab/BrownCOBRE2018/blob/master/notebooks_day2/ENCODE.png?raw=true", alt="ENCODE image", style="width:600px"></p>
<p>Such experimentally-derived genomic region sets are raising similar questions regarding functional enrichment as in gene expression data analysis.</p>
<p>Of particular interest is thereby whether experimentally-derived regions overlap more (enrichment) or less (depletion) than expected by chance with regions representing known functional features such as genes or promoters.</p>
<p>The <a href="http://bioconductor.org/packages/regioneR">regioneR</a> package implements a general framework for testing overlaps of genomic regions based on permutation sampling. This allows to repeatedly sample random regions from the genome, matching size and chromosomal distribution of the region set under study. By recomputing the overlap with the functional features in each permutation, statistical significance of the observed overlap can be assessed.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>(<span class="kw">library</span>(regioneR))</code></pre></div>
<p>To demonstrate the basic functionality of the package, we consider the overlap of gene promoter regions and CpG islands in the human genome. We expect to find an enrichment as promoter regions are known to be GC-rich. Hence, is the overlap between CpG islands and promoters greater than expected by chance?</p>
<p>We use the collection of CpG islands described in <a href="https://doi.org/10.1093/biostatistics/kxq005">Wu et al., 2010</a> and restrict them to the set of canonical chromosomes 1-23, <em>X</em>, and <em>Y</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cpgHMM &lt;-<span class="st"> </span><span class="kw">toGRanges</span>(<span class="st">&quot;http://www.haowulab.org/software/makeCGI/model-based-cpg-islands-hg19.txt&quot;</span>)
cpgHMM &lt;-<span class="st"> </span><span class="kw">filterChromosomes</span>(cpgHMM, <span class="dt">chr.type=</span><span class="st">&quot;canonical&quot;</span>)
cpgHMM &lt;-<span class="st"> </span><span class="kw">sort</span>(cpgHMM)
cpgHMM
<span class="co">#&gt; GRanges object with 63705 ranges and 5 metadata columns:</span>
<span class="co">#&gt;           seqnames            ranges strand |    length  CpGcount</span>
<span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;       [1]     chr1       10497-11241      * |       745       110</span>
<span class="co">#&gt;       [2]     chr1       28705-29791      * |      1087       115</span>
<span class="co">#&gt;       [3]     chr1     135086-135805      * |       720        42</span>
<span class="co">#&gt;       [4]     chr1     136164-137362      * |      1199        71</span>
<span class="co">#&gt;       [5]     chr1     137665-138121      * |       457        22</span>
<span class="co">#&gt;       ...      ...               ...    ... .       ...       ...</span>
<span class="co">#&gt;   [63701]     chrY 59213702-59214290      * |       589        43</span>
<span class="co">#&gt;   [63702]     chrY 59240512-59241057      * |       546        40</span>
<span class="co">#&gt;   [63703]     chrY 59348047-59348370      * |       324        17</span>
<span class="co">#&gt;   [63704]     chrY 59349137-59349565      * |       429        31</span>
<span class="co">#&gt;   [63705]     chrY 59361489-59362401      * |       913       128</span>
<span class="co">#&gt;           GCcontent     pctGC    obsExp</span>
<span class="co">#&gt;           &lt;integer&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="co">#&gt;       [1]       549     0.737     1.106</span>
<span class="co">#&gt;       [2]       792     0.729     0.818</span>
<span class="co">#&gt;       [3]       484     0.672     0.548</span>
<span class="co">#&gt;       [4]       832     0.694     0.524</span>
<span class="co">#&gt;       [5]       301     0.659     0.475</span>
<span class="co">#&gt;       ...       ...       ...       ...</span>
<span class="co">#&gt;   [63701]       366     0.621     0.765</span>
<span class="co">#&gt;   [63702]       369     0.676     0.643</span>
<span class="co">#&gt;   [63703]       193     0.596     0.593</span>
<span class="co">#&gt;   [63704]       276     0.643       0.7</span>
<span class="co">#&gt;   [63705]       650     0.712     1.108</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Analogously, we load promoter regions in the <em>hg19</em> human genome assembly as available from <a href="https://genome.ucsc.edu/">UCSC</a>:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">promoters &lt;-<span class="st"> </span><span class="kw">toGRanges</span>(<span class="st">&quot;http://gattaca.imppc.org/regioner/data/UCSC.promoters.hg19.bed&quot;</span>)
promoters &lt;-<span class="st"> </span><span class="kw">filterChromosomes</span>(promoters, <span class="dt">chr.type=</span><span class="st">&quot;canonical&quot;</span>)
promoters &lt;-<span class="st"> </span><span class="kw">sort</span>(promoters)
promoters
<span class="co">#&gt; GRanges object with 49049 ranges and 3 metadata columns:</span>
<span class="co">#&gt;           seqnames            ranges strand |       V4       V5       V6</span>
<span class="co">#&gt;              &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;</span>
<span class="co">#&gt;       [1]     chr1        9873-12073      * |      TSS        .        +</span>
<span class="co">#&gt;       [2]     chr1       16565-18765      * |      TSS        .        -</span>
<span class="co">#&gt;       [3]     chr1       17551-19751      * |      TSS        .        -</span>
<span class="co">#&gt;       [4]     chr1       17861-20061      * |      TSS        .        -</span>
<span class="co">#&gt;       [5]     chr1       19559-21759      * |      TSS        .        -</span>
<span class="co">#&gt;       ...      ...               ...    ... .      ...      ...      ...</span>
<span class="co">#&gt;   [49045]     chrY 59211948-59214148      * |      TSS        .        +</span>
<span class="co">#&gt;   [49046]     chrY 59328251-59330451      * |      TSS        .        +</span>
<span class="co">#&gt;   [49047]     chrY 59350972-59353172      * |      TSS        .        +</span>
<span class="co">#&gt;   [49048]     chrY 59352984-59355184      * |      TSS        .        +</span>
<span class="co">#&gt;   [49049]     chrY 59360654-59362854      * |      TSS        .        -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>To speed up the example, we restrict analysis to chromosomes 21 and 22. Note that this is done for demonstration only. To make an accurate claim, the complete region set should be used (which, however, runs considerably longer).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cpg &lt;-<span class="st"> </span>cpgHMM[<span class="kw">seqnames</span>(cpgHMM) <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;chr21&quot;</span>, <span class="st">&quot;chr22&quot;</span>)]
prom &lt;-<span class="st"> </span>promoters[<span class="kw">seqnames</span>(promoters) <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;chr21&quot;</span>, <span class="st">&quot;chr22&quot;</span>)]</code></pre></div>
<p>Now, we are applying an overlap permutation test with 100 permutations (<code>ntimes=100</code>), while maintaining chromosomal distribution of the CpG island region set (<code>per.chromosome=TRUE</code>). Furthermore, we use the option <code>count.once=TRUE</code> to count an overlapping CpG island only once, even if it overlaps with 2 or more promoters.</p>
<p>Note that we use 100 permutations for demonstration only. To draw robust conclusions a minimum of 1000 permutations should be carried out.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pt &lt;-<span class="st"> </span><span class="kw">overlapPermTest</span>(cpg, prom, <span class="dt">genome=</span><span class="st">&quot;hg19&quot;</span>, <span class="dt">ntimes=</span><span class="dv">100</span>, <span class="dt">per.chromosome=</span><span class="ot">TRUE</span>, <span class="dt">count.once=</span><span class="ot">TRUE</span>)
pt
<span class="co">#&gt; $numOverlaps</span>
<span class="co">#&gt; P-value: 0.0099009900990099</span>
<span class="co">#&gt; Z-score: 43.3181</span>
<span class="co">#&gt; Number of iterations: 100</span>
<span class="co">#&gt; Alternative: greater</span>
<span class="co">#&gt; Evaluation of the original region set: 719</span>
<span class="co">#&gt; Evaluation function: numOverlaps</span>
<span class="co">#&gt; Randomization function: randomizeRegions</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,&quot;class&quot;)</span>
<span class="co">#&gt; [1] &quot;permTestResultsList&quot;</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(pt[[<span class="dv">1</span>]]<span class="op">$</span>permuted)
<span class="co">#&gt;    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span>
<span class="co">#&gt;   139.0   159.0   168.0   168.9   178.2   202.0</span></code></pre></div>
<p>The resulting permutation <em>p</em>-value indicates a significant enrichment. Out of the 2859 CpG islands, 719 overlap with at least one promoter. In contrast, when repeatedly drawing random regions matching the CpG islands in size and chromosomal distribution, the mean number of overlapping regions across permutations was 117.7 <span class="math inline">\(\pm\)</span> 11.8.</p>
<p>Note #1: The function <code>regioneR::permTest</code> allows to incorporate user-defined functions for randomizing regions and evaluating additional measures of overlap such as total genomic size in bp.</p>
<p>Note #2: The <a href="http://bioconductor.org/packages/LOLA">LOLA</a> package implements a genomic region ORA, which assesses genomic region overlap based on the hypergeometric distribution using a library of pre-defined functional region sets.</p>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="workshop.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="solving-common-bioinformatic-challenges-using-genomicranges.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
