<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="index.html">
<link rel="next" href="introduction-to-bioconductor-annotation-resources.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-workshops"><i class="fa fa-check"></i><b>1.1</b> The Workshops</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html"><i class="fa fa-check"></i><b>2</b> 100: <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#description"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="2.1.7" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-goals"><i class="fa fa-check"></i><b>2.1.7</b> Learning goals</a></li>
<li class="chapter" data-level="2.1.8" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-objectives"><i class="fa fa-check"></i><b>2.1.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-r"><i class="fa fa-check"></i><b>2.2</b> Introduction to <em>R</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#rstudio-orientation"><i class="fa fa-check"></i><b>2.2.1</b> <em>RStudio</em> orientation</a></li>
<li class="chapter" data-level="2.2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#very-basics"><i class="fa fa-check"></i><b>2.2.2</b> Very basics</a></li>
<li class="chapter" data-level="2.2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#data-input-and-manipulation"><i class="fa fa-check"></i><b>2.2.3</b> Data input and manipulation</a></li>
<li class="chapter" data-level="2.2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#writing-r-scripts"><i class="fa fa-check"></i><b>2.2.4</b> Writing <em>R</em> scripts</a></li>
<li class="chapter" data-level="2.2.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#using-packages"><i class="fa fa-check"></i><b>2.2.5</b> Using packages</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.3</b> Introduction to <em>Bioconductor</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#discovering-installing-and-learning-how-to-use-bioconductor-packages"><i class="fa fa-check"></i><b>2.3.1</b> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-genomic-ranges"><i class="fa fa-check"></i><b>2.3.2</b> Working with Genomic Ranges</a></li>
<li class="chapter" data-level="2.3.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-summarized-experimental-data"><i class="fa fa-check"></i><b>2.3.3</b> Working with summarized experimental data</a></li>
<li class="chapter" data-level="2.3.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#down-stream-analysis"><i class="fa fa-check"></i><b>2.3.4</b> Down-stream analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>3</b> 101: Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>3.1</b> Instructors</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description"><i class="fa fa-check"></i><b>3.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="3.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>3.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="3.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation"><i class="fa fa-check"></i><b>3.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="3.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="3.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-1"><i class="fa fa-check"></i><b>3.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="3.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-1"><i class="fa fa-check"></i><b>3.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>3.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="3.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>3.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="3.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>3.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="3.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>3.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="3.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>3.4.4</b> Data containers</a></li>
<li class="chapter" data-level="3.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>3.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="3.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>3.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>3.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>3.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="3.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>3.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="3.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>3.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="3.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>3.4.11</b> Simple example</a></li>
<li class="chapter" data-level="3.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>3.4.12</b> Questions!</a></li>
<li class="chapter" data-level="3.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>3.4.13</b> More questions!</a></li>
<li class="chapter" data-level="3.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>3.4.14</b> Another example</a></li>
<li class="chapter" data-level="3.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>3.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="3.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>3.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="3.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>3.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="3.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>3.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="3.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>3.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="3.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>3.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="3.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>3.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="3.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>3.4.22</b> GRanges</a></li>
<li class="chapter" data-level="3.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>3.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="3.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>3.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="3.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>3.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="3.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>3.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="3.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>3.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="3.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>3.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="3.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>3.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>3.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>3.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="3.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>3.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="3.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr"><i class="fa fa-check"></i><b>3.4.33</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-1"><i class="fa fa-check"></i><b>3.4.34</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-2"><i class="fa fa-check"></i><b>3.4.35</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>3.4.36</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="3.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>3.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>3.4.38</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>3.4.39</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>3.4.40</b> BSgenome exercises</a></li>
<li class="chapter" data-level="3.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub"><i class="fa fa-check"></i><b>3.4.41</b> AnnotationHub</a></li>
<li class="chapter" data-level="3.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub"><i class="fa fa-check"></i><b>3.4.42</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="3.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>3.4.43</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="3.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>3.4.44</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="3.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>3.4.45</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="3.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>3.4.46</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="3.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>3.4.47</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>3.4.48</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>3.4.49</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="3.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>3.4.50</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="3.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>3.4.51</b> biomaRt</a></li>
<li class="chapter" data-level="3.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>3.4.52</b> biomaRt data sets</a></li>
<li class="chapter" data-level="3.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>3.4.53</b> biomaRt queries</a></li>
<li class="chapter" data-level="3.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>3.4.54</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="3.4.55" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>3.4.55</b> biomaRt query</a></li>
<li class="chapter" data-level="3.4.56" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>3.4.56</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>4</b> 102: Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>4.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-1"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-1"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-1"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-1"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="4.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-2"><i class="fa fa-check"></i><b>4.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="4.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-2"><i class="fa fa-check"></i><b>4.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#introduction-1"><i class="fa fa-check"></i><b>4.4</b> Introduction</a><ul>
<li class="chapter" data-level="4.4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#what-is-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.1</b> What is the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#why-use-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.2</b> Why use the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>4.4.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#setup"><i class="fa fa-check"></i><b>4.5</b> Setup</a></li>
<li class="chapter" data-level="4.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>4.6</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="4.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#constructing-a-granges-object-from-data.frame"><i class="fa fa-check"></i><b>4.7</b> Constructing a <em>GRanges</em> object from data.frame</a></li>
<li class="chapter" data-level="4.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#loading-a-granges-object-from-a-standard-file-format"><i class="fa fa-check"></i><b>4.8</b> Loading a <em>GRanges</em> object from a standard file format</a></li>
<li class="chapter" data-level="4.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-manipulation-of-granges-objects"><i class="fa fa-check"></i><b>4.9</b> Basic manipulation of <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>4.10</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>4.11</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#aggregating-granges-objects"><i class="fa fa-check"></i><b>4.12</b> Aggregating <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.13</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.14</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.15" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#finding-overlaps-between-granges-objects"><i class="fa fa-check"></i><b>4.15</b> Finding overlaps between <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.16" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises"><i class="fa fa-check"></i><b>4.16</b> Exercises</a></li>
<li class="chapter" data-level="4.17" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>4.17</b> Example: exploring BigWig files from AnnotationHub</a><ul>
<li class="chapter" data-level="4.17.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#extracting-data-from-annotationhub"><i class="fa fa-check"></i><b>4.17.1</b> Extracting data from AnnotationHub</a></li>
<li class="chapter" data-level="4.17.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#reading-bigwig-files"><i class="fa fa-check"></i><b>4.17.2</b> Reading BigWig files</a></li>
<li class="chapter" data-level="4.17.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-1"><i class="fa fa-check"></i><b>4.17.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.18" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>4.18</b> Worked example: coverage analysis of BAM files</a><ul>
<li class="chapter" data-level="4.18.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-2"><i class="fa fa-check"></i><b>4.18.1</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.19" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#conclusions"><i class="fa fa-check"></i><b>4.19</b> Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>5</b> 103: Public data resources and Bioconductor</a><ul>
<li class="chapter" data-level="5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>5.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>5.2</b> Syllabus</a><ul>
<li class="chapter" data-level="5.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-2"><i class="fa fa-check"></i><b>5.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="5.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-2"><i class="fa fa-check"></i><b>5.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="5.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-2"><i class="fa fa-check"></i><b>5.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>5.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="5.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-2"><i class="fa fa-check"></i><b>5.2.5</b> Time outline</a></li>
<li class="chapter" data-level="5.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>5.2.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="5.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-3"><i class="fa fa-check"></i><b>5.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="5.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-3"><i class="fa fa-check"></i><b>5.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-1"><i class="fa fa-check"></i><b>5.3</b> Overview</a></li>
<li class="chapter" data-level="5.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>5.4</b> GEOquery</a><ul>
<li class="chapter" data-level="5.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>5.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="5.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>5.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="5.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>5.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="5.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>5.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>5.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="5.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>5.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="5.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>5.5.2</b> Basic design</a></li>
<li class="chapter" data-level="5.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>5.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>5.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="5.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>5.6.1</b> Authentication</a></li>
<li class="chapter" data-level="5.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>5.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>5.7</b> Sequence Read Archive</a><ul>
<li class="chapter" data-level="5.7.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage-1"><i class="fa fa-check"></i><b>5.7.1</b> Usage</a></li>
<li class="chapter" data-level="5.7.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#query-syntax"><i class="fa fa-check"></i><b>5.7.2</b> Query syntax</a></li>
<li class="chapter" data-level="5.7.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#using-the-raw-api-without-rbioconductor"><i class="fa fa-check"></i><b>5.7.3</b> Using the raw API without R/Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>5.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="5.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>5.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="5.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>5.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="5.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>5.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="5.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>5.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="5.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>5.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="5.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>5.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="5.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>5.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>5.12</b> Pharmacogenomics</a><ul>
<li class="chapter" data-level="5.12.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started"><i class="fa fa-check"></i><b>5.12.1</b> Getting started</a></li>
<li class="chapter" data-level="5.12.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-pharmacogx-datasets-pharmacosets"><i class="fa fa-check"></i><b>5.12.2</b> Overview of PharmacoGx datasets (PharmacoSets)</a></li>
<li class="chapter" data-level="5.12.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#exploring-drug-sensitivity-datasets"><i class="fa fa-check"></i><b>5.12.3</b> Exploring drug sensitivity datasets</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>5.13</b> Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>6</b> 200: RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>6.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>6.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="6.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>6.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="6.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>6.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-3"><i class="fa fa-check"></i><b>6.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>6.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="6.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-4"><i class="fa fa-check"></i><b>6.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="6.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-4"><i class="fa fa-check"></i><b>6.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-2"><i class="fa fa-check"></i><b>6.4</b> Introduction</a></li>
<li class="chapter" data-level="6.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>6.5</b> Data packaging</a><ul>
<li class="chapter" data-level="6.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>6.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="6.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>6.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="6.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>6.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>6.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="6.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>6.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="6.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>6.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="6.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>6.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="6.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>6.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis"><i class="fa fa-check"></i><b>6.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="6.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>6.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="6.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>6.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="6.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>6.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="6.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>6.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="6.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>6.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="6.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>6.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>7</b> 201: RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview-2"><i class="fa fa-check"></i><b>7.1</b> Overview</a><ul>
<li class="chapter" data-level="7.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-1"><i class="fa fa-check"></i><b>7.1.1</b> Description</a></li>
<li class="chapter" data-level="7.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation-1"><i class="fa fa-check"></i><b>7.1.3</b> Participation</a></li>
<li class="chapter" data-level="7.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>7.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-4"><i class="fa fa-check"></i><b>7.1.5</b> Time outline</a></li>
<li class="chapter" data-level="7.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>7.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>7.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="7.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>7.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="7.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>7.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="7.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>7.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="7.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>7.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>7.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="7.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>7.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="7.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>7.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="7.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>7.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>7.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="7.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>7.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="7.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>7.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="7.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>7.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>7.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="7.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>7.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="7.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>7.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub-1"><i class="fa fa-check"></i><b>7.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="7.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>7.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="7.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>7.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>7.7</b> Building reports</a><ul>
<li class="chapter" data-level="7.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>7.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="7.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>7.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>7.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="7.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>7.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>7.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>7.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="7.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>7.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="7.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>7.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="7.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>7.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>8</b> 202: Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="8.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-3"><i class="fa fa-check"></i><b>8.1</b> Overview</a><ul>
<li class="chapter" data-level="8.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-2"><i class="fa fa-check"></i><b>8.1.1</b> Description</a></li>
<li class="chapter" data-level="8.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-2"><i class="fa fa-check"></i><b>8.1.3</b> Participation</a></li>
<li class="chapter" data-level="8.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>8.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-5"><i class="fa fa-check"></i><b>8.1.5</b> Time outline</a></li>
<li class="chapter" data-level="8.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>8.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started-1"><i class="fa fa-check"></i><b>8.2</b> Getting started</a><ul>
<li class="chapter" data-level="8.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>8.2.1</b> The data</a></li>
<li class="chapter" data-level="8.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>8.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>8.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="8.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>8.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="8.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>8.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>8.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="8.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>8.5.1</b> Normalization</a></li>
<li class="chapter" data-level="8.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>8.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>8.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="8.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>8.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="8.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>8.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>9</b> 210: Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="9.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>9.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-4"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-4"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-6"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>9.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="9.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>9.4</b> Where does it all come from?</a></li>
<li class="chapter" data-level="9.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.5</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="9.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>9.6</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="9.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>9.7</b> Data types</a></li>
<li class="chapter" data-level="9.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>9.8</b> Differential expression analysis</a></li>
<li class="chapter" data-level="9.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>9.9</b> Gene sets</a></li>
<li class="chapter" data-level="9.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>9.10</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="9.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>9.11</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="9.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.12</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="9.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>9.13</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>10</b> 220: Workflow for multi-omics analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="10.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information-2"><i class="fa fa-check"></i><b>10.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-5"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-5"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>10.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-7"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-5"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-5"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>10.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="10.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="10.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>10.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="10.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>10.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>10.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="10.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>10.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="10.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>10.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="10.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>10.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>10.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="10.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>10.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="10.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>10.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="10.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>10.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="10.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>10.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="10.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>10.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>10.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="10.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>10.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="10.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>10.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="10.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>10.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>10.9</b> Complete cases</a></li>
<li class="chapter" data-level="10.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>10.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="10.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>10.11</b> Extraction</a><ul>
<li class="chapter" data-level="10.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>10.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="10.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>10.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="10.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>10.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="10.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>10.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="10.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>10.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="10.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>10.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="10.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>10.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="10.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>10.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="10.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>10.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>10.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="10.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>10.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="10.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>10.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="10.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>10.16.1</b> “Simplification” of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="10.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>10.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="10.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>10.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="10.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>10.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="10.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>10.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="10.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>10.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="10.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>10.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="10.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>10.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="10.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>10.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="10.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>10.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="10.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>10.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="10.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>10.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>11</b> 230: Cytoscape automation in R using Rcy3</a><ul>
<li class="chapter" data-level="11.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-4"><i class="fa fa-check"></i><b>11.1</b> Overview</a><ul>
<li class="chapter" data-level="11.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructor-names-and-contact-information-3"><i class="fa fa-check"></i><b>11.1.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="11.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description-6"><i class="fa fa-check"></i><b>11.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="11.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>11.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="11.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>11.1.4</b> Background:</a></li>
<li class="chapter" data-level="11.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation-6"><i class="fa fa-check"></i><b>11.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>11.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="11.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-8"><i class="fa fa-check"></i><b>11.1.7</b> Time outline</a></li>
<li class="chapter" data-level="11.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>11.2</b> Background</a><ul>
<li class="chapter" data-level="11.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>11.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="11.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>11.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="11.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>11.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>11.3</b> Translating biological data into Cytoscape using RCy3</a></li>
<li class="chapter" data-level="11.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>11.4</b> Set up</a><ul>
<li class="chapter" data-level="11.4.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started-2"><i class="fa fa-check"></i><b>11.4.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>11.5</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="11.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>11.6</b> Example Data Set</a></li>
<li class="chapter" data-level="11.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>11.7</b> Finding Network Data</a></li>
<li class="chapter" data-level="11.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>11.8</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="11.9" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>11.9</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="11.10" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>11.10</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>12</b> 240: Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="12.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructor-names-and-contact-information-4"><i class="fa fa-check"></i><b>12.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="12.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-7"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-8"><i class="fa fa-check"></i><b>12.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-7"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>12.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="12.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-9"><i class="fa fa-check"></i><b>12.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-6"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-6"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop"><i class="fa fa-check"></i><b>12.4</b> Workshop</a></li>
<li class="chapter" data-level="12.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-3"><i class="fa fa-check"></i><b>12.5</b> Introduction</a><ul>
<li class="chapter" data-level="12.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>12.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="12.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>12.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="12.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for-1"><i class="fa fa-check"></i><b>12.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup-1"><i class="fa fa-check"></i><b>12.6</b> Setup</a></li>
<li class="chapter" data-level="12.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>12.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="12.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>12.8</b> The Grammar</a><ul>
<li class="chapter" data-level="12.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>12.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="12.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>12.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="12.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises-3"><i class="fa fa-check"></i><b>12.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>12.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="12.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>12.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="12.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files-1"><i class="fa fa-check"></i><b>12.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="12.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>12.10</b> What’s next?</a></li>
<li class="chapter" data-level="12.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>12.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html"><i class="fa fa-check"></i><b>13</b> 250: Working with genomic data in R with the DECIPHER package</a><ul>
<li class="chapter" data-level="13.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#overview-5"><i class="fa fa-check"></i><b>13.1</b> Overview</a><ul>
<li class="chapter" data-level="13.1.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-description-8"><i class="fa fa-check"></i><b>13.1.1</b> Workshop Description</a></li>
<li class="chapter" data-level="13.1.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.1.3" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-participation-8"><i class="fa fa-check"></i><b>13.1.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.1.4" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>13.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="13.1.5" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#time-outline-10"><i class="fa fa-check"></i><b>13.1.5</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.2</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.2.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-goals-7"><i class="fa fa-check"></i><b>13.2.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.2.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-objectives-7"><i class="fa fa-check"></i><b>13.2.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>14</b> 260: Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="14.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>14.1</b> Instructors:</a></li>
<li class="chapter" data-level="14.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-9"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-10"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-9"><i class="fa fa-check"></i><b>14.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>14.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="14.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-11"><i class="fa fa-check"></i><b>14.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-8"><i class="fa fa-check"></i><b>14.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-8"><i class="fa fa-check"></i><b>14.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#abstract"><i class="fa fa-check"></i><b>14.4</b> Abstract</a></li>
<li class="chapter" data-level="14.5" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#introduction-5"><i class="fa fa-check"></i><b>14.5</b> Introduction</a><ul>
<li class="chapter" data-level="14.5.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#downloading-pharmacoset-objects"><i class="fa fa-check"></i><b>14.5.1</b> Downloading PharmacoSet objects</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#reproducibility"><i class="fa fa-check"></i><b>14.6</b> Reproducibility</a><ul>
<li class="chapter" data-level="14.6.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#plotting-drug-dose-response-data"><i class="fa fa-check"></i><b>14.6.1</b> Plotting Drug-Dose Response Data</a></li>
<li class="chapter" data-level="14.6.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pharmacological-profiles"><i class="fa fa-check"></i><b>14.6.2</b> Pharmacological profiles</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#replication"><i class="fa fa-check"></i><b>14.7</b> Replication</a><ul>
<li class="chapter" data-level="14.7.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-of-pharmacological-profiles"><i class="fa fa-check"></i><b>14.7.1</b> Consistency of pharmacological profiles</a></li>
<li class="chapter" data-level="14.7.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-assessment-improved-by-modified-concordance-index"><i class="fa fa-check"></i><b>14.7.2</b> consistency assessment improved by Modified Concordance Index</a></li>
<li class="chapter" data-level="14.7.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#known-biomarkers"><i class="fa fa-check"></i><b>14.7.3</b> Known Biomarkers</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#machine-learning-and-biomarker-discovery"><i class="fa fa-check"></i><b>14.8</b> Machine Learning and Biomarker Discovery</a><ul>
<li class="chapter" data-level="14.8.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#bonus-using-the-connectivity-map-for-drug-repurposing"><i class="fa fa-check"></i><b>14.8.1</b> Bonus: Using the Connectivity Map for drug repurposing</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#session-info"><i class="fa fa-check"></i><b>14.9</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>15</b> 500: Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="15.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-6"><i class="fa fa-check"></i><b>15.1</b> Overview</a><ul>
<li class="chapter" data-level="15.1.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#description-3"><i class="fa fa-check"></i><b>15.1.1</b> Description</a></li>
<li class="chapter" data-level="15.1.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-11"><i class="fa fa-check"></i><b>15.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="15.1.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#participation-3"><i class="fa fa-check"></i><b>15.1.3</b> Participation</a></li>
<li class="chapter" data-level="15.1.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-11"><i class="fa fa-check"></i><b>15.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="15.1.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-12"><i class="fa fa-check"></i><b>15.1.5</b> Time outline</a></li>
<li class="chapter" data-level="15.1.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#introductory-material"><i class="fa fa-check"></i><b>15.2</b> Introductory material</a><ul>
<li class="chapter" data-level="15.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#discussion"><i class="fa fa-check"></i><b>15.2.1</b> Discussion</a></li>
<li class="chapter" data-level="15.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#why-learn-delayedarray"><i class="fa fa-check"></i><b>15.2.2</b> Why learn DelayedArray?</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-of-delayedarray-framework"><i class="fa fa-check"></i><b>15.3</b> Overview of DelayedArray framework</a><ul>
<li class="chapter" data-level="15.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-package"><i class="fa fa-check"></i><b>15.3.1</b> The <strong>DelayedArray</strong> package</a></li>
<li class="chapter" data-level="15.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-class"><i class="fa fa-check"></i><b>15.3.2</b> The <em>DelayedArray</em> class</a></li>
<li class="chapter" data-level="15.3.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#operating-on-delayedarray-objects"><i class="fa fa-check"></i><b>15.3.3</b> Operating on <em>DelayedArray</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#whats-out-there-already"><i class="fa fa-check"></i><b>15.4</b> What’s out there already?</a></li>
<li class="chapter" data-level="15.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#incoporating-delayedarray-into-a-package"><i class="fa fa-check"></i><b>15.5</b> Incoporating DelayedArray into a package</a><ul>
<li class="chapter" data-level="15.5.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#writing-algorithms-to-process-delayedarray-instances"><i class="fa fa-check"></i><b>15.5.1</b> Writing algorithms to process <em>DelayedArray</em> instances</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#questions-and-discussion"><i class="fa fa-check"></i><b>15.6</b> Questions and discussion</a></li>
<li class="chapter" data-level="15.7" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#todos"><i class="fa fa-check"></i><b>15.7</b> TODOs</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>16</b> 510: Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="16.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-2"><i class="fa fa-check"></i><b>16.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="16.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-10"><i class="fa fa-check"></i><b>16.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="16.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-12"><i class="fa fa-check"></i><b>16.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="16.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-4"><i class="fa fa-check"></i><b>16.2.2</b> Participation</a></li>
<li class="chapter" data-level="16.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>16.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="16.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-13"><i class="fa fa-check"></i><b>16.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-6"><i class="fa fa-check"></i><b>16.3</b> Introduction</a><ul>
<li class="chapter" data-level="16.3.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>16.3.1</b> Why maintain your package</a></li>
<li class="chapter" data-level="16.3.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>16.3.2</b> Infrastructure used for maintenance</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>16.4</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="16.4.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>16.4.1</b> Build report</a></li>
<li class="chapter" data-level="16.4.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#version-numbering"><i class="fa fa-check"></i><b>16.4.2</b> Version Numbering</a></li>
<li class="chapter" data-level="16.4.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>16.4.3</b> Landing page badges</a></li>
<li class="chapter" data-level="16.4.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>16.4.4</b> Download statistics</a></li>
<li class="chapter" data-level="16.4.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>16.4.5</b> RSS feeds</a></li>
<li class="chapter" data-level="16.4.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>16.4.6</b> Support site, Bioc-devel mailing list</a></li>
<li class="chapter" data-level="16.4.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github"><i class="fa fa-check"></i><b>16.4.7</b> GitHub</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>16.5</b> Round up of resources available</a></li>
<li class="chapter" data-level="16.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#acknowledgements"><i class="fa fa-check"></i><b>16.6</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="bibliography-1.html"><a href="bibliography-1.html"><i class="fa fa-check"></i><b>17</b> Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="r-and-bioconductor-for-everyone-an-introduction" class="section level1">
<h1><span class="header-section-number">2</span> 100: <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</h1>
<p>Authors: Martin Morgan<a href="#fn2" class="footnoteRef" id="fnref2"><sup>2</sup></a>, Lori Shepherd. <br/> Last modified: 17 July 2018</p>
<div id="overview" class="section level2">
<h2><span class="header-section-number">2.1</span> Overview</h2>
<div id="description" class="section level3">
<h3><span class="header-section-number">2.1.1</span> Description</h3>
<p>This workshop is intended for those with little or no experience using <em>R</em> or <em>Bioconductor</em>. In the first portion of the workshop, we will explore the basics of using <em>RStudio</em>, essential <em>R</em> data types, composing short scripts and using functions, and installing and using packages that extend base <em>R</em> functionality. The second portion of the workshop orients participants to the <em>Bioconductor</em> collection of <em>R</em> packages for analysis and comprehension of high-throughput genomic data. We will describe how to discover, install, and learn to use <em>Bioconductor</em> packages, and will explore some of the unique ways in which <em>Bioconductor</em> represents genomic data. The workshop will primarily be instructor-led live demos, with participants following along in their own <em>RStudio</em> sessions.</p>
</div>
<div id="pre-requisites" class="section level3">
<h3><span class="header-section-number">2.1.2</span> Pre-requisites</h3>
<p>This workshop is meant to be introductory, and has no pre-requisites.</p>
<p>Participants will maximize benefit from the workshop by pursuing elementary instructions for using <em>R</em>, such as the <a href="https://www.datacamp.com/courses/free-introduction-to-r">introductory course</a> from DataCamp.</p>
</div>
<div id="participation" class="section level3">
<h3><span class="header-section-number">2.1.3</span> Participation</h3>
<p>Participants will use <em>RStudio</em> interactively as the instructor moves carefully through short examples of <em>R</em> and <em>Bioconductor</em> code.</p>
</div>
<div id="r-bioconductor-packages-used" class="section level3">
<h3><span class="header-section-number">2.1.4</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>Base <em>R</em> packages, e.g., <code>stats</code>, <code>graphics</code>; <a href="https://cran.r-project.org/package=ggplot2">ggplot2</a></li>
<li>Essential <em>Bioconductor</em> packages, e.g., <a href="https://bioconductor.org/packages/rtracklayer">rtracklayer</a>, <a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a>, <a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a></li>
</ul>
</div>
<div id="time-outline" class="section level3">
<h3><span class="header-section-number">2.1.5</span> Time outline</h3>
<p>An example for a 45-minute workshop:</p>
<table>
<thead>
<tr class="header">
<th align="left">Activity</th>
<th align="left">Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Introduction to <em>R</em></td>
<td align="left">45m</td>
</tr>
<tr class="even">
<td align="left">- Using <em>RStudio</em></td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">- <em>R</em> vectors and data frames</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">- Data input and manipulation</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">- Scripts and functions</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">- Using <em>R</em> packages</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">Introduction to <em>Bioconductor</em></td>
<td align="left">60m</td>
</tr>
<tr class="even">
<td align="left">- Project history</td>
<td align="left"></td>
</tr>
<tr class="odd">
<td align="left">- Discovering and using packages</td>
<td align="left"></td>
</tr>
<tr class="even">
<td align="left">- Working with objects</td>
<td align="left"></td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level3">
<h3><span class="header-section-number">2.1.6</span> Workshop goals and objectives</h3>
</div>
<div id="learning-goals" class="section level3">
<h3><span class="header-section-number">2.1.7</span> Learning goals</h3>
<p>Part 1: <em>R</em></p>
<ul>
<li>Import text (e.g., ‘comma-separate value’) files, into <em>R</em>.</li>
<li>Perform manipulations of <em>R</em> data frames.</li>
<li>Apply <em>R</em> functions for statistical analysis and visualization.</li>
<li>Use <em>R</em> packages to extend basic functionality.</li>
</ul>
<p>Part 2: <em>Bioconductor</em></p>
<ul>
<li>Find, install, and learn how to use <em>Bioconductor</em> packages.</li>
<li>Import and manipulate genomic files and <em>Bioconductor</em> data objects.</li>
<li>Start an RNA-seq differential expression work flow.</li>
</ul>
</div>
<div id="learning-objectives" class="section level3">
<h3><span class="header-section-number">2.1.8</span> Learning objectives</h3>
<p>Part 1: <em>R</em></p>
<ul>
<li>Import the ‘pData.csv’ file describing samples from an experiment.</li>
<li>‘Clean’ the pData to include only a subset of values.</li>
<li>Perform a t-test assessing the relationship between ‘age’ and ‘mol.biol’ (presence of BCR/ABL) amongst samples.</li>
<li>Visualize the relationship between ‘age’ and ‘mol.biol’ using base <em>R</em>’s <code>boxplot()</code> function.</li>
<li>Load the <a href="https://cran.r-project.org/package=ggplot2">ggplot2</a> package.</li>
<li>Visualize the relationship between two variables using <code>ggplot()</code>.</li>
</ul>
<p>Part 2: <em>Bioconductor</em></p>
<ul>
<li>Discover, install, and read the vignette of the <a href="https://bioconductor.org/packages/DESeq2">DESeq2</a> package.</li>
<li>Discover the ‘single cell sequencing’ vignette</li>
<li>Import BED and GTF files into <em>Bioconductor</em></li>
<li>Find regions of overlap between the BED and GTF files.</li>
<li>Import a matrix and data.frame into <em>Bioconductor</em>’s <code>SummarizedExperiment</code> object for RNA-Seq differential expression.</li>
</ul>
</div>
</div>
<div id="introduction-to-r" class="section level2">
<h2><span class="header-section-number">2.2</span> Introduction to <em>R</em></h2>
<div id="rstudio-orientation" class="section level3">
<h3><span class="header-section-number">2.2.1</span> <em>RStudio</em> orientation</h3>
<div class="figure">
<img src="100_Morgan_RBiocForAll/RStudio.png" />

</div>
</div>
<div id="very-basics" class="section level3">
<h3><span class="header-section-number">2.2.2</span> Very basics</h3>
<p><em>R</em> works by typing into the console. A very simple example is to add the numbers 1 and 2.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="dv">1</span> <span class="op">+</span><span class="st"> </span><span class="dv">2</span>
<span class="co">#&gt; [1] 3</span></code></pre></div>
<p><em>R</em> works best on <em>vectors</em>, and we can construct vectors ‘by hand’ using the function <code>c()</code> (perhaps for _c_oncatenate). Here is a vector of the numbers 1, 3, and 5.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">5</span>)
<span class="co">#&gt; [1] 1 3 5</span></code></pre></div>
<p><em>R</em> has many shortcuts. A very commonly used shortcut is to create a vector representing a sequence of numbers using <code>:</code>. For instance, here is a vector representing the numbers 1 through 5.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="dv">1</span><span class="op">:</span><span class="dv">5</span>
<span class="co">#&gt; [1] 1 2 3 4 5</span></code></pre></div>
<p>It would be pretty tedious to continually have to enter vectors by hand. <em>R</em> allows objects such as the vector returned by <code>c()</code> to be assigned to variables. The variables can then be referenced at subsequent points in the script.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">5</span>)
x
<span class="co">#&gt; [1] 1 3 5</span></code></pre></div>
<p>Since <em>R</em> knows about vectors, it can be very easy to transform all elements of the vector, e.g., taking the <code>log()</code> of <code>x</code>;</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">log</span>(x)
<span class="co">#&gt; [1] 0.000000 1.098612 1.609438</span></code></pre></div>
<p>The return value of <code>log(x)</code> is itself a vector, and can be assigned to another variable.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">y &lt;-<span class="st"> </span><span class="kw">log</span>(x)
y
<span class="co">#&gt; [1] 0.000000 1.098612 1.609438</span></code></pre></div>
</div>
<div id="data-input-and-manipulation" class="section level3">
<h3><span class="header-section-number">2.2.3</span> Data input and manipulation</h3>
<p><strong>Read data files into <em>R</em> <code>data.frame</code> objects</strong>. We start by reading a simple file into <em>R</em>. The file is a ‘csv’ (comma-separated value) file that could have been exported from a spreadsheet. The first few lines of the file include:</p>
<pre><code>&quot;Sample&quot;,&quot;sex&quot;,&quot;age&quot;,&quot;mol.biol&quot;
&quot;01005&quot;,&quot;M&quot;,53,&quot;BCR/ABL&quot;
&quot;01010&quot;,&quot;M&quot;,19,&quot;NEG&quot;
&quot;03002&quot;,&quot;F&quot;,52,&quot;BCR/ABL&quot;
&quot;04006&quot;,&quot;M&quot;,38,&quot;ALL1/AF4&quot;
&quot;04007&quot;,&quot;M&quot;,57,&quot;NEG&quot;</code></pre>
<p>The file describes samples used in a classic microarray experiment. It consists of four columns:</p>
<ul>
<li><code>Sample</code>: a unique identifier</li>
<li><code>sex</code>: the sex of each patient</li>
<li><code>age</code>: the age of each patient</li>
<li><code>mol.biol</code>: cytological characterization of each patient, e.g., <code>&quot;BCR/ABL&quot;</code> indicates presence of the classic BCR/ABL translocation, while <code>&quot;NEG&quot;</code> indicates no special cytological information.</li>
</ul>
<p>We start by asking are to find the <em>path</em> to the file, using a function <code>file.choose()</code>. This will open a dialog box, and we will navigate to the location of a file named <code>&quot;ALL-phenoData.csv&quot;</code>. Print the value of <code>fname</code> to view the path that you choose.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fname &lt;-<span class="st"> </span><span class="kw">file.choose</span>()
fname</code></pre></div>
<p>Confirm that the file exists using the <code>file.exists()</code> command.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">file.exists</span>(fname)
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p>Read the data from the csv file using <code>read.csv()</code>; assign the input to a variable called <code>pdata</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pdata &lt;-<span class="st"> </span><span class="kw">read.csv</span>(fname)</code></pre></div>
<p><strong>Viewing the <code>data.frame</code></strong></p>
<p>We could print the entire data frame to the screen using</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pdata</code></pre></div>
<p>but a smarter way to explore the data is to ask about its dimensions (rows and columns) using <code>dim()</code>, to look at the <code>head()</code> and <code>tail()</code> of the data, and to ask <em>R</em> for a brief <code>summary()</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(pdata)   <span class="co"># 128 rows x 4 columns</span>
<span class="co">#&gt; [1] 128   4</span>
<span class="kw">head</span>(pdata)  <span class="co"># first six rows</span>
<span class="co">#&gt;   Sample sex age mol.biol</span>
<span class="co">#&gt; 1  01005   M  53  BCR/ABL</span>
<span class="co">#&gt; 2  01010   M  19      NEG</span>
<span class="co">#&gt; 3  03002   F  52  BCR/ABL</span>
<span class="co">#&gt; 4  04006   M  38 ALL1/AF4</span>
<span class="co">#&gt; 5  04007   M  57      NEG</span>
<span class="co">#&gt; 6  04008   M  17      NEG</span>
<span class="kw">tail</span>(pdata)  <span class="co"># last six rows</span>
<span class="co">#&gt;     Sample  sex age mol.biol</span>
<span class="co">#&gt; 123  49004    M  24      NEG</span>
<span class="co">#&gt; 124  56007    M  37      NEG</span>
<span class="co">#&gt; 125  64005    M  19      NEG</span>
<span class="co">#&gt; 126  65003    M  30      NEG</span>
<span class="co">#&gt; 127  83001    M  29      NEG</span>
<span class="co">#&gt; 128   LAL4 &lt;NA&gt;  NA      NEG</span>
<span class="kw">summary</span>(pdata)
<span class="co">#&gt;      Sample      sex          age            mol.biol </span>
<span class="co">#&gt;  01003  :  1   F   :42   Min.   : 5.00   ALL1/AF4:10  </span>
<span class="co">#&gt;  01005  :  1   M   :83   1st Qu.:19.00   BCR/ABL :37  </span>
<span class="co">#&gt;  01007  :  1   NA&#39;s: 3   Median :29.00   E2A/PBX1: 5  </span>
<span class="co">#&gt;  01010  :  1             Mean   :32.37   NEG     :74  </span>
<span class="co">#&gt;  02020  :  1             3rd Qu.:45.50   NUP-98  : 1  </span>
<span class="co">#&gt;  03002  :  1             Max.   :58.00   p15/p16 : 1  </span>
<span class="co">#&gt;  (Other):122             NA&#39;s   :5</span></code></pre></div>
<p>The <code>age</code> element of <code>pdata</code> is a vector of integers and the summary provides a quantitative description of the data. Some values are missing, and these have the special value <code>NA</code>.</p>
<p>The <code>sex</code> and <code>mol.biol</code> vectors are <em>factors</em>. The <code>sex</code> variable has two <em>levels</em> (<code>M</code>, <code>F</code>) while the <code>mol.biol</code> vector has 6 levels. A factor is a statistical concept central to describing data; the levels describe the universe of possible values that the variable can take.</p>
<p><code>Sample</code> is also a <code>factor</code>, but it should probably be considered a <code>character</code> vector used to identify each sample; we clean this up in just a minute.</p>
<p>In addition to <code>summary()</code>, it can be helpful to use <code>class()</code> to look at the class of each object or column, e.g.,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(fname)
<span class="co">#&gt; [1] &quot;character&quot;</span>
<span class="kw">class</span>(pdata)
<span class="co">#&gt; [1] &quot;data.frame&quot;</span></code></pre></div>
<p><strong>Reading data, round two</strong></p>
<p>We noted that <code>Sample</code> has been read by <em>R</em> as a factor. Consult the (complicated!) help page for <code>read.csv()</code> and try to understand how to use <code>colClasses</code> to specify how each column of data should be input.</p>
<p>To navigate to the help page, use</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">?read.csv</code></pre></div>
<p>Focus on the <code>colClasses</code> argument. It is a vector that describes the class each column should be interpretted as. We’d like to read the columns as character, factor, integer, factor. Start by creating a vector of desired values</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">c</span>(<span class="st">&quot;character&quot;</span>, <span class="st">&quot;factor&quot;</span>, <span class="st">&quot;integer&quot;</span>, <span class="st">&quot;factor&quot;</span>)
<span class="co">#&gt; [1] &quot;character&quot; &quot;factor&quot;    &quot;integer&quot;   &quot;factor&quot;</span></code></pre></div>
<p>Then add another argument to our use of <code>read.csv()</code>, specifying the desired column classes as this vector.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pdata &lt;-<span class="st"> </span><span class="kw">read.csv</span>(
    fname,
    <span class="dt">colClasses =</span> <span class="kw">c</span>(<span class="st">&quot;character&quot;</span>, <span class="st">&quot;factor&quot;</span>, <span class="st">&quot;integer&quot;</span>, <span class="st">&quot;factor&quot;</span>)
)
<span class="kw">summary</span>(pdata)
<span class="co">#&gt;     Sample            sex          age            mol.biol </span>
<span class="co">#&gt;  Length:128         F   :42   Min.   : 5.00   ALL1/AF4:10  </span>
<span class="co">#&gt;  Class :character   M   :83   1st Qu.:19.00   BCR/ABL :37  </span>
<span class="co">#&gt;  Mode  :character   NA&#39;s: 3   Median :29.00   E2A/PBX1: 5  </span>
<span class="co">#&gt;                               Mean   :32.37   NEG     :74  </span>
<span class="co">#&gt;                               3rd Qu.:45.50   NUP-98  : 1  </span>
<span class="co">#&gt;                               Max.   :58.00   p15/p16 : 1  </span>
<span class="co">#&gt;                               NA&#39;s   :5</span></code></pre></div>
<p><strong>Subsetting</strong></p>
<p>A basic operation in <em>R</em> is to subset data. A <code>data.frame</code> is a two-dimensional object, so it is subset by specifying the desired rows and columns. Several different ways of specifying rows and colums are possible. Row selection often involves numeric vectors, logical vectors, and character vectors (selecting the rows with the corresponding <code>rownames()</code>). Column selection is most often with a character vector, but can also be numeric.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pdata[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="kw">c</span>(<span class="st">&quot;sex&quot;</span>, <span class="st">&quot;mol.biol&quot;</span>)]
<span class="co">#&gt;   sex mol.biol</span>
<span class="co">#&gt; 1   M  BCR/ABL</span>
<span class="co">#&gt; 2   M      NEG</span>
<span class="co">#&gt; 3   F  BCR/ABL</span>
<span class="co">#&gt; 4   M ALL1/AF4</span>
<span class="co">#&gt; 5   M      NEG</span>
pdata[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, <span class="kw">c</span>(<span class="dv">2</span>, <span class="dv">3</span>)]
<span class="co">#&gt;   sex age</span>
<span class="co">#&gt; 1   M  53</span>
<span class="co">#&gt; 2   M  19</span>
<span class="co">#&gt; 3   F  52</span>
<span class="co">#&gt; 4   M  38</span>
<span class="co">#&gt; 5   M  57</span></code></pre></div>
<p>Omitting either the row or column index selects all rows or columns</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pdata[<span class="dv">1</span><span class="op">:</span><span class="dv">5</span>, ]
<span class="co">#&gt;   Sample sex age mol.biol</span>
<span class="co">#&gt; 1  01005   M  53  BCR/ABL</span>
<span class="co">#&gt; 2  01010   M  19      NEG</span>
<span class="co">#&gt; 3  03002   F  52  BCR/ABL</span>
<span class="co">#&gt; 4  04006   M  38 ALL1/AF4</span>
<span class="co">#&gt; 5  04007   M  57      NEG</span></code></pre></div>
<p>The subset operator <code>[</code> generally returns the same type of object as being subset, e.g., above all return values are <code>data.frame</code> objects. It is sometimes desirable to select a single column as a vector. This can be done using <code>$</code> or <code>[[</code>; note that some values are <code>NA</code>, indicating that the patient’s age is “not available”.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pdata<span class="op">$</span>age
<span class="co">#&gt;   [1] 53 19 52 38 57 17 18 16 15 40 33 55  5 18 41 27 27 46 37 36 53 39 53</span>
<span class="co">#&gt;  [24] 20 44 28 58 43 48 58 19 26 19 32 17 45 20 16 51 57 29 16 32 15 NA 21</span>
<span class="co">#&gt;  [47] 49 38 17 26 48 16 18 17 22 47 21 54 26 19 47 18 52 27 52 18 18 23 16</span>
<span class="co">#&gt;  [70] NA 54 25 31 19 24 23 NA 41 37 54 18 19 43 53 50 54 53 49 20 26 22 36</span>
<span class="co">#&gt;  [93] 27 50 NA 31 16 48 17 40 22 30 18 22 50 41 40 28 25 16 31 14 24 19 37</span>
<span class="co">#&gt; [116] 23 30 48 22 41 52 32 24 37 19 30 29 NA</span>
pdata[[<span class="st">&quot;age&quot;</span>]]
<span class="co">#&gt;   [1] 53 19 52 38 57 17 18 16 15 40 33 55  5 18 41 27 27 46 37 36 53 39 53</span>
<span class="co">#&gt;  [24] 20 44 28 58 43 48 58 19 26 19 32 17 45 20 16 51 57 29 16 32 15 NA 21</span>
<span class="co">#&gt;  [47] 49 38 17 26 48 16 18 17 22 47 21 54 26 19 47 18 52 27 52 18 18 23 16</span>
<span class="co">#&gt;  [70] NA 54 25 31 19 24 23 NA 41 37 54 18 19 43 53 50 54 53 49 20 26 22 36</span>
<span class="co">#&gt;  [93] 27 50 NA 31 16 48 17 40 22 30 18 22 50 41 40 28 25 16 31 14 24 19 37</span>
<span class="co">#&gt; [116] 23 30 48 22 41 52 32 24 37 19 30 29 NA</span></code></pre></div>
<p>We can use <code>class()</code> to determine the type of each column</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">class</span>(pdata<span class="op">$</span>age)
<span class="co">#&gt; [1] &quot;integer&quot;</span></code></pre></div>
<p>as well as other functions to manipulate or summarize the data. What do each of the following lines do?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">table</span>(pdata<span class="op">$</span>mol.biol)
<span class="co">#&gt; </span>
<span class="co">#&gt; ALL1/AF4  BCR/ABL E2A/PBX1      NEG   NUP-98  p15/p16 </span>
<span class="co">#&gt;       10       37        5       74        1        1</span>
<span class="kw">table</span>(<span class="kw">is.na</span>(pdata<span class="op">$</span>age))
<span class="co">#&gt; </span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt;   123     5</span>
<span class="kw">levels</span>(pdata<span class="op">$</span>sex)
<span class="co">#&gt; [1] &quot;F&quot; &quot;M&quot;</span></code></pre></div>
<p><strong>Basic data manipulations</strong></p>
<p>We’ve seen (e.g., <code>log()</code>, above) that numeric vectors can be transformed by mathematical functions. Similar functions are available for other data types. A very common scenario is to create logical vectors that are then used to subset objects. Here we compare each element of the <code>sex</code> column to <code>&quot;F&quot;</code>. The result indicates which rows in the <code>pdata</code> correspond to indivdudals with sex <code>&quot;F&quot;</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">pdata<span class="op">$</span>sex <span class="op">==</span><span class="st"> &quot;F&quot;</span>
<span class="co">#&gt;   [1] FALSE FALSE  TRUE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [12] FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE  TRUE FALSE FALSE</span>
<span class="co">#&gt;  [23] FALSE FALSE FALSE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE</span>
<span class="co">#&gt;  [34] FALSE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE FALSE  TRUE FALSE</span>
<span class="co">#&gt;  [45]    NA FALSE  TRUE FALSE  TRUE  TRUE  TRUE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [56]  TRUE FALSE  TRUE FALSE  TRUE FALSE FALSE  TRUE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [67] FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE  TRUE FALSE FALSE</span>
<span class="co">#&gt;  [78] FALSE  TRUE FALSE FALSE FALSE  TRUE  TRUE  TRUE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [89] FALSE  TRUE FALSE FALSE  TRUE FALSE    NA FALSE  TRUE  TRUE FALSE</span>
<span class="co">#&gt; [100]  TRUE FALSE FALSE FALSE FALSE FALSE FALSE  TRUE  TRUE FALSE  TRUE</span>
<span class="co">#&gt; [111] FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt; [122] FALSE FALSE FALSE FALSE FALSE FALSE    NA</span></code></pre></div>
<p>A more elaborate example identifies rows corresponding to females greater than 50 years of age. Note that some comparisons are <code>NA</code>; why is that?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">(pdata<span class="op">$</span>sex <span class="op">==</span><span class="st"> &quot;F&quot;</span>) <span class="op">&amp;</span><span class="st"> </span>(pdata<span class="op">$</span>age <span class="op">&gt;</span><span class="st"> </span><span class="dv">50</span>)
<span class="co">#&gt;   [1] FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [12] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [23] FALSE FALSE FALSE FALSE  TRUE FALSE FALSE  TRUE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [34] FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [45]    NA FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [56] FALSE FALSE  TRUE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [67] FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [78] FALSE FALSE FALSE FALSE FALSE FALSE  TRUE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt;  [89] FALSE FALSE FALSE FALSE FALSE FALSE    NA FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt; [100] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt; [111] FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE FALSE</span>
<span class="co">#&gt; [122] FALSE FALSE FALSE FALSE FALSE FALSE    NA</span></code></pre></div>
<p>Another elaborate comparison is <code>%in%</code>, which asks whether each element in the vector on the left-hand side of <code>%in%</code> is contained in the set of values on the right-hand side, for example, which elements of the <code>pdata$mol.biol</code> column are in the set <code>&quot;BCR/ABL&quot;</code> or <code>&quot;NEG&quot;</code>?</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">table</span>( pdata<span class="op">$</span>mol.biol )
<span class="co">#&gt; </span>
<span class="co">#&gt; ALL1/AF4  BCR/ABL E2A/PBX1      NEG   NUP-98  p15/p16 </span>
<span class="co">#&gt;       10       37        5       74        1        1</span>
pdata<span class="op">$</span>mol.biol <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;BCR/ABL&quot;</span>, <span class="st">&quot;NEG&quot;</span>)
<span class="co">#&gt;   [1]  TRUE  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE</span>
<span class="co">#&gt;  [12]  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE</span>
<span class="co">#&gt;  [23]  TRUE  TRUE FALSE  TRUE FALSE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE</span>
<span class="co">#&gt;  [34]  TRUE  TRUE FALSE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE</span>
<span class="co">#&gt;  [45]  TRUE  TRUE  TRUE FALSE FALSE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE</span>
<span class="co">#&gt;  [56]  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE FALSE  TRUE  TRUE  TRUE</span>
<span class="co">#&gt;  [67]  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE  TRUE FALSE  TRUE  TRUE</span>
<span class="co">#&gt;  [78]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE</span>
<span class="co">#&gt;  [89]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE FALSE  TRUE FALSE  TRUE  TRUE</span>
<span class="co">#&gt; [100]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE</span>
<span class="co">#&gt; [111]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE</span>
<span class="co">#&gt; [122]  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE</span></code></pre></div>
<p>The <code>subset()</code> function can be used to perform the subset, e.g., selecting all records of females older than 50</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">subset</span>(pdata, sex <span class="op">==</span><span class="st"> &quot;F&quot;</span> <span class="op">&amp;</span><span class="st"> </span>age <span class="op">&gt;</span><span class="st"> </span><span class="dv">50</span>)
<span class="co">#&gt;    Sample sex age mol.biol</span>
<span class="co">#&gt; 3   03002   F  52  BCR/ABL</span>
<span class="co">#&gt; 27  16004   F  58 ALL1/AF4</span>
<span class="co">#&gt; 30  20002   F  58  BCR/ABL</span>
<span class="co">#&gt; 39  24011   F  51  BCR/ABL</span>
<span class="co">#&gt; 58  28021   F  54  BCR/ABL</span>
<span class="co">#&gt; 63  28032   F  52 ALL1/AF4</span>
<span class="co">#&gt; 71  30001   F  54  BCR/ABL</span>
<span class="co">#&gt; 84  57001   F  53      NEG</span></code></pre></div>
<p>The return value of <code>subset()</code> can be assigned to a variable, so the below creates a subset of <code>pdata</code> corresponding to indivdiuals whose <code>mol.biol</code> is either <code>&quot;BCR/ABL&quot;</code> or <code>&quot;NEG&quot;</code>; the result is assigned to a variable <code>bcrabl</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bcrabl &lt;-<span class="st"> </span><span class="kw">subset</span>(pdata, mol.biol <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;BCR/ABL&quot;</span>, <span class="st">&quot;NEG&quot;</span>))
<span class="kw">dim</span>( bcrabl )
<span class="co">#&gt; [1] 111   4</span></code></pre></div>
<p>Here’s a tabular summary of the levels of the <code>mol.biol</code> factor in <code>bcrabl</code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">table</span>(bcrabl<span class="op">$</span>mol.biol)
<span class="co">#&gt; </span>
<span class="co">#&gt; ALL1/AF4  BCR/ABL E2A/PBX1      NEG   NUP-98  p15/p16 </span>
<span class="co">#&gt;        0       37        0       74        0        0</span></code></pre></div>
<p>Note that the factor includes levels that are not actually present in the subset. For our work below, we’d like to remove the unused levels. This can be done by calling the <code>factor()</code> function</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">factor</span>(bcrabl<span class="op">$</span>mol.biol)
<span class="co">#&gt;   [1] BCR/ABL NEG     BCR/ABL NEG     NEG     NEG     NEG     NEG    </span>
<span class="co">#&gt;   [9] BCR/ABL BCR/ABL NEG     NEG     BCR/ABL NEG     BCR/ABL BCR/ABL</span>
<span class="co">#&gt;  [17] BCR/ABL BCR/ABL NEG     BCR/ABL BCR/ABL NEG     BCR/ABL NEG    </span>
<span class="co">#&gt;  [25] BCR/ABL NEG     BCR/ABL NEG     BCR/ABL BCR/ABL NEG     BCR/ABL</span>
<span class="co">#&gt;  [33] BCR/ABL BCR/ABL NEG     BCR/ABL NEG     NEG     NEG     BCR/ABL</span>
<span class="co">#&gt;  [41] BCR/ABL BCR/ABL NEG     NEG     NEG     NEG     BCR/ABL BCR/ABL</span>
<span class="co">#&gt;  [49] NEG     NEG     NEG     NEG     BCR/ABL NEG     NEG     NEG    </span>
<span class="co">#&gt;  [57] NEG     NEG     BCR/ABL BCR/ABL NEG     NEG     BCR/ABL BCR/ABL</span>
<span class="co">#&gt;  [65] NEG     NEG     NEG     NEG     BCR/ABL NEG     BCR/ABL BCR/ABL</span>
<span class="co">#&gt;  [73] BCR/ABL NEG     NEG     BCR/ABL NEG     BCR/ABL BCR/ABL NEG    </span>
<span class="co">#&gt;  [81] NEG     NEG     NEG     NEG     NEG     NEG     NEG     NEG    </span>
<span class="co">#&gt;  [89] NEG     NEG     NEG     NEG     NEG     NEG     NEG     NEG    </span>
<span class="co">#&gt;  [97] NEG     NEG     NEG     NEG     NEG     NEG     NEG     NEG    </span>
<span class="co">#&gt; [105] NEG     NEG     NEG     NEG     NEG     NEG     NEG    </span>
<span class="co">#&gt; Levels: BCR/ABL NEG</span></code></pre></div>
<p>We can then update the <code>mol.biol</code> column of <code>bcrabl</code> by assigning new values to it with <code>&lt;-</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bcrabl<span class="op">$</span>mol.biol &lt;-<span class="st"> </span><span class="kw">factor</span>(bcrabl<span class="op">$</span>mol.biol)
<span class="kw">table</span>(bcrabl<span class="op">$</span>mol.biol)
<span class="co">#&gt; </span>
<span class="co">#&gt; BCR/ABL     NEG </span>
<span class="co">#&gt;      37      74</span></code></pre></div>
<p><strong>Exploratory data analysis</strong></p>
<p>A very useful concept in <em>R</em> is the <code>formula</code>. This is specified in a form like <code>lhs ~ rhs</code>. The left-hand side is typically a response (dependent) variable, and the right-hand side a description of the independent variables that one is interested in. I ‘say’ a formula like <code>age ~ mol.biol</code> as “age as a function of mol.biol”.</p>
<p>To get a sense of the use of formulas in <em>R</em>, and the ease with which one can explore data, use the <code>boxplot()</code> function to visualize the relationship between age as a function of molecular biology.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">boxplot</span>(age <span class="op">~</span><span class="st"> </span>mol.biol, bcrabl)</code></pre></div>
<p><img src="100_Morgan_RBiocForAll_files/figure-html/unnamed-chunk-31-1.png" width="672" /></p>
<p>Consult the help page <code>?boxplot</code> to figure out how to add “Age” as the y-axis label.</p>
<p>The boxplot suggests that in our sample individuals with BCR/ABL are on average older than individuals classified as NEG. Confirm this using <code>t.test()</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">t.test</span>(age <span class="op">~</span><span class="st"> </span>mol.biol, bcrabl)
<span class="co">#&gt; </span>
<span class="co">#&gt;  Welch Two Sample t-test</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; data:  age by mol.biol</span>
<span class="co">#&gt; t = 4.8172, df = 68.529, p-value = 8.401e-06</span>
<span class="co">#&gt; alternative hypothesis: true difference in means is not equal to 0</span>
<span class="co">#&gt; 95 percent confidence interval:</span>
<span class="co">#&gt;   7.13507 17.22408</span>
<span class="co">#&gt; sample estimates:</span>
<span class="co">#&gt; mean in group BCR/ABL     mean in group NEG </span>
<span class="co">#&gt;              40.25000              28.07042</span></code></pre></div>
<p>It might be surprising that the <code>df</code> (degrees of freedom) are not a round number, but 68.592. This is because by default <em>R</em> performs a t-test that allows for variances to differ between groups. Consult the <code>?t.test</code> help page to figure out to perform a simpler form of the t-test, where variances are equal.</p>
</div>
<div id="writing-r-scripts" class="section level3">
<h3><span class="header-section-number">2.2.4</span> Writing <em>R</em> scripts</h3>
</div>
<div id="using-packages" class="section level3">
<h3><span class="header-section-number">2.2.5</span> Using packages</h3>
<p><em>R</em> functions are made available in <em>packages</em>. All the functions that we have used so far are implemented in packages that are distributed with <em>R</em>. There are actually more than 15,000 <em>R</em> packages available for a wide variety of purposes. The general challenge we explore here is how to discover, install, and use one of these packages.</p>
<p><strong>Discover</strong></p>
<p>The most common place to find <em>R</em> packages in on the comprehensive <em>R</em> archive network, CRAN.</p>
<p>Visit CRAN at <a href="https://cran.r-project.org" class="uri">https://cran.r-project.org</a>, click on the ‘Packages’ link, and the table of available packages sorted by name. Find the <a href="https://cran.r-project.org/package=ggplot2">ggplot2</a> package and peruse the information you are presented with. What does this package do?</p>
<p>The large number of <em>R</em> packages can make finding the ‘best’ packages for a particular purpose difficult to identify. From <a href="https://cran.r-project.org" class="uri">https://cran.r-project.org</a>, click on the ‘Task Views’ link and explore a topic interesting to you.</p>
<p><strong>Install (once only per <em>R</em> installation)</strong></p>
<p>One useful packages have been identified, they need to be installed into <em>R</em>. This only needs to be done once per <em>R</em> installation. The following installs the <a href="https://cran.r-project.org/package=ggplot2">ggplot2</a> and <a href="https://cran.r-project.org/package=tibble">tibble</a> packages; you DO NOT need to do this command because <a href="https://cran.r-project.org/package=ggplot2">ggplot2</a> is already installed on your AMI.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">## No need to do this...
<span class="kw">install.packages</span>(<span class="kw">c</span>(<span class="st">&quot;ggplot2&quot;</span>, <span class="st">&quot;tibble&quot;</span>))</code></pre></div>
<p><strong>Use</strong></p>
<p>Packages that are installed on your system are not available for use until they are attached to the current <em>R</em> session. Use <code>library()</code> to attach the <a href="https://cran.r-project.org/package=ggplot2">ggplot2</a> package.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;ggplot2&quot;</span>)</code></pre></div>
<p>Look up help on <code>?ggplot</code>. Peruse especially the examples at the bottom of the help page. The idea is that one identifies the data set to be used, and the variables with the data to be used as the x and y ‘aesthetics’</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(brcabl, <span class="kw">aes</span>(<span class="dt">x =</span> mol.biol, <span class="dt">y =</span> age))</code></pre></div>
<p>One then adds ‘geoms’ to the plot, for instance the boxplot geom</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ggplot</span>(bcrabl, <span class="kw">aes</span>(<span class="dt">x =</span> mol.biol, <span class="dt">y =</span> age)) <span class="op">+</span><span class="st"> </span><span class="kw">geom_boxplot</span>()
<span class="co">#&gt; Warning: Removed 4 rows containing non-finite values (stat_boxplot).</span></code></pre></div>
<p><img src="100_Morgan_RBiocForAll_files/figure-html/unnamed-chunk-36-1.png" width="672" /></p>
<p>Use <code>xlab()</code> and <code>ylab()</code> to enhance the plot with more meaning x and y labels; see the help page <code>?xlab</code>, especially the example section, for hints on how to modify the graph.</p>
<p>The grammar of graphics implemented by <a href="https://cran.r-project.org/package=ggplot2">ggplot2</a> is a very powerful and expressive system for producing appealing visualizations, and is widely used in the <em>R</em> community.</p>
</div>
</div>
<div id="introduction-to-bioconductor" class="section level2">
<h2><span class="header-section-number">2.3</span> Introduction to <em>Bioconductor</em></h2>
<p><em>Bioconductor</em> is a collection of more than 1,500 packages for the statistical analysis and comprehension of high-throughput genomic data. Originally developed for microarrays, <em>Bioconductor</em> packages are now used in a wide range of analyses, including bulk and single-cell RNA-seq, ChIP seq, copy number analysis, microarray methylation and classic expression analysis, flow cytometry, and many other domains.</p>
<p>This section of the workshop introduces the essential of <em>Bioconductor</em> package discovery, installation, and use.</p>
<div id="discovering-installing-and-learning-how-to-use-bioconductor-packages" class="section level3">
<h3><span class="header-section-number">2.3.1</span> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</h3>
<p><strong>Discovery</strong></p>
<p>The web site at <a href="https://bioconductor.org" class="uri">https://bioconductor.org</a> contains descriptions of all <em>Bioconductor</em> packages, as well as essential reference material for all levels of user.</p>
<p>Packages available in <em>Bioconductor</em> are summarized at <a href="https://bioconductor.org/packages" class="uri">https://bioconductor.org/packages</a>, also linked from the front page of the web site. The widget on the left summarizes four distinct types of <em>Bioconductor</em> packages</p>
<ul>
<li><p>‘Software’ packages implement particular analyses or other functionality, e.g., querying web-based resources or importing common file formats to <em>Bioconductor</em> objects.</p></li>
<li><p>‘Annotation’ packages contain data that can be helpful in placing analysis results in context, for example: mapping between gene symbols such as “BRCA1” and Ensembl or Entrez gene identifiers; classifying genes in terms of gene ontology; describing the genomic coordinates of exons, transcripts, and genes; and representing whole genome sequences of common organisms.</p></li>
<li><p>‘Experiment data’ packages contain highly curated data sets that can be useful for learning and teaching (e.g., the <a href="https://bioconductor.org/packages/airway">airway</a> package and data set used in the <a href="https://bioconductor.org/packages/DESeq2">DESeq2</a> package for the analysis of bulk RNA-seq differential expression) or placing results in context (e.g., the <a href="https://bioconductor.org/packages/curatedTCGAData">curatedTCGAData</a> package for conveniently accessing TCGA data in a way that allows very smooth integeration into <em>Bioconductor</em> analyses).</p></li>
<li><p>‘Workflow’ packages that summarize common work flows, e.g., <a href="https://bioconductor.org/packages/simpleSingleCell">simpleSingleCell</a> for single-cell RNA-seq expression analysis.</p></li>
</ul>
<p><strong>Installation</strong></p>
<p>Like CRAN packages, <em>Bioconductor</em> packages need to be installed only once per <em>R</em> installation, and then attached to each session where they are going to be used.</p>
<p><em>Bioconductor</em> packages are installed slightly differently from CRAN packages. The first step is to install the <a href="https://cran.r-project.org/package=BiocManager">BiocManager</a> package from CRAN.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="cf">if</span> (<span class="op">!</span><span class="st">&quot;BiocManager&quot;</span> <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(<span class="kw">intalled.packages</span>()))
    <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="dt">repos=</span><span class="st">&quot;https://cran.r-project.org&quot;</span>)</code></pre></div>
<p>The next step is to install the desired <em>Bioconductor</em> packages. The syntax to install the <a href="https://bioconductor.org/packages/rtracklayer">rtracklayer</a> and <a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a> packages is</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="kw">c</span>(<span class="st">&quot;rtracklayer&quot;</span>, <span class="st">&quot;GenomicRanges&quot;</span>))</code></pre></div>
<p><em>Bioconductor</em> packages tend to depend on one another quite alot, so it is important that the correct versions of all packages are installed. Validate your installation (not necessary during the course) with</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">BiocManager<span class="op">::</span><span class="kw">valid</span>()</code></pre></div>
<p>A convenient function in <a href="https://cran.r-project.org/package=BiocManager">BiocManager</a> is <code>available()</code>, which accepts a regular expression to find matching packages. The following finds all ‘TxDb’ packages (describing exon, transcript, and gene coordinates) for <em>Homo sapiens</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">BiocManager<span class="op">::</span><span class="kw">available</span>(<span class="st">&quot;TxDb.Hsapiens&quot;</span>)
<span class="co">#&gt; [1] &quot;TxDb.Hsapiens.BioMart.igis&quot;                 </span>
<span class="co">#&gt; [2] &quot;TxDb.Hsapiens.UCSC.hg18.knownGene&quot;          </span>
<span class="co">#&gt; [3] &quot;TxDb.Hsapiens.UCSC.hg19.knownGene&quot;          </span>
<span class="co">#&gt; [4] &quot;TxDb.Hsapiens.UCSC.hg19.lincRNAsTranscripts&quot;</span>
<span class="co">#&gt; [5] &quot;TxDb.Hsapiens.UCSC.hg38.knownGene&quot;</span></code></pre></div>
<p><strong>Learning and support</strong></p>
<p>Each package is linked to a ‘landing page’, e.g., <a href="https://bioconductor.org/packages/DESeq2">DESeq2</a> that contains a description of the package, authors, perhaps literature citations where the software is described, and installation instructions.</p>
<p>An important part of <em>Bioconductor</em> packages are ‘vignettes’ which describe how the package is to be used. Vignettes are linked from package landing pages, and are available from within <em>R</em> using</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">browseVignettes</span>(<span class="st">&quot;simpleSingleCell&quot;</span>)</code></pre></div>
<p>Users can get support on using packages at <a href="https://support.bioconductor.org" class="uri">https://support.bioconductor.org</a>, a question-and-answer style forum where response usually come quickly and often from very knowledgable users or the package developer. There are many additional sources of support, including <a href="https://bioconductor.org/help/course-materials">course material</a> linked from the home page.</p>
</div>
<div id="working-with-genomic-ranges" class="section level3">
<h3><span class="header-section-number">2.3.2</span> Working with Genomic Ranges</h3>
<p>This section introduces two useful packages for general-purpose work on genomic coordinates. The <a href="https://bioconductor.org/packages/rtracklayer">rtracklayer</a> package provides the <code>import()</code> function to read many types of genomic files (e.g., BED, GTF, VCF, FASTA) into <em>Bioconductor</em> objects. The <a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a> package provides functions for manipulating genomic ranges, i.e., descriptions of exons, genes, ChIP peaks, called variants, … as coordinates in genome space.</p>
<p>Start by attaching the <a href="https://bioconductor.org/packages/rtracklayer">rtracklayer</a> and <a href="https://bioconductor.org/packages/GenomicRanges">GenomicRanges</a> packages to our session.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;rtracklayer&quot;</span>)
<span class="kw">library</span>(<span class="st">&quot;GenomicRanges&quot;</span>)</code></pre></div>
<p><strong>Importing data</strong></p>
<p>We’ll read in a BED file derived from the UCSC genome browser. The file contains the coordinates of all CpG islands in the human genome, and is described at the <a href="https://genome.ucsc.edu/cgi-bin/hgTables?hgsid=578954849_wF1QP81SIHdfr8b0kmZUOcsZcHYr&amp;clade=mammal&amp;org=Human&amp;db=hg38&amp;hgta_group=regulation&amp;hgta_track=knownGene&amp;hgta_table=0&amp;hgta_regionType=genome&amp;position=chr9%3A133252000-133280861&amp;hgta_outputType=primaryTable&amp;hgta_outFileName=">UCSC table browser</a>. Here are the first few lines of the file, giving the chromosme, start and end coordinates, and identifier of each CpG island.</p>
<pre><code>chr1    155188536   155192004   CpG:_361
chr1    2226773 2229734 CpG:_366
chr1    36306229    36307408    CpG:_110
chr1    47708822    47710847    CpG:_164
chr1    53737729    53739637    CpG:_221
chr1    144179071   144179313   CpG:_20</code></pre>
<p>Use <code>file.choose()</code> to find the file</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fname &lt;-<span class="st"> </span><span class="kw">file.choose</span>()   <span class="co"># CpGislands.Hsapiens.hg38.UCSC.bed</span></code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fname
<span class="co">#&gt; [1] &quot;100_Morgan_RBiocForAll/CpGislands.Hsapiens.hg38.UCSC.bed&quot;</span>
<span class="kw">file.exists</span>(fname)
<span class="co">#&gt; [1] TRUE</span></code></pre></div>
<p>Then use <code>import()</code> from <a href="https://bioconductor.org/packages/rtracklayer">rtracklayer</a> to read the data into <em>R</em>. The end result is a <code>GenomicRanges</code> object describing each CpG island.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cpg &lt;-<span class="st"> </span><span class="kw">import</span>(fname)
cpg
<span class="co">#&gt; GRanges object with 30477 ranges and 1 metadata column:</span>
<span class="co">#&gt;                          seqnames              ranges strand |        name</span>
<span class="co">#&gt;                             &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;       [1]                    chr1 155188537-155192004      * |    CpG:_361</span>
<span class="co">#&gt;       [2]                    chr1     2226774-2229734      * |    CpG:_366</span>
<span class="co">#&gt;       [3]                    chr1   36306230-36307408      * |    CpG:_110</span>
<span class="co">#&gt;       [4]                    chr1   47708823-47710847      * |    CpG:_164</span>
<span class="co">#&gt;       [5]                    chr1   53737730-53739637      * |    CpG:_221</span>
<span class="co">#&gt;       ...                     ...                 ...    ... .         ...</span>
<span class="co">#&gt;   [30473] chr22_KI270734v1_random       131010-132049      * |    CpG:_102</span>
<span class="co">#&gt;   [30474] chr22_KI270734v1_random       161257-161626      * |     CpG:_55</span>
<span class="co">#&gt;   [30475] chr22_KI270735v1_random         17221-18098      * |    CpG:_100</span>
<span class="co">#&gt;   [30476] chr22_KI270738v1_random           4413-5280      * |     CpG:_80</span>
<span class="co">#&gt;   [30477] chr22_KI270738v1_random           6226-6467      * |     CpG:_34</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 254 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Closely compare the coordinates of the first few ranges from the file with the first few ranges in the <em>Bioconductor</em> representation. The <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">BED format</a> specification says that coordinates are 0-based, and intervals are half-open (the ‘start’ coordinate is in the range, the ‘end’ coordinate is immediately after the range; this makes some computations easy). <em>Bioconductor</em>’s convention is that coordinates are 1-based and closed (i.e., both start and end coordinates are included in the range). <a href="https://bioconductor.org/packages/rtracklayer">rtracklayer</a>’s <code>import()</code> function has adjusted coordinates to follow <em>Bioconductor</em> conventions.</p>
<p><strong>Working with genomic ranges</strong></p>
<p>For convenience and to illustrate functionality, let’s work only with the ‘standard’ chromosomes 1 - 22 autosomal, X, and Y chromosomes. Look up the help page <code>?keepStandardChromosomes</code> for an explanation of <code>pruning.mode=</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cpg &lt;-<span class="st"> </span><span class="kw">keepStandardChromosomes</span>(cpg, <span class="dt">pruning.mode =</span> <span class="st">&quot;coarse&quot;</span>)
cpg
<span class="co">#&gt; GRanges object with 27949 ranges and 1 metadata column:</span>
<span class="co">#&gt;           seqnames              ranges strand |        name</span>
<span class="co">#&gt;              &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;       [1]     chr1 155188537-155192004      * |    CpG:_361</span>
<span class="co">#&gt;       [2]     chr1     2226774-2229734      * |    CpG:_366</span>
<span class="co">#&gt;       [3]     chr1   36306230-36307408      * |    CpG:_110</span>
<span class="co">#&gt;       [4]     chr1   47708823-47710847      * |    CpG:_164</span>
<span class="co">#&gt;       [5]     chr1   53737730-53739637      * |    CpG:_221</span>
<span class="co">#&gt;       ...      ...                 ...    ... .         ...</span>
<span class="co">#&gt;   [27945]    chr22   50704375-50704880      * |     CpG:_38</span>
<span class="co">#&gt;   [27946]    chr22   50710878-50711294      * |     CpG:_41</span>
<span class="co">#&gt;   [27947]    chr22   50719959-50721632      * |    CpG:_180</span>
<span class="co">#&gt;   [27948]    chr22   50730600-50731304      * |     CpG:_65</span>
<span class="co">#&gt;   [27949]    chr22   50783345-50783889      * |     CpG:_63</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>There are two parts to a <code>GenomicRanges</code> object. The <code>seqnames</code> (chromosomes, in the present case), start and end coordinates, and strand are <em>required</em>. Additional elements such as <code>name</code> in the current example are optional. Required components are accessed by functions such as <code>start()</code>, <code>end()</code> and <code>width()</code>. Optional components can be accessed using the <code>$</code> notation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>( <span class="kw">start</span>(cpg) )
<span class="co">#&gt; [1] 155188537   2226774  36306230  47708823  53737730 144179072</span>
<span class="kw">head</span>( cpg<span class="op">$</span>name )
<span class="co">#&gt; [1] &quot;CpG:_361&quot; &quot;CpG:_366&quot; &quot;CpG:_110&quot; &quot;CpG:_164&quot; &quot;CpG:_221&quot; &quot;CpG:_20&quot;</span></code></pre></div>
<p>Use the <code>width()</code> accessor function to extract a vector of widths of each CpG island. Transform the values using <code>log10()</code>, and visualize the distribution using <code>hist()</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(<span class="kw">log10</span>(<span class="kw">width</span>(cpg)))</code></pre></div>
<p><img src="100_Morgan_RBiocForAll_files/figure-html/unnamed-chunk-49-1.png" width="672" /></p>
<p>Use <code>subset()</code> to select the CpG islands on chromosomes 1 and 2.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">subset</span>(cpg, seqnames <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>))
<span class="co">#&gt; GRanges object with 4217 ranges and 1 metadata column:</span>
<span class="co">#&gt;          seqnames              ranges strand |        name</span>
<span class="co">#&gt;             &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">#&gt;      [1]     chr1 155188537-155192004      * |    CpG:_361</span>
<span class="co">#&gt;      [2]     chr1     2226774-2229734      * |    CpG:_366</span>
<span class="co">#&gt;      [3]     chr1   36306230-36307408      * |    CpG:_110</span>
<span class="co">#&gt;      [4]     chr1   47708823-47710847      * |    CpG:_164</span>
<span class="co">#&gt;      [5]     chr1   53737730-53739637      * |    CpG:_221</span>
<span class="co">#&gt;      ...      ...                 ...    ... .         ...</span>
<span class="co">#&gt;   [4213]     chr2 242003256-242004412      * |     CpG:_79</span>
<span class="co">#&gt;   [4214]     chr2 242006590-242010686      * |    CpG:_263</span>
<span class="co">#&gt;   [4215]     chr2 242045491-242045723      * |     CpG:_16</span>
<span class="co">#&gt;   [4216]     chr2 242046615-242047706      * |    CpG:_170</span>
<span class="co">#&gt;   [4217]     chr2 242088150-242089411      * |    CpG:_149</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 24 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p><strong>Genomic annotations</strong></p>
<p>Earlier we mentioned ‘Annotation data’ packages. An example is the TxDb.* family of packages. These packages contain information on the genomic coordinates of exons, genes, transcripts, etc. Attach the TxDb package corresponding to the <em>Homo sapiens</em> hg38 genome build using the UCSC ‘knownGene’ track.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;TxDb.Hsapiens.UCSC.hg38.knownGene&quot;</span>)</code></pre></div>
<p>Extract the coordinates of all transcripts</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tx &lt;-<span class="st"> </span><span class="kw">transcripts</span>(TxDb.Hsapiens.UCSC.hg38.knownGene)
tx
<span class="co">#&gt; GRanges object with 197782 ranges and 2 metadata columns:</span>
<span class="co">#&gt;                    seqnames        ranges strand |     tx_id     tx_name</span>
<span class="co">#&gt;                       &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt;        [1]             chr1   29554-31097      + |         1  uc057aty.1</span>
<span class="co">#&gt;        [2]             chr1   30267-31109      + |         2  uc057atz.1</span>
<span class="co">#&gt;        [3]             chr1   30366-30503      + |         3  uc031tlb.1</span>
<span class="co">#&gt;        [4]             chr1   69091-70008      + |         4  uc001aal.1</span>
<span class="co">#&gt;        [5]             chr1 160446-161525      + |         5  uc057aum.1</span>
<span class="co">#&gt;        ...              ...           ...    ... .       ...         ...</span>
<span class="co">#&gt;   [197778] chrUn_KI270750v1 148668-148843      + |    197778  uc064xrp.1</span>
<span class="co">#&gt;   [197779] chrUn_KI270752v1       144-268      + |    197779  uc064xrq.1</span>
<span class="co">#&gt;   [197780] chrUn_KI270752v1   21813-21944      + |    197780  uc064xrt.1</span>
<span class="co">#&gt;   [197781] chrUn_KI270752v1     3497-3623      - |    197781  uc064xrr.1</span>
<span class="co">#&gt;   [197782] chrUn_KI270752v1    9943-10067      - |    197782  uc064xrs.1</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>Keep only the standard chromosomes, to work smoothly with our <code>cpg</code> object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tx &lt;-<span class="st"> </span><span class="kw">keepStandardChromosomes</span>(tx, <span class="dt">pruning.mode=</span><span class="st">&quot;coarse&quot;</span>)
tx
<span class="co">#&gt; GRanges object with 182435 ranges and 2 metadata columns:</span>
<span class="co">#&gt;            seqnames        ranges strand |     tx_id     tx_name</span>
<span class="co">#&gt;               &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt;        [1]     chr1   29554-31097      + |         1  uc057aty.1</span>
<span class="co">#&gt;        [2]     chr1   30267-31109      + |         2  uc057atz.1</span>
<span class="co">#&gt;        [3]     chr1   30366-30503      + |         3  uc031tlb.1</span>
<span class="co">#&gt;        [4]     chr1   69091-70008      + |         4  uc001aal.1</span>
<span class="co">#&gt;        [5]     chr1 160446-161525      + |         5  uc057aum.1</span>
<span class="co">#&gt;        ...      ...           ...    ... .       ...         ...</span>
<span class="co">#&gt;   [182431]     chrM     5826-5891      - |    182431  uc064xpa.1</span>
<span class="co">#&gt;   [182432]     chrM     7446-7514      - |    182432  uc064xpb.1</span>
<span class="co">#&gt;   [182433]     chrM   14149-14673      - |    182433  uc064xpm.1</span>
<span class="co">#&gt;   [182434]     chrM   14674-14742      - |    182434  uc022bqv.3</span>
<span class="co">#&gt;   [182435]     chrM   15956-16023      - |    182435  uc022bqx.2</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 25 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p><strong>Overlaps</strong></p>
<p>A very useful operation is to count overlaps in two distinct genomic ranges objects. The following counts the number of CpG islands that overlap each transcript. Related functions include <code>findOverlaps()</code>, <code>nearest()</code>, <code>precede()</code>, and <code>follow()</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">olaps &lt;-<span class="st"> </span><span class="kw">countOverlaps</span>(tx, cpg)
<span class="kw">length</span>(olaps)     <span class="co"># 1 count per transcript</span>
<span class="co">#&gt; [1] 182435</span>
<span class="kw">table</span>(olaps)
<span class="co">#&gt; olaps</span>
<span class="co">#&gt;     0     1     2     3     4     5     6     7     8     9    10    11 </span>
<span class="co">#&gt; 94621 70551 10983  3228  1317   595   351   214   153    93    64    39 </span>
<span class="co">#&gt;    12    13    14    15    16    17    18    19    20    21    22    23 </span>
<span class="co">#&gt;    41    31    21    20    17     8    14     8     7     3     6     6 </span>
<span class="co">#&gt;    24    25    26    27    28    29    30    31    32    33    34    35 </span>
<span class="co">#&gt;     3     2     4     1     3     2     6     5     3     3     1     2 </span>
<span class="co">#&gt;    36    37    38    63 </span>
<span class="co">#&gt;     3     1     1     4</span></code></pre></div>
<p>Calculations such as <code>countOverlaps()</code> can be added to the <code>GRanges</code> object, tightly coupling derived data with the original annotation.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tx<span class="op">$</span>cpgOverlaps &lt;-<span class="st"> </span><span class="kw">countOverlaps</span>(tx, cpg)
tx
<span class="co">#&gt; GRanges object with 182435 ranges and 3 metadata columns:</span>
<span class="co">#&gt;            seqnames        ranges strand |     tx_id     tx_name</span>
<span class="co">#&gt;               &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt;        [1]     chr1   29554-31097      + |         1  uc057aty.1</span>
<span class="co">#&gt;        [2]     chr1   30267-31109      + |         2  uc057atz.1</span>
<span class="co">#&gt;        [3]     chr1   30366-30503      + |         3  uc031tlb.1</span>
<span class="co">#&gt;        [4]     chr1   69091-70008      + |         4  uc001aal.1</span>
<span class="co">#&gt;        [5]     chr1 160446-161525      + |         5  uc057aum.1</span>
<span class="co">#&gt;        ...      ...           ...    ... .       ...         ...</span>
<span class="co">#&gt;   [182431]     chrM     5826-5891      - |    182431  uc064xpa.1</span>
<span class="co">#&gt;   [182432]     chrM     7446-7514      - |    182432  uc064xpb.1</span>
<span class="co">#&gt;   [182433]     chrM   14149-14673      - |    182433  uc064xpm.1</span>
<span class="co">#&gt;   [182434]     chrM   14674-14742      - |    182434  uc022bqv.3</span>
<span class="co">#&gt;   [182435]     chrM   15956-16023      - |    182435  uc022bqx.2</span>
<span class="co">#&gt;            cpgOverlaps</span>
<span class="co">#&gt;              &lt;integer&gt;</span>
<span class="co">#&gt;        [1]           1</span>
<span class="co">#&gt;        [2]           0</span>
<span class="co">#&gt;        [3]           0</span>
<span class="co">#&gt;        [4]           0</span>
<span class="co">#&gt;        [5]           0</span>
<span class="co">#&gt;        ...         ...</span>
<span class="co">#&gt;   [182431]           0</span>
<span class="co">#&gt;   [182432]           0</span>
<span class="co">#&gt;   [182433]           0</span>
<span class="co">#&gt;   [182434]           0</span>
<span class="co">#&gt;   [182435]           0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 25 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>It is then possible to perform coordinated actions, e.g., subsetting the <code>GRanges</code> objects for transcripts satisfying particular conditions, in a coordinated fashion where the software does all the book-keeping to makes sure the correct ranges are selected.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">subset</span>(tx, cpgOverlaps <span class="op">&gt;</span><span class="st"> </span><span class="dv">10</span>)
<span class="co">#&gt; GRanges object with 265 ranges and 3 metadata columns:</span>
<span class="co">#&gt;         seqnames            ranges strand |     tx_id     tx_name</span>
<span class="co">#&gt;            &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt; &lt;character&gt;</span>
<span class="co">#&gt;     [1]     chr1   2050470-2185395      + |       213  uc001aiq.3</span>
<span class="co">#&gt;     [2]     chr1   2050485-2146108      + |       214  uc057bkd.1</span>
<span class="co">#&gt;     [3]     chr1   2073462-2185390      + |       219  uc001air.4</span>
<span class="co">#&gt;     [4]     chr1   2073986-2185190      + |       221  uc010nyw.3</span>
<span class="co">#&gt;     [5]     chr1   2104716-2185393      + |       227  uc001ais.4</span>
<span class="co">#&gt;     ...      ...               ...    ... .       ...         ...</span>
<span class="co">#&gt;   [261]     chrX 40051246-40177329      - |    179887  uc004deo.4</span>
<span class="co">#&gt;   [262]     chrX 40051248-40177329      - |    179889  uc004dep.5</span>
<span class="co">#&gt;   [263]     chrX 40062955-40177320      - |    179892  uc064yov.1</span>
<span class="co">#&gt;   [264]     chrY     333963-386710      - |    182211  uc004fot.4</span>
<span class="co">#&gt;   [265]     chrY     344896-386955      - |    182217  uc011nah.3</span>
<span class="co">#&gt;         cpgOverlaps</span>
<span class="co">#&gt;           &lt;integer&gt;</span>
<span class="co">#&gt;     [1]          15</span>
<span class="co">#&gt;     [2]          11</span>
<span class="co">#&gt;     [3]          13</span>
<span class="co">#&gt;     [4]          13</span>
<span class="co">#&gt;     [5]          11</span>
<span class="co">#&gt;     ...         ...</span>
<span class="co">#&gt;   [261]          11</span>
<span class="co">#&gt;   [262]          11</span>
<span class="co">#&gt;   [263]          11</span>
<span class="co">#&gt;   [264]          14</span>
<span class="co">#&gt;   [265]          12</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 25 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>Can you think of other situations where one might calculate derived values and couple these with <code>GRanges</code> or similar objects?</p>
</div>
<div id="working-with-summarized-experimental-data" class="section level3">
<h3><span class="header-section-number">2.3.3</span> Working with summarized experimental data</h3>
<p>This section introduces another broadly useful package and data structure, the <a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a> package and <code>SummarizedExperiment</code> object.</p>
<div class="figure">
<img src="100_Morgan_RBiocForAll/SummarizedExperiment.png" />

</div>
<p>The <code>SummarizedExperiment</code> object has matrix-like properties – it has two dimensions and can be subset by ‘rows’ and ‘columns’. The <code>assay()</code> data of a <code>SummarizedExperiment</code> experiment contains one or more matrix-like objects where rows represent features of interest (e.g., genes), columns represent samples, and elements of the matrix represent results of a genomic assay (e.g., counts of reads overlaps genes in each sample of an bulk RNA-seq differential expression assay.</p>
<p><strong>Object construction</strong></p>
<p>The <code>SummarizedExperiment</code> coordinates assays with (optional) descriptions of rows and columns. We start by reading in a simple <code>data.frame</code> describing 8 samples from an RNASeq experiment looking at dexamethasone treatment across 4 human smooth muscle cell lines; use <code>browseVignettes(&quot;airway&quot;)</code> for a more complete description of the experiment and data processing. Read the column data in using <code>file.choose()</code> and <code>read.csv()</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fname &lt;-<span class="st"> </span><span class="kw">file.choose</span>()  <span class="co"># airway_colData.csv</span>
fname</code></pre></div>
<p>We want the first column the the data to be treated as row names (sample identifiers) in the <code>data.frame</code>, so <code>read.csv()</code> has an extra argument to indicate this.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">colData &lt;-<span class="st"> </span><span class="kw">read.csv</span>(fname, <span class="dt">row.names =</span> <span class="dv">1</span>)
colData
<span class="co">#&gt;            SampleName    cell   dex albut        Run avgLength Experiment</span>
<span class="co">#&gt; SRR1039508 GSM1275862  N61311 untrt untrt SRR1039508       126  SRX384345</span>
<span class="co">#&gt; SRR1039509 GSM1275863  N61311   trt untrt SRR1039509       126  SRX384346</span>
<span class="co">#&gt; SRR1039512 GSM1275866 N052611 untrt untrt SRR1039512       126  SRX384349</span>
<span class="co">#&gt; SRR1039513 GSM1275867 N052611   trt untrt SRR1039513        87  SRX384350</span>
<span class="co">#&gt; SRR1039516 GSM1275870 N080611 untrt untrt SRR1039516       120  SRX384353</span>
<span class="co">#&gt; SRR1039517 GSM1275871 N080611   trt untrt SRR1039517       126  SRX384354</span>
<span class="co">#&gt; SRR1039520 GSM1275874 N061011 untrt untrt SRR1039520       101  SRX384357</span>
<span class="co">#&gt; SRR1039521 GSM1275875 N061011   trt untrt SRR1039521        98  SRX384358</span>
<span class="co">#&gt;               Sample    BioSample</span>
<span class="co">#&gt; SRR1039508 SRS508568 SAMN02422669</span>
<span class="co">#&gt; SRR1039509 SRS508567 SAMN02422675</span>
<span class="co">#&gt; SRR1039512 SRS508571 SAMN02422678</span>
<span class="co">#&gt; SRR1039513 SRS508572 SAMN02422670</span>
<span class="co">#&gt; SRR1039516 SRS508575 SAMN02422682</span>
<span class="co">#&gt; SRR1039517 SRS508576 SAMN02422673</span>
<span class="co">#&gt; SRR1039520 SRS508579 SAMN02422683</span>
<span class="co">#&gt; SRR1039521 SRS508580 SAMN02422677</span></code></pre></div>
<p>The data are from the Short Read Archive, and the row names, <code>SampleName</code>, <code>Run</code>, <code>Experiment</code>, <code>Sampel</code>, and <code>BioSample</code> columns are classifications from the archive. Additional columns include:</p>
<ul>
<li><code>cell</code>: the cell line used. There are four cell lines.</li>
<li><code>dex</code>: whether the sample was untreated, or treated with dexamethasone.</li>
<li><code>albut</code>: a second treatment, which we ignore</li>
<li><code>avgLength</code>: the sample-specific average length of the RNAseq reads estimated in the experiment.</li>
</ul>
<p><strong>Assay data</strong></p>
<p>Now import the assay data from the file “airway_counts.csv”</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">fname &lt;-<span class="st"> </span><span class="kw">file.choose</span>()  <span class="co"># airway_counts.csv</span>
fname</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">counts &lt;-<span class="st"> </span><span class="kw">read.csv</span>(fname, <span class="dt">row.names=</span><span class="dv">1</span>)</code></pre></div>
<p>Although the data are read as a <code>data.frame</code>, all columns are of the same type (integer-valued) and represent the same attribute; the data is really a <code>matrix</code> rather than <code>data.frame</code>, so we coerce to matrix using <code>as.matrix()</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">counts &lt;-<span class="st"> </span><span class="kw">as.matrix</span>(counts)</code></pre></div>
<p>We see the dimensions and first few rows of the counts matrix</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dim</span>(counts)
<span class="co">#&gt; [1] 33469     8</span>
<span class="kw">head</span>(counts)
<span class="co">#&gt;                 SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516</span>
<span class="co">#&gt; ENSG00000000003        679        448        873        408       1138</span>
<span class="co">#&gt; ENSG00000000419        467        515        621        365        587</span>
<span class="co">#&gt; ENSG00000000457        260        211        263        164        245</span>
<span class="co">#&gt; ENSG00000000460         60         55         40         35         78</span>
<span class="co">#&gt; ENSG00000000938          0          0          2          0          1</span>
<span class="co">#&gt; ENSG00000000971       3251       3679       6177       4252       6721</span>
<span class="co">#&gt;                 SRR1039517 SRR1039520 SRR1039521</span>
<span class="co">#&gt; ENSG00000000003       1047        770        572</span>
<span class="co">#&gt; ENSG00000000419        799        417        508</span>
<span class="co">#&gt; ENSG00000000457        331        233        229</span>
<span class="co">#&gt; ENSG00000000460         63         76         60</span>
<span class="co">#&gt; ENSG00000000938          0          0          0</span>
<span class="co">#&gt; ENSG00000000971      11027       5176       7995</span></code></pre></div>
<p>It’s interesting to think about what the counts mean – for ENSG00000000003, sample SRR1039508 had 679 reads that overlapped this gene, sample SRR1039509 had 448 reads, etc. Notice that for this gene there seems to be a consistent pattern – within a cell line, the read counts in the untreated group are always larger than the read counts for the treated group. This and other basic observations from ‘looking at’ the data motivate many steps in a rigorous RNASeq differential expression analysis.</p>
<p><strong>Creating a <code>SummarizedExperiment</code> object</strong></p>
<p>We saw earlier that there was considerable value in tightly coupling the count of CpG islands overlapping each transcript with the <code>GRanges</code> describing the transcripts. We can anticipate that close coupling of the column data with the assay data will have similar benefits, e.g., reducing the chances of bookkeeping errors as we work with our data.</p>
<p>Attach the <a href="https://bioconductor.org/packages/SummarizedExperiment">SummarizedExperiment</a> library to our <em>R</em> session.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;SummarizedExperiment&quot;</span>)</code></pre></div>
<p>Use the <code>SummarizedExperiment()</code> function to coordinate the assay and column data; this function uses row and column names to make sure the correct assay columns are desribed by the correct column data rows.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">se &lt;-<span class="st"> </span><span class="kw">SummarizedExperiment</span>(<span class="dt">assay =</span> counts, <span class="dt">colData =</span> colData)
se
<span class="co">#&gt; class: SummarizedExperiment </span>
<span class="co">#&gt; dim: 33469 8 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): &#39;&#39;</span>
<span class="co">#&gt; rownames(33469): ENSG00000000003 ENSG00000000419 ...</span>
<span class="co">#&gt;   ENSG00000273492 ENSG00000273493</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span>
<span class="co">#&gt; colData names(9): SampleName cell ... Sample BioSample</span></code></pre></div>
<p>It is straight-forward to use <code>subset()</code> on <code>SummarizedExperiment</code> to create subsets of the data in a coordinated way. Remember that a <code>SummarizedExperiment</code> is conceptually two-dimensional (matrix-like), and in the example below we are subsetting on the second dimension.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">subset</span>(se, , dex <span class="op">==</span><span class="st"> &quot;trt&quot;</span>)
<span class="co">#&gt; class: SummarizedExperiment </span>
<span class="co">#&gt; dim: 33469 4 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): &#39;&#39;</span>
<span class="co">#&gt; rownames(33469): ENSG00000000003 ENSG00000000419 ...</span>
<span class="co">#&gt;   ENSG00000273492 ENSG00000273493</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(4): SRR1039509 SRR1039513 SRR1039517 SRR1039521</span>
<span class="co">#&gt; colData names(9): SampleName cell ... Sample BioSample</span></code></pre></div>
<p>As with <code>GRanges</code>, there are accessors that extract data from the <code>SummarizedExperiment</code>. For instance, we can use <code>assay()</code> to extract the count matrix, and <code>colSums()</code> to calculate the library size (total number of reads overlapping genes in each sample).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">colSums</span>(<span class="kw">assay</span>(se))
<span class="co">#&gt; SRR1039508 SRR1039509 SRR1039512 SRR1039513 SRR1039516 SRR1039517 </span>
<span class="co">#&gt;   20637971   18809481   25348649   15163415   24448408   30818215 </span>
<span class="co">#&gt; SRR1039520 SRR1039521 </span>
<span class="co">#&gt;   19126151   21164133</span></code></pre></div>
<p>Note that library sizes differ by a factor of 2 from largest to smallest; how would this influence the interpretation of counts in individual cells of the assay data?</p>
<p>As with <code>GRanges</code>, it might be useful to remember important computations in a way that is robust, e.g.,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">se<span class="op">$</span>lib.size &lt;-<span class="st"> </span><span class="kw">colSums</span>(<span class="kw">assay</span>(se))
<span class="kw">colData</span>(se)
<span class="co">#&gt; DataFrame with 8 rows and 10 columns</span>
<span class="co">#&gt;            SampleName     cell      dex    albut        Run avgLength</span>
<span class="co">#&gt;              &lt;factor&gt; &lt;factor&gt; &lt;factor&gt; &lt;factor&gt;   &lt;factor&gt; &lt;integer&gt;</span>
<span class="co">#&gt; SRR1039508 GSM1275862   N61311    untrt    untrt SRR1039508       126</span>
<span class="co">#&gt; SRR1039509 GSM1275863   N61311      trt    untrt SRR1039509       126</span>
<span class="co">#&gt; SRR1039512 GSM1275866  N052611    untrt    untrt SRR1039512       126</span>
<span class="co">#&gt; SRR1039513 GSM1275867  N052611      trt    untrt SRR1039513        87</span>
<span class="co">#&gt; SRR1039516 GSM1275870  N080611    untrt    untrt SRR1039516       120</span>
<span class="co">#&gt; SRR1039517 GSM1275871  N080611      trt    untrt SRR1039517       126</span>
<span class="co">#&gt; SRR1039520 GSM1275874  N061011    untrt    untrt SRR1039520       101</span>
<span class="co">#&gt; SRR1039521 GSM1275875  N061011      trt    untrt SRR1039521        98</span>
<span class="co">#&gt;            Experiment    Sample    BioSample  lib.size</span>
<span class="co">#&gt;              &lt;factor&gt;  &lt;factor&gt;     &lt;factor&gt; &lt;numeric&gt;</span>
<span class="co">#&gt; SRR1039508  SRX384345 SRS508568 SAMN02422669  20637971</span>
<span class="co">#&gt; SRR1039509  SRX384346 SRS508567 SAMN02422675  18809481</span>
<span class="co">#&gt; SRR1039512  SRX384349 SRS508571 SAMN02422678  25348649</span>
<span class="co">#&gt; SRR1039513  SRX384350 SRS508572 SAMN02422670  15163415</span>
<span class="co">#&gt; SRR1039516  SRX384353 SRS508575 SAMN02422682  24448408</span>
<span class="co">#&gt; SRR1039517  SRX384354 SRS508576 SAMN02422673  30818215</span>
<span class="co">#&gt; SRR1039520  SRX384357 SRS508579 SAMN02422683  19126151</span>
<span class="co">#&gt; SRR1039521  SRX384358 SRS508580 SAMN02422677  21164133</span></code></pre></div>
</div>
<div id="down-stream-analysis" class="section level3">
<h3><span class="header-section-number">2.3.4</span> Down-stream analysis</h3>
<p>In this final section we quickly hint at down-stream analysis, and the way in which skills learned in working with <em>Bioconductor</em> objects in one package translate to working with objects in other packages.</p>
<p>We start by loading the <a href="https://bioconductor.org/packages/DESeq2">DESeq2</a> package, a very popular facility for analysing bulk RNAseq differential experssion data.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(<span class="st">&quot;DESeq2&quot;</span>)</code></pre></div>
<p>The package requires count data like that in the <code>SummarizedExperiment</code> we have been working with, in addition to a <code>formula</code> describing the experimental design. Some of the observations above suggest that we should include cell line as a covariate, and dexamethazone treatment as the main factor that we are interested in.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds &lt;-<span class="st"> </span><span class="kw">DESeqDataSet</span>(se, <span class="dt">design =</span> <span class="op">~</span><span class="st"> </span>cell <span class="op">+</span><span class="st"> </span>dex)
<span class="co">#&gt; renaming the first element in assays to &#39;counts&#39;</span>
dds
<span class="co">#&gt; class: DESeqDataSet </span>
<span class="co">#&gt; dim: 33469 8 </span>
<span class="co">#&gt; metadata(1): version</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(33469): ENSG00000000003 ENSG00000000419 ...</span>
<span class="co">#&gt;   ENSG00000273492 ENSG00000273493</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span>
<span class="co">#&gt; colData names(10): SampleName cell ... BioSample lib.size</span></code></pre></div>
<p>The <code>dds</code> object can be manipulated very much like a <code>SummarizedExperiment</code>.</p>
<p>The essention DESeq work flow is summarized by a single function call, which performs advanced statistical analysis on the data in the <code>dds</code> object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">dds &lt;-<span class="st"> </span><span class="kw">DESeq</span>(dds)
<span class="co">#&gt; estimating size factors</span>
<span class="co">#&gt; estimating dispersions</span>
<span class="co">#&gt; gene-wise dispersion estimates</span>
<span class="co">#&gt; mean-dispersion relationship</span>
<span class="co">#&gt; final dispersion estimates</span>
<span class="co">#&gt; fitting model and testing</span></code></pre></div>
<p>A table summarizing measures of differential expression can be extracted from the object, and visualized or manipulated using commands we learned earlier today.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">results</span>(dds)
<span class="co">#&gt; log2 fold change (MLE): dex untrt vs trt </span>
<span class="co">#&gt; Wald test p-value: dex untrt vs trt </span>
<span class="co">#&gt; DataFrame with 33469 rows and 6 columns</span>
<span class="co">#&gt;                          baseMean     log2FoldChange             lfcSE</span>
<span class="co">#&gt;                         &lt;numeric&gt;          &lt;numeric&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt; ENSG00000000003  708.602169691234  0.381253982523047 0.100654411865821</span>
<span class="co">#&gt; ENSG00000000419  520.297900552084 -0.206812601085051 0.112218646507256</span>
<span class="co">#&gt; ENSG00000000457  237.163036796015 -0.037920431205081 0.143444654934088</span>
<span class="co">#&gt; ENSG00000000460  57.9326331250967 0.0881681758701336 0.287141822937769</span>
<span class="co">#&gt; ENSG00000000938 0.318098378392895   1.37822703433213  3.49987280259267</span>
<span class="co">#&gt; ...                           ...                ...               ...</span>
<span class="co">#&gt; ENSG00000273487   8.1632349843654  -1.04640654119647 0.698966822033151</span>
<span class="co">#&gt; ENSG00000273488  8.58447903624707 -0.107830154768606 0.638099034888812</span>
<span class="co">#&gt; ENSG00000273489 0.275899382507797  -1.48373838425916  3.51394583815493</span>
<span class="co">#&gt; ENSG00000273492 0.105978355992386  0.463679009985687  3.52308267589304</span>
<span class="co">#&gt; ENSG00000273493 0.106141666408122  0.521354507611989  3.53139024554344</span>
<span class="co">#&gt;                               stat               pvalue</span>
<span class="co">#&gt;                          &lt;numeric&gt;            &lt;numeric&gt;</span>
<span class="co">#&gt; ENSG00000000003   3.78775232457058 0.000152016273044953</span>
<span class="co">#&gt; ENSG00000000419  -1.84294328546976   0.0653372915097615</span>
<span class="co">#&gt; ENSG00000000457 -0.264355832725208     0.79150574160836</span>
<span class="co">#&gt; ENSG00000000460  0.307054454722334    0.758801923982929</span>
<span class="co">#&gt; ENSG00000000938  0.393793463954221    0.693733530342183</span>
<span class="co">#&gt; ...                            ...                  ...</span>
<span class="co">#&gt; ENSG00000273487  -1.49707612466166    0.134373451371486</span>
<span class="co">#&gt; ENSG00000273488 -0.168986550477068    0.865807220420437</span>
<span class="co">#&gt; ENSG00000273489 -0.422242815511986    0.672847792938571</span>
<span class="co">#&gt; ENSG00000273492  0.131611731157615    0.895291406125436</span>
<span class="co">#&gt; ENSG00000273493  0.147634351165219    0.882631343839791</span>
<span class="co">#&gt;                                padj</span>
<span class="co">#&gt;                           &lt;numeric&gt;</span>
<span class="co">#&gt; ENSG00000000003 0.00127423101752341</span>
<span class="co">#&gt; ENSG00000000419   0.195433088560704</span>
<span class="co">#&gt; ENSG00000000457   0.910602138773154</span>
<span class="co">#&gt; ENSG00000000460   0.893977566267183</span>
<span class="co">#&gt; ENSG00000000938                  NA</span>
<span class="co">#&gt; ...                             ...</span>
<span class="co">#&gt; ENSG00000273487   0.327756206316493</span>
<span class="co">#&gt; ENSG00000273488   0.944617232242564</span>
<span class="co">#&gt; ENSG00000273489                  NA</span>
<span class="co">#&gt; ENSG00000273492                  NA</span>
<span class="co">#&gt; ENSG00000273493                  NA</span></code></pre></div>
</div>
</div>
<div id="summary" class="section level2">
<h2><span class="header-section-number">2.4</span> Summary</h2>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="2">
<li id="fn2"><p>Roswell Park Comprehensive Cancer Center, Buffalo, NY<a href="r-and-bioconductor-for-everyone-an-introduction.html#fnref2">↩</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="introduction-to-bioconductor-annotation-resources.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
