<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="public-data-resources-and-bioconductor.html">
<link rel="next" href="rna-seq-data-analysis-with-deseq2.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-122837239-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-122837239-1');
</script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-workshops"><i class="fa fa-check"></i><b>1.1</b> The Workshops</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#how-to-use-these-workshops"><i class="fa fa-check"></i><b>1.2</b> How to use these workshops</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#install-on-your-own-computer"><i class="fa fa-check"></i><b>1.2.1</b> Install on your own computer</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#use-the-exact-same-ami-as-the-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Use the exact same AMI as the workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#make-your-own-docker-image-and-ami"><i class="fa fa-check"></i><b>1.2.3</b> Make your own Docker image and AMI</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html"><i class="fa fa-check"></i><b>2</b> 100: <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#description"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="2.1.7" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-goals"><i class="fa fa-check"></i><b>2.1.7</b> Learning goals</a></li>
<li class="chapter" data-level="2.1.8" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-objectives"><i class="fa fa-check"></i><b>2.1.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-r"><i class="fa fa-check"></i><b>2.2</b> Introduction to <em>R</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#rstudio-orientation"><i class="fa fa-check"></i><b>2.2.1</b> <em>RStudio</em> orientation</a></li>
<li class="chapter" data-level="2.2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#very-basics"><i class="fa fa-check"></i><b>2.2.2</b> Very basics</a></li>
<li class="chapter" data-level="2.2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#data-input-and-manipulation"><i class="fa fa-check"></i><b>2.2.3</b> Data input and manipulation</a></li>
<li class="chapter" data-level="2.2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#writing-r-scripts"><i class="fa fa-check"></i><b>2.2.4</b> Writing <em>R</em> scripts</a></li>
<li class="chapter" data-level="2.2.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#using-packages"><i class="fa fa-check"></i><b>2.2.5</b> Using packages</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.3</b> Introduction to <em>Bioconductor</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#discovering-installing-and-learning-how-to-use-bioconductor-packages"><i class="fa fa-check"></i><b>2.3.1</b> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-genomic-ranges"><i class="fa fa-check"></i><b>2.3.2</b> Working with Genomic Ranges</a></li>
<li class="chapter" data-level="2.3.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-summarized-experimental-data"><i class="fa fa-check"></i><b>2.3.3</b> Working with summarized experimental data</a></li>
<li class="chapter" data-level="2.3.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#down-stream-analysis"><i class="fa fa-check"></i><b>2.3.4</b> Down-stream analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>3</b> 101: Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>3.1</b> Instructors</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description"><i class="fa fa-check"></i><b>3.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="3.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>3.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="3.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation"><i class="fa fa-check"></i><b>3.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="3.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="3.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-1"><i class="fa fa-check"></i><b>3.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="3.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-1"><i class="fa fa-check"></i><b>3.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>3.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="3.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>3.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="3.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>3.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="3.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>3.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="3.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>3.4.4</b> Data containers</a></li>
<li class="chapter" data-level="3.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>3.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="3.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>3.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>3.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>3.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="3.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>3.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="3.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>3.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="3.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>3.4.11</b> Simple example</a></li>
<li class="chapter" data-level="3.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>3.4.12</b> Questions!</a></li>
<li class="chapter" data-level="3.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>3.4.13</b> More questions!</a></li>
<li class="chapter" data-level="3.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>3.4.14</b> Another example</a></li>
<li class="chapter" data-level="3.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>3.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="3.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>3.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="3.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>3.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="3.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>3.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="3.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>3.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="3.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>3.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="3.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>3.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="3.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>3.4.22</b> GRanges</a></li>
<li class="chapter" data-level="3.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>3.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="3.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>3.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="3.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>3.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="3.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>3.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="3.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>3.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="3.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>3.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="3.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>3.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>3.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>3.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="3.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>3.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="3.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr"><i class="fa fa-check"></i><b>3.4.33</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-1"><i class="fa fa-check"></i><b>3.4.34</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-2"><i class="fa fa-check"></i><b>3.4.35</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>3.4.36</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="3.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>3.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>3.4.38</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>3.4.39</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>3.4.40</b> BSgenome exercises</a></li>
<li class="chapter" data-level="3.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub"><i class="fa fa-check"></i><b>3.4.41</b> AnnotationHub</a></li>
<li class="chapter" data-level="3.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub"><i class="fa fa-check"></i><b>3.4.42</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="3.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>3.4.43</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="3.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>3.4.44</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="3.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>3.4.45</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="3.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>3.4.46</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="3.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>3.4.47</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>3.4.48</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>3.4.49</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="3.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>3.4.50</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="3.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>3.4.51</b> biomaRt</a></li>
<li class="chapter" data-level="3.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>3.4.52</b> biomaRt data sets</a></li>
<li class="chapter" data-level="3.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>3.4.53</b> biomaRt queries</a></li>
<li class="chapter" data-level="3.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>3.4.54</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="3.4.55" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>3.4.55</b> biomaRt query</a></li>
<li class="chapter" data-level="3.4.56" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>3.4.56</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>4</b> 102: Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>4.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-1"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-1"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-1"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-1"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="4.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-2"><i class="fa fa-check"></i><b>4.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="4.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-2"><i class="fa fa-check"></i><b>4.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#introduction-1"><i class="fa fa-check"></i><b>4.4</b> Introduction</a><ul>
<li class="chapter" data-level="4.4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#what-is-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.1</b> What is the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#why-use-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.2</b> Why use the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>4.4.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#setup"><i class="fa fa-check"></i><b>4.5</b> Setup</a></li>
<li class="chapter" data-level="4.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>4.6</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="4.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#constructing-a-granges-object-from-data.frame"><i class="fa fa-check"></i><b>4.7</b> Constructing a <em>GRanges</em> object from data.frame</a></li>
<li class="chapter" data-level="4.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#loading-a-granges-object-from-a-standard-file-format"><i class="fa fa-check"></i><b>4.8</b> Loading a <em>GRanges</em> object from a standard file format</a></li>
<li class="chapter" data-level="4.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-manipulation-of-granges-objects"><i class="fa fa-check"></i><b>4.9</b> Basic manipulation of <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>4.10</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>4.11</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#aggregating-granges-objects"><i class="fa fa-check"></i><b>4.12</b> Aggregating <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.13</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.14</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.15" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#finding-overlaps-between-granges-objects"><i class="fa fa-check"></i><b>4.15</b> Finding overlaps between <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.16" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises"><i class="fa fa-check"></i><b>4.16</b> Exercises</a></li>
<li class="chapter" data-level="4.17" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>4.17</b> Example: exploring BigWig files from AnnotationHub</a><ul>
<li class="chapter" data-level="4.17.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#extracting-data-from-annotationhub"><i class="fa fa-check"></i><b>4.17.1</b> Extracting data from AnnotationHub</a></li>
<li class="chapter" data-level="4.17.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#reading-bigwig-files"><i class="fa fa-check"></i><b>4.17.2</b> Reading BigWig files</a></li>
<li class="chapter" data-level="4.17.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-1"><i class="fa fa-check"></i><b>4.17.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.18" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>4.18</b> Worked example: coverage analysis of BAM files</a><ul>
<li class="chapter" data-level="4.18.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-2"><i class="fa fa-check"></i><b>4.18.1</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.19" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#conclusions"><i class="fa fa-check"></i><b>4.19</b> Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>5</b> 103: Public data resources and Bioconductor</a><ul>
<li class="chapter" data-level="5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>5.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>5.2</b> Syllabus</a><ul>
<li class="chapter" data-level="5.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-2"><i class="fa fa-check"></i><b>5.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="5.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-2"><i class="fa fa-check"></i><b>5.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="5.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-2"><i class="fa fa-check"></i><b>5.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>5.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="5.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-2"><i class="fa fa-check"></i><b>5.2.5</b> Time outline</a></li>
<li class="chapter" data-level="5.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>5.2.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="5.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-3"><i class="fa fa-check"></i><b>5.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="5.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-3"><i class="fa fa-check"></i><b>5.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-1"><i class="fa fa-check"></i><b>5.3</b> Overview</a></li>
<li class="chapter" data-level="5.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>5.4</b> GEOquery</a><ul>
<li class="chapter" data-level="5.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>5.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="5.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>5.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="5.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>5.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="5.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>5.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>5.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="5.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>5.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="5.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>5.5.2</b> Basic design</a></li>
<li class="chapter" data-level="5.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>5.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>5.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="5.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>5.6.1</b> Authentication</a></li>
<li class="chapter" data-level="5.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>5.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>5.7</b> Sequence Read Archive</a><ul>
<li class="chapter" data-level="5.7.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage-1"><i class="fa fa-check"></i><b>5.7.1</b> Usage</a></li>
<li class="chapter" data-level="5.7.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#query-syntax"><i class="fa fa-check"></i><b>5.7.2</b> Query syntax</a></li>
<li class="chapter" data-level="5.7.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#using-the-raw-api-without-rbioconductor"><i class="fa fa-check"></i><b>5.7.3</b> Using the raw API without R/Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>5.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="5.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>5.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="5.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>5.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="5.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>5.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="5.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>5.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="5.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>5.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="5.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>5.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="5.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>5.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>5.12</b> Pharmacogenomics</a><ul>
<li class="chapter" data-level="5.12.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started"><i class="fa fa-check"></i><b>5.12.1</b> Getting started</a></li>
<li class="chapter" data-level="5.12.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-pharmacogx-datasets-pharmacosets"><i class="fa fa-check"></i><b>5.12.2</b> Overview of PharmacoGx datasets (PharmacoSets)</a></li>
<li class="chapter" data-level="5.12.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#exploring-drug-sensitivity-datasets"><i class="fa fa-check"></i><b>5.12.3</b> Exploring drug sensitivity datasets</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>5.13</b> Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>6</b> 200: RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>6.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>6.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="6.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>6.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="6.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>6.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-3"><i class="fa fa-check"></i><b>6.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>6.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="6.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-4"><i class="fa fa-check"></i><b>6.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="6.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-4"><i class="fa fa-check"></i><b>6.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-2"><i class="fa fa-check"></i><b>6.4</b> Introduction</a></li>
<li class="chapter" data-level="6.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>6.5</b> Data packaging</a><ul>
<li class="chapter" data-level="6.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>6.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="6.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>6.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="6.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>6.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>6.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="6.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>6.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="6.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>6.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="6.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>6.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="6.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>6.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis"><i class="fa fa-check"></i><b>6.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="6.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>6.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="6.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>6.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="6.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>6.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="6.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>6.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="6.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>6.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="6.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>6.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>7</b> 201: RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview-2"><i class="fa fa-check"></i><b>7.1</b> Overview</a><ul>
<li class="chapter" data-level="7.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-1"><i class="fa fa-check"></i><b>7.1.1</b> Description</a></li>
<li class="chapter" data-level="7.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation-1"><i class="fa fa-check"></i><b>7.1.3</b> Participation</a></li>
<li class="chapter" data-level="7.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>7.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-4"><i class="fa fa-check"></i><b>7.1.5</b> Time outline</a></li>
<li class="chapter" data-level="7.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>7.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>7.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="7.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>7.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="7.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>7.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="7.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>7.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="7.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>7.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>7.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="7.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>7.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="7.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>7.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="7.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>7.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>7.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="7.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>7.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="7.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>7.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="7.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>7.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>7.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="7.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>7.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="7.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>7.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub-1"><i class="fa fa-check"></i><b>7.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="7.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>7.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="7.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>7.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>7.7</b> Building reports</a><ul>
<li class="chapter" data-level="7.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>7.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="7.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>7.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>7.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="7.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>7.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>7.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>7.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="7.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>7.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="7.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>7.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="7.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>7.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>8</b> 202: Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="8.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-3"><i class="fa fa-check"></i><b>8.1</b> Overview</a><ul>
<li class="chapter" data-level="8.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-2"><i class="fa fa-check"></i><b>8.1.1</b> Description</a></li>
<li class="chapter" data-level="8.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-2"><i class="fa fa-check"></i><b>8.1.3</b> Participation</a></li>
<li class="chapter" data-level="8.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>8.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-5"><i class="fa fa-check"></i><b>8.1.5</b> Time outline</a></li>
<li class="chapter" data-level="8.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>8.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started-1"><i class="fa fa-check"></i><b>8.2</b> Getting started</a><ul>
<li class="chapter" data-level="8.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>8.2.1</b> The data</a></li>
<li class="chapter" data-level="8.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>8.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>8.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="8.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>8.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="8.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>8.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>8.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="8.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>8.5.1</b> Normalization</a></li>
<li class="chapter" data-level="8.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>8.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>8.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="8.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>8.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="8.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>8.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>9</b> 210: Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="9.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>9.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-4"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-4"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-6"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>9.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="9.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>9.4</b> Where does it all come from?</a></li>
<li class="chapter" data-level="9.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.5</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="9.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>9.6</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="9.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>9.7</b> Data types</a></li>
<li class="chapter" data-level="9.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>9.8</b> Differential expression analysis</a></li>
<li class="chapter" data-level="9.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>9.9</b> Gene sets</a></li>
<li class="chapter" data-level="9.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>9.10</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="9.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>9.11</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="9.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.12</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="9.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>9.13</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>10</b> 220: Workflow for multi-omics analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="10.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information-2"><i class="fa fa-check"></i><b>10.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-5"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-5"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>10.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-7"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-5"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-5"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>10.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="10.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="10.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>10.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="10.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>10.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>10.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="10.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>10.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="10.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>10.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="10.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>10.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>10.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="10.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>10.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="10.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>10.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="10.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>10.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="10.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>10.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="10.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>10.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>10.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="10.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>10.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="10.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>10.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="10.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>10.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>10.9</b> Complete cases</a></li>
<li class="chapter" data-level="10.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>10.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="10.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>10.11</b> Extraction</a><ul>
<li class="chapter" data-level="10.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>10.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="10.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>10.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="10.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>10.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="10.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>10.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="10.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>10.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="10.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>10.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="10.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>10.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="10.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>10.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="10.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>10.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>10.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="10.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>10.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="10.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>10.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="10.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>10.16.1</b> “Simplification” of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="10.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>10.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="10.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>10.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="10.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>10.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="10.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>10.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="10.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>10.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="10.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>10.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="10.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>10.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="10.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>10.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="10.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>10.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="10.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>10.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="10.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>10.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>11</b> 230: Cytoscape automation in R using Rcy3</a><ul>
<li class="chapter" data-level="11.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-4"><i class="fa fa-check"></i><b>11.1</b> Overview</a><ul>
<li class="chapter" data-level="11.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructor-names-and-contact-information-3"><i class="fa fa-check"></i><b>11.1.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="11.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description-6"><i class="fa fa-check"></i><b>11.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="11.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>11.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="11.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>11.1.4</b> Background:</a></li>
<li class="chapter" data-level="11.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation-6"><i class="fa fa-check"></i><b>11.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>11.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="11.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-8"><i class="fa fa-check"></i><b>11.1.7</b> Time outline</a></li>
<li class="chapter" data-level="11.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>11.2</b> Background</a><ul>
<li class="chapter" data-level="11.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>11.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="11.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>11.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="11.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>11.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>11.3</b> Translating biological data into Cytoscape using RCy3</a></li>
<li class="chapter" data-level="11.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>11.4</b> Set up</a><ul>
<li class="chapter" data-level="11.4.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started-2"><i class="fa fa-check"></i><b>11.4.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>11.5</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="11.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>11.6</b> Example Data Set</a></li>
<li class="chapter" data-level="11.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>11.7</b> Finding Network Data</a></li>
<li class="chapter" data-level="11.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>11.8</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="11.9" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>11.9</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="11.10" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>11.10</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>12</b> 240: Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="12.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructor-names-and-contact-information-4"><i class="fa fa-check"></i><b>12.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="12.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-7"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-8"><i class="fa fa-check"></i><b>12.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-7"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>12.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="12.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-9"><i class="fa fa-check"></i><b>12.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-6"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-6"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop"><i class="fa fa-check"></i><b>12.4</b> Workshop</a></li>
<li class="chapter" data-level="12.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-3"><i class="fa fa-check"></i><b>12.5</b> Introduction</a><ul>
<li class="chapter" data-level="12.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>12.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="12.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>12.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="12.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for-1"><i class="fa fa-check"></i><b>12.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup-1"><i class="fa fa-check"></i><b>12.6</b> Setup</a></li>
<li class="chapter" data-level="12.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>12.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="12.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>12.8</b> The Grammar</a><ul>
<li class="chapter" data-level="12.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>12.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="12.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>12.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="12.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises-3"><i class="fa fa-check"></i><b>12.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>12.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="12.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>12.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="12.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files-1"><i class="fa fa-check"></i><b>12.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="12.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>12.10</b> What’s next?</a></li>
<li class="chapter" data-level="12.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>12.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html"><i class="fa fa-check"></i><b>13</b> 250: Working with genomic data in R with the DECIPHER package</a><ul>
<li class="chapter" data-level="13.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#overview-5"><i class="fa fa-check"></i><b>13.1</b> Overview</a><ul>
<li class="chapter" data-level="13.1.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-description-8"><i class="fa fa-check"></i><b>13.1.1</b> Workshop Description</a></li>
<li class="chapter" data-level="13.1.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.1.3" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-participation-8"><i class="fa fa-check"></i><b>13.1.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.1.4" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>13.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="13.1.5" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#time-outline-10"><i class="fa fa-check"></i><b>13.1.5</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.2</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.2.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-goals-7"><i class="fa fa-check"></i><b>13.2.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.2.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-objectives-7"><i class="fa fa-check"></i><b>13.2.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>14</b> 260: Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="14.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>14.1</b> Instructors:</a></li>
<li class="chapter" data-level="14.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-9"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-10"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-9"><i class="fa fa-check"></i><b>14.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>14.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="14.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-11"><i class="fa fa-check"></i><b>14.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-8"><i class="fa fa-check"></i><b>14.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-8"><i class="fa fa-check"></i><b>14.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#abstract"><i class="fa fa-check"></i><b>14.4</b> Abstract</a></li>
<li class="chapter" data-level="14.5" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#introduction-5"><i class="fa fa-check"></i><b>14.5</b> Introduction</a><ul>
<li class="chapter" data-level="14.5.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#downloading-pharmacoset-objects"><i class="fa fa-check"></i><b>14.5.1</b> Downloading PharmacoSet objects</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#reproducibility"><i class="fa fa-check"></i><b>14.6</b> Reproducibility</a><ul>
<li class="chapter" data-level="14.6.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#plotting-drug-dose-response-data"><i class="fa fa-check"></i><b>14.6.1</b> Plotting Drug-Dose Response Data</a></li>
<li class="chapter" data-level="14.6.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pharmacological-profiles"><i class="fa fa-check"></i><b>14.6.2</b> Pharmacological profiles</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#replication"><i class="fa fa-check"></i><b>14.7</b> Replication</a><ul>
<li class="chapter" data-level="14.7.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-of-pharmacological-profiles"><i class="fa fa-check"></i><b>14.7.1</b> Consistency of pharmacological profiles</a></li>
<li class="chapter" data-level="14.7.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-assessment-improved-by-modified-concordance-index"><i class="fa fa-check"></i><b>14.7.2</b> consistency assessment improved by Modified Concordance Index</a></li>
<li class="chapter" data-level="14.7.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#known-biomarkers"><i class="fa fa-check"></i><b>14.7.3</b> Known Biomarkers</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#machine-learning-and-biomarker-discovery"><i class="fa fa-check"></i><b>14.8</b> Machine Learning and Biomarker Discovery</a><ul>
<li class="chapter" data-level="14.8.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#bonus-using-the-connectivity-map-for-drug-repurposing"><i class="fa fa-check"></i><b>14.8.1</b> Bonus: Using the Connectivity Map for drug repurposing</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#session-info"><i class="fa fa-check"></i><b>14.9</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>15</b> 500: Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="15.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-6"><i class="fa fa-check"></i><b>15.1</b> Overview</a><ul>
<li class="chapter" data-level="15.1.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#description-3"><i class="fa fa-check"></i><b>15.1.1</b> Description</a></li>
<li class="chapter" data-level="15.1.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-11"><i class="fa fa-check"></i><b>15.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="15.1.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#participation-3"><i class="fa fa-check"></i><b>15.1.3</b> Participation</a></li>
<li class="chapter" data-level="15.1.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-11"><i class="fa fa-check"></i><b>15.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="15.1.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-12"><i class="fa fa-check"></i><b>15.1.5</b> Time outline</a></li>
<li class="chapter" data-level="15.1.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#introductory-material"><i class="fa fa-check"></i><b>15.2</b> Introductory material</a><ul>
<li class="chapter" data-level="15.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#discussion"><i class="fa fa-check"></i><b>15.2.1</b> Discussion</a></li>
<li class="chapter" data-level="15.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#why-learn-delayedarray"><i class="fa fa-check"></i><b>15.2.2</b> Why learn DelayedArray?</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-of-delayedarray-framework"><i class="fa fa-check"></i><b>15.3</b> Overview of DelayedArray framework</a><ul>
<li class="chapter" data-level="15.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-package"><i class="fa fa-check"></i><b>15.3.1</b> The <strong>DelayedArray</strong> package</a></li>
<li class="chapter" data-level="15.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-class"><i class="fa fa-check"></i><b>15.3.2</b> The <em>DelayedArray</em> class</a></li>
<li class="chapter" data-level="15.3.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#operating-on-delayedarray-objects"><i class="fa fa-check"></i><b>15.3.3</b> Operating on <em>DelayedArray</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#whats-out-there-already"><i class="fa fa-check"></i><b>15.4</b> What’s out there already?</a></li>
<li class="chapter" data-level="15.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#incoporating-delayedarray-into-a-package"><i class="fa fa-check"></i><b>15.5</b> Incoporating DelayedArray into a package</a><ul>
<li class="chapter" data-level="15.5.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#writing-algorithms-to-process-delayedarray-instances"><i class="fa fa-check"></i><b>15.5.1</b> Writing algorithms to process <em>DelayedArray</em> instances</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#questions-and-discussion"><i class="fa fa-check"></i><b>15.6</b> Questions and discussion</a></li>
<li class="chapter" data-level="15.7" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#todos"><i class="fa fa-check"></i><b>15.7</b> TODOs</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>16</b> 510: Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="16.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-2"><i class="fa fa-check"></i><b>16.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="16.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-10"><i class="fa fa-check"></i><b>16.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="16.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-12"><i class="fa fa-check"></i><b>16.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="16.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-4"><i class="fa fa-check"></i><b>16.2.2</b> Participation</a></li>
<li class="chapter" data-level="16.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>16.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="16.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-13"><i class="fa fa-check"></i><b>16.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-6"><i class="fa fa-check"></i><b>16.3</b> Introduction</a><ul>
<li class="chapter" data-level="16.3.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>16.3.1</b> Why maintain your package</a></li>
<li class="chapter" data-level="16.3.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>16.3.2</b> Infrastructure used for maintenance</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>16.4</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="16.4.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>16.4.1</b> Build report</a></li>
<li class="chapter" data-level="16.4.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#version-numbering"><i class="fa fa-check"></i><b>16.4.2</b> Version Numbering</a></li>
<li class="chapter" data-level="16.4.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>16.4.3</b> Landing page badges</a></li>
<li class="chapter" data-level="16.4.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>16.4.4</b> Download statistics</a></li>
<li class="chapter" data-level="16.4.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>16.4.5</b> RSS feeds</a></li>
<li class="chapter" data-level="16.4.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>16.4.6</b> Support site, Bioc-devel mailing list</a></li>
<li class="chapter" data-level="16.4.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github"><i class="fa fa-check"></i><b>16.4.7</b> GitHub</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>16.5</b> Round up of resources available</a></li>
<li class="chapter" data-level="16.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#acknowledgements"><i class="fa fa-check"></i><b>16.6</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="bibliography-1.html"><a href="bibliography-1.html"><i class="fa fa-check"></i><b>17</b> Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger" class="section level1">
<h1><span class="header-section-number">6</span> 200: RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</h1>
<div id="instructor-name-and-contact-information-1" class="section level2">
<h2><span class="header-section-number">6.1</span> Instructor name and contact information</h2>
<ul>
<li>Charity Law (<a href="mailto:law@wehi.edu.au">law@wehi.edu.au</a>)</li>
</ul>
</div>
<div id="workshop-description-3" class="section level2">
<h2><span class="header-section-number">6.2</span> Workshop Description</h2>
<p>In this instructor-led live demo, we analyse RNA-sequencing data from the mouse mammary gland, demonstrating use of the popular <strong>edgeR</strong> package to import, organise, filter and normalise the data, followed by the <strong>limma</strong> package with its voom method, linear modelling and empirical Bayes moderation to assess differential expression and graphical representations. This pipeline is further enhanced by the <strong>Glimma</strong> package which enables interactive exploration of the results so that individual samples and genes can be examined by the user. The complete analysis offered by these three packages highlights the ease with which researchers can turn the raw counts from an RNA-sequencing experiment into biological insights using Bioconductor. The complete workflow is available at <a href="http://master.bioconductor.org/packages/release/workflows/html/RNAseq123.html" class="uri">http://master.bioconductor.org/packages/release/workflows/html/RNAseq123.html</a> .</p>
<div id="pre-requisites-3" class="section level3">
<h3><span class="header-section-number">6.2.1</span> Pre-requisites</h3>
<ul>
<li>Basic knowledge of RNA-sequencing</li>
<li>Basic knowledge of R syntax, R object classes and object manipulation</li>
</ul>
</div>
<div id="workshop-participation-3" class="section level3">
<h3><span class="header-section-number">6.2.2</span> Workshop Participation</h3>
<p>Participants can watch the live demo, or may prefer to follow the demonstration by bringing their laptops along. To follow the analysis on their own laptops, participants need to install the <em>RNAseq123</em> workflow by running</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">source</span>(<span class="st">&quot;https://bioconductor.org/biocLite.R&quot;</span>)  
<span class="kw">biocLite</span>(<span class="st">&quot;RNAseq123&quot;</span>)</code></pre></div>
<p>in R. The relevant sequencing data should also be download in advance.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">url &lt;-<span class="st"> &quot;https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE63310&amp;format=file&quot;</span>  
utils<span class="op">::</span><span class="kw">download.file</span>(url, <span class="dt">destfile=</span><span class="st">&quot;GSE63310_RAW.tar&quot;</span>, <span class="dt">mode=</span><span class="st">&quot;wb&quot;</span>)   
utils<span class="op">::</span><span class="kw">untar</span>(<span class="st">&quot;GSE63310_RAW.tar&quot;</span>, <span class="dt">exdir =</span> <span class="st">&quot;.&quot;</span>)  
files &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;GSM1545535_10_6_5_11.txt&quot;</span>, <span class="st">&quot;GSM1545536_9_6_5_11.txt&quot;</span>, <span class="st">&quot;GSM1545538_purep53.txt&quot;</span>,
  <span class="st">&quot;GSM1545539_JMS8-2.txt&quot;</span>, <span class="st">&quot;GSM1545540_JMS8-3.txt&quot;</span>, <span class="st">&quot;GSM1545541_JMS8-4.txt&quot;</span>,
  <span class="st">&quot;GSM1545542_JMS8-5.txt&quot;</span>, <span class="st">&quot;GSM1545544_JMS9-P7c.txt&quot;</span>, <span class="st">&quot;GSM1545545_JMS9-P8c.txt&quot;</span>)
<span class="cf">for</span>(i <span class="cf">in</span> <span class="kw">paste</span>(files, <span class="st">&quot;.gz&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>))  
  R.utils<span class="op">::</span><span class="kw">gunzip</span>(i, <span class="dt">overwrite=</span><span class="ot">TRUE</span>)  </code></pre></div>
<p>Due to time restraints, extra help regarding R package installation and coding errors will not be addressed during the workshop.</p>
</div>
<div id="r-bioconductor-packages-used-3" class="section level3">
<h3><span class="header-section-number">6.2.3</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<p>Bioconductor: limma, Glimma, edgeR, Mus.musculus<br />
CRAN: RColorBrewer, gplots</p>
</div>
<div id="time-outline-3" class="section level3">
<h3><span class="header-section-number">6.2.4</span> Time outline</h3>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Introduction</td>
<td>5mins</td>
</tr>
<tr class="even">
<td>Data packaging</td>
<td>10mins</td>
</tr>
<tr class="odd">
<td>Data pre-processing</td>
<td>15mins</td>
</tr>
<tr class="even">
<td>Differential expression analysis</td>
<td>30mins</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="workshop-goals-and-objectives-4" class="section level2">
<h2><span class="header-section-number">6.3</span> Workshop goals and objectives</h2>
<p>The key steps to RNA-seq data analysis are described in this workshop with basic statistical theory of methods used. The goal is to allow beginner-analysts of RNA-seq data to become familiar with each of the steps involved, as well as completing a standard analysis pipeline from start to finish.</p>
<div id="learning-goals-4" class="section level3">
<h3><span class="header-section-number">6.3.1</span> Learning goals</h3>
<ul>
<li>learn how to analyse RNA-seq data</li>
<li>identify methods for pre-processing data</li>
<li>understand linear models used in differential expression analysis</li>
<li>examine plots for data exploration and result representation</li>
</ul>
</div>
<div id="learning-objectives-4" class="section level3">
<h3><span class="header-section-number">6.3.2</span> Learning objectives</h3>
<ul>
<li>read in count data and format as a DGEList-object</li>
<li>annotate Entrez gene identifiers with gene information</li>
<li>filter out lowly expressed genes</li>
<li>normalise gene expression values</li>
<li>unsupervised clustering of samples (standard and interactive plots)</li>
<li>linear modelling for comparisons of interest</li>
<li>remove heteroscedascity</li>
<li>examine the number of differentially expressed genes</li>
<li>mean-difference plots (standard and interactive plots)</li>
<li>heatmaps</li>
</ul>
</div>
</div>
<div id="introduction-2" class="section level2">
<h2><span class="header-section-number">6.4</span> Introduction</h2>
<p>RNA-sequencing (RNA-seq) has become the primary technology used for gene expression profiling, with the genome-wide detection of differentially expressed genes between two or more conditions of interest one of the most commonly asked questions by researchers. The <strong>edgeR</strong> and <strong>limma</strong> packages available from the Bioconductor project offer a well-developed suite of statistical methods for dealing with this question for RNA-seq data.</p>
<p>In this article, we describe an <strong>edgeR</strong> - <strong>limma</strong> workflow for analysing RNA-seq data that takes gene-level counts as its input, and moves through pre-processing and exploratory data analysis before obtaining lists of differentially expressed (DE) genes and gene signatures. This analysis is enhanced through the use of interactive graphics from the <strong>Glimma</strong> package, that allows for a more detailed exploration of the data at both the sample and gene-level than is possible using static <strong>R</strong> plots.</p>
<p>The experiment analysed in this workflow is from Sheridan <em>et al.</em> (2015) and consists of three cell populations (basal, luminal progenitor (LP) and mature luminal (ML)) sorted from the mammary glands of female virgin mice, each profiled in triplicate. RNA samples were sequenced across three batches on an Illumina HiSeq 2000 to obtain 100 base-pair single-end reads. The analysis outlined in this article assumes that reads obtained from an RNA-seq experiment have been aligned to an appropriate reference genome and summarised into counts associated with gene-specific regions. In this instance, reads were aligned to the mouse reference genome (mm10) using the <strong>R</strong> based pipeline available in the <strong>Rsubread</strong> package (specifically the <code>align</code> function followed by <code>featureCounts</code> for gene-level summarisation based on the in-built <em>mm10</em> RefSeq-based annotation).</p>
<p>Count data for these samples can be downloaded from the Gene Expression Omnibus (GEO) <a href="http://www.ncbi.nlm.nih.gov/geo/" class="uri">http://www.ncbi.nlm.nih.gov/geo/</a> using GEO Series accession number GSE63310. Further information on experimental design and sample preparation is also available from GEO under this accession number.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>({
  <span class="kw">library</span>(limma)
  <span class="kw">library</span>(Glimma)
  <span class="kw">library</span>(edgeR)
  <span class="kw">library</span>(Mus.musculus)
})</code></pre></div>
</div>
<div id="data-packaging" class="section level2">
<h2><span class="header-section-number">6.5</span> Data packaging</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">dir.create</span>(<span class="st">&quot;Law_RNAseq123&quot;</span>)
<span class="co">#&gt; Warning in dir.create(&quot;Law_RNAseq123&quot;): &#39;Law_RNAseq123&#39; already exists</span>
<span class="kw">setwd</span>(<span class="st">&quot;Law_RNAseq123&quot;</span>)

url &lt;-<span class="st"> &quot;https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE63310&amp;format=file&quot;</span>
utils<span class="op">::</span><span class="kw">download.file</span>(url, <span class="dt">destfile=</span><span class="st">&quot;GSE63310_RAW.tar&quot;</span>, <span class="dt">mode=</span><span class="st">&quot;wb&quot;</span>)
utils<span class="op">::</span><span class="kw">untar</span>(<span class="st">&quot;GSE63310_RAW.tar&quot;</span>, <span class="dt">exdir =</span> <span class="st">&quot;.&quot;</span>)

files &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;GSM1545535_10_6_5_11.txt&quot;</span>, <span class="st">&quot;GSM1545536_9_6_5_11.txt&quot;</span>,
    <span class="st">&quot;GSM1545538_purep53.txt&quot;</span>, <span class="st">&quot;GSM1545539_JMS8-2.txt&quot;</span>, <span class="st">&quot;GSM1545540_JMS8-3.txt&quot;</span>,
    <span class="st">&quot;GSM1545541_JMS8-4.txt&quot;</span>, <span class="st">&quot;GSM1545542_JMS8-5.txt&quot;</span>, <span class="st">&quot;GSM1545544_JMS9-P7c.txt&quot;</span>,
    <span class="st">&quot;GSM1545545_JMS9-P8c.txt&quot;</span>)

<span class="cf">for</span>(i <span class="cf">in</span> <span class="kw">paste</span>(files, <span class="st">&quot;.gz&quot;</span>, <span class="dt">sep=</span><span class="st">&quot;&quot;</span>))
    R.utils<span class="op">::</span><span class="kw">gunzip</span>(i, <span class="dt">overwrite=</span><span class="ot">TRUE</span>)</code></pre></div>
<div id="reading-in-count-data" class="section level3">
<h3><span class="header-section-number">6.5.1</span> Reading in count-data</h3>
<p>To get started with this analysis, download the file <em>GSE63310_RAW.tar</em> available online from <a href="https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE63310&amp;format=file" class="uri">https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE63310&amp;format=file</a>, and extract the relevant files from this archive. Each of these text files contains the raw gene-level counts for a given sample. Note that our analysis only includes the basal, LP and ML samples from this experiment (see associated file names below).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">read.delim</span>(<span class="kw">file.path</span>(<span class="st">&quot;Law_RNAseq123&quot;</span>, files[<span class="dv">1</span>]), <span class="dt">nrow=</span><span class="dv">5</span>)
<span class="co">#&gt;    EntrezID GeneLength Count</span>
<span class="co">#&gt; 1    497097       3634     1</span>
<span class="co">#&gt; 2 100503874       3259     0</span>
<span class="co">#&gt; 3 100038431       1634     0</span>
<span class="co">#&gt; 4     19888       9747     0</span>
<span class="co">#&gt; 5     20671       3130     1</span></code></pre></div>
<p>Whilst each of the nine text files can be read into <strong>R</strong> separately and combined into a matrix of counts, <strong>edgeR</strong> offers a convenient way to do this in one step using the <code>readDGE</code> function. The resulting DGEList-object contains a matrix of counts with 27,179 rows associated with unique Entrez gene identifiers (IDs) and nine columns associated with the individual samples in the experiment.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">readDGE</span>(<span class="kw">file.path</span>(<span class="st">&quot;Law_RNAseq123&quot;</span>, files), <span class="dt">columns=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">3</span>))
<span class="kw">class</span>(x)
<span class="co">#&gt; [1] &quot;DGEList&quot;</span>
<span class="co">#&gt; attr(,&quot;package&quot;)</span>
<span class="co">#&gt; [1] &quot;edgeR&quot;</span>
<span class="kw">dim</span>(x)
<span class="co">#&gt; [1] 27179     9</span></code></pre></div>
<p>If the counts from all samples were stored in a single file, the data can be read into <strong>R</strong> and then converted into a DGEList-object using the <code>DGEList</code> function.</p>
</div>
<div id="organising-sample-information" class="section level3">
<h3><span class="header-section-number">6.5.2</span> Organising sample information</h3>
<p>For downstream analysis, sample-level information related to the experimental design needs to be associated with the columns of the counts matrix. This should include experimental variables, both biological and technical, that could have an effect on expression levels. Examples include cell type (basal, LP and ML in this experiment), genotype (wild-type, knock-out), phenotype (disease status, sex, age), sample treatment (drug, control) and batch information (date experiment was performed if samples were collected and analysed at distinct time points) to name just a few.</p>
<p>Our DGEList-object contains a <code>samples</code> data frame that stores both cell type (or <code>group</code>) and batch (sequencing <code>lane</code>) information, each of which consists of three distinct levels. Note that within <code>x$samples</code>, library sizes are automatically calculated for each sample and normalisation factors are set to 1. For simplicity, we remove the GEO sample IDs (GSM*) from the column names of our DGEList-object <code>x</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">samplenames &lt;-<span class="st"> </span><span class="kw">substring</span>(<span class="kw">colnames</span>(x), <span class="dv">12</span>, <span class="kw">nchar</span>(<span class="kw">colnames</span>(x)))
samplenames
<span class="co">#&gt; [1] &quot;23/GSM1545535_10_6_5_11&quot; &quot;23/GSM1545536_9_6_5_11&quot; </span>
<span class="co">#&gt; [3] &quot;23/GSM1545538_purep53&quot;   &quot;23/GSM1545539_JMS8-2&quot;   </span>
<span class="co">#&gt; [5] &quot;23/GSM1545540_JMS8-3&quot;    &quot;23/GSM1545541_JMS8-4&quot;   </span>
<span class="co">#&gt; [7] &quot;23/GSM1545542_JMS8-5&quot;    &quot;23/GSM1545544_JMS9-P7c&quot; </span>
<span class="co">#&gt; [9] &quot;23/GSM1545545_JMS9-P8c&quot;</span>
<span class="kw">colnames</span>(x) &lt;-<span class="st"> </span>samplenames
group &lt;-<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">c</span>(<span class="st">&quot;LP&quot;</span>, <span class="st">&quot;ML&quot;</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;ML&quot;</span>, <span class="st">&quot;LP&quot;</span>, 
                     <span class="st">&quot;Basal&quot;</span>, <span class="st">&quot;ML&quot;</span>, <span class="st">&quot;LP&quot;</span>))
x<span class="op">$</span>samples<span class="op">$</span>group &lt;-<span class="st"> </span>group
lane &lt;-<span class="st"> </span><span class="kw">as.factor</span>(<span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;L004&quot;</span>,<span class="st">&quot;L006&quot;</span>,<span class="st">&quot;L008&quot;</span>), <span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">2</span>)))
x<span class="op">$</span>samples<span class="op">$</span>lane &lt;-<span class="st"> </span>lane
x<span class="op">$</span>samples
<span class="co">#&gt;                                                          files group</span>
<span class="co">#&gt; 23/GSM1545535_10_6_5_11 Law_RNAseq123/GSM1545535_10_6_5_11.txt    LP</span>
<span class="co">#&gt; 23/GSM1545536_9_6_5_11   Law_RNAseq123/GSM1545536_9_6_5_11.txt    ML</span>
<span class="co">#&gt; 23/GSM1545538_purep53     Law_RNAseq123/GSM1545538_purep53.txt Basal</span>
<span class="co">#&gt; 23/GSM1545539_JMS8-2       Law_RNAseq123/GSM1545539_JMS8-2.txt Basal</span>
<span class="co">#&gt; 23/GSM1545540_JMS8-3       Law_RNAseq123/GSM1545540_JMS8-3.txt    ML</span>
<span class="co">#&gt; 23/GSM1545541_JMS8-4       Law_RNAseq123/GSM1545541_JMS8-4.txt    LP</span>
<span class="co">#&gt; 23/GSM1545542_JMS8-5       Law_RNAseq123/GSM1545542_JMS8-5.txt Basal</span>
<span class="co">#&gt; 23/GSM1545544_JMS9-P7c   Law_RNAseq123/GSM1545544_JMS9-P7c.txt    ML</span>
<span class="co">#&gt; 23/GSM1545545_JMS9-P8c   Law_RNAseq123/GSM1545545_JMS9-P8c.txt    LP</span>
<span class="co">#&gt;                         lib.size norm.factors lane</span>
<span class="co">#&gt; 23/GSM1545535_10_6_5_11 32863052            1 L004</span>
<span class="co">#&gt; 23/GSM1545536_9_6_5_11  35335491            1 L004</span>
<span class="co">#&gt; 23/GSM1545538_purep53   57160817            1 L004</span>
<span class="co">#&gt; 23/GSM1545539_JMS8-2    51368625            1 L006</span>
<span class="co">#&gt; 23/GSM1545540_JMS8-3    75795034            1 L006</span>
<span class="co">#&gt; 23/GSM1545541_JMS8-4    60517657            1 L006</span>
<span class="co">#&gt; 23/GSM1545542_JMS8-5    55086324            1 L006</span>
<span class="co">#&gt; 23/GSM1545544_JMS9-P7c  21311068            1 L008</span>
<span class="co">#&gt; 23/GSM1545545_JMS9-P8c  19958838            1 L008</span></code></pre></div>
</div>
<div id="organising-gene-annotations" class="section level3">
<h3><span class="header-section-number">6.5.3</span> Organising gene annotations</h3>
<p>A second data frame named <code>genes</code> in the DGEList-object is used to store gene-level information associated with rows of the counts matrix. This information can be retrieved using organism specific packages such as <strong>Mus.musculus</strong> for mouse (or <strong>Homo.sapiens</strong> for human) or the <strong>biomaRt</strong> package which interfaces the Ensembl genome databases in order to perform gene annotation.</p>
<p>The type of information that can be retrieved includes gene symbols, gene names, chromosome names and locations, Entrez gene IDs, Refseq gene IDs and Ensembl gene IDs to name just a few. <strong>biomaRt</strong> primarily works off Ensembl gene IDs, whereas <strong>Mus.musculus</strong> packages information from various sources and allows users to choose between many different gene IDs as the key.</p>
<p>The Entrez gene IDs available in our dataset were annotated using the <strong>Mus.musculus</strong> package to retrieve associated gene symbols and chromosome information.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">geneid &lt;-<span class="st"> </span><span class="kw">rownames</span>(x)
genes &lt;-<span class="st"> </span><span class="kw">select</span>(Mus.musculus, <span class="dt">keys=</span>geneid, <span class="dt">columns=</span><span class="kw">c</span>(<span class="st">&quot;SYMBOL&quot;</span>, <span class="st">&quot;TXCHROM&quot;</span>), 
                <span class="dt">keytype=</span><span class="st">&quot;ENTREZID&quot;</span>)
<span class="kw">head</span>(genes)
<span class="co">#&gt;    ENTREZID  SYMBOL TXCHROM</span>
<span class="co">#&gt; 1    497097    Xkr4    chr1</span>
<span class="co">#&gt; 2 100503874 Gm19938    &lt;NA&gt;</span>
<span class="co">#&gt; 3 100038431 Gm10568    &lt;NA&gt;</span>
<span class="co">#&gt; 4     19888     Rp1    chr1</span>
<span class="co">#&gt; 5     20671   Sox17    chr1</span>
<span class="co">#&gt; 6     27395  Mrpl15    chr1</span></code></pre></div>
<p>As with any gene ID, Entrez gene IDs may not map one-to-one to the gene information of interest. It is important to check for duplicated gene IDs and to understand the source of duplication before resolving them. Our gene annotation contains 28 genes that map to multiple chromosomes (e.g. gene Gm1987 is associated with <em>chr4</em> and <em>chr4_JH584294_random</em> and microRNA Mir5098 is associated with <em>chr2</em>, <em>chr5</em>, <em>chr8</em>, <em>chr11</em> and <em>chr17</em>). To resolve duplicate gene IDs one could combine all chromosome information from the multi-mapped genes, such that gene Gm1987 would be is assigned to <em>chr4 and chr4_JH584294_random</em>, or select one of the chromosomes to represent the gene with duplicate annotation. For simplicity we do the latter, keeping only the first occurrence of each gene ID.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">genes &lt;-<span class="st"> </span>genes[<span class="op">!</span><span class="kw">duplicated</span>(genes<span class="op">$</span>ENTREZID),]</code></pre></div>
<p>In this example, the gene order is the same in both the annotation and the data object. If this is not the case due to missing and/or rearranged gene IDs, the <code>match</code> function can be used to order genes correctly. The data frame of gene annotations is then added to the data object and neatly packaged in a DGEList-object containing raw count data with associated sample information and gene annotations.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x<span class="op">$</span>genes &lt;-<span class="st"> </span>genes
x
<span class="co">#&gt; An object of class &quot;DGEList&quot;</span>
<span class="co">#&gt; $samples</span>
<span class="co">#&gt;                                                          files group</span>
<span class="co">#&gt; 23/GSM1545535_10_6_5_11 Law_RNAseq123/GSM1545535_10_6_5_11.txt    LP</span>
<span class="co">#&gt; 23/GSM1545536_9_6_5_11   Law_RNAseq123/GSM1545536_9_6_5_11.txt    ML</span>
<span class="co">#&gt; 23/GSM1545538_purep53     Law_RNAseq123/GSM1545538_purep53.txt Basal</span>
<span class="co">#&gt; 23/GSM1545539_JMS8-2       Law_RNAseq123/GSM1545539_JMS8-2.txt Basal</span>
<span class="co">#&gt; 23/GSM1545540_JMS8-3       Law_RNAseq123/GSM1545540_JMS8-3.txt    ML</span>
<span class="co">#&gt; 23/GSM1545541_JMS8-4       Law_RNAseq123/GSM1545541_JMS8-4.txt    LP</span>
<span class="co">#&gt; 23/GSM1545542_JMS8-5       Law_RNAseq123/GSM1545542_JMS8-5.txt Basal</span>
<span class="co">#&gt; 23/GSM1545544_JMS9-P7c   Law_RNAseq123/GSM1545544_JMS9-P7c.txt    ML</span>
<span class="co">#&gt; 23/GSM1545545_JMS9-P8c   Law_RNAseq123/GSM1545545_JMS9-P8c.txt    LP</span>
<span class="co">#&gt;                         lib.size norm.factors lane</span>
<span class="co">#&gt; 23/GSM1545535_10_6_5_11 32863052            1 L004</span>
<span class="co">#&gt; 23/GSM1545536_9_6_5_11  35335491            1 L004</span>
<span class="co">#&gt; 23/GSM1545538_purep53   57160817            1 L004</span>
<span class="co">#&gt; 23/GSM1545539_JMS8-2    51368625            1 L006</span>
<span class="co">#&gt; 23/GSM1545540_JMS8-3    75795034            1 L006</span>
<span class="co">#&gt; 23/GSM1545541_JMS8-4    60517657            1 L006</span>
<span class="co">#&gt; 23/GSM1545542_JMS8-5    55086324            1 L006</span>
<span class="co">#&gt; 23/GSM1545544_JMS9-P7c  21311068            1 L008</span>
<span class="co">#&gt; 23/GSM1545545_JMS9-P8c  19958838            1 L008</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $counts</span>
<span class="co">#&gt;            Samples</span>
<span class="co">#&gt; Tags        23/GSM1545535_10_6_5_11 23/GSM1545536_9_6_5_11</span>
<span class="co">#&gt;   497097                          1                      2</span>
<span class="co">#&gt;   100503874                       0                      0</span>
<span class="co">#&gt;   100038431                       0                      0</span>
<span class="co">#&gt;   19888                           0                      1</span>
<span class="co">#&gt;   20671                           1                      1</span>
<span class="co">#&gt;            Samples</span>
<span class="co">#&gt; Tags        23/GSM1545538_purep53 23/GSM1545539_JMS8-2</span>
<span class="co">#&gt;   497097                      342                  526</span>
<span class="co">#&gt;   100503874                     5                    6</span>
<span class="co">#&gt;   100038431                     0                    0</span>
<span class="co">#&gt;   19888                         0                    0</span>
<span class="co">#&gt;   20671                        76                   40</span>
<span class="co">#&gt;            Samples</span>
<span class="co">#&gt; Tags        23/GSM1545540_JMS8-3 23/GSM1545541_JMS8-4 23/GSM1545542_JMS8-5</span>
<span class="co">#&gt;   497097                       3                    3                  535</span>
<span class="co">#&gt;   100503874                    0                    0                    5</span>
<span class="co">#&gt;   100038431                    0                    0                    1</span>
<span class="co">#&gt;   19888                       17                    2                    0</span>
<span class="co">#&gt;   20671                       33                   14                   98</span>
<span class="co">#&gt;            Samples</span>
<span class="co">#&gt; Tags        23/GSM1545544_JMS9-P7c 23/GSM1545545_JMS9-P8c</span>
<span class="co">#&gt;   497097                         2                      0</span>
<span class="co">#&gt;   100503874                      0                      0</span>
<span class="co">#&gt;   100038431                      0                      0</span>
<span class="co">#&gt;   19888                          1                      0</span>
<span class="co">#&gt;   20671                         18                      8</span>
<span class="co">#&gt; 27174 more rows ...</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $genes</span>
<span class="co">#&gt;    ENTREZID  SYMBOL TXCHROM</span>
<span class="co">#&gt; 1    497097    Xkr4    chr1</span>
<span class="co">#&gt; 2 100503874 Gm19938    &lt;NA&gt;</span>
<span class="co">#&gt; 3 100038431 Gm10568    &lt;NA&gt;</span>
<span class="co">#&gt; 4     19888     Rp1    chr1</span>
<span class="co">#&gt; 5     20671   Sox17    chr1</span>
<span class="co">#&gt; 27174 more rows ...</span></code></pre></div>
</div>
</div>
<div id="data-pre-processing" class="section level2">
<h2><span class="header-section-number">6.6</span> Data pre-processing</h2>
<div id="transformations-from-the-raw-scale" class="section level3">
<h3><span class="header-section-number">6.6.1</span> Transformations from the raw-scale</h3>
<p>For differential expression and related analyses, gene expression is rarely considered at the level of raw counts since libraries sequenced at a greater depth will result in higher counts. Rather, it is common practice to transform raw counts onto a scale that accounts for such library size differences. Popular transformations include counts per million (CPM), log2-counts per million (log-CPM), reads per kilobase of transcript per million (RPKM), and fragments per kilobase of transcript per million (FPKM).</p>
<p>In our analyses, CPM and log-CPM transformations are used regularly although they do not account for feature length differences which RPKM and FPKM values do. Whilst RPKM and FPKM values can just as well be used, CPM and log-CPM values can be calculated using a counts matrix alone and will suffice for the type of comparisons we are interested in. Assuming that there are no differences in isoform usage between conditions, differential expression analyses look at gene expression changes between conditions rather than comparing expression across multiple genes or drawing conclusions on absolute levels of expression. In other words, gene lengths remain constant for comparisons of interest and any observed differences are a result of changes in condition rather than changes in gene length.</p>
<p>Here raw counts are converted to CPM and log-CPM values using the <code>cpm</code> function in <strong>edgeR</strong>, where log-transformations use a prior count of 0.25 to avoid taking the log of zero. RPKM values are just as easily calculated as CPM values using the <code>rpkm</code> function in <strong>edgeR</strong> if gene lengths are available.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cpm &lt;-<span class="st"> </span><span class="kw">cpm</span>(x)
lcpm &lt;-<span class="st"> </span><span class="kw">cpm</span>(x, <span class="dt">log=</span><span class="ot">TRUE</span>)</code></pre></div>
</div>
<div id="removing-genes-that-are-lowly-expressed" class="section level3">
<h3><span class="header-section-number">6.6.2</span> Removing genes that are lowly expressed</h3>
<p>All datasets will include a mix of genes that are expressed and those that are not expressed. Whilst it is of interest to examine genes that are expressed in one condition but not in another, some genes are unexpressed throughout all samples. In fact, 19% of genes in this dataset have zero counts across all nine samples.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">table</span>(<span class="kw">rowSums</span>(x<span class="op">$</span>counts<span class="op">==</span><span class="dv">0</span>)<span class="op">==</span><span class="dv">9</span>)
<span class="co">#&gt; </span>
<span class="co">#&gt; FALSE  TRUE </span>
<span class="co">#&gt; 22026  5153</span></code></pre></div>
<p>Genes that are not expressed at a biologically meaningful level in any condition should be discarded to reduce the subset of genes to those that are of interest, and to reduce the number of tests carried out downstream when looking at differential expression. Upon examination of log-CPM values, it can be seen that a large proportion of genes within each sample is unexpressed or lowly-expressed (shown in panel A of the next figure). Using a nominal CPM value of 1 (which is equivalent to a log-CPM value of 0) genes are deemed to be expressed if their expression is above this threshold, and unexpressed otherwise. Genes must be expressed in at least one group (or in at least three samples across the entire experiment) to be kept for downstream analysis.</p>
<p>Although any sensible value can be used as the expression cutoff, typically a CPM value of 1 is used in our analyses as it separates expressed genes from unexpressed genes well for most datasets. Here, a CPM value of 1 means that a gene is <em>expressed</em> if it has at least 20 counts in the sample with the lowest sequencing depth (JMS9-P8c, library size approx. 20 million) or at least 76 counts in the sample with the greatest sequencing depth (JMS8-3, library size approx. 76 million). If sequence reads are summarised by exons rather than genes and/or experiments have low sequencing depth, a lower CPM cutoff may be considered.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">keep.exprs &lt;-<span class="st"> </span><span class="kw">rowSums</span>(cpm<span class="op">&gt;</span><span class="dv">1</span>)<span class="op">&gt;=</span><span class="dv">3</span>
x &lt;-<span class="st"> </span>x[keep.exprs,, keep.lib.sizes=<span class="ot">FALSE</span>]
<span class="kw">dim</span>(x)
<span class="co">#&gt; [1] 14165     9</span></code></pre></div>
<p>Using this criterion, the number of genes is reduced to approximately half the number that we started with (14,165 genes, panel B of the next figure). Note that subsetting the entire DGEList-object removes both the counts as well as the associated gene information. Code to produce the figure is given below.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(RColorBrewer)
nsamples &lt;-<span class="st"> </span><span class="kw">ncol</span>(x)
col &lt;-<span class="st"> </span><span class="kw">brewer.pal</span>(nsamples, <span class="st">&quot;Paired&quot;</span>)
<span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
<span class="kw">plot</span>(<span class="kw">density</span>(lcpm[,<span class="dv">1</span>]), <span class="dt">col=</span>col[<span class="dv">1</span>], <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="fl">0.21</span>), <span class="dt">las=</span><span class="dv">2</span>, 
     <span class="dt">main=</span><span class="st">&quot;&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;&quot;</span>)
<span class="kw">title</span>(<span class="dt">main=</span><span class="st">&quot;A. Raw data&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Log-cpm&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span><span class="dv">0</span>, <span class="dt">lty=</span><span class="dv">3</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="op">:</span>nsamples){
 den &lt;-<span class="st"> </span><span class="kw">density</span>(lcpm[,i])
 <span class="kw">lines</span>(den<span class="op">$</span>x, den<span class="op">$</span>y, <span class="dt">col=</span>col[i], <span class="dt">lwd=</span><span class="dv">2</span>)
}
<span class="kw">legend</span>(<span class="st">&quot;topright&quot;</span>, samplenames, <span class="dt">text.col=</span>col, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>)
lcpm &lt;-<span class="st"> </span><span class="kw">cpm</span>(x, <span class="dt">log=</span><span class="ot">TRUE</span>)
<span class="kw">plot</span>(<span class="kw">density</span>(lcpm[,<span class="dv">1</span>]), <span class="dt">col=</span>col[<span class="dv">1</span>], <span class="dt">lwd=</span><span class="dv">2</span>, <span class="dt">ylim=</span><span class="kw">c</span>(<span class="dv">0</span>,<span class="fl">0.21</span>), <span class="dt">las=</span><span class="dv">2</span>, 
     <span class="dt">main=</span><span class="st">&quot;&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;&quot;</span>)
<span class="kw">title</span>(<span class="dt">main=</span><span class="st">&quot;B. Filtered data&quot;</span>, <span class="dt">xlab=</span><span class="st">&quot;Log-cpm&quot;</span>)
<span class="kw">abline</span>(<span class="dt">v=</span><span class="dv">0</span>, <span class="dt">lty=</span><span class="dv">3</span>)
<span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="op">:</span>nsamples){
   den &lt;-<span class="st"> </span><span class="kw">density</span>(lcpm[,i])
   <span class="kw">lines</span>(den<span class="op">$</span>x, den<span class="op">$</span>y, <span class="dt">col=</span>col[i], <span class="dt">lwd=</span><span class="dv">2</span>)
}
<span class="kw">legend</span>(<span class="st">&quot;topright&quot;</span>, samplenames, <span class="dt">text.col=</span>col, <span class="dt">bty=</span><span class="st">&quot;n&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:filterplot1"></span>
<img src="200_Law_RNAseq123_files/figure-html/filterplot1-1.png" alt="The density of log-CPM values for raw pre-filtered data (A) and post-filtered data (B) are shown for each sample. Dotted vertical lines mark the log-CPM of zero threshold (equivalent to a CPM value of 1) used in the filtering step." width="768" />
<p class="caption">
Figure 6.1: The density of log-CPM values for raw pre-filtered data (A) and post-filtered data (B) are shown for each sample. Dotted vertical lines mark the log-CPM of zero threshold (equivalent to a CPM value of 1) used in the filtering step.
</p>
</div>
</div>
<div id="normalising-gene-expression-distributions" class="section level3">
<h3><span class="header-section-number">6.6.3</span> Normalising gene expression distributions</h3>
<p>During the sample preparation or sequencing process, external factors that are not of biological interest can affect the expression of individual samples. For example, samples processed in the first batch of an experiment can have higher expression overall when compared to samples processed in a second batch. It is assumed that all samples should have a similar range and distribution of expression values. Normalisation is required to ensure that the expression distributions of each sample are similar across the entire experiment.</p>
<p>Any plot showing the per sample expression distributions, such as a density or boxplot, is useful in determining whether any samples are dissimilar to others. Distributions of log-CPM values are similar throughout all samples within this dataset (panel B of the figure above).</p>
<p>Nonetheless, normalisation by the method of trimmed mean of M-values (TMM) is performed using the <code>calcNormFactors</code> function in <strong>edgeR</strong>. The normalisation factors calculated here are used as a scaling factor for the library sizes. When working with DGEList-objects, these normalisation factors are automatically stored in <code>x$samples$norm.factors</code>. For this dataset the effect of TMM-normalisation is mild, as evident in the magnitude of the scaling factors, which are all relatively close to 1.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">calcNormFactors</span>(x, <span class="dt">method =</span> <span class="st">&quot;TMM&quot;</span>)
x<span class="op">$</span>samples<span class="op">$</span>norm.factors
<span class="co">#&gt; [1] 0.8957309 1.0349196 1.0439552 1.0405040 1.0323599 0.9223424 0.9836603</span>
<span class="co">#&gt; [8] 1.0827381 0.9792607</span></code></pre></div>
<p>To give a better visual representation of the effects of normalisation, the data was duplicated then adjusted so that the counts of the first sample are reduced to 5% of their original values, and in the second sample they are inflated to be 5-times larger.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x2 &lt;-<span class="st"> </span>x
x2<span class="op">$</span>samples<span class="op">$</span>norm.factors &lt;-<span class="st"> </span><span class="dv">1</span>
x2<span class="op">$</span>counts[,<span class="dv">1</span>] &lt;-<span class="st"> </span><span class="kw">ceiling</span>(x2<span class="op">$</span>counts[,<span class="dv">1</span>]<span class="op">*</span><span class="fl">0.05</span>)
x2<span class="op">$</span>counts[,<span class="dv">2</span>] &lt;-<span class="st"> </span>x2<span class="op">$</span>counts[,<span class="dv">2</span>]<span class="op">*</span><span class="dv">5</span></code></pre></div>
<p>The figure below shows the expression distribution of samples for unnormalised and normalised data, where distributions are noticeably different pre-normalisation and are similar post-normalisation. Here the first sample has a small TMM scaling factor of 0.05, whereas the second sample has a large scaling factor of 6.13 – neither values are close to 1.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
lcpm &lt;-<span class="st"> </span><span class="kw">cpm</span>(x2, <span class="dt">log=</span><span class="ot">TRUE</span>)
<span class="kw">boxplot</span>(lcpm, <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">col=</span>col, <span class="dt">main=</span><span class="st">&quot;&quot;</span>)
<span class="kw">title</span>(<span class="dt">main=</span><span class="st">&quot;A. Example: Unnormalised data&quot;</span>,<span class="dt">ylab=</span><span class="st">&quot;Log-cpm&quot;</span>)
x2 &lt;-<span class="st"> </span><span class="kw">calcNormFactors</span>(x2)  
x2<span class="op">$</span>samples<span class="op">$</span>norm.factors
<span class="co">#&gt; [1] 0.05472223 6.13059440 1.22927355 1.17051887 1.21487709 1.05622968</span>
<span class="co">#&gt; [7] 1.14587663 1.26129350 1.11702264</span>
lcpm &lt;-<span class="st"> </span><span class="kw">cpm</span>(x2, <span class="dt">log=</span><span class="ot">TRUE</span>)
<span class="kw">boxplot</span>(lcpm, <span class="dt">las=</span><span class="dv">2</span>, <span class="dt">col=</span>col, <span class="dt">main=</span><span class="st">&quot;&quot;</span>)
<span class="kw">title</span>(<span class="dt">main=</span><span class="st">&quot;B. Example: Normalised data&quot;</span>,<span class="dt">ylab=</span><span class="st">&quot;Log-cpm&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:plotmodifieddata"></span>
<img src="200_Law_RNAseq123_files/figure-html/plotmodifieddata-1.png" alt="Example data: Boxplots of log-CPM values showing expression distributions for unnormalised data (A) and normalised data (B) for each sample in the modified dataset where the counts in samples 1 and 2 have been scaled to 5% and 500% of their original values respectively." width="768" />
<p class="caption">
Figure 6.2: Example data: Boxplots of log-CPM values showing expression distributions for unnormalised data (A) and normalised data (B) for each sample in the modified dataset where the counts in samples 1 and 2 have been scaled to 5% and 500% of their original values respectively.
</p>
</div>
</div>
<div id="unsupervised-clustering-of-samples" class="section level3">
<h3><span class="header-section-number">6.6.4</span> Unsupervised clustering of samples</h3>
<p>In our opinion, one of the most important exploratory plots to examine for gene expression analyses is the multi-dimensional scaling (MDS) plot, or similar. The plot shows similarities and dissimilarities between samples in an unsupervised manner so that one can have an idea of the extent to which differential expression can be detected before carrying out formal tests. Ideally, samples would cluster well within the primary condition of interest, and any sample straying far from its group could be identified and followed up for sources of error or extra variation. If present, technical replicates should lie very close to one another.</p>
<p>Such a plot can be made in <strong>limma</strong> using the <code>plotMDS</code> function. The first dimension represents the leading-fold-change that best separates samples and explains the largest proportion of variation in the data, with subsequent dimensions having a smaller effect and being orthogonal to the ones before it. When experimental design involves multiple factors, it is recommended that each factor is examined over several dimensions. If samples cluster by a given factor in any of these dimensions, it suggests that the factor contributes to expression differences and is worth including in the linear modelling. On the other hand, factors that show little or no effect may be left out of downstream analysis.</p>
<p>In this dataset, samples can be seen to cluster well within experimental groups over dimension 1 and 2, and then separate by sequencing lane (sample batch) over dimension 3 (shown in the plot below). Keeping in mind that the first dimension explains the largest proportion of variation in the data, notice that the range of values over the dimensions become smaller as we move to higher dimensions.</p>
<p>Whilst all samples cluster by groups, the largest transcriptional difference is observed between basal and LP, and basal and ML over dimension 1. For this reason, it is expected that pairwise comparisons between cell populations will result in a greater number of DE genes for comparisons involving basal samples, and relatively small numbers of DE genes when comparing ML to LP. In other datasets, samples that do not cluster by their groups of interest may also show little or no evidence of differential expression in the downstream analysis.</p>
<p>To create the MDS plots, different colour groupings are assigned to factors of interest. Dimensions 1 and 2 are examined using the color grouping defined by cell types.</p>
<p>Dimensions 3 and 4 are examined using the colour grouping defined by sequencing lanes (batch).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">lcpm &lt;-<span class="st"> </span><span class="kw">cpm</span>(x, <span class="dt">log=</span><span class="ot">TRUE</span>)
<span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
col.group &lt;-<span class="st"> </span>group
<span class="kw">levels</span>(col.group) &lt;-<span class="st">  </span><span class="kw">brewer.pal</span>(<span class="kw">nlevels</span>(col.group), <span class="st">&quot;Set1&quot;</span>)
col.group &lt;-<span class="st"> </span><span class="kw">as.character</span>(col.group)
col.lane &lt;-<span class="st"> </span>lane
<span class="kw">levels</span>(col.lane) &lt;-<span class="st">  </span><span class="kw">brewer.pal</span>(<span class="kw">nlevels</span>(col.lane), <span class="st">&quot;Set2&quot;</span>)
col.lane &lt;-<span class="st"> </span><span class="kw">as.character</span>(col.lane)
<span class="kw">plotMDS</span>(lcpm, <span class="dt">labels=</span>group, <span class="dt">col=</span>col.group)
<span class="kw">title</span>(<span class="dt">main=</span><span class="st">&quot;A. Sample groups&quot;</span>)
<span class="kw">plotMDS</span>(lcpm, <span class="dt">labels=</span>lane, <span class="dt">col=</span>col.lane, <span class="dt">dim=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">4</span>))
<span class="kw">title</span>(<span class="dt">main=</span><span class="st">&quot;B. Sequencing lanes&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:MDS1"></span>
<img src="200_Law_RNAseq123_files/figure-html/MDS1-1.png" alt="MDS plots of log-CPM values over dimensions 1 and 2 with samples coloured and labeled by sample groups (A) and over dimensions 3 and 4 with samples coloured and labeled by sequencing lane (B). Distances on the plot correspond to the leading fold-change, which is the average (root-mean-square) log2-fold-change for the 500 genes most divergent between each pair of samples by default." width="768" />
<p class="caption">
Figure 6.3: MDS plots of log-CPM values over dimensions 1 and 2 with samples coloured and labeled by sample groups (A) and over dimensions 3 and 4 with samples coloured and labeled by sequencing lane (B). Distances on the plot correspond to the leading fold-change, which is the average (root-mean-square) log2-fold-change for the 500 genes most divergent between each pair of samples by default.
</p>
</div>
<p>Alternatively, the <strong>Glimma</strong> package offers the convenience of an interactive MDS plot where multiple dimensions can be explored. The <code>glMDSPlot</code> function generates an html page (that is opened in a browser if <code>launch=TRUE</code>) with an MDS plot in the left panel and a barplot showing the proportion of variation explained by each dimension in the right panel. Clicking on the bars of the bar plot changes the pair of dimensions plotted in the MDS plot, and hovering over the individual points reveals the sample label. The colour scheme can be changed as well to highlight cell population or sequencing lane (batch). An interactive MDS plot of this dataset can be found at <a href="http://bioinf.wehi.edu.au/folders/limmaWorkflow/glimma-plots/MDS-Plot.html" class="uri">http://bioinf.wehi.edu.au/folders/limmaWorkflow/glimma-plots/MDS-Plot.html</a>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">glMDSPlot</span>(lcpm, <span class="dt">labels=</span><span class="kw">paste</span>(group, lane, <span class="dt">sep=</span><span class="st">&quot;_&quot;</span>), 
          <span class="dt">groups=</span>x<span class="op">$</span>samples[,<span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">5</span>)], <span class="dt">launch=</span><span class="ot">FALSE</span>)</code></pre></div>
</div>
</div>
<div id="differential-expression-analysis" class="section level2">
<h2><span class="header-section-number">6.7</span> Differential expression analysis</h2>
<div id="creating-a-design-matrix-and-contrasts" class="section level3">
<h3><span class="header-section-number">6.7.1</span> Creating a design matrix and contrasts</h3>
<p>In this study, it is of interest to see which genes are expressed at different levels between the three cell populations profiled. In our analysis, linear models are fitted to the data with the assumption that the underlying data is normally distributed. To get started, a design matrix is set up with both the cell population and sequencing lane (batch) information.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">design &lt;-<span class="st"> </span><span class="kw">model.matrix</span>(<span class="op">~</span><span class="dv">0</span><span class="op">+</span>group<span class="op">+</span>lane)
<span class="kw">colnames</span>(design) &lt;-<span class="st"> </span><span class="kw">gsub</span>(<span class="st">&quot;group&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">colnames</span>(design))
design
<span class="co">#&gt;   Basal LP ML laneL006 laneL008</span>
<span class="co">#&gt; 1     0  1  0        0        0</span>
<span class="co">#&gt; 2     0  0  1        0        0</span>
<span class="co">#&gt; 3     1  0  0        0        0</span>
<span class="co">#&gt; 4     1  0  0        1        0</span>
<span class="co">#&gt; 5     0  0  1        1        0</span>
<span class="co">#&gt; 6     0  1  0        1        0</span>
<span class="co">#&gt; 7     1  0  0        1        0</span>
<span class="co">#&gt; 8     0  0  1        0        1</span>
<span class="co">#&gt; 9     0  1  0        0        1</span>
<span class="co">#&gt; attr(,&quot;assign&quot;)</span>
<span class="co">#&gt; [1] 1 1 1 2 2</span>
<span class="co">#&gt; attr(,&quot;contrasts&quot;)</span>
<span class="co">#&gt; attr(,&quot;contrasts&quot;)$group</span>
<span class="co">#&gt; [1] &quot;contr.treatment&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,&quot;contrasts&quot;)$lane</span>
<span class="co">#&gt; [1] &quot;contr.treatment&quot;</span></code></pre></div>
<p>For a given experiment, there are usually several equivalent ways to set up an appropriate design matrix. For example, <code>~0+group+lane</code> removes the intercept from the first factor, <code>group</code>, but an intercept remains in the second factor <code>lane</code>. Alternatively, <code>~group+lane</code> could be used to keep the intercepts in both <code>group</code> and <code>lane</code>. Understanding how to interpret the coefficients estimated in a given model is key here. We choose the first model for our analysis, as setting up model contrasts is more straight forward in the absence of an intercept for <code>group</code>. Contrasts for pairwise comparisons between cell populations are set up in <strong>limma</strong> using the <code>makeContrasts</code> function.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">contr.matrix &lt;-<span class="st"> </span><span class="kw">makeContrasts</span>(
   <span class="dt">BasalvsLP =</span> Basal<span class="op">-</span>LP, 
   <span class="dt">BasalvsML =</span> Basal <span class="op">-</span><span class="st"> </span>ML, 
   <span class="dt">LPvsML =</span> LP <span class="op">-</span><span class="st"> </span>ML, 
   <span class="dt">levels =</span> <span class="kw">colnames</span>(design))
contr.matrix
<span class="co">#&gt;           Contrasts</span>
<span class="co">#&gt; Levels     BasalvsLP BasalvsML LPvsML</span>
<span class="co">#&gt;   Basal            1         1      0</span>
<span class="co">#&gt;   LP              -1         0      1</span>
<span class="co">#&gt;   ML               0        -1     -1</span>
<span class="co">#&gt;   laneL006         0         0      0</span>
<span class="co">#&gt;   laneL008         0         0      0</span></code></pre></div>
<p>A key strength of <strong>limma</strong>’s linear modelling approach, is the ability accommodate arbitrary experimental complexity. Simple designs, such as the one in this workflow, with cell type and batch, through to more complicated factorial designs and models with interaction terms can be handled relatively easily. Where experimental or technical effects can be modelled using a random effect, another possibility in <strong>limma</strong> is to estimate correlations using <code>duplicateCorrelation</code> by specifying a <code>block</code> argument for both this function and in the <code>lmFit</code> linear modelling step.</p>
</div>
<div id="removing-heteroscedascity-from-count-data" class="section level3">
<h3><span class="header-section-number">6.7.2</span> Removing heteroscedascity from count data</h3>
<p>It has been shown that for RNA-seq count data, the variance is not independent of the mean – this is true of raw counts or when transformed to log-CPM values. Methods that model counts using a Negative Binomial distribution assume a quadratic mean-variance relationship. In <strong>limma</strong>, linear modelling is carried out on the log-CPM values which are assumed to be normally distributed and the mean-variance relationship is accommodated using precision weights calculated by the <code>voom</code> function.</p>
<p>When operating on a DGEList-object, <code>voom</code> converts raw counts to log-CPM values by automatically extracting library sizes and normalisation factors from <code>x</code> itself. Additional normalisation to log-CPM values can be specified within <code>voom</code> using the <code>normalize.method</code> argument.</p>
<p>The mean-variance relationship of log-CPM values for this dataset is shown in the left-hand panel of the next figure. Typically, the <em>voom-plot</em> shows a decreasing trend between the means and variances resulting from a combination of technical variation in the sequencing experiment and biological variation amongst the replicate samples from different cell populations. Experiments with high biological variation usually result in flatter trends, where variance values plateau at high expression values. Experiments with low biological variation tend to result in sharp decreasing trends.</p>
<p>Moreover, the voom-plot provides a visual check on the level of filtering performed upstream. If filtering of lowly-expressed genes is insufficient, a drop in variance levels can be observed at the low end of the expression scale due to very small counts. If this is observed, one should return to the earlier filtering step and increase the expression threshold applied to the dataset.</p>
<p>Where sample-level variation is evident from earlier inspections of the MDS plot, the <code>voomWithQualityWeights</code> function can be used to simultaneously incorporate sample-level weights together with the abundance dependent weights estimated by <code>voom</code>. For an example of this, see Liu <em>et al.</em> (2016).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(<span class="dt">mfrow=</span><span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>))
v &lt;-<span class="st"> </span><span class="kw">voom</span>(x, design, <span class="dt">plot=</span><span class="ot">TRUE</span>)
v
<span class="co">#&gt; An object of class &quot;EList&quot;</span>
<span class="co">#&gt; $genes</span>
<span class="co">#&gt;    ENTREZID SYMBOL TXCHROM</span>
<span class="co">#&gt; 1    497097   Xkr4    chr1</span>
<span class="co">#&gt; 6     27395 Mrpl15    chr1</span>
<span class="co">#&gt; 7     18777 Lypla1    chr1</span>
<span class="co">#&gt; 9     21399  Tcea1    chr1</span>
<span class="co">#&gt; 10    58175  Rgs20    chr1</span>
<span class="co">#&gt; 14160 more rows ...</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $targets</span>
<span class="co">#&gt;                                                          files group</span>
<span class="co">#&gt; 23/GSM1545535_10_6_5_11 Law_RNAseq123/GSM1545535_10_6_5_11.txt    LP</span>
<span class="co">#&gt; 23/GSM1545536_9_6_5_11   Law_RNAseq123/GSM1545536_9_6_5_11.txt    ML</span>
<span class="co">#&gt; 23/GSM1545538_purep53     Law_RNAseq123/GSM1545538_purep53.txt Basal</span>
<span class="co">#&gt; 23/GSM1545539_JMS8-2       Law_RNAseq123/GSM1545539_JMS8-2.txt Basal</span>
<span class="co">#&gt; 23/GSM1545540_JMS8-3       Law_RNAseq123/GSM1545540_JMS8-3.txt    ML</span>
<span class="co">#&gt; 23/GSM1545541_JMS8-4       Law_RNAseq123/GSM1545541_JMS8-4.txt    LP</span>
<span class="co">#&gt; 23/GSM1545542_JMS8-5       Law_RNAseq123/GSM1545542_JMS8-5.txt Basal</span>
<span class="co">#&gt; 23/GSM1545544_JMS9-P7c   Law_RNAseq123/GSM1545544_JMS9-P7c.txt    ML</span>
<span class="co">#&gt; 23/GSM1545545_JMS9-P8c   Law_RNAseq123/GSM1545545_JMS9-P8c.txt    LP</span>
<span class="co">#&gt;                         lib.size norm.factors lane</span>
<span class="co">#&gt; 23/GSM1545535_10_6_5_11 29409426    0.8957309 L004</span>
<span class="co">#&gt; 23/GSM1545536_9_6_5_11  36528591    1.0349196 L004</span>
<span class="co">#&gt; 23/GSM1545538_purep53   59598629    1.0439552 L004</span>
<span class="co">#&gt; 23/GSM1545539_JMS8-2    53382070    1.0405040 L006</span>
<span class="co">#&gt; 23/GSM1545540_JMS8-3    78175314    1.0323599 L006</span>
<span class="co">#&gt; 23/GSM1545541_JMS8-4    55762781    0.9223424 L006</span>
<span class="co">#&gt; 23/GSM1545542_JMS8-5    54115150    0.9836603 L006</span>
<span class="co">#&gt; 23/GSM1545544_JMS9-P7c  23043111    1.0827381 L008</span>
<span class="co">#&gt; 23/GSM1545545_JMS9-P8c  19525423    0.9792607 L008</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $E</span>
<span class="co">#&gt;         Samples</span>
<span class="co">#&gt; Tags     23/GSM1545535_10_6_5_11 23/GSM1545536_9_6_5_11</span>
<span class="co">#&gt;   497097               -4.293244              -3.869026</span>
<span class="co">#&gt;   27395                 3.875010               4.400568</span>
<span class="co">#&gt;   18777                 4.707695               5.559334</span>
<span class="co">#&gt;   21399                 4.784462               4.741999</span>
<span class="co">#&gt;   58175                 3.943567               3.294875</span>
<span class="co">#&gt;         Samples</span>
<span class="co">#&gt; Tags     23/GSM1545538_purep53 23/GSM1545539_JMS8-2 23/GSM1545540_JMS8-3</span>
<span class="co">#&gt;   497097              2.522753             3.302006            -4.481286</span>
<span class="co">#&gt;   27395               4.521172             4.570624             4.322845</span>
<span class="co">#&gt;   18777               5.400569             5.171235             5.627798</span>
<span class="co">#&gt;   21399               5.374548             5.130925             4.848030</span>
<span class="co">#&gt;   58175              -1.767924            -1.880302             2.993289</span>
<span class="co">#&gt;         Samples</span>
<span class="co">#&gt; Tags     23/GSM1545541_JMS8-4 23/GSM1545542_JMS8-5 23/GSM1545544_JMS9-P7c</span>
<span class="co">#&gt;   497097            -3.993876             3.306782              -3.204336</span>
<span class="co">#&gt;   27395              3.786547             3.918878               4.345642</span>
<span class="co">#&gt;   18777              5.081794             5.080061               5.757404</span>
<span class="co">#&gt;   21399              4.944024             5.158292               5.036933</span>
<span class="co">#&gt;   58175              3.357379            -2.114104               3.142621</span>
<span class="co">#&gt;         Samples</span>
<span class="co">#&gt; Tags     23/GSM1545545_JMS9-P8c</span>
<span class="co">#&gt;   497097              -5.287282</span>
<span class="co">#&gt;   27395                4.132678</span>
<span class="co">#&gt;   18777                5.150470</span>
<span class="co">#&gt;   21399                4.987679</span>
<span class="co">#&gt;   58175                3.523290</span>
<span class="co">#&gt; 14160 more rows ...</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $weights</span>
<span class="co">#&gt;           [,1]      [,2]      [,3]     [,4]      [,5]      [,6]      [,7]</span>
<span class="co">#&gt; [1,]  1.183974  1.183974 20.526779 20.97747  1.773562  1.217142 21.125740</span>
<span class="co">#&gt; [2,] 20.879554 26.561871 31.596323 29.66102 32.558344 26.745293 29.792090</span>
<span class="co">#&gt; [3,] 28.003202 33.695540 34.845507 34.45673 35.148529 33.550527 34.517259</span>
<span class="co">#&gt; [4,] 27.670233 29.595778 34.901302 34.43298 34.841349 33.159425 34.493456</span>
<span class="co">#&gt; [5,] 19.737381 18.658333  3.184207  2.62986 24.191635 24.014937  2.648747</span>
<span class="co">#&gt;           [,8]      [,9]</span>
<span class="co">#&gt; [1,]  1.183974  1.183974</span>
<span class="co">#&gt; [2,] 21.900102 17.150677</span>
<span class="co">#&gt; [3,] 31.440457 25.228325</span>
<span class="co">#&gt; [4,] 26.136796 24.502247</span>
<span class="co">#&gt; [5,] 13.149278 14.351930</span>
<span class="co">#&gt; 14160 more rows ...</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $design</span>
<span class="co">#&gt;   Basal LP ML laneL006 laneL008</span>
<span class="co">#&gt; 1     0  1  0        0        0</span>
<span class="co">#&gt; 2     0  0  1        0        0</span>
<span class="co">#&gt; 3     1  0  0        0        0</span>
<span class="co">#&gt; 4     1  0  0        1        0</span>
<span class="co">#&gt; 5     0  0  1        1        0</span>
<span class="co">#&gt; 6     0  1  0        1        0</span>
<span class="co">#&gt; 7     1  0  0        1        0</span>
<span class="co">#&gt; 8     0  0  1        0        1</span>
<span class="co">#&gt; 9     0  1  0        0        1</span>
<span class="co">#&gt; attr(,&quot;assign&quot;)</span>
<span class="co">#&gt; [1] 1 1 1 2 2</span>
<span class="co">#&gt; attr(,&quot;contrasts&quot;)</span>
<span class="co">#&gt; attr(,&quot;contrasts&quot;)$group</span>
<span class="co">#&gt; [1] &quot;contr.treatment&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; attr(,&quot;contrasts&quot;)$lane</span>
<span class="co">#&gt; [1] &quot;contr.treatment&quot;</span>
vfit &lt;-<span class="st"> </span><span class="kw">lmFit</span>(v, design)
vfit &lt;-<span class="st"> </span><span class="kw">contrasts.fit</span>(vfit, <span class="dt">contrasts=</span>contr.matrix)
efit &lt;-<span class="st"> </span><span class="kw">eBayes</span>(vfit)
<span class="kw">plotSA</span>(efit, <span class="dt">main=</span><span class="st">&quot;Final model: Mean-variance trend&quot;</span>)</code></pre></div>
<p><img src="200_Law_RNAseq123_files/figure-html/voom-1.png" width="768" /></p>
<p>Note that the other data frames stored within the DGEList-object that contain gene- and sample-level information, are retained in the EList-object <code>v</code> created by <code>voom</code>. The <code>v$genes</code> data frame is equivalent to <code>x$genes</code>, <code>v$targets</code> is equivalent to <code>x$samples</code>, and the expression values stored in <code>v$E</code> is analogous to <code>x$counts</code>, albeit on a transformed scale. In addition to this, the <code>voom</code> EList-object has a matrix of precision weights <code>v$weights</code> and stores the design matrix in <code>v$design</code>.</p>
</div>
<div id="fitting-linear-models-for-comparisons-of-interest" class="section level3">
<h3><span class="header-section-number">6.7.3</span> Fitting linear models for comparisons of interest</h3>
<p>Linear modelling in <strong>limma</strong> is carried out using the <code>lmFit</code> and <code>contrasts.fit</code> functions originally written for application to microarrays. The functions can be used for both microarray and RNA-seq data and fit a separate model to the expression values for each gene. Next, empirical Bayes moderation is carried out by borrowing information across all the genes to obtain more precise estimates of gene-wise variability. The model’s residual variances are plotted against average expression values in the next figure. It can be seen from this plot that the variance is no longer dependent on the mean expression level.</p>
</div>
<div id="examining-the-number-of-de-genes" class="section level3">
<h3><span class="header-section-number">6.7.4</span> Examining the number of DE genes</h3>
<p>For a quick look at differential expression levels, the number of significantly up- and down-regulated genes can be summarised in a table. Significance is defined using an adjusted <em>p</em>-value cutoff that is set at 5% by default. For the comparison between expression levels in basal and LP, 4,127 genes are found to be down-regulated in basal relative to LP and 4,298 genes are up-regulated in basal relative to LP – a total of 8,425 DE genes. A total of 8,510 DE genes are found between basal and ML (4,338 down- and 4,172 up-regulated genes), and a total of 5,340 DE genes are found between LP and ML (2,895 down- and 2,445 up-regulated). The larger numbers of DE genes observed for comparisons involving the basal population are consistent with our observations from the MDS plots.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">summary</span>(<span class="kw">decideTests</span>(efit))
<span class="co">#&gt;        BasalvsLP BasalvsML LPvsML</span>
<span class="co">#&gt; Down        4127      4338   2895</span>
<span class="co">#&gt; NotSig      5740      5655   8825</span>
<span class="co">#&gt; Up          4298      4172   2445</span></code></pre></div>
<p>Some studies require more than an adjusted <em>p</em>-value cut-off. For a stricter definition on significance, one may require log-fold-changes (log-FCs) to be above a minimum value. The <em>treat</em> method can be used to calculate <em>p</em>-values from empirical Bayes moderated <em>t</em>-statistics with a minimum log-FC requirement. The number of differentially expressed genes are reduced to a total of 3,135 DE genes for basal versus LP, 3,270 DE genes for basal versus ML, and 385 DE genes for LP versus ML when testing requires genes to have a log-FC that is significantly greater than 1 (equivalent to a 2-fold difference between cell types on the original scale).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tfit &lt;-<span class="st"> </span><span class="kw">treat</span>(vfit, <span class="dt">lfc=</span><span class="dv">1</span>)
dt &lt;-<span class="st"> </span><span class="kw">decideTests</span>(tfit)
<span class="kw">summary</span>(dt)
<span class="co">#&gt;        BasalvsLP BasalvsML LPvsML</span>
<span class="co">#&gt; Down        1417      1512    203</span>
<span class="co">#&gt; NotSig     11030     10895  13780</span>
<span class="co">#&gt; Up          1718      1758    182</span></code></pre></div>
<p>Genes that are DE in multiple comparisons can be extracted using the results from <code>decideTests</code>, where 0s represent genes that are not DE, 1s represent genes that are up-regulated, and -1s represent genes that are down-regulated. A total of 2,409 genes are DE in both basal versus LP and basal versus ML, twenty of which are listed below. The <code>write.fit</code> function can be used to extract and write results for all three comparisons to a single output file.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">de.common &lt;-<span class="st"> </span><span class="kw">which</span>(dt[,<span class="dv">1</span>]<span class="op">!=</span><span class="dv">0</span> <span class="op">&amp;</span><span class="st"> </span>dt[,<span class="dv">2</span>]<span class="op">!=</span><span class="dv">0</span>)
<span class="kw">length</span>(de.common)
<span class="co">#&gt; [1] 2409</span>
<span class="kw">head</span>(tfit<span class="op">$</span>genes<span class="op">$</span>SYMBOL[de.common], <span class="dt">n=</span><span class="dv">20</span>)
<span class="co">#&gt;  [1] &quot;Xkr4&quot;          &quot;Rgs20&quot;         &quot;Cpa6&quot;          &quot;Sulf1&quot;        </span>
<span class="co">#&gt;  [5] &quot;Eya1&quot;          &quot;Msc&quot;           &quot;Sbspon&quot;        &quot;Pi15&quot;         </span>
<span class="co">#&gt;  [9] &quot;Crispld1&quot;      &quot;Kcnq5&quot;         &quot;Ptpn18&quot;        &quot;Arhgef4&quot;      </span>
<span class="co">#&gt; [13] &quot;2010300C02Rik&quot; &quot;Aff3&quot;          &quot;Npas2&quot;         &quot;Tbc1d8&quot;       </span>
<span class="co">#&gt; [17] &quot;Creg2&quot;         &quot;Il1r1&quot;         &quot;Il18r1&quot;        &quot;Il18rap&quot;</span>
<span class="kw">vennDiagram</span>(dt[,<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], <span class="dt">circle.col=</span><span class="kw">c</span>(<span class="st">&quot;turquoise&quot;</span>, <span class="st">&quot;salmon&quot;</span>))</code></pre></div>
<div class="figure"><span id="fig:venn"></span>
<img src="200_Law_RNAseq123_files/figure-html/venn-1.png" alt="Venn diagram showing the number of genes DE in the comparison between basal versus LP only (left), basal versus ML only (right), and the number of genes that are DE in both comparisons (center). The number of genes that are not DE in either comparison are marked in the bottom-right." width="576" />
<p class="caption">
Figure 6.4: Venn diagram showing the number of genes DE in the comparison between basal versus LP only (left), basal versus ML only (right), and the number of genes that are DE in both comparisons (center). The number of genes that are not DE in either comparison are marked in the bottom-right.
</p>
</div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">write.fit</span>(tfit, dt, <span class="dt">file=</span><span class="st">&quot;results.txt&quot;</span>)</code></pre></div>
</div>
<div id="examining-individual-de-genes-from-top-to-bottom" class="section level3">
<h3><span class="header-section-number">6.7.5</span> Examining individual DE genes from top to bottom</h3>
<p>The top DE genes can be listed using <code>topTreat</code> for results using <code>treat</code> (or <code>topTable</code> for results using <code>eBayes</code>). By default <code>topTreat</code> arranges genes from smallest to largest adjusted <em>p</em>-value with associated gene information, log-FC, average log-CPM, moderated <em>t</em>-statistic, raw and adjusted <em>p</em>-value for each gene. The number of top genes displayed can be specified, where <code>n=Inf</code> includes all genes. Genes <em>Cldn7</em> and <em>Rasef</em> are amongst the top DE genes for both basal versus LP and basal versus ML.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">basal.vs.lp &lt;-<span class="st"> </span><span class="kw">topTreat</span>(tfit, <span class="dt">coef=</span><span class="dv">1</span>, <span class="dt">n=</span><span class="ot">Inf</span>)
basal.vs.ml &lt;-<span class="st"> </span><span class="kw">topTreat</span>(tfit, <span class="dt">coef=</span><span class="dv">2</span>, <span class="dt">n=</span><span class="ot">Inf</span>)
<span class="kw">head</span>(basal.vs.lp)
<span class="co">#&gt;        ENTREZID SYMBOL TXCHROM     logFC  AveExpr         t      P.Value</span>
<span class="co">#&gt; 12759     12759    Clu   chr14 -5.442877 8.857907 -33.44429 3.990899e-10</span>
<span class="co">#&gt; 53624     53624  Cldn7   chr11 -5.514605 6.296762 -32.94533 4.503694e-10</span>
<span class="co">#&gt; 242505   242505  Rasef    chr4 -5.921741 5.119585 -31.77625 6.063249e-10</span>
<span class="co">#&gt; 67451     67451   Pkp2   chr16 -5.724823 4.420495 -30.65370 8.010456e-10</span>
<span class="co">#&gt; 228543   228543   Rhov    chr2 -6.253427 5.486640 -29.46244 1.112729e-09</span>
<span class="co">#&gt; 70350     70350  Basp1   chr15 -6.073297 5.248349 -28.64890 1.380545e-09</span>
<span class="co">#&gt;           adj.P.Val</span>
<span class="co">#&gt; 12759  2.703871e-06</span>
<span class="co">#&gt; 53624  2.703871e-06</span>
<span class="co">#&gt; 242505 2.703871e-06</span>
<span class="co">#&gt; 67451  2.703871e-06</span>
<span class="co">#&gt; 228543 2.703871e-06</span>
<span class="co">#&gt; 70350  2.703871e-06</span>
<span class="kw">head</span>(basal.vs.ml)
<span class="co">#&gt;        ENTREZID  SYMBOL TXCHROM     logFC  AveExpr         t      P.Value</span>
<span class="co">#&gt; 242505   242505   Rasef    chr4 -6.510470 5.119585 -35.49093 2.573575e-10</span>
<span class="co">#&gt; 53624     53624   Cldn7   chr11 -5.469160 6.296762 -32.52520 4.978446e-10</span>
<span class="co">#&gt; 12521     12521    Cd82    chr2 -4.667737 7.070963 -31.82187 5.796191e-10</span>
<span class="co">#&gt; 71740     71740 Nectin4    chr1 -5.556046 5.166292 -31.29987 6.760578e-10</span>
<span class="co">#&gt; 20661     20661   Sort1    chr3 -4.908119 6.705784 -31.23083 6.761331e-10</span>
<span class="co">#&gt; 15375     15375   Foxa1   chr12 -5.753884 5.625064 -28.34612 1.487280e-09</span>
<span class="co">#&gt;           adj.P.Val</span>
<span class="co">#&gt; 242505 1.915485e-06</span>
<span class="co">#&gt; 53624  1.915485e-06</span>
<span class="co">#&gt; 12521  1.915485e-06</span>
<span class="co">#&gt; 71740  1.915485e-06</span>
<span class="co">#&gt; 20661  1.915485e-06</span>
<span class="co">#&gt; 15375  2.281914e-06</span></code></pre></div>
</div>
<div id="useful-graphical-representations-of-differential-expression-results" class="section level3">
<h3><span class="header-section-number">6.7.6</span> Useful graphical representations of differential expression results</h3>
<p>To summarise results for all genes visually, mean-difference plots, which display log-FCs from the linear model fit against the average log-CPM values can be generated using the <code>plotMD</code> function, with the differentially expressed genes highlighted.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plotMD</span>(tfit, <span class="dt">column=</span><span class="dv">1</span>, <span class="dt">status=</span>dt[,<span class="dv">1</span>], <span class="dt">main=</span><span class="kw">colnames</span>(tfit)[<span class="dv">1</span>], 
       <span class="dt">xlim=</span><span class="kw">c</span>(<span class="op">-</span><span class="dv">8</span>,<span class="dv">13</span>))</code></pre></div>
<p><strong>Glimma</strong> extends this functionality by providing an interactive mean-difference plot via the <code>glMDPlot</code> function. The output of this function is an html page, with summarised results in the left panel (similar to what is output by <code>plotMD</code>), and the log-CPM values from individual samples in the right panel, with a table of results below the plots. This interactive display allows the user to search for particular genes based on their Gene symbol, which is not possible in a static <strong>R</strong> plot. The <code>glMDPlot</code> function is not limited to mean-difference plots, with a default version allowing a data frame to be passed with the user able to select the columns of interest to plot in the left panel.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">glMDPlot</span>(tfit, <span class="dt">coef=</span><span class="dv">1</span>, <span class="dt">status=</span>dt, <span class="dt">main=</span><span class="kw">colnames</span>(tfit)[<span class="dv">1</span>],
         <span class="dt">side.main=</span><span class="st">&quot;ENTREZID&quot;</span>, <span class="dt">counts=</span>x<span class="op">$</span>counts, <span class="dt">groups=</span>group, <span class="dt">launch=</span><span class="ot">FALSE</span>)</code></pre></div>
<p>The mean-difference plot generated by the command above is available online (see <a href="http://bioinf.wehi.edu.au/folders/limmaWorkflow/glimma-plots/MD-Plot.html" class="uri">http://bioinf.wehi.edu.au/folders/limmaWorkflow/glimma-plots/MD-Plot.html</a>). The interactivity provided by the <strong>Glimma</strong> package allows additional information to be presented in a single graphical window. <strong>Glimma</strong> is implemented in <strong>R</strong> and Javascript, with the <strong>R</strong> code generating the data which is converted into graphics using the Javascript library D3 (<a href="https://d3js.org" class="uri">https://d3js.org</a>), with the Bootstrap library handling layouts and Datatables generating the interactive searchable tables. This allows plots to be viewed in any modern browser, which is convenient for including them as linked files from an Rmarkdown report of the analysis.</p>
<p>Plots shown previously include either all of the genes that are expressed in any one condition (such as the Venn diagram of common DE genes or mean-difference plot) or look at genes individually (log-CPM values shown in right panel of the interactive mean-difference plot). Heatmaps allow users to look at the expression of a subset of genes. This can be give useful insight into the expression of individual groups and samples without losing perspective of the overall study when focusing on individual genes, or losing resolution when examining patterns averaged over thousands of genes at the same time.</p>
<p>A heatmap is created for the top 100 DE genes (as ranked by adjusted p-value) from the basal versus LP contrast using the <code>heatmap.2</code> function from the <strong>gplots</strong> package. The heatmap correctly clusters samples into cell type and rearranges the order of genes to form blocks of similar expression. From the heatmap, we observe that the expression of ML and LP samples are very similar for the top 100 DE genes between basal and LP.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(gplots)
basal.vs.lp.topgenes &lt;-<span class="st"> </span>basal.vs.lp<span class="op">$</span>ENTREZID[<span class="dv">1</span><span class="op">:</span><span class="dv">100</span>]
i &lt;-<span class="st"> </span><span class="kw">which</span>(v<span class="op">$</span>genes<span class="op">$</span>ENTREZID <span class="op">%in%</span><span class="st"> </span>basal.vs.lp.topgenes)
mycol &lt;-<span class="st"> </span><span class="kw">colorpanel</span>(<span class="dv">1000</span>,<span class="st">&quot;blue&quot;</span>,<span class="st">&quot;white&quot;</span>,<span class="st">&quot;red&quot;</span>)
<span class="kw">heatmap.2</span>(v<span class="op">$</span>E[i,], <span class="dt">scale=</span><span class="st">&quot;row&quot;</span>,
   <span class="dt">labRow=</span>v<span class="op">$</span>genes<span class="op">$</span>SYMBOL[i], <span class="dt">labCol=</span>group, 
   <span class="dt">col=</span>mycol, <span class="dt">trace=</span><span class="st">&quot;none&quot;</span>, <span class="dt">density.info=</span><span class="st">&quot;none&quot;</span>, 
   <span class="dt">margin=</span><span class="kw">c</span>(<span class="dv">8</span>,<span class="dv">6</span>), <span class="dt">lhei=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">10</span>), <span class="dt">dendrogram=</span><span class="st">&quot;column&quot;</span>)</code></pre></div>
<div class="figure"><span id="fig:heatmap"></span>
<img src="200_Law_RNAseq123_files/figure-html/heatmap-1.png" alt="Heatmap of log-CPM values for top 100 genes DE in basal versus LP. Expression across each gene (or row) have been scaled so that mean expression is zero and standard deviation is one. Samples with relatively high expression of a given gene are marked in red and samples with relatively low expression are marked in blue. Lighter shades and white represent genes with intermediate expression levels. Samples and genes have been reordered by the method of hierarchical clustering. A dendrogram is shown for the sample clustering." width="672" />
<p class="caption">
Figure 6.5: Heatmap of log-CPM values for top 100 genes DE in basal versus LP. Expression across each gene (or row) have been scaled so that mean expression is zero and standard deviation is one. Samples with relatively high expression of a given gene are marked in red and samples with relatively low expression are marked in blue. Lighter shades and white represent genes with intermediate expression levels. Samples and genes have been reordered by the method of hierarchical clustering. A dendrogram is shown for the sample clustering.
</p>
</div>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="public-data-resources-and-bioconductor.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="rna-seq-data-analysis-with-deseq2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
