<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book contains all the workshops presented at the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="introduction-to-bioconductor-annotation-resources.html">
<link rel="next" href="public-data-resources-and-bioconductor.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />







<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-122837239-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-122837239-1');
</script>


<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#the-workshops"><i class="fa fa-check"></i><b>1.1</b> The Workshops</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html"><i class="fa fa-check"></i><b>2</b> 100: <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#description"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="2.1.7" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-goals"><i class="fa fa-check"></i><b>2.1.7</b> Learning goals</a></li>
<li class="chapter" data-level="2.1.8" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-objectives"><i class="fa fa-check"></i><b>2.1.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-r"><i class="fa fa-check"></i><b>2.2</b> Introduction to <em>R</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#rstudio-orientation"><i class="fa fa-check"></i><b>2.2.1</b> <em>RStudio</em> orientation</a></li>
<li class="chapter" data-level="2.2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#very-basics"><i class="fa fa-check"></i><b>2.2.2</b> Very basics</a></li>
<li class="chapter" data-level="2.2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#data-input-and-manipulation"><i class="fa fa-check"></i><b>2.2.3</b> Data input and manipulation</a></li>
<li class="chapter" data-level="2.2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#writing-r-scripts"><i class="fa fa-check"></i><b>2.2.4</b> Writing <em>R</em> scripts</a></li>
<li class="chapter" data-level="2.2.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#using-packages"><i class="fa fa-check"></i><b>2.2.5</b> Using packages</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.3</b> Introduction to <em>Bioconductor</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#discovering-installing-and-learning-how-to-use-bioconductor-packages"><i class="fa fa-check"></i><b>2.3.1</b> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-genomic-ranges"><i class="fa fa-check"></i><b>2.3.2</b> Working with Genomic Ranges</a></li>
<li class="chapter" data-level="2.3.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-summarized-experimental-data"><i class="fa fa-check"></i><b>2.3.3</b> Working with summarized experimental data</a></li>
<li class="chapter" data-level="2.3.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#down-stream-analysis"><i class="fa fa-check"></i><b>2.3.4</b> Down-stream analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>3</b> 101: Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>3.1</b> Instructors</a></li>
<li class="chapter" data-level="3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description"><i class="fa fa-check"></i><b>3.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="3.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>3.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="3.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation"><i class="fa fa-check"></i><b>3.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="3.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="3.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-1"><i class="fa fa-check"></i><b>3.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="3.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-1"><i class="fa fa-check"></i><b>3.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>3.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="3.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>3.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="3.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>3.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="3.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>3.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="3.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>3.4.4</b> Data containers</a></li>
<li class="chapter" data-level="3.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>3.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="3.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>3.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>3.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="3.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>3.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="3.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>3.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="3.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>3.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="3.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>3.4.11</b> Simple example</a></li>
<li class="chapter" data-level="3.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>3.4.12</b> Questions!</a></li>
<li class="chapter" data-level="3.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>3.4.13</b> More questions!</a></li>
<li class="chapter" data-level="3.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>3.4.14</b> Another example</a></li>
<li class="chapter" data-level="3.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>3.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="3.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>3.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="3.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>3.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="3.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>3.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="3.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>3.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="3.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>3.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="3.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>3.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="3.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>3.4.22</b> GRanges</a></li>
<li class="chapter" data-level="3.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>3.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="3.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>3.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="3.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>3.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="3.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>3.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="3.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>3.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="3.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>3.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="3.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>3.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>3.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="3.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>3.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="3.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>3.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="3.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr"><i class="fa fa-check"></i><b>3.4.33</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-1"><i class="fa fa-check"></i><b>3.4.34</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-2"><i class="fa fa-check"></i><b>3.4.35</b> Organism.dplyr</a></li>
<li class="chapter" data-level="3.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>3.4.36</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="3.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>3.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>3.4.38</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>3.4.39</b> BSgenome packages</a></li>
<li class="chapter" data-level="3.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>3.4.40</b> BSgenome exercises</a></li>
<li class="chapter" data-level="3.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub"><i class="fa fa-check"></i><b>3.4.41</b> AnnotationHub</a></li>
<li class="chapter" data-level="3.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub"><i class="fa fa-check"></i><b>3.4.42</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="3.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>3.4.43</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="3.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>3.4.44</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="3.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>3.4.45</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="3.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>3.4.46</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="3.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>3.4.47</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>3.4.48</b> AnnotationHub query</a></li>
<li class="chapter" data-level="3.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>3.4.49</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="3.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>3.4.50</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="3.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>3.4.51</b> biomaRt</a></li>
<li class="chapter" data-level="3.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>3.4.52</b> biomaRt data sets</a></li>
<li class="chapter" data-level="3.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>3.4.53</b> biomaRt queries</a></li>
<li class="chapter" data-level="3.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>3.4.54</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="3.4.55" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>3.4.55</b> biomaRt query</a></li>
<li class="chapter" data-level="3.4.56" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>3.4.56</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>4</b> 102: Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>4.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-1"><i class="fa fa-check"></i><b>4.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="4.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-1"><i class="fa fa-check"></i><b>4.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-1"><i class="fa fa-check"></i><b>4.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="4.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-1"><i class="fa fa-check"></i><b>4.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="4.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-2"><i class="fa fa-check"></i><b>4.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="4.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-2"><i class="fa fa-check"></i><b>4.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#introduction-1"><i class="fa fa-check"></i><b>4.4</b> Introduction</a><ul>
<li class="chapter" data-level="4.4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#what-is-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.1</b> What is the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#why-use-the-ranges-infrastructure"><i class="fa fa-check"></i><b>4.4.2</b> Why use the Ranges infrastructure?</a></li>
<li class="chapter" data-level="4.4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>4.4.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#setup"><i class="fa fa-check"></i><b>4.5</b> Setup</a></li>
<li class="chapter" data-level="4.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>4.6</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="4.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#constructing-a-granges-object-from-data.frame"><i class="fa fa-check"></i><b>4.7</b> Constructing a <em>GRanges</em> object from data.frame</a></li>
<li class="chapter" data-level="4.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#loading-a-granges-object-from-a-standard-file-format"><i class="fa fa-check"></i><b>4.8</b> Loading a <em>GRanges</em> object from a standard file format</a></li>
<li class="chapter" data-level="4.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-manipulation-of-granges-objects"><i class="fa fa-check"></i><b>4.9</b> Basic manipulation of <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>4.10</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>4.11</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#aggregating-granges-objects"><i class="fa fa-check"></i><b>4.12</b> Aggregating <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.13</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>4.14</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.15" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#finding-overlaps-between-granges-objects"><i class="fa fa-check"></i><b>4.15</b> Finding overlaps between <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="4.16" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises"><i class="fa fa-check"></i><b>4.16</b> Exercises</a></li>
<li class="chapter" data-level="4.17" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>4.17</b> Example: exploring BigWig files from AnnotationHub</a><ul>
<li class="chapter" data-level="4.17.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#extracting-data-from-annotationhub"><i class="fa fa-check"></i><b>4.17.1</b> Extracting data from AnnotationHub</a></li>
<li class="chapter" data-level="4.17.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#reading-bigwig-files"><i class="fa fa-check"></i><b>4.17.2</b> Reading BigWig files</a></li>
<li class="chapter" data-level="4.17.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-1"><i class="fa fa-check"></i><b>4.17.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.18" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>4.18</b> Worked example: coverage analysis of BAM files</a><ul>
<li class="chapter" data-level="4.18.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-2"><i class="fa fa-check"></i><b>4.18.1</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4.19" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#conclusions"><i class="fa fa-check"></i><b>4.19</b> Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>5</b> 103: Public data resources and Bioconductor</a><ul>
<li class="chapter" data-level="5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>5.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>5.2</b> Syllabus</a><ul>
<li class="chapter" data-level="5.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-2"><i class="fa fa-check"></i><b>5.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="5.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-2"><i class="fa fa-check"></i><b>5.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="5.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-2"><i class="fa fa-check"></i><b>5.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>5.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="5.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-2"><i class="fa fa-check"></i><b>5.2.5</b> Time outline</a></li>
<li class="chapter" data-level="5.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>5.2.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="5.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-3"><i class="fa fa-check"></i><b>5.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="5.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-3"><i class="fa fa-check"></i><b>5.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-1"><i class="fa fa-check"></i><b>5.3</b> Overview</a></li>
<li class="chapter" data-level="5.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>5.4</b> GEOquery</a><ul>
<li class="chapter" data-level="5.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>5.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="5.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>5.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="5.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>5.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="5.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>5.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>5.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="5.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>5.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="5.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>5.5.2</b> Basic design</a></li>
<li class="chapter" data-level="5.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>5.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="5.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>5.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="5.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>5.6.1</b> Authentication</a></li>
<li class="chapter" data-level="5.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>5.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>5.7</b> Sequence Read Archive</a><ul>
<li class="chapter" data-level="5.7.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage-1"><i class="fa fa-check"></i><b>5.7.1</b> Usage</a></li>
<li class="chapter" data-level="5.7.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#query-syntax"><i class="fa fa-check"></i><b>5.7.2</b> Query syntax</a></li>
<li class="chapter" data-level="5.7.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#using-the-raw-api-without-rbioconductor"><i class="fa fa-check"></i><b>5.7.3</b> Using the raw API without R/Bioconductor</a></li>
</ul></li>
<li class="chapter" data-level="5.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>5.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="5.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>5.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="5.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>5.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="5.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>5.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="5.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>5.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="5.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>5.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="5.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>5.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="5.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>5.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="5.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>5.12</b> Pharmacogenomics</a><ul>
<li class="chapter" data-level="5.12.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started"><i class="fa fa-check"></i><b>5.12.1</b> Getting started</a></li>
<li class="chapter" data-level="5.12.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-pharmacogx-datasets-pharmacosets"><i class="fa fa-check"></i><b>5.12.2</b> Overview of PharmacoGx datasets (PharmacoSets)</a></li>
<li class="chapter" data-level="5.12.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#exploring-drug-sensitivity-datasets"><i class="fa fa-check"></i><b>5.12.3</b> Exploring drug sensitivity datasets</a></li>
</ul></li>
<li class="chapter" data-level="5.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>5.13</b> Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>6</b> 200: RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>6.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-3"><i class="fa fa-check"></i><b>6.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="6.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-3"><i class="fa fa-check"></i><b>6.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="6.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>6.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-3"><i class="fa fa-check"></i><b>6.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>6.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="6.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-4"><i class="fa fa-check"></i><b>6.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="6.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-4"><i class="fa fa-check"></i><b>6.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-2"><i class="fa fa-check"></i><b>6.4</b> Introduction</a></li>
<li class="chapter" data-level="6.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>6.5</b> Data packaging</a><ul>
<li class="chapter" data-level="6.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>6.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="6.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>6.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="6.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>6.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>6.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="6.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>6.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="6.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>6.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="6.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>6.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="6.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>6.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="6.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis"><i class="fa fa-check"></i><b>6.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="6.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>6.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="6.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>6.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="6.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>6.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="6.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>6.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="6.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>6.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="6.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>6.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>7</b> 201: RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview-2"><i class="fa fa-check"></i><b>7.1</b> Overview</a><ul>
<li class="chapter" data-level="7.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-1"><i class="fa fa-check"></i><b>7.1.1</b> Description</a></li>
<li class="chapter" data-level="7.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation-1"><i class="fa fa-check"></i><b>7.1.3</b> Participation</a></li>
<li class="chapter" data-level="7.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>7.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-4"><i class="fa fa-check"></i><b>7.1.5</b> Time outline</a></li>
<li class="chapter" data-level="7.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>7.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>7.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="7.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>7.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="7.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>7.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="7.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>7.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="7.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>7.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>7.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="7.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>7.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="7.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>7.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="7.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>7.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="7.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>7.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="7.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>7.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="7.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>7.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="7.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>7.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="7.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>7.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="7.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>7.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="7.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>7.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="7.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub-1"><i class="fa fa-check"></i><b>7.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="7.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>7.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="7.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>7.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="7.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>7.7</b> Building reports</a><ul>
<li class="chapter" data-level="7.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>7.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="7.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>7.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="7.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>7.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="7.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>7.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>7.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="7.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>7.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="7.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>7.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="7.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>7.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="7.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>7.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>8</b> 202: Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="8.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-3"><i class="fa fa-check"></i><b>8.1</b> Overview</a><ul>
<li class="chapter" data-level="8.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-2"><i class="fa fa-check"></i><b>8.1.1</b> Description</a></li>
<li class="chapter" data-level="8.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-2"><i class="fa fa-check"></i><b>8.1.3</b> Participation</a></li>
<li class="chapter" data-level="8.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>8.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-5"><i class="fa fa-check"></i><b>8.1.5</b> Time outline</a></li>
<li class="chapter" data-level="8.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>8.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started-1"><i class="fa fa-check"></i><b>8.2</b> Getting started</a><ul>
<li class="chapter" data-level="8.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>8.2.1</b> The data</a></li>
<li class="chapter" data-level="8.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>8.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="8.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>8.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="8.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>8.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="8.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>8.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="8.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>8.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="8.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>8.5.1</b> Normalization</a></li>
<li class="chapter" data-level="8.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>8.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="8.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>8.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="8.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>8.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="8.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>8.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>9</b> 210: Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="9.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>9.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-4"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-4"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-6"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>9.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="9.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>9.4</b> Where does it all come from?</a></li>
<li class="chapter" data-level="9.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.5</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="9.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>9.6</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="9.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>9.7</b> Data types</a></li>
<li class="chapter" data-level="9.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>9.8</b> Differential expression analysis</a></li>
<li class="chapter" data-level="9.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>9.9</b> Gene sets</a></li>
<li class="chapter" data-level="9.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>9.10</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="9.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>9.11</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="9.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.12</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="9.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>9.13</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>10</b> 220: Workflow for multi-omics analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="10.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information-2"><i class="fa fa-check"></i><b>10.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-5"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-5"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>10.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-7"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-5"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-5"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>10.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="10.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="10.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>10.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="10.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>10.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="10.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>10.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="10.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>10.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="10.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>10.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="10.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>10.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>10.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>10.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="10.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>10.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="10.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>10.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="10.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>10.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="10.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>10.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="10.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>10.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="10.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>10.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="10.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>10.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="10.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>10.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="10.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>10.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="10.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>10.9</b> Complete cases</a></li>
<li class="chapter" data-level="10.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>10.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="10.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>10.11</b> Extraction</a><ul>
<li class="chapter" data-level="10.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>10.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="10.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>10.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="10.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>10.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="10.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>10.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="10.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>10.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="10.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>10.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="10.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>10.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="10.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>10.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="10.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>10.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="10.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>10.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="10.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>10.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="10.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>10.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="10.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>10.16.1</b> “Simplification” of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="10.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>10.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="10.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>10.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="10.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>10.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="10.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>10.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="10.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>10.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="10.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>10.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="10.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>10.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="10.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>10.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="10.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>10.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="10.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>10.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="10.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>10.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>11</b> 230: Cytoscape automation in R using Rcy3</a><ul>
<li class="chapter" data-level="11.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-4"><i class="fa fa-check"></i><b>11.1</b> Overview</a><ul>
<li class="chapter" data-level="11.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructor-names-and-contact-information-3"><i class="fa fa-check"></i><b>11.1.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="11.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description-6"><i class="fa fa-check"></i><b>11.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="11.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>11.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="11.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>11.1.4</b> Background:</a></li>
<li class="chapter" data-level="11.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation-6"><i class="fa fa-check"></i><b>11.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>11.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="11.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-8"><i class="fa fa-check"></i><b>11.1.7</b> Time outline</a></li>
<li class="chapter" data-level="11.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>11.2</b> Background</a><ul>
<li class="chapter" data-level="11.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>11.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="11.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>11.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="11.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>11.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>11.3</b> Translating biological data into Cytoscape using RCy3</a></li>
<li class="chapter" data-level="11.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>11.4</b> Set up</a><ul>
<li class="chapter" data-level="11.4.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started-2"><i class="fa fa-check"></i><b>11.4.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>11.5</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="11.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>11.6</b> Example Data Set</a></li>
<li class="chapter" data-level="11.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>11.7</b> Finding Network Data</a></li>
<li class="chapter" data-level="11.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>11.8</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="11.9" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>11.9</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="11.10" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>11.10</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>12</b> 240: Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="12.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructor-names-and-contact-information-4"><i class="fa fa-check"></i><b>12.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="12.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-7"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-8"><i class="fa fa-check"></i><b>12.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-7"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>12.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="12.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-9"><i class="fa fa-check"></i><b>12.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-6"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-6"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop"><i class="fa fa-check"></i><b>12.4</b> Workshop</a></li>
<li class="chapter" data-level="12.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-3"><i class="fa fa-check"></i><b>12.5</b> Introduction</a><ul>
<li class="chapter" data-level="12.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>12.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="12.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>12.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="12.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for-1"><i class="fa fa-check"></i><b>12.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup-1"><i class="fa fa-check"></i><b>12.6</b> Setup</a></li>
<li class="chapter" data-level="12.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>12.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="12.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>12.8</b> The Grammar</a><ul>
<li class="chapter" data-level="12.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>12.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="12.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>12.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="12.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises-3"><i class="fa fa-check"></i><b>12.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>12.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="12.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>12.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="12.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files-1"><i class="fa fa-check"></i><b>12.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="12.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>12.10</b> What’s next?</a></li>
<li class="chapter" data-level="12.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>12.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html"><i class="fa fa-check"></i><b>13</b> 250: Working with genomic data in R with the DECIPHER package</a><ul>
<li class="chapter" data-level="13.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#overview-5"><i class="fa fa-check"></i><b>13.1</b> Overview</a><ul>
<li class="chapter" data-level="13.1.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-description-8"><i class="fa fa-check"></i><b>13.1.1</b> Workshop Description</a></li>
<li class="chapter" data-level="13.1.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#pre-requisites-9"><i class="fa fa-check"></i><b>13.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="13.1.3" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-participation-8"><i class="fa fa-check"></i><b>13.1.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.1.4" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>13.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="13.1.5" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#time-outline-10"><i class="fa fa-check"></i><b>13.1.5</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.2</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.2.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-goals-7"><i class="fa fa-check"></i><b>13.2.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.2.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-objectives-7"><i class="fa fa-check"></i><b>13.2.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>14</b> 260: Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="14.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>14.1</b> Instructors:</a></li>
<li class="chapter" data-level="14.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-9"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-10"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-9"><i class="fa fa-check"></i><b>14.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>14.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="14.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-11"><i class="fa fa-check"></i><b>14.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-8"><i class="fa fa-check"></i><b>14.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-8"><i class="fa fa-check"></i><b>14.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#abstract"><i class="fa fa-check"></i><b>14.4</b> Abstract</a></li>
<li class="chapter" data-level="14.5" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#introduction-5"><i class="fa fa-check"></i><b>14.5</b> Introduction</a><ul>
<li class="chapter" data-level="14.5.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#downloading-pharmacoset-objects"><i class="fa fa-check"></i><b>14.5.1</b> Downloading PharmacoSet objects</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#reproducibility"><i class="fa fa-check"></i><b>14.6</b> Reproducibility</a><ul>
<li class="chapter" data-level="14.6.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#plotting-drug-dose-response-data"><i class="fa fa-check"></i><b>14.6.1</b> Plotting Drug-Dose Response Data</a></li>
<li class="chapter" data-level="14.6.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pharmacological-profiles"><i class="fa fa-check"></i><b>14.6.2</b> Pharmacological profiles</a></li>
</ul></li>
<li class="chapter" data-level="14.7" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#replication"><i class="fa fa-check"></i><b>14.7</b> Replication</a><ul>
<li class="chapter" data-level="14.7.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-of-pharmacological-profiles"><i class="fa fa-check"></i><b>14.7.1</b> Consistency of pharmacological profiles</a></li>
<li class="chapter" data-level="14.7.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#consistency-assessment-improved-by-modified-concordance-index"><i class="fa fa-check"></i><b>14.7.2</b> consistency assessment improved by Modified Concordance Index</a></li>
<li class="chapter" data-level="14.7.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#known-biomarkers"><i class="fa fa-check"></i><b>14.7.3</b> Known Biomarkers</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#machine-learning-and-biomarker-discovery"><i class="fa fa-check"></i><b>14.8</b> Machine Learning and Biomarker Discovery</a><ul>
<li class="chapter" data-level="14.8.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#bonus-using-the-connectivity-map-for-drug-repurposing"><i class="fa fa-check"></i><b>14.8.1</b> Bonus: Using the Connectivity Map for drug repurposing</a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#session-info"><i class="fa fa-check"></i><b>14.9</b> Session Info</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>15</b> 500: Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="15.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-6"><i class="fa fa-check"></i><b>15.1</b> Overview</a><ul>
<li class="chapter" data-level="15.1.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#description-3"><i class="fa fa-check"></i><b>15.1.1</b> Description</a></li>
<li class="chapter" data-level="15.1.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-11"><i class="fa fa-check"></i><b>15.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="15.1.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#participation-3"><i class="fa fa-check"></i><b>15.1.3</b> Participation</a></li>
<li class="chapter" data-level="15.1.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-11"><i class="fa fa-check"></i><b>15.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="15.1.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-12"><i class="fa fa-check"></i><b>15.1.5</b> Time outline</a></li>
<li class="chapter" data-level="15.1.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="15.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#introductory-material"><i class="fa fa-check"></i><b>15.2</b> Introductory material</a><ul>
<li class="chapter" data-level="15.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#discussion"><i class="fa fa-check"></i><b>15.2.1</b> Discussion</a></li>
<li class="chapter" data-level="15.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#why-learn-delayedarray"><i class="fa fa-check"></i><b>15.2.2</b> Why learn DelayedArray?</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-of-delayedarray-framework"><i class="fa fa-check"></i><b>15.3</b> Overview of DelayedArray framework</a><ul>
<li class="chapter" data-level="15.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-package"><i class="fa fa-check"></i><b>15.3.1</b> The <strong>DelayedArray</strong> package</a></li>
<li class="chapter" data-level="15.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-class"><i class="fa fa-check"></i><b>15.3.2</b> The <em>DelayedArray</em> class</a></li>
<li class="chapter" data-level="15.3.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#operating-on-delayedarray-objects"><i class="fa fa-check"></i><b>15.3.3</b> Operating on <em>DelayedArray</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="15.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#whats-out-there-already"><i class="fa fa-check"></i><b>15.4</b> What’s out there already?</a></li>
<li class="chapter" data-level="15.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#incoporating-delayedarray-into-a-package"><i class="fa fa-check"></i><b>15.5</b> Incoporating DelayedArray into a package</a><ul>
<li class="chapter" data-level="15.5.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#writing-algorithms-to-process-delayedarray-instances"><i class="fa fa-check"></i><b>15.5.1</b> Writing algorithms to process <em>DelayedArray</em> instances</a></li>
</ul></li>
<li class="chapter" data-level="15.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#questions-and-discussion"><i class="fa fa-check"></i><b>15.6</b> Questions and discussion</a></li>
<li class="chapter" data-level="15.7" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#todos"><i class="fa fa-check"></i><b>15.7</b> TODOs</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>16</b> 510: Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="16.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-2"><i class="fa fa-check"></i><b>16.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="16.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-10"><i class="fa fa-check"></i><b>16.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="16.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-12"><i class="fa fa-check"></i><b>16.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="16.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-4"><i class="fa fa-check"></i><b>16.2.2</b> Participation</a></li>
<li class="chapter" data-level="16.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>16.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="16.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-13"><i class="fa fa-check"></i><b>16.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-6"><i class="fa fa-check"></i><b>16.3</b> Introduction</a><ul>
<li class="chapter" data-level="16.3.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>16.3.1</b> Why maintain your package</a></li>
<li class="chapter" data-level="16.3.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>16.3.2</b> Infrastructure used for maintenance</a></li>
</ul></li>
<li class="chapter" data-level="16.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>16.4</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="16.4.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>16.4.1</b> Build report</a></li>
<li class="chapter" data-level="16.4.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#version-numbering"><i class="fa fa-check"></i><b>16.4.2</b> Version Numbering</a></li>
<li class="chapter" data-level="16.4.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>16.4.3</b> Landing page badges</a></li>
<li class="chapter" data-level="16.4.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>16.4.4</b> Download statistics</a></li>
<li class="chapter" data-level="16.4.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>16.4.5</b> RSS feeds</a></li>
<li class="chapter" data-level="16.4.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>16.4.6</b> Support site, Bioc-devel mailing list</a></li>
<li class="chapter" data-level="16.4.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github"><i class="fa fa-check"></i><b>16.4.7</b> GitHub</a></li>
</ul></li>
<li class="chapter" data-level="16.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>16.5</b> Round up of resources available</a></li>
<li class="chapter" data-level="16.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#acknowledgements"><i class="fa fa-check"></i><b>16.6</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="bibliography-1.html"><a href="bibliography-1.html"><i class="fa fa-check"></i><b>17</b> Bibliography</a></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="solving-common-bioinformatic-challenges-using-genomicranges" class="section level1">
<h1><span class="header-section-number">4</span> 102: Solving common bioinformatic challenges using GenomicRanges</h1>
<div id="instructor-name-and-contact-information" class="section level2">
<h2><span class="header-section-number">4.1</span> Instructor name and contact information</h2>
<ul>
<li>Michael Lawrence (<a href="mailto:michafla@gene.com">michafla@gene.com</a>)</li>
</ul>
</div>
<div id="workshop-description-1" class="section level2">
<h2><span class="header-section-number">4.2</span> Workshop Description</h2>
<p>We will introduce the fundamental concepts underlying the GenomicRanges package and related infrastructure. After a structured introduction, we will follow a realistic workflow, along the way exploring the central data structures, including GRanges and SummarizedExperiment, and useful operations in the ranges algebra. Topics will include data import/export, computing and summarizing data on genomic features, overlap detection, integration with reference annotations, scaling strategies, and visualization. Students can follow along, and there will be plenty of time for students to ask questions about how to apply the infrastructure to their particular use case. Michael Lawrence (Genentech).</p>
<div id="pre-requisites-1" class="section level3">
<h3><span class="header-section-number">4.2.1</span> Pre-requisites</h3>
<ul>
<li>Solid understanding of R</li>
<li>Basic familiarity with GRanges objects</li>
<li>Basic familiarity with packages like S4Vectors, IRanges, GenomicRanges, rtracklayer, etc.</li>
</ul>
</div>
<div id="workshop-participation-1" class="section level3">
<h3><span class="header-section-number">4.2.2</span> Workshop Participation</h3>
<p>Describe how students will be expected to participate in the workshop.</p>
</div>
<div id="r-bioconductor-packages-used-2" class="section level3">
<h3><span class="header-section-number">4.2.3</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>S4Vectors</li>
<li>IRanges</li>
<li>GenomicRanges</li>
<li>rtracklayer</li>
<li>GenomicFeatures</li>
<li>SummarizedExperiment</li>
<li>GenomicAlignments</li>
</ul>
</div>
<div id="time-outline-1" class="section level3">
<h3><span class="header-section-number">4.2.4</span> Time outline</h3>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Intro slides</td>
<td>30m</td>
</tr>
<tr class="even">
<td>Workflow(s)</td>
<td>1hr</td>
</tr>
<tr class="odd">
<td>Remaining questions</td>
<td>30m</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="workshop-goals-and-objectives-2" class="section level2">
<h2><span class="header-section-number">4.3</span> Workshop goals and objectives</h2>
<div id="learning-goals-2" class="section level3">
<h3><span class="header-section-number">4.3.1</span> Learning goals</h3>
<ul>
<li>Understand how to apply the *Ranges infrastructure to real-world problems</li>
<li>Gain insight into the design principles of the infrastructure and how it was meant to be used</li>
</ul>
</div>
<div id="learning-objectives-2" class="section level3">
<h3><span class="header-section-number">4.3.2</span> Learning objectives</h3>
<ul>
<li>Manipulate GRanges and related objects</li>
<li>Use the ranges algebra to analyze genomic ranges</li>
<li>Implement efficient workflows based on the *Ranges infrastructure</li>
</ul>
</div>
</div>
<div id="introduction-1" class="section level2">
<h2><span class="header-section-number">4.4</span> Introduction</h2>
<div id="what-is-the-ranges-infrastructure" class="section level3">
<h3><span class="header-section-number">4.4.1</span> What is the Ranges infrastructure?</h3>
<p>The Ranges framework of packages provide data structures and algorithms for analyzing genomic data. This includes standard genomic data containers like GRanges and SummarizedExperiment, optimized data representations like Rle, and fast algorithms for computing overlaps, finding nearest neighbors, summarizing ranges and metadata, etc.</p>
</div>
<div id="why-use-the-ranges-infrastructure" class="section level3">
<h3><span class="header-section-number">4.4.2</span> Why use the Ranges infrastructure?</h3>
<p>Hundreds of Bioconductor packages operate on Ranges data structures, enabling the construction of complex workflows integrating multiple packages and data types. The API directly supports data analysis as well the construction of new genomic software. Code evolves easily from analysis script to generalized package extending the Bioconductor ecosystem.</p>
</div>
<div id="who-is-this-workshop-for" class="section level3">
<h3><span class="header-section-number">4.4.3</span> Who is this workshop for?</h3>
<p>If you still think of R as a programming language and want to write new bioinformatics algorithms and/or build interoperable software on top of formal genomic data structures, this workshop is for you. For the tidyverse analog of this workshop, see the plyranges tutorial by Stuart Lee.</p>
</div>
</div>
<div id="setup" class="section level2">
<h2><span class="header-section-number">4.5</span> Setup</h2>
<p>To participate in this workshop you’ll need to have R &gt;= 3.5 and install the GenomicRanges, AnnotationHub, and airway Bioconductor 3.7 packages (<span class="citation">Morgan (2018)</span>; <span class="citation">Love (2018)</span>). You can achieve this by installing the BiocManager package from CRAN, loading it then running the install command:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)
<span class="kw">library</span>(BiocManager)
<span class="kw">install</span>(<span class="kw">c</span>(<span class="st">&quot;GenomicRanges&quot;</span>, <span class="st">&quot;AnnotationHub&quot;</span>, <span class="st">&quot;airway&quot;</span>))</code></pre></div>
</div>
<div id="granges-genomic-ranges" class="section level2">
<h2><span class="header-section-number">4.6</span> <em>GRanges</em>: Genomic Ranges</h2>
<div class="figure"><span id="fig:GRanges"></span>
<img src="Lawrence_GenomicRanges/granges.pdf" alt="An illustration of genomic ranges. GRanges represents a set genomic ranges in terms of the sequence name (typically the chromosome), start and end coordinates (as an IRanges object), and strand (either positive, negative, or unstranded). GRanges holds information about its universe of sequences (typically a genome) and an arbitrary set of metadata columns with information particular to the dataset." width="\textwidth" />
<p class="caption">
Figure 4.1: An illustration of genomic ranges. GRanges represents a set genomic ranges in terms of the sequence name (typically the chromosome), start and end coordinates (as an IRanges object), and strand (either positive, negative, or unstranded). GRanges holds information about its universe of sequences (typically a genome) and an arbitrary set of metadata columns with information particular to the dataset.
</p>
</div>
<p>The central genomic data structure is the <em>GRanges</em> class, which represents a collection of genomic ranges that each have a single start and end location on the genome. It can be used to store the location of genomic features such as binding sites, read alignments and transcripts.</p>
</div>
<div id="constructing-a-granges-object-from-data.frame" class="section level2">
<h2><span class="header-section-number">4.7</span> Constructing a <em>GRanges</em> object from data.frame</h2>
<p>If we have a data.frame containing scores on a set of genomic ranges, we can call <code>makeGRangesFromDataFrame()</code> to promote the data.frame to a GRanges, thus adding semantics, formal constraints, and range-specific functionality. For example,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressPackageStartupMessages</span>({
 <span class="kw">library</span>(BiocStyle)
 <span class="kw">library</span>(GenomicRanges)
})</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(
    <span class="dt">seqnames =</span> <span class="kw">rep</span>(<span class="kw">c</span>(<span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr2&quot;</span>, <span class="st">&quot;chr1&quot;</span>, <span class="st">&quot;chr3&quot;</span>), <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">4</span>)),
    <span class="dt">start =</span> <span class="kw">c</span>(<span class="dv">101</span>, <span class="dv">105</span>, <span class="dv">125</span>, <span class="dv">132</span>, <span class="dv">134</span>, <span class="dv">152</span>, <span class="dv">153</span>, <span class="dv">160</span>, <span class="dv">166</span>, <span class="dv">170</span>),
    <span class="dt">end =</span> <span class="kw">c</span>(<span class="dv">104</span>, <span class="dv">120</span>, <span class="dv">133</span>, <span class="dv">132</span>, <span class="dv">155</span>, <span class="dv">154</span>, <span class="dv">159</span>, <span class="dv">166</span>, <span class="dv">171</span>, <span class="dv">190</span>),
    <span class="dt">strand =</span> <span class="kw">rep</span>(<span class="kw">strand</span>(<span class="kw">c</span>(<span class="st">&quot;-&quot;</span>, <span class="st">&quot;+&quot;</span>, <span class="st">&quot;*&quot;</span>, <span class="st">&quot;+&quot;</span>, <span class="st">&quot;-&quot;</span>)), <span class="kw">c</span>(<span class="dv">1</span>, <span class="dv">2</span>, <span class="dv">2</span>, <span class="dv">3</span>, <span class="dv">2</span>)),
    <span class="dt">score =</span> <span class="dv">1</span><span class="op">:</span><span class="dv">10</span>,
    <span class="dt">GC =</span> <span class="kw">seq</span>(<span class="dv">1</span>, <span class="dv">0</span>, <span class="dt">length=</span><span class="dv">10</span>),
    <span class="dt">row.names =</span> <span class="kw">head</span>(letters, <span class="dv">10</span>))
gr &lt;-<span class="st"> </span><span class="kw">makeGRangesFromDataFrame</span>(df, <span class="dt">keep.extra.columns=</span><span class="ot">TRUE</span>)</code></pre></div>
<p>creates a <em>GRanges</em> object with 10 genomic ranges. The output of the <em>GRanges</em> <code>show()</code> method separates the information into a left and right hand region that are separated by <code>|</code> symbols. The genomic coordinates (seqnames, ranges, and strand) are located on the left-hand side and the metadata columns (annotation) are located on the right. For this example, the metadata is comprised of <code>&quot;score&quot;</code> and <code>&quot;GC&quot;</code> information, but almost anything can be stored in the metadata portion of a <em>GRanges</em> object.</p>
</div>
<div id="loading-a-granges-object-from-a-standard-file-format" class="section level2">
<h2><span class="header-section-number">4.8</span> Loading a <em>GRanges</em> object from a standard file format</h2>
<p>We often obtain data on genomic ranges from standard track formats, like BED, GFF and BigWig. The rtracklayer package parses those files directly into GRanges objects. The GenomicAlignments package parses BAM files into GAlignments objects, which behave much like GRanges, and it is easy to convert a GAlignments to a GRanges. We will see some examples of loading data from files later in the tutorial.</p>
<p>The <code>seqnames()</code>, <code>ranges()</code>, and <code>strand()</code> accessor functions extract the components of the genomic coordinates,</p>
</div>
<div id="basic-manipulation-of-granges-objects" class="section level2">
<h2><span class="header-section-number">4.9</span> Basic manipulation of <em>GRanges</em> objects</h2>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">seqnames</span>(gr)
<span class="co">#&gt; factor-Rle of length 10 with 4 runs</span>
<span class="co">#&gt;   Lengths:    1    3    2    4</span>
<span class="co">#&gt;   Values : chr1 chr2 chr1 chr3</span>
<span class="co">#&gt; Levels(3): chr1 chr2 chr3</span>
<span class="kw">ranges</span>(gr)
<span class="co">#&gt; IRanges object with 10 ranges and 0 metadata columns:</span>
<span class="co">#&gt;         start       end     width</span>
<span class="co">#&gt;     &lt;integer&gt; &lt;integer&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   a       101       104         4</span>
<span class="co">#&gt;   b       105       120        16</span>
<span class="co">#&gt;   c       125       133         9</span>
<span class="co">#&gt;   d       132       132         1</span>
<span class="co">#&gt;   e       134       155        22</span>
<span class="co">#&gt;   f       152       154         3</span>
<span class="co">#&gt;   g       153       159         7</span>
<span class="co">#&gt;   h       160       166         7</span>
<span class="co">#&gt;   i       166       171         6</span>
<span class="co">#&gt;   j       170       190        21</span>
<span class="kw">strand</span>(gr)
<span class="co">#&gt; factor-Rle of length 10 with 5 runs</span>
<span class="co">#&gt;   Lengths: 1 2 2 3 2</span>
<span class="co">#&gt;   Values : - + * + -</span>
<span class="co">#&gt; Levels(3): + - *</span></code></pre></div>
<p>The <code>granges()</code> function extracts genomic ranges without corresponding metadata,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">granges</span>(gr)
<span class="co">#&gt; GRanges object with 10 ranges and 0 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   a     chr1   101-104      -</span>
<span class="co">#&gt;   b     chr2   105-120      +</span>
<span class="co">#&gt;   c     chr2   125-133      +</span>
<span class="co">#&gt;   d     chr2       132      *</span>
<span class="co">#&gt;   e     chr1   134-155      *</span>
<span class="co">#&gt;   f     chr1   152-154      +</span>
<span class="co">#&gt;   g     chr3   153-159      +</span>
<span class="co">#&gt;   h     chr3   160-166      +</span>
<span class="co">#&gt;   i     chr3   166-171      -</span>
<span class="co">#&gt;   j     chr3   170-190      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>The <code>start()</code>, <code>end()</code>, <code>width()</code>, and <code>range</code> functions extract basic interval characteristics,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">start</span>(gr)
<span class="co">#&gt;  [1] 101 105 125 132 134 152 153 160 166 170</span>
<span class="kw">end</span>(gr)
<span class="co">#&gt;  [1] 104 120 133 132 155 154 159 166 171 190</span>
<span class="kw">width</span>(gr)
<span class="co">#&gt;  [1]  4 16  9  1 22  3  7  7  6 21</span></code></pre></div>
<p>The <code>mcols()</code> accessor extracts the metadata as a <em>DataFrame</em>,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">mcols</span>(gr)
<span class="co">#&gt; DataFrame with 10 rows and 2 columns</span>
<span class="co">#&gt;       score                GC</span>
<span class="co">#&gt;   &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt; a         1                 1</span>
<span class="co">#&gt; b         2 0.888888888888889</span>
<span class="co">#&gt; c         3 0.777777777777778</span>
<span class="co">#&gt; d         4 0.666666666666667</span>
<span class="co">#&gt; e         5 0.555555555555556</span>
<span class="co">#&gt; f         6 0.444444444444444</span>
<span class="co">#&gt; g         7 0.333333333333333</span>
<span class="co">#&gt; h         8 0.222222222222222</span>
<span class="co">#&gt; i         9 0.111111111111111</span>
<span class="co">#&gt; j        10                 0</span>
<span class="kw">mcols</span>(gr)<span class="op">$</span>score
<span class="co">#&gt;  [1]  1  2  3  4  5  6  7  8  9 10</span>
<span class="kw">score</span>(gr)
<span class="co">#&gt;  [1]  1  2  3  4  5  6  7  8  9 10</span></code></pre></div>
<p>The lengths and other properties of the sequences containing the ranges can (and should) be stored in the <em>GRanges</em> object. Formal tracking of the sequence universe, typically the genome build, ensures data integrity and prevents accidental mixing of ranges from incompatible contexts. Assuming these data are of <em>Homo sapiens</em>, we could add the sequence information like this:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">seqinfo</span>(gr) &lt;-<span class="st"> </span><span class="kw">Seqinfo</span>(<span class="dt">genome=</span><span class="st">&quot;hg38&quot;</span>)</code></pre></div>
<p>The <code>Seqinfo()</code> function automatically loads the sequence information for the specified <code>genome=</code> by querying the UCSC database.</p>
<p>And then retrieves as:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">seqinfo</span>(gr)
<span class="co">#&gt; Seqinfo object with 455 sequences (1 circular) from hg38 genome:</span>
<span class="co">#&gt;   seqnames         seqlengths isCircular genome</span>
<span class="co">#&gt;   chr1              248956422      FALSE   hg38</span>
<span class="co">#&gt;   chr2              242193529      FALSE   hg38</span>
<span class="co">#&gt;   chr3              198295559      FALSE   hg38</span>
<span class="co">#&gt;   chr4              190214555      FALSE   hg38</span>
<span class="co">#&gt;   chr5              181538259      FALSE   hg38</span>
<span class="co">#&gt;   ...                     ...        ...    ...</span>
<span class="co">#&gt;   chrUn_KI270753v1      62944      FALSE   hg38</span>
<span class="co">#&gt;   chrUn_KI270754v1      40191      FALSE   hg38</span>
<span class="co">#&gt;   chrUn_KI270755v1      36723      FALSE   hg38</span>
<span class="co">#&gt;   chrUn_KI270756v1      79590      FALSE   hg38</span>
<span class="co">#&gt;   chrUn_KI270757v1      71251      FALSE   hg38</span></code></pre></div>
<p>Methods for accessing the <code>length</code> and <code>names</code> have also been defined.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">names</span>(gr)
<span class="co">#&gt;  [1] &quot;a&quot; &quot;b&quot; &quot;c&quot; &quot;d&quot; &quot;e&quot; &quot;f&quot; &quot;g&quot; &quot;h&quot; &quot;i&quot; &quot;j&quot;</span>
<span class="kw">length</span>(gr)
<span class="co">#&gt; [1] 10</span></code></pre></div>
</div>
<div id="subsetting-granges-objects" class="section level2">
<h2><span class="header-section-number">4.10</span> Subsetting <em>GRanges</em> objects</h2>
<p><em>GRanges</em> objects act like vectors of ranges, with the expected vector-like subsetting operations available</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gr[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>]
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   125-133      + |         3 0.777777777777778</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>A second argument to the <code>[</code> subset operator specifies which metadata columns to extract from the <em>GRanges</em> object. For example,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">gr[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>, <span class="st">&quot;GC&quot;</span>]
<span class="co">#&gt; GRanges object with 2 ranges and 1 metadata column:</span>
<span class="co">#&gt;     seqnames    ranges strand |                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |         &lt;numeric&gt;</span>
<span class="co">#&gt;   b     chr2   105-120      + | 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   125-133      + | 0.777777777777778</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>The <code>subset()</code> function provides an easy way to subset based on attributes of the ranges and columns in the metadata. For example,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">subset</span>(gr, strand <span class="op">==</span><span class="st"> &quot;+&quot;</span> <span class="op">&amp;</span><span class="st"> </span>score <span class="op">&gt;</span><span class="st"> </span><span class="dv">5</span>, <span class="dt">select =</span> score)
<span class="co">#&gt; GRanges object with 3 ranges and 1 metadata column:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="co">#&gt;   f     chr1   152-154      + |         6</span>
<span class="co">#&gt;   g     chr3   153-159      + |         7</span>
<span class="co">#&gt;   h     chr3   160-166      + |         8</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>Elements can also be assigned to the <em>GRanges</em> object. This example replaces the the second row of a <em>GRanges</em> object with the first row of <code>gr</code>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">grMod &lt;-<span class="st"> </span>gr
grMod[<span class="dv">2</span>] &lt;-<span class="st"> </span>gr[<span class="dv">1</span>]
<span class="kw">head</span>(grMod, <span class="dt">n=</span><span class="dv">3</span>)
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   101-104      - |         1                 1</span>
<span class="co">#&gt;   b     chr1   101-104      - |         1                 1</span>
<span class="co">#&gt;   c     chr2   125-133      + |         3 0.777777777777778</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>There are methods to repeat, reverse, or select specific portions of <em>GRanges</em> objects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">rep</span>(gr[<span class="dv">2</span>], <span class="dt">times =</span> <span class="dv">3</span>)
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span>
<span class="kw">rev</span>(gr)
<span class="co">#&gt; GRanges object with 10 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   j     chr3   170-190      - |        10                 0</span>
<span class="co">#&gt;   i     chr3   166-171      - |         9 0.111111111111111</span>
<span class="co">#&gt;   h     chr3   160-166      + |         8 0.222222222222222</span>
<span class="co">#&gt;   g     chr3   153-159      + |         7 0.333333333333333</span>
<span class="co">#&gt;   f     chr1   152-154      + |         6 0.444444444444444</span>
<span class="co">#&gt;   e     chr1   134-155      * |         5 0.555555555555556</span>
<span class="co">#&gt;   d     chr2       132      * |         4 0.666666666666667</span>
<span class="co">#&gt;   c     chr2   125-133      + |         3 0.777777777777778</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889</span>
<span class="co">#&gt;   a     chr1   101-104      - |         1                 1</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span>
<span class="kw">head</span>(gr,<span class="dt">n=</span><span class="dv">2</span>)
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   101-104      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span>
<span class="kw">tail</span>(gr,<span class="dt">n=</span><span class="dv">2</span>)
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   i     chr3   166-171      - |         9 0.111111111111111</span>
<span class="co">#&gt;   j     chr3   170-190      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span>
<span class="kw">window</span>(gr, <span class="dt">start=</span><span class="dv">2</span>,<span class="dt">end=</span><span class="dv">4</span>)
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   125-133      + |         3 0.777777777777778</span>
<span class="co">#&gt;   d     chr2       132      * |         4 0.666666666666667</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span>
gr[<span class="kw">IRanges</span>(<span class="dt">start=</span><span class="kw">c</span>(<span class="dv">2</span>,<span class="dv">7</span>), <span class="dt">end=</span><span class="kw">c</span>(<span class="dv">3</span>,<span class="dv">9</span>))]
<span class="co">#&gt; GRanges object with 5 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   125-133      + |         3 0.777777777777778</span>
<span class="co">#&gt;   g     chr3   153-159      + |         7 0.333333333333333</span>
<span class="co">#&gt;   h     chr3   160-166      + |         8 0.222222222222222</span>
<span class="co">#&gt;   i     chr3   166-171      - |         9 0.111111111111111</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
</div>
<div id="splitting-and-combining-granges-objects" class="section level2">
<h2><span class="header-section-number">4.11</span> Splitting and combining <em>GRanges</em> objects</h2>
<p>THe <code>split()</code> function divides a <em>GRanges</em> into groups, returning a <em>GRangesList</em>, a class that we will describe and demonstrate later.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">sp &lt;-<span class="st"> </span><span class="kw">split</span>(gr, <span class="kw">rep</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>, <span class="dt">each=</span><span class="dv">5</span>))
sp
<span class="co">#&gt; GRangesList object of length 2:</span>
<span class="co">#&gt; $1 </span>
<span class="co">#&gt; GRanges object with 5 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   101-104      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   125-133      + |         3 0.777777777777778</span>
<span class="co">#&gt;   d     chr2       132      * |         4 0.666666666666667</span>
<span class="co">#&gt;   e     chr1   134-155      * |         5 0.555555555555556</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $2 </span>
<span class="co">#&gt; GRanges object with 5 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames  ranges strand | score                GC</span>
<span class="co">#&gt;   f     chr1 152-154      + |     6 0.444444444444444</span>
<span class="co">#&gt;   g     chr3 153-159      + |     7 0.333333333333333</span>
<span class="co">#&gt;   h     chr3 160-166      + |     8 0.222222222222222</span>
<span class="co">#&gt;   i     chr3 166-171      - |     9 0.111111111111111</span>
<span class="co">#&gt;   j     chr3 170-190      - |    10                 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>We can split the ranges by metadata columns, like strand,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">split</span>(gr, <span class="op">~</span><span class="st"> </span>strand)
<span class="co">#&gt; GRangesList object of length 3:</span>
<span class="co">#&gt; $+ </span>
<span class="co">#&gt; GRanges object with 5 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   125-133      + |         3 0.777777777777778</span>
<span class="co">#&gt;   f     chr1   152-154      + |         6 0.444444444444444</span>
<span class="co">#&gt;   g     chr3   153-159      + |         7 0.333333333333333</span>
<span class="co">#&gt;   h     chr3   160-166      + |         8 0.222222222222222</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $- </span>
<span class="co">#&gt; GRanges object with 3 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames  ranges strand | score                GC</span>
<span class="co">#&gt;   a     chr1 101-104      - |     1                 1</span>
<span class="co">#&gt;   i     chr3 166-171      - |     9 0.111111111111111</span>
<span class="co">#&gt;   j     chr3 170-190      - |    10                 0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $* </span>
<span class="co">#&gt; GRanges object with 2 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames  ranges strand | score                GC</span>
<span class="co">#&gt;   d     chr2     132      * |     4 0.666666666666667</span>
<span class="co">#&gt;   e     chr1 134-155      * |     5 0.555555555555556</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>The <code>c()</code> and <code>append()</code> functions combine two (or more in the case of <code>c()</code>) <em>GRanges</em> objects.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">c</span>(sp[[<span class="dv">1</span>]], sp[[<span class="dv">2</span>]])
<span class="co">#&gt; GRanges object with 10 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   101-104      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   125-133      + |         3 0.777777777777778</span>
<span class="co">#&gt;   d     chr2       132      * |         4 0.666666666666667</span>
<span class="co">#&gt;   e     chr1   134-155      * |         5 0.555555555555556</span>
<span class="co">#&gt;   f     chr1   152-154      + |         6 0.444444444444444</span>
<span class="co">#&gt;   g     chr3   153-159      + |         7 0.333333333333333</span>
<span class="co">#&gt;   h     chr3   160-166      + |         8 0.222222222222222</span>
<span class="co">#&gt;   i     chr3   166-171      - |         9 0.111111111111111</span>
<span class="co">#&gt;   j     chr3   170-190      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>The <code>stack()</code> function stacks the elements of a <em>GRangesList</em> into a single <em>GRanges</em> and adds a column indicating the origin of each element,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">stack</span>(sp, <span class="dt">index.var=</span><span class="st">&quot;group&quot;</span>)
<span class="co">#&gt; GRanges object with 10 ranges and 3 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand | group     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;Rle&gt; &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   101-104      - |     1         1                 1</span>
<span class="co">#&gt;   b     chr2   105-120      + |     1         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   125-133      + |     1         3 0.777777777777778</span>
<span class="co">#&gt;   d     chr2       132      * |     1         4 0.666666666666667</span>
<span class="co">#&gt;   e     chr1   134-155      * |     1         5 0.555555555555556</span>
<span class="co">#&gt;   f     chr1   152-154      + |     2         6 0.444444444444444</span>
<span class="co">#&gt;   g     chr3   153-159      + |     2         7 0.333333333333333</span>
<span class="co">#&gt;   h     chr3   160-166      + |     2         8 0.222222222222222</span>
<span class="co">#&gt;   i     chr3   166-171      - |     2         9 0.111111111111111</span>
<span class="co">#&gt;   j     chr3   170-190      - |     2        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
</div>
<div id="aggregating-granges-objects" class="section level2">
<h2><span class="header-section-number">4.12</span> Aggregating <em>GRanges</em> objects</h2>
<p>Like other tabular data structures, we can aggregate <em>GRanges</em> objects, for example,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">aggregate</span>(gr, score <span class="op">~</span><span class="st"> </span>strand, mean)
<span class="co">#&gt; DataFrame with 3 rows and 2 columns</span>
<span class="co">#&gt;     strand            score</span>
<span class="co">#&gt;   &lt;factor&gt;        &lt;numeric&gt;</span>
<span class="co">#&gt; 1        +              5.2</span>
<span class="co">#&gt; 2        - 6.66666666666667</span>
<span class="co">#&gt; 3        *              4.5</span></code></pre></div>
<p>The <code>aggregate()</code> function also supports a syntax similar to <code>summarize()</code> from dplyr,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">aggregate</span>(gr, <span class="op">~</span><span class="st"> </span>strand, <span class="dt">n_score =</span> <span class="kw">lengths</span>(score), <span class="dt">mean_score =</span> <span class="kw">mean</span>(score))
<span class="co">#&gt; DataFrame with 3 rows and 4 columns</span>
<span class="co">#&gt;              grouping   strand   n_score       mean_score</span>
<span class="co">#&gt;   &lt;ManyToOneGrouping&gt; &lt;factor&gt; &lt;integer&gt;        &lt;numeric&gt;</span>
<span class="co">#&gt; 1           2,3,6,...        +         5              5.2</span>
<span class="co">#&gt; 2              1,9,10        -         3 6.66666666666667</span>
<span class="co">#&gt; 3                 4,5        *         2              4.5</span></code></pre></div>
<p>Note that we need to call <code>lengths(score)</code> instead of <code>length(score)</code> because <code>score</code> is actually a list-like object in the aggregation expression.</p>
</div>
<div id="basic-interval-operations-for-granges-objects" class="section level2">
<h2><span class="header-section-number">4.13</span> Basic interval operations for <em>GRanges</em> objects</h2>
<p>There are many functions for manipulating <em>GRanges</em> objects. The functions can be classified as <em>intra-range functions</em>, <em>inter-range functions</em>, and <em>between-range functions</em>.</p>
<p><em>Intra-range functions</em> operate on each element of a <em>GRanges</em> object independent of the other ranges in the object. For example, the <code>flank</code> function can be used to recover regions flanking the set of ranges represented by the <em>GRanges</em> object. So to get a <em>GRanges</em> object containing the ranges that include the 10 bases upstream according to the direction of “transcription” (indicated by the strand):</p>
<p><code>r  g &lt;- gr[1:3]  g &lt;- append(g, gr[10])  flank(g, 10)  #&gt; GRanges object with 4 ranges and 2 metadata columns:  #&gt;     seqnames    ranges strand |     score                GC  #&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;  #&gt;   a     chr1   105-114      - |         1                 1  #&gt;   b     chr2    95-104      + |         2 0.888888888888889  #&gt;   c     chr2   115-124      + |         3 0.777777777777778  #&gt;   j     chr3   191-200      - |        10                 0  #&gt;   -------  #&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</code></p>
<p>And to include the downstream bases:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">flank</span>(g, <span class="dv">10</span>, <span class="dt">start=</span><span class="ot">FALSE</span>)
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1    91-100      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   121-130      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   134-143      + |         3 0.777777777777778</span>
<span class="co">#&gt;   j     chr3   160-169      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>A common use case for <code>flank()</code> is generating promoter regions based on the transcript ranges. There is a convenience function that by default generates a region starting 2000bp upstream and 200bp downstream of the TSS,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">promoters</span>(g)
<span class="co">#&gt; Warning in valid.GenomicRanges.seqinfo(x, suggest.trim = TRUE): GRanges object contains 4 out-of-bound ranges located on sequences</span>
<span class="co">#&gt;   chr1, chr2, and chr3. Note that ranges located on a sequence whose</span>
<span class="co">#&gt;   length is unknown (NA) or on a circular sequence are not</span>
<span class="co">#&gt;   considered out-of-bound (use seqlengths() and isCircular() to get</span>
<span class="co">#&gt;   the lengths and circularity flags of the underlying sequences).</span>
<span class="co">#&gt;   You can use trim() to trim these ranges. See</span>
<span class="co">#&gt;   ?`trim,GenomicRanges-method` for more information.</span>
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1  -95-2104      - |         1                 1</span>
<span class="co">#&gt;   b     chr2 -1895-304      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2 -1875-324      + |         3 0.777777777777778</span>
<span class="co">#&gt;   j     chr3   -9-2190      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>To ignore strand/transcription and assume the orientation of left to right use <code>unstrand()</code>,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">flank</span>(<span class="kw">unstrand</span>(g), <span class="dv">10</span>)
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1    91-100      * |         1                 1</span>
<span class="co">#&gt;   b     chr2    95-104      * |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   115-124      * |         3 0.777777777777778</span>
<span class="co">#&gt;   j     chr3   160-169      * |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>Other examples of intra-range functions include <code>resize()</code> and <code>shift()</code>. The <code>shift()</code> function will move the ranges by a specific number of base pairs, and the <code>resize()</code> function will set a specific width, by default fixing the “transcription” start (or just the start when strand is “*“). The <code>fix=</code> argument controls whether the”start“,”end&quot; or “center” is held constant.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">shift</span>(g, <span class="dv">5</span>)
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1   106-109      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   110-125      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   130-138      + |         3 0.777777777777778</span>
<span class="co">#&gt;   j     chr3   175-195      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span>
<span class="kw">resize</span>(g, <span class="dv">30</span>)
<span class="co">#&gt; GRanges object with 4 ranges and 2 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt;</span>
<span class="co">#&gt;   a     chr1    75-104      - |         1                 1</span>
<span class="co">#&gt;   b     chr2   105-134      + |         2 0.888888888888889</span>
<span class="co">#&gt;   c     chr2   125-154      + |         3 0.777777777777778</span>
<span class="co">#&gt;   j     chr3   161-190      - |        10                 0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>The <em><a href="http://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></em> help page <code>?&quot;intra-range-methods&quot;</code> summarizes these methods.</p>
<p><em>Inter-range functions</em> involve comparisons between ranges in a single <em>GRanges</em> object and typically aggregate ranges. For instance, the <code>reduce()</code> function will merge overlapping and adjacent ranges to produce a minimal set of ranges representing the regions covered by the original set.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">reduce</span>(gr)
<span class="co">#&gt; GRanges object with 8 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1   152-154      +</span>
<span class="co">#&gt;   [2]     chr1   101-104      -</span>
<span class="co">#&gt;   [3]     chr1   134-155      *</span>
<span class="co">#&gt;   [4]     chr2   105-120      +</span>
<span class="co">#&gt;   [5]     chr2   125-133      +</span>
<span class="co">#&gt;   [6]     chr2       132      *</span>
<span class="co">#&gt;   [7]     chr3   153-166      +</span>
<span class="co">#&gt;   [8]     chr3   166-190      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span>
<span class="kw">reduce</span>(gr, <span class="dt">ignore.strand=</span><span class="ot">TRUE</span>)
<span class="co">#&gt; GRanges object with 5 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1   101-104      *</span>
<span class="co">#&gt;   [2]     chr1   134-155      *</span>
<span class="co">#&gt;   [3]     chr2   105-120      *</span>
<span class="co">#&gt;   [4]     chr2   125-133      *</span>
<span class="co">#&gt;   [5]     chr3   153-190      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>Rarely, it useful to complement the (reduced) ranges. Note that the universe is taken as the entire sequence span in all three strands (+, -, *), which is often surprising when working with unstranded ranges.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">gaps</span>(g)
<span class="co">#&gt; GRanges object with 1369 ranges and 0 metadata columns:</span>
<span class="co">#&gt;                  seqnames        ranges strand</span>
<span class="co">#&gt;                     &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;      [1]             chr1   1-248956422      +</span>
<span class="co">#&gt;      [2]             chr1         1-100      -</span>
<span class="co">#&gt;      [3]             chr1 105-248956422      -</span>
<span class="co">#&gt;      [4]             chr1   1-248956422      *</span>
<span class="co">#&gt;      [5]             chr2         1-104      +</span>
<span class="co">#&gt;      ...              ...           ...    ...</span>
<span class="co">#&gt;   [1365] chrUn_KI270756v1       1-79590      -</span>
<span class="co">#&gt;   [1366] chrUn_KI270756v1       1-79590      *</span>
<span class="co">#&gt;   [1367] chrUn_KI270757v1       1-71251      +</span>
<span class="co">#&gt;   [1368] chrUn_KI270757v1       1-71251      -</span>
<span class="co">#&gt;   [1369] chrUn_KI270757v1       1-71251      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>The <code>disjoin</code> function breaks up the ranges so that they do not overlap but still cover the same regions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">disjoin</span>(g)
<span class="co">#&gt; GRanges object with 4 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1   101-104      -</span>
<span class="co">#&gt;   [2]     chr2   105-120      +</span>
<span class="co">#&gt;   [3]     chr2   125-133      +</span>
<span class="co">#&gt;   [4]     chr3   170-190      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>The <code>coverage</code> function counts how many ranges overlap each position in the sequence universe of a <em>GRanges</em> object.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cov &lt;-<span class="st"> </span><span class="kw">coverage</span>(g)
cov[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>]
<span class="co">#&gt; RleList of length 3</span>
<span class="co">#&gt; $chr1</span>
<span class="co">#&gt; integer-Rle of length 248956422 with 3 runs</span>
<span class="co">#&gt;   Lengths:       100         4 248956318</span>
<span class="co">#&gt;   Values :         0         1         0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $chr2</span>
<span class="co">#&gt; integer-Rle of length 242193529 with 5 runs</span>
<span class="co">#&gt;   Lengths:       104        16         4         9 242193396</span>
<span class="co">#&gt;   Values :         0         1         0         1         0</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $chr3</span>
<span class="co">#&gt; integer-Rle of length 198295559 with 3 runs</span>
<span class="co">#&gt;   Lengths:       169        21 198295369</span>
<span class="co">#&gt;   Values :         0         1         0</span></code></pre></div>
<p>The coverage is stored compactly as an <em>RleList</em>, with one <em>Rle</em> vector per sequence. We can convert it to a <em>GRanges</em>,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cov_gr &lt;-<span class="st"> </span><span class="kw">GRanges</span>(cov)
cov_gr
<span class="co">#&gt; GRanges object with 463 ranges and 1 metadata column:</span>
<span class="co">#&gt;                 seqnames        ranges strand |     score</span>
<span class="co">#&gt;                    &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;</span>
<span class="co">#&gt;     [1]             chr1         1-100      * |         0</span>
<span class="co">#&gt;     [2]             chr1       101-104      * |         1</span>
<span class="co">#&gt;     [3]             chr1 105-248956422      * |         0</span>
<span class="co">#&gt;     [4]             chr2         1-104      * |         0</span>
<span class="co">#&gt;     [5]             chr2       105-120      * |         1</span>
<span class="co">#&gt;     ...              ...           ...    ... .       ...</span>
<span class="co">#&gt;   [459] chrUn_KI270753v1       1-62944      * |         0</span>
<span class="co">#&gt;   [460] chrUn_KI270754v1       1-40191      * |         0</span>
<span class="co">#&gt;   [461] chrUn_KI270755v1       1-36723      * |         0</span>
<span class="co">#&gt;   [462] chrUn_KI270756v1       1-79590      * |         0</span>
<span class="co">#&gt;   [463] chrUn_KI270757v1       1-71251      * |         0</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences from an unspecified genome</span></code></pre></div>
<p>and even convert the <em>GRanges</em> form back to an <em>RleList</em> by computing a weighted coverage,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cov &lt;-<span class="st"> </span><span class="kw">coverage</span>(cov_gr, <span class="dt">weight=</span><span class="st">&quot;score&quot;</span>)</code></pre></div>
<p>The <em>GRanges</em> derivative <em>GPos</em>, a compact representation of width 1 ranges, is useful for representing coverage, although it cannot yet represent the coverage for the entire human genome (or any genome with over ~ 2 billion bp).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">GPos</span>(cov[<span class="dv">1</span><span class="op">:</span><span class="dv">3</span>])
<span class="co">#&gt; GPos object with 689445510 positions and 0 metadata columns:</span>
<span class="co">#&gt;               seqnames       pos strand</span>
<span class="co">#&gt;                  &lt;Rle&gt; &lt;integer&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;           [1]     chr1         1      *</span>
<span class="co">#&gt;           [2]     chr1         2      *</span>
<span class="co">#&gt;           [3]     chr1         3      *</span>
<span class="co">#&gt;           [4]     chr1         4      *</span>
<span class="co">#&gt;           [5]     chr1         5      *</span>
<span class="co">#&gt;           ...      ...       ...    ...</span>
<span class="co">#&gt;   [689445506]     chr3 198295555      *</span>
<span class="co">#&gt;   [689445507]     chr3 198295556      *</span>
<span class="co">#&gt;   [689445508]     chr3 198295557      *</span>
<span class="co">#&gt;   [689445509]     chr3 198295558      *</span>
<span class="co">#&gt;   [689445510]     chr3 198295559      *</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome</span></code></pre></div>
<p>These inter-range functions all generate entirely new sets of ranges. The return value is left unannotated, since there is no obvious way to carry the metadata across the operation. The user is left to map the metadata to the new ranges. Functions like <code>reduce()</code> and <code>disjoin()</code> facilitate this by optionally including in the returned metadata a one-to-many reverse mapping from the aggregate ranges to input ranges. For example, to average the score over a reduction,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">rg &lt;-<span class="st"> </span><span class="kw">reduce</span>(gr, <span class="dt">with.revmap=</span><span class="ot">TRUE</span>)
rg<span class="op">$</span>score &lt;-<span class="st"> </span><span class="kw">mean</span>(<span class="kw">extractList</span>(gr<span class="op">$</span>score, rg<span class="op">$</span>revmap))</code></pre></div>
<p>See the <em><a href="http://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></em> help page <code>?&quot;inter-range-methods&quot;</code> for additional help.</p>
</div>
<div id="interval-set-operations-for-granges-objects" class="section level2">
<h2><span class="header-section-number">4.14</span> Interval set operations for <em>GRanges</em> objects</h2>
<p><em>Between-range functions</em> calculate relationships between different <em>GRanges</em> objects. Of central importance are <code>findOverlaps</code> and related operations; these are discussed below. Additional operations treat <em>GRanges</em> as mathematical sets of coordinates; <code>union(g, g2)</code> is the union of the coordinates in <code>g</code> and <code>g2</code>. Here are examples for calculating the <code>union</code>, the <code>intersect</code> and the asymmetric difference (using <code>setdiff</code>).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">g2 &lt;-<span class="st"> </span><span class="kw">head</span>(gr, <span class="dt">n=</span><span class="dv">2</span>)
<span class="kw">union</span>(g, g2)
<span class="co">#&gt; GRanges object with 4 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1   101-104      -</span>
<span class="co">#&gt;   [2]     chr2   105-120      +</span>
<span class="co">#&gt;   [3]     chr2   125-133      +</span>
<span class="co">#&gt;   [4]     chr3   170-190      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span>
<span class="kw">intersect</span>(g, g2)
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr1   101-104      -</span>
<span class="co">#&gt;   [2]     chr2   105-120      +</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span>
<span class="kw">setdiff</span>(g, g2)
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames    ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]     chr2   125-133      +</span>
<span class="co">#&gt;   [2]     chr3   170-190      -</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>Related functions are available when the structure of the <em>GRanges</em> objects are ‘parallel’ to one another, i.e., element 1 of object 1 is related to element 1 of object 2, and so on. These operations all begin with a <code>p</code>, which is short for parallel. The functions then perform element-wise, e.g., the union of element 1 of object 1 with element 1 of object 2, etc. A requirement for these operations is that the number of elements in each <em>GRanges</em> object is the same, and that both of the objects have the same seqnames and strand assignments throughout.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">g3 &lt;-<span class="st"> </span>g[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]
<span class="kw">ranges</span>(g3[<span class="dv">1</span>]) &lt;-<span class="st"> </span><span class="kw">IRanges</span>(<span class="dt">start=</span><span class="dv">105</span>, <span class="dt">end=</span><span class="dv">112</span>)
<span class="kw">punion</span>(g2, g3)
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   a     chr1   101-112      -</span>
<span class="co">#&gt;   b     chr2   105-120      +</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span>
<span class="kw">pintersect</span>(g2, g3)
<span class="co">#&gt; GRanges object with 2 ranges and 3 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand |     score                GC       hit</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;integer&gt;         &lt;numeric&gt; &lt;logical&gt;</span>
<span class="co">#&gt;   a     chr1   105-104      - |         1                 1      TRUE</span>
<span class="co">#&gt;   b     chr2   105-120      + |         2 0.888888888888889      TRUE</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span>
<span class="kw">psetdiff</span>(g2, g3)
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;     seqnames    ranges strand</span>
<span class="co">#&gt;        &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   a     chr1   101-104      -</span>
<span class="co">#&gt;   b     chr2   105-104      +</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 455 sequences (1 circular) from hg38 genome</span></code></pre></div>
<p>For more information on the <code>GRanges</code> classes be sure to consult the manual page.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">?GRanges</code></pre></div>
<p>A relatively comprehensive list of available functions is discovered with</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">methods</span>(<span class="dt">class=</span><span class="st">&quot;GRanges&quot;</span>)</code></pre></div>
</div>
<div id="finding-overlaps-between-granges-objects" class="section level2">
<h2><span class="header-section-number">4.15</span> Finding overlaps between <em>GRanges</em> objects</h2>
<p>Interval overlapping is the process of comparing the ranges in two objects to determine if and when they overlap. As such, it is perhaps the most common operation performed on <em>GRanges</em> objects. To this end, the <em><a href="http://bioconductor.org/packages/GenomicRanges">GenomicRanges</a></em> package provides a family of interval overlap functions. The most general of these functions is <code>findOverlaps()</code>, which takes a query and a subject as inputs and returns a <em>Hits</em> object containing the index pairings for the overlapping elements.</p>
<p>Let us assume that we have three random data.frame objects, each with annoyingly differing ways of naming the columns defining the ranges,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set.seed</span>(<span class="dv">66</span><span class="op">+</span><span class="dv">105</span><span class="op">+</span><span class="dv">111</span><span class="op">+</span><span class="dv">99</span><span class="op">+</span><span class="dv">49</span><span class="op">+</span><span class="dv">56</span>)

pos &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">200</span>, <span class="dt">size =</span> 30L)
size &lt;-<span class="st"> </span>10L
end &lt;-<span class="st"> </span>size <span class="op">+</span><span class="st"> </span>pos <span class="op">-</span><span class="st"> </span>1L
chrom &lt;-<span class="st"> </span><span class="kw">sample</span>(<span class="kw">paste0</span>(<span class="st">&quot;chr&quot;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>), <span class="dt">size =</span> 30L, <span class="dt">replace =</span> <span class="ot">TRUE</span>)
query_df &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">chrom =</span> chrom, 
                       <span class="dt">start =</span> pos,
                       <span class="dt">end =</span> end)
query_dfs &lt;-<span class="st"> </span><span class="kw">split</span>(query_df, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>)
q1 &lt;-<span class="st"> </span><span class="kw">rename</span>(query_dfs[[1L]], <span class="dt">start =</span> <span class="st">&quot;pos&quot;</span>)
q2 &lt;-<span class="st"> </span><span class="kw">rename</span>(query_dfs[[2L]], <span class="dt">chrom =</span> <span class="st">&quot;ch&quot;</span>, <span class="dt">start =</span> <span class="st">&quot;st&quot;</span>)
q3 &lt;-<span class="st"> </span><span class="kw">rename</span>(query_dfs[[3L]], <span class="dt">end =</span> <span class="st">&quot;last&quot;</span>)</code></pre></div>
<p>The <code>makeGRangesFromDataFrame()</code> function can guess some of these, but not all of them, so we help it out,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">q1 &lt;-<span class="st"> </span><span class="kw">makeGRangesFromDataFrame</span>(q1, <span class="dt">start.field =</span> <span class="st">&quot;pos&quot;</span>)
q2 &lt;-<span class="st"> </span><span class="kw">makeGRangesFromDataFrame</span>(q2, <span class="dt">seqnames.field =</span> <span class="st">&quot;ch&quot;</span>,
                                 <span class="dt">start.field =</span> <span class="st">&quot;st&quot;</span>)
q3 &lt;-<span class="st"> </span><span class="kw">makeGRangesFromDataFrame</span>(q3, <span class="dt">end.field =</span> <span class="st">&quot;last&quot;</span>)
query &lt;-<span class="st"> </span><span class="kw">mstack</span>(q1, q2, q3, <span class="dt">.index.var=</span><span class="st">&quot;replicate&quot;</span>)
<span class="kw">sort</span>(query, <span class="dt">by =</span> <span class="op">~</span><span class="st"> </span>start)
<span class="co">#&gt; GRanges object with 30 ranges and 1 metadata column:</span>
<span class="co">#&gt;      seqnames    ranges strand | replicate</span>
<span class="co">#&gt;         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;Rle&gt;</span>
<span class="co">#&gt;   25     chr1     11-20      * |         1</span>
<span class="co">#&gt;   22     chr1     16-25      * |         1</span>
<span class="co">#&gt;    2     chr2     21-30      * |         2</span>
<span class="co">#&gt;   30     chr3     50-59      * |         3</span>
<span class="co">#&gt;    6     chr2     51-60      * |         3</span>
<span class="co">#&gt;   ..      ...       ...    ... .       ...</span>
<span class="co">#&gt;   26     chr3   160-169      * |         2</span>
<span class="co">#&gt;   13     chr1   169-178      * |         1</span>
<span class="co">#&gt;   29     chr2   183-192      * |         2</span>
<span class="co">#&gt;   27     chr3   190-199      * |         3</span>
<span class="co">#&gt;   24     chr3   197-206      * |         3</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>Above, we use the convenient <code>mstack()</code> function, which stacks its arguments, populating the <code>.index.var=</code> column with the origin of each range (using the argument names or positions).</p>
<p>Perhaps the simplest overlap-based operation is <code>subsetByOverlaps()</code>, which extracts the elements in the query (the first argument) that overlap at least one element in the subject (the second).</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">subject &lt;-<span class="st"> </span>gr
<span class="kw">subsetByOverlaps</span>(query, subject, <span class="dt">ignore.strand=</span><span class="ot">TRUE</span>)
<span class="co">#&gt; GRanges object with 9 ranges and 1 metadata column:</span>
<span class="co">#&gt;      seqnames    ranges strand | replicate</span>
<span class="co">#&gt;         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;Rle&gt;</span>
<span class="co">#&gt;   10     chr2   120-129      * |         1</span>
<span class="co">#&gt;   28     chr1   151-160      * |         1</span>
<span class="co">#&gt;    5     chr2   128-137      * |         2</span>
<span class="co">#&gt;   14     chr1   149-158      * |         2</span>
<span class="co">#&gt;   23     chr3   153-162      * |         2</span>
<span class="co">#&gt;   26     chr3   160-169      * |         2</span>
<span class="co">#&gt;    9     chr1    92-101      * |         3</span>
<span class="co">#&gt;   21     chr1   148-157      * |         3</span>
<span class="co">#&gt;   27     chr3   190-199      * |         3</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>In every call to an overlap operation, it is necessary to specify <code>ignore.strand=TRUE</code>, except in rare cases when we do not want ranges on opposite strands to be considered overlapping.</p>
<p>To generally compute on the overlaps, we call <code>findOverlaps()</code> to return a <code>Hits</code> object, which is essentially a bipartite graph matching query ranges to overlapping subject ranges.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hits &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(query, subject, <span class="dt">ignore.strand=</span><span class="ot">TRUE</span>)</code></pre></div>
<p>We typically use the hits to perform one of two operations: join and aggregate. For example, we could inner join the scores from the subject using the query and subject indexes,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">joined &lt;-<span class="st"> </span>query[<span class="kw">queryHits</span>(hits)]
joined<span class="op">$</span>score &lt;-<span class="st"> </span>subject<span class="op">$</span>score[<span class="kw">subjectHits</span>(hits)]</code></pre></div>
<p>The above carries over a single metadata column from the subject. Similar code would carry over other columns and even the ranges themselves.</p>
<p>Sometimes, we want to merge the matched query and subject ranges, typically by finding their intersection,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">ranges</span>(joined) &lt;-<span class="st"> </span><span class="kw">ranges</span>(<span class="kw">pintersect</span>(joined, subject[<span class="kw">subjectHits</span>(hits)]))</code></pre></div>
<p>The typical aggregation is counting the number of hits overlapping a query. In general, aggregation starts by grouping the subject hits by query hits, which we express as a coercion to a <em>List</em>,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hitsByQuery &lt;-<span class="st"> </span><span class="kw">as</span>(hits, <span class="st">&quot;List&quot;</span>)</code></pre></div>
<p>The result is an <em>IntegerList</em>, a type of <em>AtomicList</em>. <em>AtomicList</em> objects have many methods for efficient aggregation. In this case, we just call <code>lengths()</code> to get the count:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">counts &lt;-<span class="st"> </span><span class="kw">lengths</span>(hitsByQuery)</code></pre></div>
<p>Since this a common operation, there are shortcuts,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">counts &lt;-<span class="st"> </span><span class="kw">countQueryHits</span>(hits)</code></pre></div>
<p>or even shorter and more efficient,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">counts &lt;-<span class="st"> </span><span class="kw">countOverlaps</span>(query, subject, <span class="dt">ignore.strand=</span><span class="ot">TRUE</span>)
<span class="kw">unname</span>(counts)
<span class="co">#&gt;  [1] 0 0 0 2 0 0 0 0 0 2 0 2 0 0 2 0 0 2 2 0 0 0 1 0 0 0 2 0 1 0</span></code></pre></div>
<p>Often, we want to combine joins and aggregations. For example, we may want to annotate each query with the maximum score among the subject hits,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">query<span class="op">$</span>maxScore &lt;-<span class="st"> </span><span class="kw">max</span>(<span class="kw">extractList</span>(subject<span class="op">$</span>score, hitsByQuery))
<span class="kw">subset</span>(query, maxScore <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>)
<span class="co">#&gt; GRanges object with 9 ranges and 2 metadata columns:</span>
<span class="co">#&gt;      seqnames    ranges strand | replicate  maxScore</span>
<span class="co">#&gt;         &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |     &lt;Rle&gt; &lt;integer&gt;</span>
<span class="co">#&gt;   10     chr2   120-129      * |         1         3</span>
<span class="co">#&gt;   28     chr1   151-160      * |         1         6</span>
<span class="co">#&gt;    5     chr2   128-137      * |         2         4</span>
<span class="co">#&gt;   14     chr1   149-158      * |         2         6</span>
<span class="co">#&gt;   23     chr3   153-162      * |         2         8</span>
<span class="co">#&gt;   26     chr3   160-169      * |         2         9</span>
<span class="co">#&gt;    9     chr1    92-101      * |         3         1</span>
<span class="co">#&gt;   21     chr1   148-157      * |         3         6</span>
<span class="co">#&gt;   27     chr3   190-199      * |         3        10</span>
<span class="co">#&gt;   -------</span>
<span class="co">#&gt;   seqinfo: 3 sequences from an unspecified genome; no seqlengths</span></code></pre></div>
<p>In rare cases, we can more or less arbitrarily select one of the subject hits. The <code>select=</code> argument to <code>findOverlaps()</code> automatically selects an “arbitrary”, “first” (in subject order) or “last” subject range,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">hits &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(query, subject, <span class="dt">select=</span><span class="st">&quot;first&quot;</span>, <span class="dt">ignore.strand=</span><span class="ot">TRUE</span>)
hits &lt;-<span class="st"> </span><span class="kw">findOverlaps</span>(query, subject, <span class="dt">select=</span><span class="st">&quot;arbitrary&quot;</span>, <span class="dt">ignore.strand=</span><span class="ot">TRUE</span>)
hits
<span class="co">#&gt;  [1] NA NA NA  2 NA NA NA NA NA  5 NA  3 NA NA  5 NA NA  7  8 NA NA NA  1</span>
<span class="co">#&gt; [24] NA NA NA  5 NA 10 NA</span></code></pre></div>
</div>
<div id="exercises" class="section level2">
<h2><span class="header-section-number">4.16</span> Exercises</h2>
<ol style="list-style-type: decimal">
<li>Find the average intensity of the X and Y measurements for each each replicate over all positions in the query object</li>
<li>Add a new column to the intensities object that is the distance from each position to its closest gene (hint <code>IRanges::distance()</code>)</li>
<li>Find flanking regions downstream of the genes in gr that have width of 8bp</li>
<li>Are any of the intensities positions within the flanking region?</li>
</ol>
</div>
<div id="example-exploring-bigwig-files-from-annotationhub" class="section level2">
<h2><span class="header-section-number">4.17</span> Example: exploring BigWig files from AnnotationHub</h2>
<p>In the workflow of ChIP-seq data analysis, we are often interested in finding peaks from islands of coverage over a chromosome. Here we will use plyranges to explore ChiP-seq data from the Human Epigenome Roadmap project <span class="citation">Roadmap Epigenomics Consortium et al. (2015)</span>.</p>
<div id="extracting-data-from-annotationhub" class="section level3">
<h3><span class="header-section-number">4.17.1</span> Extracting data from AnnotationHub</h3>
<p>This data is available on Bioconductor’s AnnotationHub. First we construct an AnnotationHub, and then <code>query()</code> for all bigWigFiles related to the project that correspond to the following conditions:</p>
<ol style="list-style-type: decimal">
<li>are from methylation marks (H3K4ME in the title)</li>
<li>correspond to primary T CD8+ memory cells from peripheral blood</li>
<li>correspond to unimputed log10 P-values</li>
</ol>
<p>First we construct a hub that contains all references to the EpigenomeRoadMap data and extract the metadata as a data.frame:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(AnnotationHub)
ah &lt;-<span class="st"> </span><span class="kw">AnnotationHub</span>()
<span class="co">#&gt; snapshotDate(): 2018-06-27</span>
roadmap_hub &lt;-<span class="st"> </span><span class="kw">query</span>(ah, <span class="st">&quot;EpigenomeRoadMap&quot;</span>) 
metadata &lt;-<span class="st"> </span><span class="kw">query</span>(ah, <span class="st">&quot;Metadata&quot;</span>)[[1L]]
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.AnnotationHub/47270&#39;</span>
<span class="kw">head</span>(metadata)
<span class="co">#&gt;    EID    GROUP   COLOR          MNEMONIC</span>
<span class="co">#&gt; 1 E001      ESC #924965            ESC.I3</span>
<span class="co">#&gt; 2 E002      ESC #924965           ESC.WA7</span>
<span class="co">#&gt; 3 E003      ESC #924965            ESC.H1</span>
<span class="co">#&gt; 4 E004 ES-deriv #4178AE ESDR.H1.BMP4.MESO</span>
<span class="co">#&gt; 5 E005 ES-deriv #4178AE ESDR.H1.BMP4.TROP</span>
<span class="co">#&gt; 6 E006 ES-deriv #4178AE       ESDR.H1.MSC</span>
<span class="co">#&gt;                                     STD_NAME</span>
<span class="co">#&gt; 1                                ES-I3 Cells</span>
<span class="co">#&gt; 2                               ES-WA7 Cells</span>
<span class="co">#&gt; 3                                   H1 Cells</span>
<span class="co">#&gt; 4 H1 BMP4 Derived Mesendoderm Cultured Cells</span>
<span class="co">#&gt; 5 H1 BMP4 Derived Trophoblast Cultured Cells</span>
<span class="co">#&gt; 6          H1 Derived Mesenchymal Stem Cells</span>
<span class="co">#&gt;                                   EDACC_NAME     ANATOMY           TYPE</span>
<span class="co">#&gt; 1                            ES-I3_Cell_Line         ESC PrimaryCulture</span>
<span class="co">#&gt; 2                           ES-WA7_Cell_Line         ESC PrimaryCulture</span>
<span class="co">#&gt; 3                               H1_Cell_Line         ESC PrimaryCulture</span>
<span class="co">#&gt; 4 H1_BMP4_Derived_Mesendoderm_Cultured_Cells ESC_DERIVED     ESCDerived</span>
<span class="co">#&gt; 5 H1_BMP4_Derived_Trophoblast_Cultured_Cells ESC_DERIVED     ESCDerived</span>
<span class="co">#&gt; 6          H1_Derived_Mesenchymal_Stem_Cells ESC_DERIVED     ESCDerived</span>
<span class="co">#&gt;   AGE    SEX SOLID_LIQUID ETHNICITY SINGLEDONOR_COMPOSITE</span>
<span class="co">#&gt; 1  CL Female         &lt;NA&gt;      &lt;NA&gt;                    SD</span>
<span class="co">#&gt; 2  CL Female         &lt;NA&gt;      &lt;NA&gt;                    SD</span>
<span class="co">#&gt; 3  CL   Male         &lt;NA&gt;      &lt;NA&gt;                    SD</span>
<span class="co">#&gt; 4  CL   Male         &lt;NA&gt;      &lt;NA&gt;                    SD</span>
<span class="co">#&gt; 5  CL   Male         &lt;NA&gt;      &lt;NA&gt;                    SD</span>
<span class="co">#&gt; 6  CL   Male         &lt;NA&gt;      &lt;NA&gt;                    SD</span></code></pre></div>
<p>To find out the name of the sample corresponding to primary memory T-cells we can filter the data.frame. We extract the sample ID corresponding to our filter.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">primary_tcells &lt;-<span class="st"> </span><span class="kw">subset</span>(metadata,
                         ANATOMY <span class="op">==</span><span class="st"> &quot;BLOOD&quot;</span> <span class="op">&amp;</span><span class="st"> </span>TYPE <span class="op">==</span><span class="st"> &quot;PrimaryCell&quot;</span> <span class="op">&amp;</span>
<span class="st">                             </span>EDACC_NAME <span class="op">==</span><span class="st"> &quot;CD8_Memory_Primary_Cells&quot;</span>)<span class="op">$</span>EID
primary_tcells &lt;-<span class="st"> </span><span class="kw">as.character</span>(primary_tcells)</code></pre></div>
<p>Now we can take our roadmap hub and query it based on our other conditions:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">methylation_files &lt;-<span class="st">  </span><span class="kw">query</span>(roadmap_hub,
                            <span class="kw">c</span>(<span class="st">&quot;BigWig&quot;</span>, primary_tcells, <span class="st">&quot;H3K4ME[1-3]&quot;</span>,
                              <span class="st">&quot;pval.signal&quot;</span>))
methylation_files
<span class="co">#&gt; AnnotationHub with 5 records</span>
<span class="co">#&gt; # snapshotDate(): 2018-06-27 </span>
<span class="co">#&gt; # $dataprovider: BroadInstitute</span>
<span class="co">#&gt; # $species: Homo sapiens</span>
<span class="co">#&gt; # $rdataclass: BigWigFile</span>
<span class="co">#&gt; # additional mcols(): taxonomyid, genome, description,</span>
<span class="co">#&gt; #   coordinate_1_based, maintainer, rdatadateadded, preparerclass,</span>
<span class="co">#&gt; #   tags, rdatapath, sourceurl, sourcetype </span>
<span class="co">#&gt; # retrieve records with, e.g., &#39;object[[&quot;AH33454&quot;]]&#39; </span>
<span class="co">#&gt; </span>
<span class="co">#&gt;             title                                  </span>
<span class="co">#&gt;   AH33454 | E048-H3K4me1.pval.signal.bigwig        </span>
<span class="co">#&gt;   AH33455 | E048-H3K4me3.pval.signal.bigwig        </span>
<span class="co">#&gt;   AH39974 | E048-H3K4me1.imputed.pval.signal.bigwig</span>
<span class="co">#&gt;   AH40101 | E048-H3K4me2.imputed.pval.signal.bigwig</span>
<span class="co">#&gt;   AH40228 | E048-H3K4me3.imputed.pval.signal.bigwig</span></code></pre></div>
<p>So we’ll take the first two entries and download them as BigWigFiles:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">bw_files &lt;-<span class="st"> </span><span class="kw">lapply</span>(methylation_files[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>], <span class="st">`</span><span class="dt">[[</span><span class="st">`</span>, 1L)
<span class="co">#&gt; require(&quot;rtracklayer&quot;)</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.AnnotationHub/38894&#39;</span>
<span class="co">#&gt; downloading 0 resources</span>
<span class="co">#&gt; loading from cache </span>
<span class="co">#&gt;     &#39;/home/mramos//.AnnotationHub/38895&#39;</span></code></pre></div>
<p>We have our desired BigWig files so now we can we can start analyzing them.</p>
</div>
<div id="reading-bigwig-files" class="section level3">
<h3><span class="header-section-number">4.17.2</span> Reading BigWig files</h3>
<p>For this analysis, we will call peaks from a score vector over chromosome 10.</p>
<p>First, we extract the genome information from the first BigWig file and filter to get the range for chromosome 10. This range will be used as a filter when reading the file.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">chr10_ranges &lt;-<span class="st"> </span><span class="kw">Seqinfo</span>(<span class="dt">genome=</span><span class="st">&quot;hg19&quot;</span>)[<span class="st">&quot;chr10&quot;</span>]</code></pre></div>
<p>Then we read the BigWig file only extracting scores if they overlap chromosome 10.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(rtracklayer)
chr10_scores &lt;-<span class="st"> </span><span class="kw">lapply</span>(bw_files, import, <span class="dt">which =</span> chr10_ranges,
                       <span class="dt">as =</span> <span class="st">&quot;RleList&quot;</span>) 
chr10_scores[[<span class="dv">1</span>]]<span class="op">$</span>chr10
<span class="co">#&gt; numeric-Rle of length 135534747 with 5641879 runs</span>
<span class="co">#&gt;   Lengths:               60612                 172 ...                9907</span>
<span class="co">#&gt;   Values :  0.0394200012087822   0.154219999909401 ...                   0</span></code></pre></div>
<p>Each of element of the list is a run-length encoded vector of the scores for a particular signal type.</p>
<p>We find the islands by slicing the vectors,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">islands &lt;-<span class="st"> </span><span class="kw">lapply</span>(chr10_scores, slice, <span class="dt">lower=</span>1L)</code></pre></div>
<p>where the islands are represented as <em>Views</em> objects, i.e., ranges of interest over a genomic vector. Then we find the summit within each island,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">summits &lt;-<span class="st"> </span><span class="kw">lapply</span>(islands, viewRangeMaxs)</code></pre></div>
<p>using the optimized <code>viewRangeMaxs()</code> function. Each element of the <code>summits</code> list is a <em>RangesList</em> object, holding the ranges for each summit. The structure of the <em>RangesList</em> keeps track of the chromosome (10) of the summits (there could be multiple chromosomes in general). We broaden the summits and reduce them in order to smooth the peak calls and provide some context,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">summits &lt;-<span class="st"> </span><span class="kw">lapply</span>(<span class="kw">lapply</span>(summits, <span class="st">`</span><span class="dt">+</span><span class="st">`</span>, 50L), reduce)</code></pre></div>
<p>After this preprocessing, we want to convert the result to a more familiar and convenient GRanges object containing an <em>RleList</em> “score” column containing the score vector for each summit,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">summits_grs &lt;-<span class="st"> </span><span class="kw">lapply</span>(summits, GRanges)
score_grs &lt;-<span class="st"> </span><span class="kw">mapply</span>(<span class="cf">function</span>(scores, summits) {
    summits<span class="op">$</span>score &lt;-<span class="st"> </span>scores[summits]
    <span class="kw">seqlengths</span>(summits) &lt;-<span class="st"> </span><span class="kw">lengths</span>(scores)
    summits
}, chr10_scores, summits_grs)
score_gr &lt;-<span class="st"> </span><span class="kw">stack</span>(<span class="kw">GenomicRangesList</span>(score_grs), <span class="dt">index.var=</span><span class="st">&quot;signal_type&quot;</span>)</code></pre></div>
<p>One problem with <em>RangesList</em> is that it does not keep track of the sequence lengths, so we need to add those after forming the <em>GRanges</em>.</p>
<p>We could then find summits with the maximum summit height within each signal type:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">score_gr<span class="op">$</span>score_max &lt;-<span class="st"> </span><span class="kw">max</span>(score_gr<span class="op">$</span>score)
chr10_max_score_region &lt;-<span class="st"> </span><span class="kw">aggregate</span>(score_gr, score_max <span class="op">~</span><span class="st"> </span>signal_type, max)</code></pre></div>
</div>
<div id="exercises-1" class="section level3">
<h3><span class="header-section-number">4.17.3</span> Exercises</h3>
<ol style="list-style-type: decimal">
<li>Use the <code>reduce_ranges()</code> function to find all peaks for each signal type.</li>
<li>How could you annotate the scores to find out which genes overlap each peak found in 1.?</li>
<li>Plot a 1000nt window centred around the maximum scores for each signal type using the <code>ggbio</code> or <code>Gviz</code> package.</li>
</ol>
</div>
</div>
<div id="worked-example-coverage-analysis-of-bam-files" class="section level2">
<h2><span class="header-section-number">4.18</span> Worked example: coverage analysis of BAM files</h2>
<p>A common quality control check in a genomics workflow is to perform coverage analysis over features of interest or over the entire genome. Here we use the data from the airway package to operate on read alignment data and compute coverage histograms.</p>
<p>First let’s gather all the BAM files available to use in airway (see <code>browseVignettes(&quot;airway&quot;)</code> for more information about the data and how it was prepared):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(tools)
bams &lt;-<span class="st"> </span><span class="kw">list_files_with_exts</span>(<span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;airway&quot;</span>), <span class="st">&quot;bam&quot;</span>)
<span class="kw">names</span>(bams) &lt;-<span class="st"> </span><span class="kw">sub</span>(<span class="st">&quot;_[^_]+$&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="kw">basename</span>(bams))
<span class="kw">library</span>(Rsamtools)
<span class="co">#&gt; Loading required package: Biostrings</span>
<span class="co">#&gt; Loading required package: XVector</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;Biostrings&#39;</span>
<span class="co">#&gt; The following object is masked from &#39;package:base&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     strsplit</span>
bams &lt;-<span class="st"> </span><span class="kw">BamFileList</span>(bams)</code></pre></div>
<p>Casting the vector of filenames to a formal <em>BamFileList</em> is critical for informing the following code about the nature of the files.</p>
<p>To start let’s look at a single BAM file (containing only reads from chr1). We can compute the coverage of the alignments over all contigs in the BAM as follows:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">first_bam &lt;-<span class="st"> </span>bams[[1L]]
first_bam_cvg &lt;-<span class="st"> </span><span class="kw">coverage</span>(first_bam)</code></pre></div>
<p>The result is a list of <em>Rle</em> objects, one per chromosome. Like other <em>AtomicList</em> objects, we call pass our <em>RleList</em> to <code>table()</code> to compute the coverage histogram by chromosome,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">head</span>(<span class="kw">table</span>(first_bam_cvg)[1L,])
<span class="co">#&gt;         0         1         2         3         4         5 </span>
<span class="co">#&gt; 249202844     15607      5247      3055      2030      1280</span></code></pre></div>
<p>For RNA-seq experiments we are often interested in splitting up alignments based on whether the alignment has skipped a region from the reference (that is, there is an “N” in the cigar string, indicating an intron). We can represent the nested structure using a <em>GRangesList</em> object.</p>
<p>To begin we read the BAM file into a <em>GAlignments</em> object using <code>readGAlignments()</code> and extract the ranges, chopping by introns, using <code>grglist()</code>,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(GenomicAlignments)
<span class="co">#&gt; Loading required package: SummarizedExperiment</span>
<span class="co">#&gt; Loading required package: Biobase</span>
<span class="co">#&gt; Welcome to Bioconductor</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     Vignettes contain introductory material; view with</span>
<span class="co">#&gt;     &#39;browseVignettes()&#39;. To cite Bioconductor, see</span>
<span class="co">#&gt;     &#39;citation(&quot;Biobase&quot;)&#39;, and for packages &#39;citation(&quot;pkgname&quot;)&#39;.</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;Biobase&#39;</span>
<span class="co">#&gt; The following object is masked from &#39;package:AnnotationHub&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     cache</span>
<span class="co">#&gt; Loading required package: DelayedArray</span>
<span class="co">#&gt; Loading required package: matrixStats</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;matrixStats&#39;</span>
<span class="co">#&gt; The following objects are masked from &#39;package:Biobase&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     anyMissing, rowMedians</span>
<span class="co">#&gt; Loading required package: BiocParallel</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;DelayedArray&#39;</span>
<span class="co">#&gt; The following objects are masked from &#39;package:matrixStats&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     colMaxs, colMins, colRanges, rowMaxs, rowMins, rowRanges</span>
<span class="co">#&gt; The following object is masked from &#39;package:Biostrings&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     type</span>
<span class="co">#&gt; The following objects are masked from &#39;package:base&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     aperm, apply</span>
reads &lt;-<span class="st"> </span><span class="kw">grglist</span>(<span class="kw">readGAlignments</span>(first_bam))</code></pre></div>
<p>Finally, we can find the junction reads:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">reads[<span class="kw">lengths</span>(reads) <span class="op">&gt;=</span><span class="st"> </span>2L]
<span class="co">#&gt; GRangesList object of length 3833:</span>
<span class="co">#&gt; [[1]] </span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames            ranges strand</span>
<span class="co">#&gt;          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="co">#&gt;   [1]        1 11072744-11072800      +</span>
<span class="co">#&gt;   [2]        1 11073773-11073778      +</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]] </span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames            ranges strand</span>
<span class="co">#&gt;   [1]        1 11072745-11072800      -</span>
<span class="co">#&gt;   [2]        1 11073773-11073779      -</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[3]] </span>
<span class="co">#&gt; GRanges object with 2 ranges and 0 metadata columns:</span>
<span class="co">#&gt;       seqnames            ranges strand</span>
<span class="co">#&gt;   [1]        1 11072746-11072800      +</span>
<span class="co">#&gt;   [2]        1 11073773-11073780      +</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; ...</span>
<span class="co">#&gt; &lt;3830 more elements&gt;</span>
<span class="co">#&gt; -------</span>
<span class="co">#&gt; seqinfo: 84 sequences from an unspecified genome</span></code></pre></div>
<p>We typically want to count how many reads overlap each gene. First, we get the transcript structures as a <em>GRangesList</em> from Ensembl,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(GenomicFeatures)
<span class="co">#&gt; Loading required package: AnnotationDbi</span>
<span class="kw">library</span>(EnsDb.Hsapiens.v75)
<span class="co">#&gt; Loading required package: ensembldb</span>
<span class="co">#&gt; Loading required package: AnnotationFilter</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Attaching package: &#39;ensembldb&#39;</span>
<span class="co">#&gt; The following object is masked from &#39;package:stats&#39;:</span>
<span class="co">#&gt; </span>
<span class="co">#&gt;     filter</span>
tx &lt;-<span class="st"> </span><span class="kw">exonsBy</span>(EnsDb.Hsapiens.v75, <span class="st">&quot;gene&quot;</span>)</code></pre></div>
<p>Finally, we count how many reads overlap each transcript,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">reads &lt;-<span class="st"> </span><span class="kw">keepStandardChromosomes</span>(reads)
counts &lt;-<span class="st"> </span><span class="kw">countOverlaps</span>(tx, reads, <span class="dt">ignore.strand=</span><span class="ot">TRUE</span>)
<span class="kw">head</span>(counts[counts <span class="op">&gt;</span><span class="st"> </span><span class="dv">0</span>])
<span class="co">#&gt; ENSG00000009724 ENSG00000116649 ENSG00000120942 ENSG00000120948 </span>
<span class="co">#&gt;              89            2006             436            5495 </span>
<span class="co">#&gt; ENSG00000171819 ENSG00000171824 </span>
<span class="co">#&gt;               8            2368</span></code></pre></div>
<p>To do this over every sample, we use the <code>summarizeOverlaps()</code> convenience function,</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">airway &lt;-<span class="st"> </span><span class="kw">summarizeOverlaps</span>(<span class="dt">features=</span>tx, <span class="dt">reads=</span>bams,
                            <span class="dt">mode=</span><span class="st">&quot;Union&quot;</span>, <span class="dt">singleEnd=</span><span class="ot">FALSE</span>,
                            <span class="dt">ignore.strand=</span><span class="ot">TRUE</span>, <span class="dt">fragments=</span><span class="ot">TRUE</span>)
airway
<span class="co">#&gt; class: RangedSummarizedExperiment </span>
<span class="co">#&gt; dim: 64102 8 </span>
<span class="co">#&gt; metadata(0):</span>
<span class="co">#&gt; assays(1): counts</span>
<span class="co">#&gt; rownames(64102): ENSG00000000003 ENSG00000000005 ... LRG_98 LRG_99</span>
<span class="co">#&gt; rowData names(0):</span>
<span class="co">#&gt; colnames(8): SRR1039508 SRR1039509 ... SRR1039520 SRR1039521</span>
<span class="co">#&gt; colData names(0):</span></code></pre></div>
<p>The <code>airway</code> object is a <em>SummarizedExperiment</em> object, the central Bioconductor data structure for storing results summarized per feature and sample, along with sample and feature metadata. It is at the point of summarization that workflows switch focus from the ranges infrastructure to Bioconductor modeling packages, most of which consume the <em>SummarizedExperiment</em> data structure, so this is an appropriate point to end this tutorial.</p>
<div id="exercises-2" class="section level3">
<h3><span class="header-section-number">4.18.1</span> Exercises</h3>
<ol style="list-style-type: decimal">
<li>Compute the total depth of coverage across all features.</li>
<li>How could you compute the proportion of bases covered over an entire genome? (hint: <code>seqinfo</code> and <code>S4Vectors::merge</code>)</li>
<li>How could you compute the strand specific genome wide coverage?</li>
<li>Create a workflow for computing the strand specific coverage for all BAM files.</li>
<li>For each sample plot total breadth of coverage against the number of bases covered faceted by each sample name.</li>
</ol>
</div>
</div>
<div id="conclusions" class="section level2">
<h2><span class="header-section-number">4.19</span> Conclusions</h2>
<p>The Bioconductor ranges infrastructure is rich and complex, and it can be intimidating to new users. However, the effort invested will pay dividends, especially when generalizing a series of bespoke analyses into a reusable contribution to Bioconductor.</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="introduction-to-bioconductor-annotation-resources.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="public-data-resources-and-bioconductor.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
