<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>The Bioconductor 2018 Workshop Compilation</title>
  <meta name="description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference">
  <meta name="generator" content="bookdown 0.7 and GitBook 2.6.7">

  <meta property="og:title" content="The Bioconductor 2018 Workshop Compilation" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  <meta name="github-repo" content="Bioconductor/BiocWorkshops" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Bioconductor 2018 Workshop Compilation" />
  
  <meta name="twitter:description" content="This book is a central repository for all the workshops submitted to the Bioconductor 2018 Conference" />
  




  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html">
<link rel="next" href="functional-enrichment-analysis-of-high-throughput-omics-data.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
div.sourceCode { overflow-x: auto; }
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; } /* Keyword */
code > span.dt { color: #902000; } /* DataType */
code > span.dv { color: #40a070; } /* DecVal */
code > span.bn { color: #40a070; } /* BaseN */
code > span.fl { color: #40a070; } /* Float */
code > span.ch { color: #4070a0; } /* Char */
code > span.st { color: #4070a0; } /* String */
code > span.co { color: #60a0b0; font-style: italic; } /* Comment */
code > span.ot { color: #007020; } /* Other */
code > span.al { color: #ff0000; font-weight: bold; } /* Alert */
code > span.fu { color: #06287e; } /* Function */
code > span.er { color: #ff0000; font-weight: bold; } /* Error */
code > span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
code > span.cn { color: #880000; } /* Constant */
code > span.sc { color: #4070a0; } /* SpecialChar */
code > span.vs { color: #4070a0; } /* VerbatimString */
code > span.ss { color: #bb6688; } /* SpecialString */
code > span.im { } /* Import */
code > span.va { color: #19177c; } /* Variable */
code > span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code > span.op { color: #666666; } /* Operator */
code > span.bu { } /* BuiltIn */
code > span.ex { } /* Extension */
code > span.pp { color: #bc7a00; } /* Preprocessor */
code > span.at { color: #7d9029; } /* Attribute */
code > span.do { color: #ba2121; font-style: italic; } /* Documentation */
code > span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code > span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code > span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Bioconductor 2018 Workshops</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#for-everyone"><i class="fa fa-check"></i><b>1.1</b> For Everyone</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#for-workshop-authors"><i class="fa fa-check"></i><b>1.2</b> For Workshop Authors</a><ul>
<li class="chapter" data-level="1.2.1" data-path="index.html"><a href="index.html#description"><i class="fa fa-check"></i><b>1.2.1</b> DESCRIPTION</a></li>
<li class="chapter" data-level="1.2.2" data-path="index.html"><a href="index.html#classifying-your-workshop"><i class="fa fa-check"></i><b>1.2.2</b> Classifying your workshop</a></li>
<li class="chapter" data-level="1.2.3" data-path="index.html"><a href="index.html#bookdown"><i class="fa fa-check"></i><b>1.2.3</b> Bookdown</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#deadlines-for-bioc2018"><i class="fa fa-check"></i><b>1.3</b> Deadlines for Bioc2018</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html"><i class="fa fa-check"></i><b>2</b> <em>R</em> and <em>Bioconductor</em> for everyone: an introduction</a><ul>
<li class="chapter" data-level="2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#overview"><i class="fa fa-check"></i><b>2.1</b> Overview</a><ul>
<li class="chapter" data-level="2.1.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#description-1"><i class="fa fa-check"></i><b>2.1.1</b> Description</a></li>
<li class="chapter" data-level="2.1.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#pre-requisites"><i class="fa fa-check"></i><b>2.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="2.1.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#participation"><i class="fa fa-check"></i><b>2.1.3</b> Participation</a></li>
<li class="chapter" data-level="2.1.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#r-bioconductor-packages-used"><i class="fa fa-check"></i><b>2.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="2.1.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#time-outline"><i class="fa fa-check"></i><b>2.1.5</b> Time outline</a></li>
<li class="chapter" data-level="2.1.6" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#workshop-goals-and-objectives"><i class="fa fa-check"></i><b>2.1.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="2.1.7" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-goals"><i class="fa fa-check"></i><b>2.1.7</b> Learning goals</a></li>
<li class="chapter" data-level="2.1.8" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#learning-objectives"><i class="fa fa-check"></i><b>2.1.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-r"><i class="fa fa-check"></i><b>2.2</b> Introduction to <em>R</em></a><ul>
<li class="chapter" data-level="2.2.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#rstudio-orientation"><i class="fa fa-check"></i><b>2.2.1</b> <em>RStudio</em> orientation</a></li>
<li class="chapter" data-level="2.2.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#very-basics"><i class="fa fa-check"></i><b>2.2.2</b> Very basics</a></li>
<li class="chapter" data-level="2.2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#data-input-and-manipulation"><i class="fa fa-check"></i><b>2.2.3</b> Data input and manipulation</a></li>
<li class="chapter" data-level="2.2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#writing-r-scripts"><i class="fa fa-check"></i><b>2.2.4</b> Writing <em>R</em> scripts</a></li>
<li class="chapter" data-level="2.2.5" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#using-packages"><i class="fa fa-check"></i><b>2.2.5</b> Using packages</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#introduction-to-bioconductor"><i class="fa fa-check"></i><b>2.3</b> Introduction to <em>Bioconductor</em></a><ul>
<li class="chapter" data-level="2.3.1" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#discovering-installing-and-learning-how-to-use-bioconductor-packages"><i class="fa fa-check"></i><b>2.3.1</b> Discovering, installing, and learning how to use <em>Bioconductor</em> packages</a></li>
<li class="chapter" data-level="2.3.2" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-genomic-ranges"><i class="fa fa-check"></i><b>2.3.2</b> Working with Genomic Ranges</a></li>
<li class="chapter" data-level="2.3.3" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#working-with-summarized-experimental-data"><i class="fa fa-check"></i><b>2.3.3</b> Working with summarized experimental data</a></li>
<li class="chapter" data-level="2.3.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#down-stream-analysis"><i class="fa fa-check"></i><b>2.3.4</b> Down-stream analysis</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="r-and-bioconductor-for-everyone-an-introduction.html"><a href="r-and-bioconductor-for-everyone-an-introduction.html#summary"><i class="fa fa-check"></i><b>2.4</b> Summary</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html"><i class="fa fa-check"></i><b>3</b> RNA-seq data analysis with DESeq2</a><ul>
<li class="chapter" data-level="3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#overview-1"><i class="fa fa-check"></i><b>3.1</b> Overview</a><ul>
<li class="chapter" data-level="3.1.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#description-2"><i class="fa fa-check"></i><b>3.1.1</b> Description</a></li>
<li class="chapter" data-level="3.1.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#pre-requisites-1"><i class="fa fa-check"></i><b>3.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="3.1.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#participation-1"><i class="fa fa-check"></i><b>3.1.3</b> Participation</a></li>
<li class="chapter" data-level="3.1.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#r-bioconductor-packages-used-1"><i class="fa fa-check"></i><b>3.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="3.1.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#time-outline-1"><i class="fa fa-check"></i><b>3.1.5</b> Time outline</a></li>
<li class="chapter" data-level="3.1.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#workshop-goals-and-objectives-1"><i class="fa fa-check"></i><b>3.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#preparing-data-for-deseq2"><i class="fa fa-check"></i><b>3.2</b> Preparing data for <em>DESeq2</em></a><ul>
<li class="chapter" data-level="3.2.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#experimental-data"><i class="fa fa-check"></i><b>3.2.1</b> Experimental data</a></li>
<li class="chapter" data-level="3.2.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#modeling-count-data"><i class="fa fa-check"></i><b>3.2.2</b> Modeling count data</a></li>
<li class="chapter" data-level="3.2.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#transcript-abundances"><i class="fa fa-check"></i><b>3.2.3</b> Transcript abundances</a></li>
<li class="chapter" data-level="3.2.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#salmon-quantification"><i class="fa fa-check"></i><b>3.2.4</b> <em>Salmon</em> quantification</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#importing-into-r-with-tximport"><i class="fa fa-check"></i><b>3.3</b> Importing into R with <em>tximport</em></a><ul>
<li class="chapter" data-level="3.3.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#specifying-file-locations"><i class="fa fa-check"></i><b>3.3.1</b> Specifying file locations</a></li>
<li class="chapter" data-level="3.3.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-transcripts-to-genes"><i class="fa fa-check"></i><b>3.3.2</b> Mapping transcripts to genes</a></li>
<li class="chapter" data-level="3.3.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#tximport-command"><i class="fa fa-check"></i><b>3.3.3</b> <em>tximport</em> command</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>3.4</b> Exploratory data analysis</a><ul>
<li class="chapter" data-level="3.4.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simple-eda"><i class="fa fa-check"></i><b>3.4.1</b> Simple EDA</a></li>
<li class="chapter" data-level="3.4.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#data-transformation-for-eda"><i class="fa fa-check"></i><b>3.4.2</b> Data transformation for EDA</a></li>
<li class="chapter" data-level="3.4.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#principal-components-plot"><i class="fa fa-check"></i><b>3.4.3</b> Principal components plot</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#differential-expression-analysis"><i class="fa fa-check"></i><b>3.5</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="3.5.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#standard-de-steps"><i class="fa fa-check"></i><b>3.5.1</b> Standard DE steps</a></li>
<li class="chapter" data-level="3.5.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#minimum-effect-size"><i class="fa fa-check"></i><b>3.5.2</b> Minimum effect size</a></li>
</ul></li>
<li class="chapter" data-level="3.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#annotationhub"><i class="fa fa-check"></i><b>3.6</b> <em>AnnotationHub</em></a><ul>
<li class="chapter" data-level="3.6.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#querying-annotationhub"><i class="fa fa-check"></i><b>3.6.1</b> Querying <em>AnnotationHub</em></a></li>
<li class="chapter" data-level="3.6.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#mapping-ids"><i class="fa fa-check"></i><b>3.6.2</b> Mapping IDs</a></li>
</ul></li>
<li class="chapter" data-level="3.7" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#building-reports"><i class="fa fa-check"></i><b>3.7</b> Building reports</a><ul>
<li class="chapter" data-level="3.7.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#reportingtools"><i class="fa fa-check"></i><b>3.7.1</b> <em>ReportingTools</em></a></li>
<li class="chapter" data-level="3.7.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#glimma"><i class="fa fa-check"></i><b>3.7.2</b> <em>Glimma</em></a></li>
</ul></li>
<li class="chapter" data-level="3.8" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#integration-with-zinb-wave"><i class="fa fa-check"></i><b>3.8</b> Integration with <em>ZINB-WaVE</em></a><ul>
<li class="chapter" data-level="3.8.1" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-with-splatter"><i class="fa fa-check"></i><b>3.8.1</b> Simulate with <em>splatter</em></a></li>
<li class="chapter" data-level="3.8.2" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#simulate-single-cell-count-data-with-splatter"><i class="fa fa-check"></i><b>3.8.2</b> Simulate single-cell count data with <em>splatter</em></a></li>
<li class="chapter" data-level="3.8.3" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-zeros-with-zinbwave"><i class="fa fa-check"></i><b>3.8.3</b> Model zeros with <em>zinbwave</em></a></li>
<li class="chapter" data-level="3.8.4" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#model-non-zeros-with-deseq2"><i class="fa fa-check"></i><b>3.8.4</b> Model non-zeros with <em>DESeq2</em></a></li>
<li class="chapter" data-level="3.8.5" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#plot-dispersion-estimates"><i class="fa fa-check"></i><b>3.8.5</b> Plot dispersion estimates</a></li>
<li class="chapter" data-level="3.8.6" data-path="rna-seq-data-analysis-with-deseq2.html"><a href="rna-seq-data-analysis-with-deseq2.html#evaluation-against-truth"><i class="fa fa-check"></i><b>3.8.6</b> Evaluation against truth</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><i class="fa fa-check"></i><b>4</b> Analysis of single-cell RNA-seq data: Dimensionality reduction, clustering, and lineage inference</a><ul>
<li class="chapter" data-level="4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#overview-2"><i class="fa fa-check"></i><b>4.1</b> Overview</a><ul>
<li class="chapter" data-level="4.1.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#description-3"><i class="fa fa-check"></i><b>4.1.1</b> Description</a></li>
<li class="chapter" data-level="4.1.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-requisites-2"><i class="fa fa-check"></i><b>4.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="4.1.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#participation-2"><i class="fa fa-check"></i><b>4.1.3</b> Participation</a></li>
<li class="chapter" data-level="4.1.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#r-bioconductor-packages-used-2"><i class="fa fa-check"></i><b>4.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="4.1.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#time-outline-2"><i class="fa fa-check"></i><b>4.1.5</b> Time outline</a></li>
<li class="chapter" data-level="4.1.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#workshop-goals-and-objectives-2"><i class="fa fa-check"></i><b>4.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#getting-started"><i class="fa fa-check"></i><b>4.2</b> Getting started</a><ul>
<li class="chapter" data-level="4.2.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-data"><i class="fa fa-check"></i><b>4.2.1</b> The data</a></li>
<li class="chapter" data-level="4.2.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#parallel-computing"><i class="fa fa-check"></i><b>4.2.2</b> Parallel computing</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#the-singlecellexperiment-class"><i class="fa fa-check"></i><b>4.3</b> The <code>SingleCellExperiment</code> class</a></li>
<li class="chapter" data-level="4.4" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#pre-processing"><i class="fa fa-check"></i><b>4.4</b> Pre-processing</a><ul>
<li class="chapter" data-level="4.4.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#sample-filtering"><i class="fa fa-check"></i><b>4.4.1</b> Sample filtering</a></li>
</ul></li>
<li class="chapter" data-level="4.5" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization-and-dimensionality-reduction-zinb-wave"><i class="fa fa-check"></i><b>4.5</b> Normalization and dimensionality reduction: ZINB-WaVE</a><ul>
<li class="chapter" data-level="4.5.1" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#normalization"><i class="fa fa-check"></i><b>4.5.1</b> Normalization</a></li>
<li class="chapter" data-level="4.5.2" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#dimensionality-reduction"><i class="fa fa-check"></i><b>4.5.2</b> Dimensionality reduction</a></li>
</ul></li>
<li class="chapter" data-level="4.6" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-clustering-rsec"><i class="fa fa-check"></i><b>4.6</b> Cell clustering: RSEC</a></li>
<li class="chapter" data-level="4.7" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#cell-lineage-and-pseudotime-inference-slingshot"><i class="fa fa-check"></i><b>4.7</b> Cell lineage and pseudotime inference: Slingshot</a></li>
<li class="chapter" data-level="4.8" data-path="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html"><a href="analysis-of-single-cell-rna-seq-data-dimensionality-reduction-clustering-and-lineage-inference.html#differential-expression-analysis-along-lineages"><i class="fa fa-check"></i><b>4.8</b> Differential expression analysis along lineages</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html"><i class="fa fa-check"></i><b>5</b> Cytoscape Automation in R using Rcy3</a><ul>
<li class="chapter" data-level="5.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-3"><i class="fa fa-check"></i><b>5.1</b> Overview</a><ul>
<li class="chapter" data-level="5.1.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#instructors-names-and-contact-information"><i class="fa fa-check"></i><b>5.1.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="5.1.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-description"><i class="fa fa-check"></i><b>5.1.2</b> Workshop Description</a></li>
<li class="chapter" data-level="5.1.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-prerequisites"><i class="fa fa-check"></i><b>5.1.3</b> Workshop prerequisites:</a></li>
<li class="chapter" data-level="5.1.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background"><i class="fa fa-check"></i><b>5.1.4</b> Background:</a></li>
<li class="chapter" data-level="5.1.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-participation"><i class="fa fa-check"></i><b>5.1.5</b> Workshop Participation</a></li>
<li class="chapter" data-level="5.1.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#r-bioconductor-packages-used-3"><i class="fa fa-check"></i><b>5.1.6</b> R / Bioconductor packages used</a></li>
<li class="chapter" data-level="5.1.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#time-outline-3"><i class="fa fa-check"></i><b>5.1.7</b> Time outline</a></li>
<li class="chapter" data-level="5.1.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#workshop-goals-and-objectives-3"><i class="fa fa-check"></i><b>5.1.8</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#background-1"><i class="fa fa-check"></i><b>5.2</b> Background</a><ul>
<li class="chapter" data-level="5.2.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape"><i class="fa fa-check"></i><b>5.2.1</b> Cytoscape</a></li>
<li class="chapter" data-level="5.2.2" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#overview-of-network-biology-themes-and-concepts"><i class="fa fa-check"></i><b>5.2.2</b> Overview of network biology themes and concepts</a></li>
<li class="chapter" data-level="5.2.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#networks-as-tools"><i class="fa fa-check"></i><b>5.2.3</b> Networks as Tools</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#translating-biological-data-into-cytoscape-using-rcy3"><i class="fa fa-check"></i><b>5.3</b> Translating biological data into Cytoscape using RCy3</a></li>
<li class="chapter" data-level="5.4" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#set-up"><i class="fa fa-check"></i><b>5.4</b> Set up</a><ul>
<li class="chapter" data-level="5.4.1" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#getting-started-1"><i class="fa fa-check"></i><b>5.4.1</b> Getting started</a></li>
</ul></li>
<li class="chapter" data-level="5.5" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#cytoscape-basics"><i class="fa fa-check"></i><b>5.5</b> Cytoscape Basics</a></li>
<li class="chapter" data-level="5.6" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#example-data-set"><i class="fa fa-check"></i><b>5.6</b> Example Data Set</a></li>
<li class="chapter" data-level="5.7" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#finding-network-data"><i class="fa fa-check"></i><b>5.7</b> Finding Network Data</a></li>
<li class="chapter" data-level="5.8" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-1---how-are-my-top-genes-related"><i class="fa fa-check"></i><b>5.8</b> Use Case 1 - How are my top genes related?</a></li>
<li class="chapter" data-level="5.9" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-2---which-genes-have-similar-expression."><i class="fa fa-check"></i><b>5.9</b> Use Case 2 - Which genes have similar expression.</a></li>
<li class="chapter" data-level="5.10" data-path="cytoscape-automation-in-r-using-rcy3.html"><a href="cytoscape-automation-in-r-using-rcy3.html#use-case-3---functional-enrichment-of-omics-set."><i class="fa fa-check"></i><b>5.10</b> Use Case 3 - Functional Enrichment of Omics set.</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><i class="fa fa-check"></i><b>6</b> Effectively using the DelayedArray framework to support the analysis of large datasets</a><ul>
<li class="chapter" data-level="6.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-4"><i class="fa fa-check"></i><b>6.1</b> Overview</a><ul>
<li class="chapter" data-level="6.1.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#description-4"><i class="fa fa-check"></i><b>6.1.1</b> Description</a></li>
<li class="chapter" data-level="6.1.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#pre-requisites-3"><i class="fa fa-check"></i><b>6.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="6.1.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#participation-3"><i class="fa fa-check"></i><b>6.1.3</b> Participation</a></li>
<li class="chapter" data-level="6.1.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#r-bioconductor-packages-used-4"><i class="fa fa-check"></i><b>6.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="6.1.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#time-outline-4"><i class="fa fa-check"></i><b>6.1.5</b> Time outline</a></li>
<li class="chapter" data-level="6.1.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#workshop-goals-and-objectives-4"><i class="fa fa-check"></i><b>6.1.6</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#introductory-material"><i class="fa fa-check"></i><b>6.2</b> Introductory material</a><ul>
<li class="chapter" data-level="6.2.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#discussion"><i class="fa fa-check"></i><b>6.2.1</b> Discussion</a></li>
<li class="chapter" data-level="6.2.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#why-learn-delayedarray"><i class="fa fa-check"></i><b>6.2.2</b> Why learn DelayedArray?</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#overview-of-delayedarray-framework"><i class="fa fa-check"></i><b>6.3</b> Overview of DelayedArray framework</a><ul>
<li class="chapter" data-level="6.3.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-package"><i class="fa fa-check"></i><b>6.3.1</b> The <strong>DelayedArray</strong> package</a></li>
<li class="chapter" data-level="6.3.2" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#the-delayedarray-class"><i class="fa fa-check"></i><b>6.3.2</b> The <em>DelayedArray</em> class</a></li>
<li class="chapter" data-level="6.3.3" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#operating-on-delayedarray-objects"><i class="fa fa-check"></i><b>6.3.3</b> Operating on <em>DelayedArray</em> objects</a></li>
</ul></li>
<li class="chapter" data-level="6.4" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#whats-out-there-already"><i class="fa fa-check"></i><b>6.4</b> What’s out there already?</a></li>
<li class="chapter" data-level="6.5" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#incoporating-delayedarray-into-a-package"><i class="fa fa-check"></i><b>6.5</b> Incoporating DelayedArray into a package</a><ul>
<li class="chapter" data-level="6.5.1" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#writing-algorithms-to-process-delayedarray-instances"><i class="fa fa-check"></i><b>6.5.1</b> Writing algorithms to process <em>DelayedArray</em> instances</a></li>
</ul></li>
<li class="chapter" data-level="6.6" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#questions-and-discussion"><i class="fa fa-check"></i><b>6.6</b> Questions and discussion</a></li>
<li class="chapter" data-level="6.7" data-path="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html"><a href="effectively-using-the-delayedarray-framework-to-support-the-analysis-of-large-datasets.html#todos"><i class="fa fa-check"></i><b>6.7</b> TODOs</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><i class="fa fa-check"></i><b>7</b> Variant Functional Annotation using StatePaintR, FunciVar &amp; MotifBreakR</a><ul>
<li class="chapter" data-level="7.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#instructors-names-and-contact-information-1"><i class="fa fa-check"></i><b>7.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="7.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-description-1"><i class="fa fa-check"></i><b>7.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="7.2.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#pre-requisites-4"><i class="fa fa-check"></i><b>7.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="7.2.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-participation-1"><i class="fa fa-check"></i><b>7.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="7.2.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#r-bioconductor-packages-used-5"><i class="fa fa-check"></i><b>7.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="7.2.4" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#time-outline-5"><i class="fa fa-check"></i><b>7.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#workshop-goals-and-objectives-5"><i class="fa fa-check"></i><b>7.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="7.3.1" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#key-concepts"><i class="fa fa-check"></i><b>7.3.1</b> Key Concepts</a></li>
<li class="chapter" data-level="7.3.2" data-path="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html"><a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html#learning-objectives-4"><i class="fa fa-check"></i><b>7.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="8" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html"><i class="fa fa-check"></i><b>8</b> Working with Genomic Data in R with the DECIPHER package</a><ul>
<li class="chapter" data-level="8.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#overview-5"><i class="fa fa-check"></i><b>8.1</b> Overview</a><ul>
<li class="chapter" data-level="8.1.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-description-2"><i class="fa fa-check"></i><b>8.1.1</b> Workshop Description</a></li>
<li class="chapter" data-level="8.1.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#pre-requisites-5"><i class="fa fa-check"></i><b>8.1.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="8.1.3" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-participation-2"><i class="fa fa-check"></i><b>8.1.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="8.1.4" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#r-bioconductor-packages-used-6"><i class="fa fa-check"></i><b>8.1.4</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="8.1.5" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#time-outline-6"><i class="fa fa-check"></i><b>8.1.5</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="8.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#workshop-goals-and-objectives-6"><i class="fa fa-check"></i><b>8.2</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="8.2.1" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-goals-4"><i class="fa fa-check"></i><b>8.2.1</b> Learning goals</a></li>
<li class="chapter" data-level="8.2.2" data-path="working-with-genomic-data-in-r-with-the-decipher-package.html"><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#learning-objectives-5"><i class="fa fa-check"></i><b>8.2.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html"><i class="fa fa-check"></i><b>9</b> Functional enrichment analysis of high-throughput omics data</a><ul>
<li class="chapter" data-level="9.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#instructors-names-and-contact-information-2"><i class="fa fa-check"></i><b>9.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="9.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-description-3"><i class="fa fa-check"></i><b>9.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="9.2.1" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#pre-requisites-6"><i class="fa fa-check"></i><b>9.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="9.2.2" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop-participation-3"><i class="fa fa-check"></i><b>9.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="9.2.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#r-bioconductor-packages-used-7"><i class="fa fa-check"></i><b>9.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="9.2.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#time-outline-7"><i class="fa fa-check"></i><b>9.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="9.3" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#goals-and-objectives"><i class="fa fa-check"></i><b>9.3</b> Goals and objectives</a></li>
<li class="chapter" data-level="9.4" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#workshop"><i class="fa fa-check"></i><b>9.4</b> Workshop</a></li>
<li class="chapter" data-level="9.5" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#where-does-it-all-come-from"><i class="fa fa-check"></i><b>9.5</b> Where does it all come from?</a></li>
<li class="chapter" data-level="9.6" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-expression-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.6</b> Gene expression-based enrichment analysis</a></li>
<li class="chapter" data-level="9.7" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#a-primer-on-terminology-existing-methods-statistical-theory"><i class="fa fa-check"></i><b>9.7</b> A primer on terminology, existing methods &amp; statistical theory</a></li>
<li class="chapter" data-level="9.8" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#data-types"><i class="fa fa-check"></i><b>9.8</b> Data types</a></li>
<li class="chapter" data-level="9.9" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#differential-expression-analysis-1"><i class="fa fa-check"></i><b>9.9</b> Differential expression analysis</a></li>
<li class="chapter" data-level="9.10" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gene-sets"><i class="fa fa-check"></i><b>9.10</b> Gene sets</a></li>
<li class="chapter" data-level="9.11" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#gokegg-overrepresentation-analysis"><i class="fa fa-check"></i><b>9.11</b> GO/KEGG overrepresentation analysis</a></li>
<li class="chapter" data-level="9.12" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#functional-class-scoring-permutation-testing"><i class="fa fa-check"></i><b>9.12</b> Functional class scoring &amp; permutation testing</a></li>
<li class="chapter" data-level="9.13" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#network-based-enrichment-analysis"><i class="fa fa-check"></i><b>9.13</b> Network-based enrichment analysis</a></li>
<li class="chapter" data-level="9.14" data-path="functional-enrichment-analysis-of-high-throughput-omics-data.html"><a href="functional-enrichment-analysis-of-high-throughput-omics-data.html#genomic-region-enrichment-analysis"><i class="fa fa-check"></i><b>9.14</b> Genomic region enrichment analysis</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><i class="fa fa-check"></i><b>10</b> RNA-seq analysis is easy as 1-2-3 with limma, Glimma and edgeR</a><ul>
<li class="chapter" data-level="10.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#instructor-name-and-contact-information"><i class="fa fa-check"></i><b>10.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="10.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-description-4"><i class="fa fa-check"></i><b>10.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="10.2.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#pre-requisites-7"><i class="fa fa-check"></i><b>10.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="10.2.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-participation-4"><i class="fa fa-check"></i><b>10.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="10.2.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#r-bioconductor-packages-used-8"><i class="fa fa-check"></i><b>10.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="10.2.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#time-outline-8"><i class="fa fa-check"></i><b>10.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#workshop-goals-and-objectives-7"><i class="fa fa-check"></i><b>10.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="10.3.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-goals-5"><i class="fa fa-check"></i><b>10.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="10.3.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#learning-objectives-6"><i class="fa fa-check"></i><b>10.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="10.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#introduction-2"><i class="fa fa-check"></i><b>10.4</b> Introduction</a></li>
<li class="chapter" data-level="10.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-packaging"><i class="fa fa-check"></i><b>10.5</b> Data packaging</a><ul>
<li class="chapter" data-level="10.5.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#reading-in-count-data"><i class="fa fa-check"></i><b>10.5.1</b> Reading in count-data</a></li>
<li class="chapter" data-level="10.5.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-sample-information"><i class="fa fa-check"></i><b>10.5.2</b> Organising sample information</a></li>
<li class="chapter" data-level="10.5.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#organising-gene-annotations"><i class="fa fa-check"></i><b>10.5.3</b> Organising gene annotations</a></li>
</ul></li>
<li class="chapter" data-level="10.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#data-pre-processing"><i class="fa fa-check"></i><b>10.6</b> Data pre-processing</a><ul>
<li class="chapter" data-level="10.6.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#transformations-from-the-raw-scale"><i class="fa fa-check"></i><b>10.6.1</b> Transformations from the raw-scale</a></li>
<li class="chapter" data-level="10.6.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-genes-that-are-lowly-expressed"><i class="fa fa-check"></i><b>10.6.2</b> Removing genes that are lowly expressed</a></li>
<li class="chapter" data-level="10.6.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#normalising-gene-expression-distributions"><i class="fa fa-check"></i><b>10.6.3</b> Normalising gene expression distributions</a></li>
<li class="chapter" data-level="10.6.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#unsupervised-clustering-of-samples"><i class="fa fa-check"></i><b>10.6.4</b> Unsupervised clustering of samples</a></li>
</ul></li>
<li class="chapter" data-level="10.7" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#differential-expression-analysis-2"><i class="fa fa-check"></i><b>10.7</b> Differential expression analysis</a><ul>
<li class="chapter" data-level="10.7.1" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#creating-a-design-matrix-and-contrasts"><i class="fa fa-check"></i><b>10.7.1</b> Creating a design matrix and contrasts</a></li>
<li class="chapter" data-level="10.7.2" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#removing-heteroscedascity-from-count-data"><i class="fa fa-check"></i><b>10.7.2</b> Removing heteroscedascity from count data</a></li>
<li class="chapter" data-level="10.7.3" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#fitting-linear-models-for-comparisons-of-interest"><i class="fa fa-check"></i><b>10.7.3</b> Fitting linear models for comparisons of interest</a></li>
<li class="chapter" data-level="10.7.4" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-the-number-of-de-genes"><i class="fa fa-check"></i><b>10.7.4</b> Examining the number of DE genes</a></li>
<li class="chapter" data-level="10.7.5" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#examining-individual-de-genes-from-top-to-bottom"><i class="fa fa-check"></i><b>10.7.5</b> Examining individual DE genes from top to bottom</a></li>
<li class="chapter" data-level="10.7.6" data-path="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html"><a href="rna-seq-analysis-is-easy-as-1-2-3-with-limma-glimma-and-edger.html#useful-graphical-representations-of-differential-expression-results"><i class="fa fa-check"></i><b>10.7.6</b> Useful graphical representations of differential expression results</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html"><i class="fa fa-check"></i><b>11</b> Solving common bioinformatic challenges using GenomicRanges</a><ul>
<li class="chapter" data-level="11.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#instructors-names-and-contact-information-3"><i class="fa fa-check"></i><b>11.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="11.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-description-5"><i class="fa fa-check"></i><b>11.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="11.2.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#pre-requisites-8"><i class="fa fa-check"></i><b>11.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="11.2.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-participation-5"><i class="fa fa-check"></i><b>11.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="11.2.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#r-bioconductor-packages-used-9"><i class="fa fa-check"></i><b>11.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="11.2.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#time-outline-9"><i class="fa fa-check"></i><b>11.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#workshop-goals-and-objectives-8"><i class="fa fa-check"></i><b>11.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="11.3.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-goals-6"><i class="fa fa-check"></i><b>11.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="11.3.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#learning-objectives-7"><i class="fa fa-check"></i><b>11.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="11.4" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#introduction-3"><i class="fa fa-check"></i><b>11.4</b> Introduction</a><ul>
<li class="chapter" data-level="11.4.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#what-is-the-ranges-infrastructure"><i class="fa fa-check"></i><b>11.4.1</b> What is the Ranges infrastructure?</a></li>
<li class="chapter" data-level="11.4.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#why-use-the-ranges-infrastructure"><i class="fa fa-check"></i><b>11.4.2</b> Why use the Ranges infrastructure?</a></li>
<li class="chapter" data-level="11.4.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#who-is-this-workshop-for"><i class="fa fa-check"></i><b>11.4.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="11.5" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#setup"><i class="fa fa-check"></i><b>11.5</b> Setup</a></li>
<li class="chapter" data-level="11.6" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#granges-genomic-ranges"><i class="fa fa-check"></i><b>11.6</b> <em>GRanges</em>: Genomic Ranges</a></li>
<li class="chapter" data-level="11.7" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#constructing-a-granges-object-from-data.frame"><i class="fa fa-check"></i><b>11.7</b> Constructing a <em>GRanges</em> object from data.frame</a></li>
<li class="chapter" data-level="11.8" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#loading-a-granges-object-from-a-standard-file-format"><i class="fa fa-check"></i><b>11.8</b> Loading a <em>GRanges</em> object from a standard file format</a></li>
<li class="chapter" data-level="11.9" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-manipulation-of-granges-objects"><i class="fa fa-check"></i><b>11.9</b> Basic manipulation of <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="11.10" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#subsetting-granges-objects"><i class="fa fa-check"></i><b>11.10</b> Subsetting <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="11.11" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#splitting-and-combining-granges-objects"><i class="fa fa-check"></i><b>11.11</b> Splitting and combining <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="11.12" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#aggregating-granges-objects"><i class="fa fa-check"></i><b>11.12</b> Aggregating <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="11.13" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#basic-interval-operations-for-granges-objects"><i class="fa fa-check"></i><b>11.13</b> Basic interval operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="11.14" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#interval-set-operations-for-granges-objects"><i class="fa fa-check"></i><b>11.14</b> Interval set operations for <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="11.15" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#finding-overlaps-between-granges-objects"><i class="fa fa-check"></i><b>11.15</b> Finding overlaps between <em>GRanges</em> objects</a></li>
<li class="chapter" data-level="11.16" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises"><i class="fa fa-check"></i><b>11.16</b> Exercises</a></li>
<li class="chapter" data-level="11.17" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>11.17</b> Example: exploring BigWig files from AnnotationHub</a><ul>
<li class="chapter" data-level="11.17.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#extracting-data-from-annotationhub"><i class="fa fa-check"></i><b>11.17.1</b> Extracting data from AnnotationHub</a></li>
<li class="chapter" data-level="11.17.2" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#reading-bigwig-files"><i class="fa fa-check"></i><b>11.17.2</b> Reading BigWig files</a></li>
<li class="chapter" data-level="11.17.3" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-1"><i class="fa fa-check"></i><b>11.17.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="11.18" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#worked-example-coverage-analysis-of-bam-files"><i class="fa fa-check"></i><b>11.18</b> Worked example: coverage analysis of BAM files</a><ul>
<li class="chapter" data-level="11.18.1" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#exercises-2"><i class="fa fa-check"></i><b>11.18.1</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="11.19" data-path="solving-common-bioinformatic-challenges-using-genomicranges.html"><a href="solving-common-bioinformatic-challenges-using-genomicranges.html#conclusions"><i class="fa fa-check"></i><b>11.19</b> Conclusions</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html"><i class="fa fa-check"></i><b>12</b> Fluent genomic data analysis with plyranges</a><ul>
<li class="chapter" data-level="12.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#instructors-names-and-contact-information-4"><i class="fa fa-check"></i><b>12.1</b> Instructor(s) name(s) and contact information</a></li>
<li class="chapter" data-level="12.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-description-6"><i class="fa fa-check"></i><b>12.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="12.2.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#pre-requisites-9"><i class="fa fa-check"></i><b>12.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="12.2.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-participation-6"><i class="fa fa-check"></i><b>12.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="12.2.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#r-bioconductor-packages-used-10"><i class="fa fa-check"></i><b>12.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="12.2.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#time-outline-10"><i class="fa fa-check"></i><b>12.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-goals-and-objectives-9"><i class="fa fa-check"></i><b>12.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="12.3.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-goals-7"><i class="fa fa-check"></i><b>12.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="12.3.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#learning-objectives-8"><i class="fa fa-check"></i><b>12.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#workshop-1"><i class="fa fa-check"></i><b>12.4</b> Workshop</a></li>
<li class="chapter" data-level="12.5" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#introduction-4"><i class="fa fa-check"></i><b>12.5</b> Introduction</a><ul>
<li class="chapter" data-level="12.5.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-is-plyranges"><i class="fa fa-check"></i><b>12.5.1</b> What is plyranges?</a></li>
<li class="chapter" data-level="12.5.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#why-use-plyranges"><i class="fa fa-check"></i><b>12.5.2</b> Why use plyranges?</a></li>
<li class="chapter" data-level="12.5.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#who-is-this-workshop-for-1"><i class="fa fa-check"></i><b>12.5.3</b> Who is this workshop for?</a></li>
</ul></li>
<li class="chapter" data-level="12.6" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#setup-1"><i class="fa fa-check"></i><b>12.6</b> Setup</a></li>
<li class="chapter" data-level="12.7" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#what-are-granges-objects"><i class="fa fa-check"></i><b>12.7</b> What are GRanges objects?</a></li>
<li class="chapter" data-level="12.8" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#the-grammar"><i class="fa fa-check"></i><b>12.8</b> The Grammar</a><ul>
<li class="chapter" data-level="12.8.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#core-verbs"><i class="fa fa-check"></i><b>12.8.1</b> Core verbs</a></li>
<li class="chapter" data-level="12.8.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#verbs-specific-to-granges"><i class="fa fa-check"></i><b>12.8.2</b> Verbs specific to GRanges</a></li>
<li class="chapter" data-level="12.8.3" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#exercises-3"><i class="fa fa-check"></i><b>12.8.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="12.9" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#data-import-and-creating-pipelines"><i class="fa fa-check"></i><b>12.9</b> Data import and creating pipelines</a><ul>
<li class="chapter" data-level="12.9.1" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-exploring-bigwig-files-from-annotationhub"><i class="fa fa-check"></i><b>12.9.1</b> Worked example: exploring BigWig files from AnnotationHub</a></li>
<li class="chapter" data-level="12.9.2" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#worked-example-coverage-analysis-of-bam-files-1"><i class="fa fa-check"></i><b>12.9.2</b> Worked example: coverage analysis of BAM files</a></li>
</ul></li>
<li class="chapter" data-level="12.10" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#whats-next"><i class="fa fa-check"></i><b>12.10</b> What’s next?</a></li>
<li class="chapter" data-level="12.11" data-path="fluent-genomic-data-analysis-with-plyranges.html"><a href="fluent-genomic-data-analysis-with-plyranges.html#appendix"><i class="fa fa-check"></i><b>12.11</b> Appendix</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html"><i class="fa fa-check"></i><b>13</b> Introduction to Bioconductor annotation resources</a><ul>
<li class="chapter" data-level="13.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#instructors"><i class="fa fa-check"></i><b>13.1</b> Instructors</a></li>
<li class="chapter" data-level="13.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-description-7"><i class="fa fa-check"></i><b>13.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="13.2.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#prerequisites"><i class="fa fa-check"></i><b>13.2.1</b> Prerequisites</a></li>
<li class="chapter" data-level="13.2.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-participation-7"><i class="fa fa-check"></i><b>13.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="13.2.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#r-bioconductor-packages-used-11"><i class="fa fa-check"></i><b>13.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
</ul></li>
<li class="chapter" data-level="13.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#workshop-goals-and-objectives-10"><i class="fa fa-check"></i><b>13.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="13.3.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-goals-8"><i class="fa fa-check"></i><b>13.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="13.3.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#learning-objectives-9"><i class="fa fa-check"></i><b>13.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="13.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-workshop"><i class="fa fa-check"></i><b>13.4</b> Annotation Workshop</a><ul>
<li class="chapter" data-level="13.4.1" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#goals-for-this-workshop"><i class="fa fa-check"></i><b>13.4.1</b> Goals for this workshop</a></li>
<li class="chapter" data-level="13.4.2" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#what-do-we-mean-by-annotation"><i class="fa fa-check"></i><b>13.4.2</b> What do we mean by annotation?</a></li>
<li class="chapter" data-level="13.4.3" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#specific-goal"><i class="fa fa-check"></i><b>13.4.3</b> Specific goal</a></li>
<li class="chapter" data-level="13.4.4" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#data-containers"><i class="fa fa-check"></i><b>13.4.4</b> Data containers</a></li>
<li class="chapter" data-level="13.4.5" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset"><i class="fa fa-check"></i><b>13.4.5</b> ExpressionSet</a></li>
<li class="chapter" data-level="13.4.6" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued"><i class="fa fa-check"></i><b>13.4.6</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="13.4.7" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#expressionset-continued-1"><i class="fa fa-check"></i><b>13.4.7</b> ExpressionSet (continued)</a></li>
<li class="chapter" data-level="13.4.8" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bioc-containers-vs-basic-structures"><i class="fa fa-check"></i><b>13.4.8</b> BioC containers vs basic structures</a></li>
<li class="chapter" data-level="13.4.9" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotation-sources"><i class="fa fa-check"></i><b>13.4.9</b> Annotation sources</a></li>
<li class="chapter" data-level="13.4.10" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#interacting-with-annodb-packages"><i class="fa fa-check"></i><b>13.4.10</b> Interacting with AnnoDb packages</a></li>
<li class="chapter" data-level="13.4.11" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#simple-example"><i class="fa fa-check"></i><b>13.4.11</b> Simple example</a></li>
<li class="chapter" data-level="13.4.12" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#questions"><i class="fa fa-check"></i><b>13.4.12</b> Questions!</a></li>
<li class="chapter" data-level="13.4.13" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#more-questions"><i class="fa fa-check"></i><b>13.4.13</b> More questions!</a></li>
<li class="chapter" data-level="13.4.14" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#another-example"><i class="fa fa-check"></i><b>13.4.14</b> Another example</a></li>
<li class="chapter" data-level="13.4.15" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#the-mapids-function"><i class="fa fa-check"></i><b>13.4.15</b> The <code>mapIds</code> function</a></li>
<li class="chapter" data-level="13.4.16" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals"><i class="fa fa-check"></i><b>13.4.16</b> Choices for multiVals</a></li>
<li class="chapter" data-level="13.4.17" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#choices-for-multivals-continued"><i class="fa fa-check"></i><b>13.4.17</b> Choices for multiVals (continued)</a></li>
<li class="chapter" data-level="13.4.18" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#chipdborgdb-questions"><i class="fa fa-check"></i><b>13.4.18</b> ChipDb/OrgDb questions</a></li>
<li class="chapter" data-level="13.4.19" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-packages"><i class="fa fa-check"></i><b>13.4.19</b> TxDb packages</a></li>
<li class="chapter" data-level="13.4.20" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ensdb-packages"><i class="fa fa-check"></i><b>13.4.20</b> EnsDb packages</a></li>
<li class="chapter" data-level="13.4.21" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#transcript-packages"><i class="fa fa-check"></i><b>13.4.21</b> Transcript packages</a></li>
<li class="chapter" data-level="13.4.22" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#granges"><i class="fa fa-check"></i><b>13.4.22</b> GRanges</a></li>
<li class="chapter" data-level="13.4.23" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#grangeslist"><i class="fa fa-check"></i><b>13.4.23</b> GRangesList</a></li>
<li class="chapter" data-level="13.4.24" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#other-accessors"><i class="fa fa-check"></i><b>13.4.24</b> Other accessors</a></li>
<li class="chapter" data-level="13.4.25" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#why-ranges-objects"><i class="fa fa-check"></i><b>13.4.25</b> Why *Ranges objects</a></li>
<li class="chapter" data-level="13.4.26" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#ranges-use-cases"><i class="fa fa-check"></i><b>13.4.26</b> *Ranges use cases</a></li>
<li class="chapter" data-level="13.4.27" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#summarizedexperiment-objects"><i class="fa fa-check"></i><b>13.4.27</b> SummarizedExperiment objects</a></li>
<li class="chapter" data-level="13.4.28" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#txdb-exercises"><i class="fa fa-check"></i><b>13.4.28</b> TxDb exercises</a></li>
<li class="chapter" data-level="13.4.29" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages"><i class="fa fa-check"></i><b>13.4.29</b> OrganismDb packages</a></li>
<li class="chapter" data-level="13.4.30" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-packages-1"><i class="fa fa-check"></i><b>13.4.30</b> OrganismDb packages</a></li>
<li class="chapter" data-level="13.4.31" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organismdb-exercises"><i class="fa fa-check"></i><b>13.4.31</b> OrganismDb exercises</a></li>
<li class="chapter" data-level="13.4.32" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-package"><i class="fa fa-check"></i><b>13.4.32</b> Organism.dplyr package</a></li>
<li class="chapter" data-level="13.4.33" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr"><i class="fa fa-check"></i><b>13.4.33</b> Organism.dplyr</a></li>
<li class="chapter" data-level="13.4.34" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-1"><i class="fa fa-check"></i><b>13.4.34</b> Organism.dplyr</a></li>
<li class="chapter" data-level="13.4.35" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-2"><i class="fa fa-check"></i><b>13.4.35</b> Organism.dplyr</a></li>
<li class="chapter" data-level="13.4.36" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#organism.dplyr-exercises"><i class="fa fa-check"></i><b>13.4.36</b> Organism.dplyr exercises</a></li>
<li class="chapter" data-level="13.4.37" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages"><i class="fa fa-check"></i><b>13.4.37</b> BSgenome packages</a></li>
<li class="chapter" data-level="13.4.38" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-1"><i class="fa fa-check"></i><b>13.4.38</b> BSgenome packages</a></li>
<li class="chapter" data-level="13.4.39" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-packages-2"><i class="fa fa-check"></i><b>13.4.39</b> BSgenome packages</a></li>
<li class="chapter" data-level="13.4.40" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#bsgenome-exercises"><i class="fa fa-check"></i><b>13.4.40</b> BSgenome exercises</a></li>
<li class="chapter" data-level="13.4.41" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-1"><i class="fa fa-check"></i><b>13.4.41</b> AnnotationHub</a></li>
<li class="chapter" data-level="13.4.42" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#querying-annotationhub-1"><i class="fa fa-check"></i><b>13.4.42</b> Querying AnnotationHub</a></li>
<li class="chapter" data-level="13.4.43" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-providers"><i class="fa fa-check"></i><b>13.4.43</b> AnnotationHub Data providers</a></li>
<li class="chapter" data-level="13.4.44" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-classes"><i class="fa fa-check"></i><b>13.4.44</b> AnnotationHub Data classes</a></li>
<li class="chapter" data-level="13.4.45" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-species"><i class="fa fa-check"></i><b>13.4.45</b> AnnotationHub Species</a></li>
<li class="chapter" data-level="13.4.46" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-data-sources"><i class="fa fa-check"></i><b>13.4.46</b> AnnotationHub Data sources</a></li>
<li class="chapter" data-level="13.4.47" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query"><i class="fa fa-check"></i><b>13.4.47</b> AnnotationHub query</a></li>
<li class="chapter" data-level="13.4.48" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-query-1"><i class="fa fa-check"></i><b>13.4.48</b> AnnotationHub query</a></li>
<li class="chapter" data-level="13.4.49" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#selecting-annotationhub-resource"><i class="fa fa-check"></i><b>13.4.49</b> Selecting AnnotationHub resource</a></li>
<li class="chapter" data-level="13.4.50" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#annotationhub-exercises"><i class="fa fa-check"></i><b>13.4.50</b> AnnotationHub exercises</a></li>
<li class="chapter" data-level="13.4.51" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart"><i class="fa fa-check"></i><b>13.4.51</b> biomaRt</a></li>
<li class="chapter" data-level="13.4.52" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-data-sets"><i class="fa fa-check"></i><b>13.4.52</b> biomaRt data sets</a></li>
<li class="chapter" data-level="13.4.53" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-queries"><i class="fa fa-check"></i><b>13.4.53</b> biomaRt queries</a></li>
<li class="chapter" data-level="13.4.54" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-attributes-and-filters"><i class="fa fa-check"></i><b>13.4.54</b> biomaRt attributes and filters</a></li>
<li class="chapter" data-level="13.4.55" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-query"><i class="fa fa-check"></i><b>13.4.55</b> biomaRt query</a></li>
<li class="chapter" data-level="13.4.56" data-path="introduction-to-bioconductor-annotation-resources.html"><a href="introduction-to-bioconductor-annotation-resources.html#biomart-exercises"><i class="fa fa-check"></i><b>13.4.56</b> biomaRt exercises</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><i class="fa fa-check"></i><b>14</b> Workflow for Multi-omics Analysis with MultiAssayExperiment</a><ul>
<li class="chapter" data-level="14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#instructor-names-and-contact-information"><i class="fa fa-check"></i><b>14.1</b> Instructor names and contact information</a></li>
<li class="chapter" data-level="14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-description-8"><i class="fa fa-check"></i><b>14.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="14.2.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#pre-requisites-10"><i class="fa fa-check"></i><b>14.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="14.2.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-participation-8"><i class="fa fa-check"></i><b>14.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="14.2.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rbioconductor-packages-used"><i class="fa fa-check"></i><b>14.2.3</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="14.2.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#time-outline-11"><i class="fa fa-check"></i><b>14.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#workshop-goals-and-objectives-11"><i class="fa fa-check"></i><b>14.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="14.3.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-goals-9"><i class="fa fa-check"></i><b>14.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="14.3.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#learning-objectives-10"><i class="fa fa-check"></i><b>14.3.2</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#overview-of-key-data-classes"><i class="fa fa-check"></i><b>14.4</b> Overview of key data classes</a><ul>
<li class="chapter" data-level="14.4.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#rangedsummarizedexperiment"><i class="fa fa-check"></i><b>14.4.1</b> <code>(Ranged)SummarizedExperiment</code></a></li>
<li class="chapter" data-level="14.4.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#raggedexperiment"><i class="fa fa-check"></i><b>14.4.2</b> <code>RaggedExperiment</code></a></li>
<li class="chapter" data-level="14.4.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment"><i class="fa fa-check"></i><b>14.4.3</b> <code>MultiAssayExperiment</code></a></li>
</ul></li>
<li class="chapter" data-level="14.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-raggedexperiment"><i class="fa fa-check"></i><b>14.5</b> Working with RaggedExperiment</a><ul>
<li class="chapter" data-level="14.5.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#constructing-a-raggedexperiment-object"><i class="fa fa-check"></i><b>14.5.1</b> Constructing a <code>RaggedExperiment</code> object</a></li>
<li class="chapter" data-level="14.5.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-functions"><i class="fa fa-check"></i><b>14.5.2</b> *Assay functions</a></li>
<li class="chapter" data-level="14.5.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#conversion-to-rangedsummarizedexperiment"><i class="fa fa-check"></i><b>14.5.3</b> Conversion to RangedSummarizedExperiment</a></li>
</ul></li>
<li class="chapter" data-level="14.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#working-with-multiassayexperiment"><i class="fa fa-check"></i><b>14.6</b> Working with MultiAssayExperiment</a></li>
<li class="chapter" data-level="14.7" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#api-cheat-sheet"><i class="fa fa-check"></i><b>14.7</b> API cheat sheet</a><ul>
<li class="chapter" data-level="14.7.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-multiassayexperiment-miniacc-demo-object"><i class="fa fa-check"></i><b>14.7.1</b> The MultiAssayExperiment miniACC demo object</a></li>
<li class="chapter" data-level="14.7.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#coldata---information-biological-units"><i class="fa fa-check"></i><b>14.7.2</b> colData - information biological units</a></li>
<li class="chapter" data-level="14.7.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#experimentlist---experiment-data"><i class="fa fa-check"></i><b>14.7.3</b> ExperimentList - experiment data</a></li>
<li class="chapter" data-level="14.7.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#samplemap---relationship-graph"><i class="fa fa-check"></i><b>14.7.4</b> sampleMap - relationship graph</a></li>
<li class="chapter" data-level="14.7.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#metadata"><i class="fa fa-check"></i><b>14.7.5</b> metadata</a></li>
</ul></li>
<li class="chapter" data-level="14.8" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-subsetting"><i class="fa fa-check"></i><b>14.8</b> MultiAssayExperiment Subsetting</a><ul>
<li class="chapter" data-level="14.8.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#single-bracket"><i class="fa fa-check"></i><b>14.8.1</b> Single bracket <code>[</code></a></li>
<li class="chapter" data-level="14.8.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#subsetting-by-genomic-ranges"><i class="fa fa-check"></i><b>14.8.2</b> Subsetting by genomic ranges</a></li>
<li class="chapter" data-level="14.8.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#double-bracket"><i class="fa fa-check"></i><b>14.8.3</b> Double bracket <code>[[</code></a></li>
</ul></li>
<li class="chapter" data-level="14.9" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#complete-cases"><i class="fa fa-check"></i><b>14.9</b> Complete cases</a></li>
<li class="chapter" data-level="14.10" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#row-names-that-are-common-across-assays"><i class="fa fa-check"></i><b>14.10</b> Row names that are common across assays</a></li>
<li class="chapter" data-level="14.11" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#extraction"><i class="fa fa-check"></i><b>14.11</b> Extraction</a><ul>
<li class="chapter" data-level="14.11.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#assay-and-assays"><i class="fa fa-check"></i><b>14.11.1</b> assay and assays</a></li>
</ul></li>
<li class="chapter" data-level="14.12" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#summary-of-slots-and-accessors"><i class="fa fa-check"></i><b>14.12</b> Summary of slots and accessors</a></li>
<li class="chapter" data-level="14.13" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#transformation-reshaping"><i class="fa fa-check"></i><b>14.13</b> Transformation / reshaping</a><ul>
<li class="chapter" data-level="14.13.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#longformat"><i class="fa fa-check"></i><b>14.13.1</b> <code>longFormat</code></a></li>
<li class="chapter" data-level="14.13.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#wideformat"><i class="fa fa-check"></i><b>14.13.2</b> <code>wideFormat</code></a></li>
</ul></li>
<li class="chapter" data-level="14.14" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-class-construction-and-concatenation"><i class="fa fa-check"></i><b>14.14</b> MultiAssayExperiment class construction and concatenation</a><ul>
<li class="chapter" data-level="14.14.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multiassayexperiment-constructor-function"><i class="fa fa-check"></i><b>14.14.1</b> MultiAssayExperiment constructor function</a></li>
<li class="chapter" data-level="14.14.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#prepmultiassay---constructor-function-helper"><i class="fa fa-check"></i><b>14.14.2</b> <code>prepMultiAssay</code> - Constructor function helper</a></li>
<li class="chapter" data-level="14.14.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#c---concatenate-to-multiassayexperiment"><i class="fa fa-check"></i><b>14.14.3</b> <code>c</code> - concatenate to MultiAssayExperiment</a></li>
<li class="chapter" data-level="14.14.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#building-a-multiassayexperiment-from-scratch"><i class="fa fa-check"></i><b>14.14.4</b> Building a MultiAssayExperiment from scratch</a></li>
</ul></li>
<li class="chapter" data-level="14.15" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#the-cancer-genome-atlas-tcga-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>14.15</b> The Cancer Genome Atlas (TCGA) as MultiAssayExperiment objects</a></li>
<li class="chapter" data-level="14.16" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#utilities-for-tcga"><i class="fa fa-check"></i><b>14.16</b> Utilities for TCGA</a><ul>
<li class="chapter" data-level="14.16.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#simplification-of-curatedtcgadata-objects"><i class="fa fa-check"></i><b>14.16.1</b> “Simplification” of <code>curatedTCGAData</code> objects</a></li>
<li class="chapter" data-level="14.16.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#what-types-of-samples-are-in-the-data"><i class="fa fa-check"></i><b>14.16.2</b> What types of samples are in the data?</a></li>
<li class="chapter" data-level="14.16.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#curated-molecular-subtypes"><i class="fa fa-check"></i><b>14.16.3</b> Curated molecular subtypes</a></li>
<li class="chapter" data-level="14.16.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#converting-tcga-uuids-to-barcodes-and-back"><i class="fa fa-check"></i><b>14.16.4</b> Converting TCGA UUIDs to barcodes and back</a></li>
<li class="chapter" data-level="14.16.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#other-tcga-data-types"><i class="fa fa-check"></i><b>14.16.5</b> Other TCGA data types</a></li>
</ul></li>
<li class="chapter" data-level="14.17" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#plotting-correlation-and-other-analyses"><i class="fa fa-check"></i><b>14.17</b> Plotting, correlation, and other analyses</a><ul>
<li class="chapter" data-level="14.17.1" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#how-many-miniacc-samples-have-data-for-each-combination-of-assays"><i class="fa fa-check"></i><b>14.17.1</b> How many <code>miniACC</code> samples have data for each combination of assays?</a></li>
<li class="chapter" data-level="14.17.2" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#kaplan-meier-plot-stratified-by-pathology_n_stage"><i class="fa fa-check"></i><b>14.17.2</b> Kaplan-meier plot stratified by pathology_N_stage</a></li>
<li class="chapter" data-level="14.17.3" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#multivariate-cox-regression-including-rna-seq-copy-number-and-pathology"><i class="fa fa-check"></i><b>14.17.3</b> Multivariate Cox regression including RNA-seq, copy number, and pathology</a></li>
<li class="chapter" data-level="14.17.4" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#correlation-between-rna-seq-and-copy-number"><i class="fa fa-check"></i><b>14.17.4</b> Correlation between RNA-seq and copy number</a></li>
<li class="chapter" data-level="14.17.5" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#identifying-correlated-principal-components"><i class="fa fa-check"></i><b>14.17.5</b> Identifying correlated principal components</a></li>
<li class="chapter" data-level="14.17.6" data-path="workflow-for-multi-omics-analysis-with-multiassayexperiment.html"><a href="workflow-for-multi-omics-analysis-with-multiassayexperiment.html#annotating-with-ranges"><i class="fa fa-check"></i><b>14.17.6</b> Annotating with ranges</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="15" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><i class="fa fa-check"></i><b>15</b> Biomarker discovery from large pharmacogenomics datasets</a><ul>
<li class="chapter" data-level="15.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#instructors-1"><i class="fa fa-check"></i><b>15.1</b> Instructors:</a></li>
<li class="chapter" data-level="15.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-description-9"><i class="fa fa-check"></i><b>15.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="15.2.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#pre-requisites-11"><i class="fa fa-check"></i><b>15.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="15.2.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-participation-9"><i class="fa fa-check"></i><b>15.2.2</b> Workshop Participation</a></li>
<li class="chapter" data-level="15.2.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#r-bioconductor-packages-used-12"><i class="fa fa-check"></i><b>15.2.3</b> <em>R</em> / <em>Bioconductor</em> packages used</a></li>
<li class="chapter" data-level="15.2.4" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#time-outline-12"><i class="fa fa-check"></i><b>15.2.4</b> Time outline</a></li>
</ul></li>
<li class="chapter" data-level="15.3" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#workshop-goals-and-objectives-12"><i class="fa fa-check"></i><b>15.3</b> Workshop goals and objectives</a><ul>
<li class="chapter" data-level="15.3.1" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-goals-10"><i class="fa fa-check"></i><b>15.3.1</b> Learning goals</a></li>
<li class="chapter" data-level="15.3.2" data-path="biomarker-discovery-from-large-pharmacogenomics-datasets.html"><a href="biomarker-discovery-from-large-pharmacogenomics-datasets.html#learning-objectives-11"><i class="fa fa-check"></i><b>15.3.2</b> Learning objectives</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="16" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html"><i class="fa fa-check"></i><b>16</b> Maintaining your Bioconductor package</a><ul>
<li class="chapter" data-level="16.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#instructor-name-and-contact-information-1"><i class="fa fa-check"></i><b>16.1</b> Instructor name and contact information</a></li>
<li class="chapter" data-level="16.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-description-10"><i class="fa fa-check"></i><b>16.2</b> Workshop Description</a><ul>
<li class="chapter" data-level="16.2.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#pre-requisites-12"><i class="fa fa-check"></i><b>16.2.1</b> Pre-requisites</a></li>
<li class="chapter" data-level="16.2.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#participation-4"><i class="fa fa-check"></i><b>16.2.2</b> Participation</a></li>
<li class="chapter" data-level="16.2.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#time-outline-50-mins-short-workshop"><i class="fa fa-check"></i><b>16.2.3</b> Time outline: 50 mins short workshop</a></li>
<li class="chapter" data-level="16.2.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#workshop-goals-and-objectives-13"><i class="fa fa-check"></i><b>16.2.4</b> Workshop goals and objectives</a></li>
</ul></li>
<li class="chapter" data-level="16.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#introduction-6"><i class="fa fa-check"></i><b>16.3</b> Introduction</a></li>
<li class="chapter" data-level="16.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#why-maintain-your-package"><i class="fa fa-check"></i><b>16.4</b> Why maintain your package</a></li>
<li class="chapter" data-level="16.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#infrastructure-used-for-maintenance"><i class="fa fa-check"></i><b>16.5</b> Infrastructure used for maintenance</a></li>
<li class="chapter" data-level="16.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#outline-how-to-use-infrastructure"><i class="fa fa-check"></i><b>16.6</b> Outline how to use infrastructure</a><ul>
<li class="chapter" data-level="16.6.1" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#build-report"><i class="fa fa-check"></i><b>16.6.1</b> Build report</a></li>
<li class="chapter" data-level="16.6.2" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#landing-page-badges"><i class="fa fa-check"></i><b>16.6.2</b> Landing page badges</a></li>
<li class="chapter" data-level="16.6.3" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#download-statistics"><i class="fa fa-check"></i><b>16.6.3</b> Download statistics</a></li>
<li class="chapter" data-level="16.6.4" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#rss-feeds"><i class="fa fa-check"></i><b>16.6.4</b> RSS feeds</a></li>
<li class="chapter" data-level="16.6.5" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#support-site-bioc-devel-mailing-list"><i class="fa fa-check"></i><b>16.6.5</b> Support site, Bioc-devel mailing list</a></li>
<li class="chapter" data-level="16.6.6" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#github"><i class="fa fa-check"></i><b>16.6.6</b> GitHub</a></li>
</ul></li>
<li class="chapter" data-level="16.7" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#round-up-of-resources-available"><i class="fa fa-check"></i><b>16.7</b> Round up of resources available</a></li>
<li class="chapter" data-level="16.8" data-path="maintaining-your-bioconductor-package.html"><a href="maintaining-your-bioconductor-package.html#acknowledgements"><i class="fa fa-check"></i><b>16.8</b> Acknowledgements</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html"><i class="fa fa-check"></i><b>17</b> Public Data Resources and Bioconductor</a><ul>
<li class="chapter" data-level="17.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#instructor-names-and-contact-information-1"><i class="fa fa-check"></i><b>17.1</b> Instructor name(s) and contact information</a></li>
<li class="chapter" data-level="17.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#syllabus"><i class="fa fa-check"></i><b>17.2</b> Syllabus</a><ul>
<li class="chapter" data-level="17.2.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-description-11"><i class="fa fa-check"></i><b>17.2.1</b> Workshop Description</a></li>
<li class="chapter" data-level="17.2.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pre-requisites-13"><i class="fa fa-check"></i><b>17.2.2</b> Pre-requisites</a></li>
<li class="chapter" data-level="17.2.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-participation-10"><i class="fa fa-check"></i><b>17.2.3</b> Workshop Participation</a></li>
<li class="chapter" data-level="17.2.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#rbioconductor-packages-used-1"><i class="fa fa-check"></i><b>17.2.4</b> R/Bioconductor packages used</a></li>
<li class="chapter" data-level="17.2.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#time-outline-13"><i class="fa fa-check"></i><b>17.2.5</b> Time outline</a></li>
<li class="chapter" data-level="17.2.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#workshop-goals-and-objectives-14"><i class="fa fa-check"></i><b>17.2.6</b> Workshop goals and objectives</a></li>
<li class="chapter" data-level="17.2.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-goals-11"><i class="fa fa-check"></i><b>17.2.7</b> Learning goals</a></li>
<li class="chapter" data-level="17.2.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#learning-objectives-12"><i class="fa fa-check"></i><b>17.2.8</b> Learning objectives</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-6"><i class="fa fa-check"></i><b>17.3</b> Overview</a></li>
<li class="chapter" data-level="17.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#geoquery"><i class="fa fa-check"></i><b>17.4</b> GEOquery</a><ul>
<li class="chapter" data-level="17.4.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-geo"><i class="fa fa-check"></i><b>17.4.1</b> Overview of GEO</a></li>
<li class="chapter" data-level="17.4.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-using-geoquery"><i class="fa fa-check"></i><b>17.4.2</b> Getting Started using GEOquery</a></li>
<li class="chapter" data-level="17.4.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#use-case-mds-plot-of-cancer-data"><i class="fa fa-check"></i><b>17.4.3</b> Use case: MDS plot of cancer data</a></li>
<li class="chapter" data-level="17.4.4" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-raw-data-from-geo"><i class="fa fa-check"></i><b>17.4.4</b> Accessing Raw Data from GEO</a></li>
</ul></li>
<li class="chapter" data-level="17.5" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#genomicdatacommons"><i class="fa fa-check"></i><b>17.5</b> GenomicDataCommons</a><ul>
<li class="chapter" data-level="17.5.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#quickstart"><i class="fa fa-check"></i><b>17.5.1</b> Quickstart</a></li>
<li class="chapter" data-level="17.5.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#basic-design"><i class="fa fa-check"></i><b>17.5.2</b> Basic design</a></li>
<li class="chapter" data-level="17.5.3" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#usage"><i class="fa fa-check"></i><b>17.5.3</b> Usage</a></li>
</ul></li>
<li class="chapter" data-level="17.6" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#querying-metadata"><i class="fa fa-check"></i><b>17.6</b> Querying metadata</a><ul>
<li class="chapter" data-level="17.6.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#authentication"><i class="fa fa-check"></i><b>17.6.1</b> Authentication</a></li>
<li class="chapter" data-level="17.6.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#datafile-access-and-download"><i class="fa fa-check"></i><b>17.6.2</b> Datafile access and download</a></li>
</ul></li>
<li class="chapter" data-level="17.7" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#sequence-read-archive"><i class="fa fa-check"></i><b>17.7</b> Sequence Read Archive</a></li>
<li class="chapter" data-level="17.8" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#accessing-the-cancer-genome-atlas-tcga"><i class="fa fa-check"></i><b>17.8</b> Accessing The Cancer Genome Atlas (TCGA)</a><ul>
<li class="chapter" data-level="17.8.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#tcgabiolinks"><i class="fa fa-check"></i><b>17.8.1</b> TCGAbiolinks</a></li>
<li class="chapter" data-level="17.8.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedtcgadata-curated-data-from-the-cancer-genome-atlas-as-multiassayexperiment-objects"><i class="fa fa-check"></i><b>17.8.2</b> curatedTCGAData: Curated Data From The Cancer Genome Atlas as MultiAssayExperiment Objects</a></li>
</ul></li>
<li class="chapter" data-level="17.9" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#recount-reproducible-rna-seq-analysis-using-recount2"><i class="fa fa-check"></i><b>17.9</b> recount: Reproducible RNA-seq Analysis Using recount2</a></li>
<li class="chapter" data-level="17.10" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curateddata-packages-for-standardized-cancer-transcriptomes"><i class="fa fa-check"></i><b>17.10</b> curated*Data packages for standardized cancer transcriptomes</a></li>
<li class="chapter" data-level="17.11" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#microbiome-data"><i class="fa fa-check"></i><b>17.11</b> Microbiome data</a><ul>
<li class="chapter" data-level="17.11.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#curatedmetagenomicdata-curated-and-processed-metagenomic-data-through-experimenthub"><i class="fa fa-check"></i><b>17.11.1</b> curatedMetagenomicData: Curated and processed metagenomic data through ExperimentHub</a></li>
<li class="chapter" data-level="17.11.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#hmp16sdata-16s-rrna-sequencing-data-from-the-human-microbiome-project"><i class="fa fa-check"></i><b>17.11.2</b> HMP16SData: 16S rRNA Sequencing Data from the Human Microbiome Project</a></li>
</ul></li>
<li class="chapter" data-level="17.12" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#pharmacogenomics"><i class="fa fa-check"></i><b>17.12</b> Pharmacogenomics</a><ul>
<li class="chapter" data-level="17.12.1" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#getting-started-2"><i class="fa fa-check"></i><b>17.12.1</b> Getting started</a></li>
<li class="chapter" data-level="17.12.2" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#overview-of-pharmacogx-datasets-pharmacosets"><i class="fa fa-check"></i><b>17.12.2</b> Overview of PharmacoGx datasets (PharmacoSets)</a></li>
</ul></li>
<li class="chapter" data-level="17.13" data-path="public-data-resources-and-bioconductor.html"><a href="public-data-resources-and-bioconductor.html#bibliography"><i class="fa fa-check"></i><b>17.13</b> Bibliography</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://bioc2018.bioconductor.org/" target="blank">Bioconductor Conference</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Bioconductor 2018 Workshop Compilation</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="working-with-genomic-data-in-r-with-the-decipher-package" class="section level1">
<h1><span class="header-section-number">8</span> Working with Genomic Data in R with the DECIPHER package</h1>
<p>Authors: Nicholas Cooley<a href="#fn18" class="footnoteRef" id="fnref18"><sup>18</sup></a> Last Modified: 18 July, 2018</p>
<!--
  bookdown::render_book("", "bookdown::gitbook")
-->
<div id="overview-5" class="section level2">
<h2><span class="header-section-number">8.1</span> Overview</h2>
<div id="workshop-description-2" class="section level3">
<h3><span class="header-section-number">8.1.1</span> Workshop Description</h3>
<p>In this workshop we will give an introduction to working with biological sequence data in R using the Biostrings and DECIPHER packages. We will cover:</p>
<ul>
<li>Importing, viewing, and manipulating genomes in R</li>
<li>Construction of sequence databases for organizing genomes</li>
<li>Mapping syntenic regions between genomes with the FindSynteny function</li>
<li>Understanding, accessing, and viewing objects of class Synteny</li>
<li>Using syntenic information to predict orthologous genes</li>
<li>Alignment of sequences and genomes with DECIPHER</li>
<li>Downstream analyses enabled by syntenic mapping</li>
</ul>
</div>
<div id="pre-requisites-5" class="section level3">
<h3><span class="header-section-number">8.1.2</span> Pre-requisites</h3>
<ul>
<li>Familiarity with Biostrings</li>
<li>Familiarity with DECIPHER Databases<a href="#fn19" class="footnoteRef" id="fnref19"><sup>19</sup></a></li>
</ul>
</div>
<div id="workshop-participation-2" class="section level3">
<h3><span class="header-section-number">8.1.3</span> Workshop Participation</h3>
<p>This will be a lab where participants follow along on their computers.</p>
</div>
<div id="r-bioconductor-packages-used-6" class="section level3">
<h3><span class="header-section-number">8.1.4</span> <em>R</em> / <em>Bioconductor</em> packages used</h3>
<ul>
<li>Biostrings</li>
<li>DECIPHER</li>
<li>stringr</li>
<li>FindHomology</li>
</ul>
</div>
<div id="time-outline-6" class="section level3">
<h3><span class="header-section-number">8.1.5</span> Time outline</h3>
<table>
<thead>
<tr class="header">
<th>Activity</th>
<th>Time</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Packages/Introduction</td>
<td>5m</td>
</tr>
<tr class="even">
<td>Basic commands for sequences</td>
<td>5m</td>
</tr>
<tr class="odd">
<td>Sequence databases</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Demonstration of synteny mapping</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>Explanation of function arguments</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Dissecting Synteny objects</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>Visualization of syntenic blocks</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Alignment of syntenic regions</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>Ortholog prediction from Synteny</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Constructing phylogenetic trees</td>
<td>10m</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="workshop-goals-and-objectives-6" class="section level2">
<h2><span class="header-section-number">8.2</span> Workshop goals and objectives</h2>
<div id="learning-goals-4" class="section level3">
<h3><span class="header-section-number">8.2.1</span> Learning goals</h3>
<ul>
<li>Understand a simple workflow for analysis of sequences in R and DECIPHER</li>
<li>Learn the basic use and appropriate application of functions within DECIPHER</li>
</ul>
</div>
<div id="learning-objectives-5" class="section level3">
<h3><span class="header-section-number">8.2.2</span> Learning objectives</h3>
<ul>
<li>Learn basic commands for working with sequences in R</li>
<li>Import genomes from online repositories or local files</li>
<li>Map synteny between genomes</li>
<li>Analyze a synteny map among multiple genomes</li>
<li>Develop an understanding of the data structures involved</li>
<li>Predict orthologs from syntenic maps</li>
<li>Select core and pan genomes from predicted orthologs</li>
<li>Construct and interpret phylogenetic trees</li>
</ul>
<div id="introduction-1" class="section level4">
<h4><span class="header-section-number">8.2.2.1</span> Introduction</h4>
<p>In this worshop we will be walking through a comparative genomics pipeline using functions within the R package DECIPHER, and within a package that is currently under construction under the name FindHomology.</p>
</div>
<div id="context" class="section level4">
<h4><span class="header-section-number">8.2.2.2</span> Context</h4>
<p>Lipid membranes are ubiquitious in living organisms, however the individual lipid molecules that compose membranes have a wide variety of structures. The biosynthesis of common lipid molecules is well understood, conversely the biosynthesis of more exotic lipid molecules is not. The genomes selected today are from the archaeal phylum Thaumarchaeota, which are known for being anammox archaea, and are important in the nitrogen cycle. In addition to that, these archaea all produce a series of unusual lipids. Since these lipids are shared across the entire phylum from which these genomes are pulled, it is possible that the biosynthetic genes responsible for lipid construction are shared across the selected genomes, and part of their ‘core’ genome.</p>
<p>Archaea are relatively newly characterized as organisms. First classified in the ’70s<a href="#fn20" class="footnoteRef" id="fnref20"><sup>20</sup></a>, for their deviations from Prokaryotes, many Archaea are noteable for being isolated from extreme environments. Despite the colorful beginnings of their classification, not all Archaea are extremophiles.</p>
<div class="figure" style="text-align: center">
<img src="Cooley_DECIPHER/GrandPrismatic.png" alt="Grand Prismatic Spring in Yellowstone National Park[^3]" width="75%" height="75%" />
<p class="caption">
(#fig:Grand Prismatic)Grand Prismatic Spring in Yellowstone National Park<a href="#fn21" class="footnoteRef" id="fnref21"><sup>21</sup></a>
</p>
</div>
<p>The aforementioned unusual lipds<a href="#fn22" class="footnoteRef" id="fnref22"><sup>22</sup></a> are striking in their differences from those common in both Eukaryotes and Prokaryotes.</p>
<div class="figure" style="text-align: center"><span id="fig:Lipids"></span>
<img src="Cooley_DECIPHER/ArchaealLipids.png" alt="A. A generic lipid with phosphatidyl serine head group B. A generic Archaeal lipid C. A more exotic Archaeal lipid" width="50%" height="50%" />
<p class="caption">
Figure 8.1: A. A generic lipid with phosphatidyl serine head group B. A generic Archaeal lipid C. A more exotic Archaeal lipid
</p>
</div>
<p>Whereas Lipid construction in eucharyotes and prokaryotes relies on iterative incorporation of malonyl-CoA units, archaea utilize the melvonate pathway to construct their lipids through iterative incorporation of dimethylallyl pyrophosphate. Which is responsible for the iterative methyl groups extant to the lipid chain in <strong>B</strong> above. Archaeal lipids are also notable for their lack of ester linkages between the alkyl chain, and glyerol. Even more interesting, and as yet uncharacterized, Archaea incorporate a range of ring structures into their lipids, and additionally appear to <em>fuse lipid tails together in the bilayer</em>, as shown above in <strong>C</strong>.</p>
</div>
<div id="the-above-is-interesting-but-so-what" class="section level4">
<h4><span class="header-section-number">8.2.2.3</span> The above is interesting, but so what?</h4>
<p>Our goal with this workshop is to show how functions within DECIPHER, in addition to those found within FindHomology can be used to predict homologs between genomes, as well as predict the core genome of a given set of genomes. As stated above, the chemistry and biology involved in some of the construction of archaeal lipids is unknown. Hypothetically, as this chemistry is universal in these organisms, the genes responsible could all be homologs, and be identified as part of the core genome. Further we can compare the core and pan genomes of this set.</p>
<p>To begin with, we will load the required packages: DECIPHER and FindHomology will provide tools to complete the phylogenetic analysis, while phytools will provide visualization tools, specifically for the visualization of tangelograms, and stringr is used for a specific string operation.</p>
</div>
<div id="packages-and-sequences" class="section level4">
<h4><span class="header-section-number">8.2.2.4</span> Packages and Sequences</h4>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">suppressMessages</span>(<span class="kw">library</span>(DECIPHER))
<span class="kw">library</span>(FindHomology)
<span class="kw">library</span>(stringr)</code></pre></div>
</div>
<div id="genomes-and-decipher-databases" class="section level4">
<h4><span class="header-section-number">8.2.2.5</span> Genomes, and DECIPHER databases</h4>
<p>The first step in this process is selecting the genomes we want to work with. There are several ways to do this, especially when accessing publically available genomes through the NCBI. Entrez Direct<a href="#fn23" class="footnoteRef" id="fnref23"><sup>23</sup></a> provides powerful unix command line tools, if that’s your flavor, but the NCBI Genome List, and other search tools are easily accessible, and have a lower barrier to to entry.</p>
<p>For the purposes of this workshop, we will be using a set of preselected genomes that will be accessed via ftp addresses. These are loaded with the FindHomology Package and can be accessed easily.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">data</span>(<span class="st">&quot;GeneCallAdds&quot;</span>)
<span class="co">#&gt; Warning in data(&quot;GeneCallAdds&quot;): data set &#39;GeneCallAdds&#39; not found</span>
<span class="kw">data</span>(<span class="st">&quot;GenomeAdds&quot;</span>)
<span class="co">#&gt; Warning in data(&quot;GenomeAdds&quot;): data set &#39;GenomeAdds&#39; not found</span>
<span class="kw">data</span>(<span class="st">&quot;GenomeIDs&quot;</span>)
<span class="co">#&gt; Warning in data(&quot;GenomeIDs&quot;): data set &#39;GenomeIDs&#39; not found</span></code></pre></div>
<p>We have now loaded in a character vector for ftp addresses for genomic fasta files <code>GenomeAdds</code>, a character vector of ftp addresses for genomic GFF files <code>GeneCallAdds</code> that we will be using to collect gene boundaries, strandedness, and annotations. And lastly a vector of abbreviated names for the genomes we are working with.</p>
<p>FindHomology contains a simple parser function for GFF files that allows us to grab gene boundaries and annotations.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">GeneCalls &lt;-<span class="st"> </span>FindHomology<span class="op">::</span><span class="kw">GFFParser</span>(<span class="dt">GFFAddress =</span> GeneCallAdds,
                                     <span class="dt">Verbose =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; ===========================================================================</span>
<span class="co">#&gt; Time difference of 7.620104 secs</span></code></pre></div>
<p>If you choose to mimic this work on your own, there are a few ways to get gene calls(the gene boundaries, and strandedness), and annotations, such as Prodigal<a href="#fn24" class="footnoteRef" id="fnref24"><sup>24</sup></a> (for gene calls) and Prokka<a href="#fn25" class="footnoteRef" id="fnref25"><sup>25</sup></a> (for annotations). Using the NCBI annotations is simple and conducive for this workshop. However if you are working on sequence data that you have generated or collected yourself, NCBI annotations will not be available. Additionally, a Gene Caller within the DECIPHER package is currently under construction.</p>
</div>
<div id="decipher-and-databases" class="section level4">
<h4><span class="header-section-number">8.2.2.6</span> DECIPHER and Databases</h4>
<p>DECIPHER works with sequence data through SQL databases. There are several reasons for this.</p>
<div class="figure" style="text-align: center"><span id="fig:DataBases"></span>
<img src="Cooley_DECIPHER/DataBases.png" alt="Databases are compressable and easily shareable" width="75%" height="75%" />
<p class="caption">
Figure 8.2: Databases are compressable and easily shareable
</p>
</div>
<p>Databases provide large improvements in file organization. Currently, a personalized analysis pipeline could include anywhere from one to tens of programs, all of which may take in similar data, or outputs of previous steps, or may output more files than a user needs. DECIPHER’s use of databases allows for sequence data to be stored in an organized format that eliminates this kind of redundancy.</p>
<p>Use of databases also allows for a non-destructive workflow. Sequence data in the database can be accessed as needed, but cannot be accidentally manipulated. Databases are also shareable, allowing for accession by multiple users at once.</p>
<p>Additionally, constructing a database in DECIPHER is relatively easy.</p>
<p>In your own work, it can be inadvisable to use <code>tempfile()</code> to specify the location of your .sqlite file, especially if it will be a file you use repeatedly, or takes a significant amount of time to create, or that you will eventually need to share with collaborators. However for this workhshop, it is fine.</p>
<p>We can access our databases through either a database connection, or simply a filepath to the database. In this code chunk below, we will use a database connection.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
DBPath &lt;-<span class="st"> </span><span class="kw">tempfile</span>()

DBConn &lt;-<span class="st"> </span><span class="kw">dbConnect</span>(<span class="kw">SQLite</span>(),
                    DBPath)

<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_along</span>(GenomeAdds)) {
  <span class="kw">Seqs2DB</span>(<span class="dt">seqs =</span> GenomeAdds[i],
          <span class="dt">type =</span> <span class="st">&quot;FASTA&quot;</span>,
          <span class="dt">dbFile =</span> DBConn,
          <span class="dt">identifier =</span> <span class="kw">as.character</span>(i),
          <span class="dt">tblName =</span> <span class="st">&quot;Seqs&quot;</span>,
          <span class="dt">verbose =</span> <span class="ot">FALSE</span>)
}

<span class="kw">dbDisconnect</span>(DBConn)</code></pre></div>
<p>Identifying details of the database can be viewed in your default browser: <code>BrowseDB(DBPath)</code></p>
</div>
<div id="comparison-of-genomes" class="section level4">
<h4><span class="header-section-number">8.2.2.7</span> Comparison of genomes</h4>
<p>Comparison of genomes will be accomplished with Synteny. Generically when regions in a genome are the same, they are syntenic matches. Often observing this phenomenon is accomplished through matching runs of reciprocal best blast hits. If in genome 1, genes A-B-C are reciprocal best blast hits to genes Z-Y-X in genome 2, the region encompassing those genes would be considered syntenic.</p>
<p>In DECIPHER this is accomplished using k-mer matching, runs of an exact match of length k are identified and recorded as a syntenic hit.</p>
<div class="figure" style="text-align: center">
<img src="Cooley_DECIPHER/SyntenyHits.png" alt="A syntenic hit in nucleotide space" width="100%" height="100%" />
<p class="caption">
(#fig:Simple Synteny Example)A syntenic hit in nucleotide space
</p>
</div>
<p>We can visualize syntenic hits as above, with co-linear representations of genomes with some graphical links to signify the locations and lengths of syntenic hits. Or we can employ a dotplot, which utilizes the cumulative nucleotide positions of two genomes as the x and y axes in a generic plot.</p>
<div class="figure" style="text-align: center">
<img src="Cooley_DECIPHER/SyntenyBlocks.png" alt="A syntenic hit, and several hits being chained toegher into a syntenic block" width="100%" height="100%" />
<p class="caption">
(#fig:Simple DotPlot)A syntenic hit, and several hits being chained toegher into a syntenic block
</p>
</div>
<p>Syntenic Hits can be chained together, if they are close <em>enough</em>, to form larger syntenic blocks. The enough part, has to be measured in nucleotide distance between the hits in question, in both genomes. This also means that syntenic hits can be chained together with differing distances between them in each genome. Making syntenic blocks different lengths in corresponding genomes. Additionally, chaining of hits allows for utilizing smaller syntenic hits that may appear insignificant on their own, to be grouped in with other nearby hits to increase their significance.</p>
<p>DECIPHER’s <code>FindSynteny</code> function allows users to control the degree to which hits and blocks are recorded through the argument <code>minScore</code>, or a minimum score required for a block, or hit, to be kept as a real syntenic match.</p>
<p>In this workflow, we will be determining Synteny between genomes before we consider the gene calls we have already collected. Which stands in contrast to how many current synteny finding programs work. In the code below <code>FindSyntey()</code> will perform pairways analysis of every set of genomes provided to it.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="co"># Here we will use a file path instead of a dbconnection</span>
SyntenyObject &lt;-<span class="st"> </span><span class="kw">FindSynteny</span>(<span class="dt">dbFile =</span> DBPath,
                             <span class="dt">verbose =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; ===========================================================================</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; Time difference of 52.94 secs</span></code></pre></div>
<p>There are multiple ways to view objects of class <code>synteny</code>. It has default view options for <code>plot()</code>. By default, colors are assigned as a ramp from the first nucleotide position, to the last nucleotide position of the <em>query</em> genome, or the first genome selected, and wherever syntenic hits are observed, the bar representing the <em>subject</em> genome is marked with the section of the color ramp indicating the syntenic hit in the <em>query</em>.</p>
<p>Plotting genomes 2 and 3, which are very syntenic, provides a good example of this. Grey uncolored space in the subject genome indicates a lack of syntenic blocks in that region. In some places, the ramp reverses, indicating an inversion, while in others, blocks are reordered from the default color bar, indicating possible rearrangements.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(SyntenyObject[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">2</span><span class="op">:</span><span class="dv">3</span>])</code></pre></div>
<p><img src="Cooley_DECIPHER_files/figure-html/plot%20default-1.png" width="672" /></p>
<p>Because visualizing large numbers of syntenic hits in genome sized sequences can be non-trivial, additional commands can be passed to plot in order to modify coloring schemes. <code>frequency</code> causes bars representing genomes to be colored by <strong>how syntenic</strong> that particular region (as determined by cumulative nucleotide position) of the genome is, indiscriminate of where that hit is in a corresponding genome. Once again genomes 2 and 3 in our set are good examples of this. They are highly syntenic, which this option shows well, but the rearrangements present are no longer shown.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(SyntenyObject[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],
     <span class="st">&quot;frequency&quot;</span>)</code></pre></div>
<p><img src="Cooley_DECIPHER_files/figure-html/plot%20frequency-1.png" width="672" /></p>
<p>Because visualizing large numbers of syntenic hits in genome sized sequences can be non-trivial, additional commands can be passed to plot in order to modify coloring schemes. <code>neighbor</code> tries to provide each syntenic hit with a unique color, and additionally draws linear connections between them to facilitate better visualization. This can be a useful alternative to the default plotting, if many rearrangements are present in a way that you want to view.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(SyntenyObject[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">2</span><span class="op">:</span><span class="dv">3</span>],
     <span class="st">&quot;neighbor&quot;</span>)</code></pre></div>
<p><img src="Cooley_DECIPHER_files/figure-html/plot%20neighbors-1.png" width="672" /></p>
<p>Plotting is a useful tool in analzying sequence data, however sometimes plots of genome sized sequence data can be a bit overwhelming. Plotting our entire database, 8 sequences, though good at representing the complexity present here in this workshop, can be daunting to make sense of quickly if you are unused to looking at figures of this type frequently.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">plot</span>(SyntenyObject,
     <span class="st">&quot;neighbor&quot;</span>)</code></pre></div>
<p><img src="Cooley_DECIPHER_files/figure-html/plot%20neighbors%202-1.png" width="672" /></p>
<p>The most useful way to visualize Synteny data <em>in my opinion</em> is through a dot plot, which can be accessed via the <code>pairs()</code>. Dot plots, as explained above, utilize each genome being compared, as an axis in a normal x-y scatter plot. Where a <em>dot</em> - really a <strong>very</strong> short line, representing a run of nucleotides or amino acids - is shown, that x,y coordinate is a syntenic hit. Hits can also be chained together into blocks of syntenic hits.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pairs</span>(SyntenyObject[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">2</span><span class="op">:</span><span class="dv">3</span>], <span class="dt">labels =</span> GenomeIDs[<span class="dv">2</span><span class="op">:</span><span class="dv">3</span>])</code></pre></div>
<p><img src="Cooley_DECIPHER_files/figure-html/small%20pairs%20plot-1.png" width="672" /></p>
<p>Syntenic hits are shown in the upper left, while syntenic blocks are shown in the bottom left. In the upper triangle,hits in black are co-linear, while hits in red are inverted. In the bottom triangle, blocks are colored by score.</p>
<p>Pairs plots are <em>my</em> personal favorite way to visualize syntenic information, but, they can become overwhelming as the number of genomes compared increases. They provide very clear depictions of rearrangements and inversions.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">pairs</span>(SyntenyObject,
      <span class="dt">labels =</span> GenomeIDs)</code></pre></div>
<p><img src="Cooley_DECIPHER_files/figure-html/large%20pairs%20plot-1.png" width="672" /></p>
</div>
<div id="predict-homology-between-genes-that-are-linked-by-syntenic-hits" class="section level4">
<h4><span class="header-section-number">8.2.2.8</span> Predict homology between genes that are linked by syntenic hits</h4>
<p>The FindHomology package contains 5 functions, the second of which is below. <code>NucleotideOverlap</code> takes in an object of class <code>Synteny</code>, and a list of gene calls. We previously generated the object <code>SyntenyObject</code>, using DECIPHER’s function <code>FindSynteny</code>, and a list of gene calls and annotations using <code>GFFParser</code>. Here we will use these two objects to build what is functionally a list of pairs of genes that are linked by syntenic hits. These linking of hits are used as predictions of homology. A single position in the object of class <code>Synteny</code> represents a pairwise comparison between two genomes. The corresponding position in <code>NucleotideOverlap</code>’s output object, is generated from that pairwise comparison, and the gene calls for each genome involved.</p>
<p>This is a fun object to scroll through, but is often very long, representing thousands of paired genes and is not represented well visually, a simple example would be:</p>
<table>
<thead>
<tr class="header">
<th>Genome 1</th>
<th>Genome 2</th>
<th># of Nucleotides</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Gene A</td>
<td>Gene Z</td>
<td>25L</td>
</tr>
<tr class="even">
<td>Gene B</td>
<td>Gene Y</td>
<td>1251L</td>
</tr>
<tr class="odd">
<td>Gene C</td>
<td>Gene X</td>
<td>145L</td>
</tr>
<tr class="even">
<td>Gene …</td>
<td>Gene …</td>
<td>…</td>
</tr>
</tbody>
</table>
<p>The reason we name this “MatrixObject” below is because the shape and dimensions of our data are dependent upon our synteny object. As in, if our Synteny object was built from 5 genomes, that object is a 5 x 5 matrix. NucleotideOverlap accesses the upper triangle of that object to build a 5 x 5 matrix where each position is built from data in the analogous position from the Synteny object. <code>MatrixObject[[1]][1, 2]</code> below was created from <code>SyntenyObject[[1]][1, 2]</code> and so on and so forth.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">MatrixObject &lt;-<span class="st"> </span><span class="kw">NucleotideOverlap</span>(<span class="dt">SyntenyObject =</span> SyntenyObject,
                                  <span class="dt">GeneCalls =</span> GeneCalls,
                                  <span class="dt">Verbose =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; ===========================================================================</span>
<span class="co">#&gt; Time difference of 1.015444 mins</span></code></pre></div>
<p>The function <code>Catalog</code> takes in the output of <code>NucleotideOverlap</code> and returns a list of matrices. These matrices represent agglomerated sets of pairs. The idea being, if gene <strong>A</strong> in genome <strong>1</strong> was paired by nucleotide overlap to genes <strong>B</strong> and <strong>C</strong> in genomes <strong>2</strong> and <strong>3</strong> respectively by <code>NucleotideOverLap</code>, and additionally gene <strong>B</strong> in genome <strong>2</strong> was similarly paired with gene <strong>C</strong> in genome <strong>3</strong>. With the assumption that these linkages indicate homology, gene <strong>A</strong> is predicted to be homologous to genes <strong>B</strong> and <strong>C</strong>, and gene <strong>B</strong> is predicted to be homologous to gene <strong>C</strong>.</p>
<table>
<thead>
<tr class="header">
<th>Genome 1</th>
<th>Genome 2</th>
<th>Genome 3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>Gene A</td>
<td>Gene B</td>
<td>NA</td>
</tr>
<tr class="even">
<td>NA</td>
<td>Gene B</td>
<td>Gene C</td>
</tr>
<tr class="odd">
<td>Gene A</td>
<td>NA</td>
<td>Gene C</td>
</tr>
</tbody>
</table>
<p><code>Catalog</code> will return every agglomerated set of pairs, of every size possible for the given set of genomes. From a matrix with a single row, indicating a pair of genes that are predicted to have no other homologs anywhere else in the set of genomes, to a large matrix indicating that homologs are located in every genome, and are all predicted to be homologous to each other.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">Homologs &lt;-<span class="st"> </span><span class="kw">Catalog</span>(MatrixObject,
                    <span class="dt">Verbose =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; ===========================================================================</span>
<span class="co">#&gt; Time difference of 1.882072 mins</span></code></pre></div>
<p>We can visualize this object as a histogram of the size of these agglomerations, by the number of pairs included in each agglomerated group. Where <strong>1</strong> represents pairs of genes where neither gene is linked to any other gene, <em>from other genomes</em>. Where <strong>28</strong> <em>in this case</em> represents fully linked sets of pairs, where each gene in the set is predicted to be a homolog of <em>every other gene in the set</em>.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">hist</span>(<span class="kw">sapply</span>(Homologs,
            <span class="cf">function</span>(x) <span class="kw">nrow</span>(x)),
     <span class="dt">main =</span> <span class="st">&quot;Size of Agglomerations&quot;</span>,
     <span class="dt">ylab =</span> <span class="st">&quot;Number of Sets&quot;</span>,
     <span class="dt">xlab =</span> <span class="st">&quot;Numer of Gene Pairs&quot;</span>,
     <span class="dt">breaks =</span> 27L)</code></pre></div>
<p><img src="Cooley_DECIPHER_files/figure-html/find%20homology%20objects%20part%201-1.png" width="672" /></p>
<p>We can collect these fully linked sets. For any <code>N</code> number of genomes, a set of genes where each genome contains a homolog, and those homologs in turn have agreeing homologs in every other genome, the number of rows of this matrix will be <code>N * (N-1) / 2</code>.</p>
<p>You could additionally query this list for any other presence absence pattern that you may be interested in, especially if it can be easily linked to the number of rows in each matrix.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">MaxRows &lt;-<span class="st"> </span><span class="kw">max</span>(<span class="kw">sapply</span>(Homologs,
                      <span class="cf">function</span>(x) <span class="kw">nrow</span>(x)))
CoreSet &lt;-<span class="st"> </span><span class="kw">which</span>(<span class="kw">sapply</span>(Homologs,
                        <span class="cf">function</span>(x) <span class="kw">nrow</span>(x)) <span class="op">==</span><span class="st"> </span>MaxRows)</code></pre></div>
<p>The second to last function in FindHomology, <code>CoreAligner</code>, collects the genes in these sets, from their respective genomes, aligns the respective sets, and then concatonates the alignments. Creating a concatonated core genome for this set of genomes. This function relies on a few DECIPHER functions internally, and utilizes DECIPHER’s database functionality to perform that collection of genes. Additionally, we can use a few more DECIPHER functions to create a distance matrix from our concatonated core genome, and create a phylogenetic tree from that concatonated core.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">CoreGenome &lt;-<span class="st"> </span><span class="kw">CoreAligner</span>(Homologs[CoreSet],
                          <span class="dt">PATH =</span> DBPath,
                          <span class="dt">GeneCalls =</span> GeneCalls,
                          <span class="dt">Verbose =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; ===========================================================================</span>
<span class="co">#&gt; Time difference of 36.76989 secs</span>
CoreDist &lt;-<span class="st"> </span><span class="kw">DistanceMatrix</span>(<span class="dt">myXStringSet =</span> CoreGenome,
                           <span class="dt">verbose =</span> <span class="ot">FALSE</span>,
                           <span class="dt">correction =</span> <span class="st">&quot;Jukes-Cantor&quot;</span>)
CoreDend &lt;-<span class="st"> </span><span class="kw">IdClusters</span>(<span class="dt">myDistMatrix =</span> CoreDist,
                       <span class="dt">myXStringSet =</span> CoreGenome,
                       <span class="dt">method =</span> <span class="st">&quot;NJ&quot;</span>,
                       <span class="dt">verbose =</span> <span class="ot">FALSE</span>,
                       <span class="dt">showPlot =</span> <span class="ot">TRUE</span>,
                       <span class="dt">type =</span> <span class="st">&quot;dendrogram&quot;</span>)</code></pre></div>
<p><img src="Cooley_DECIPHER_files/figure-html/align%20core%20genes-1.png" width="672" /></p>
<p>We can now finally tidy up our workspace and unlink our temp file.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">unlink</span>(DBPath)</code></pre></div>
<p>We are also interested in the pan genome, and FindHomolog’s <code>LogicalPan</code> function allows us to utilize the Homologs object, and the list of GeneCall dataframes, to create a presence absence matrix of the pan genome.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">PanGenomeMatrix &lt;-<span class="st"> </span><span class="kw">LogicalPan</span>(<span class="dt">HomologList =</span> Homologs,
                              <span class="dt">GeneCalls =</span> GeneCalls,
                              <span class="dt">Verbose =</span> <span class="ot">TRUE</span>,
                              <span class="dt">Plot =</span> <span class="ot">FALSE</span>)
<span class="co">#&gt; ===========================================================================</span>
<span class="co">#&gt; Time difference of 1.157171 secs</span></code></pre></div>
<p>We can visualize this matrix if we so choose.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">image</span>(<span class="kw">t</span>(PanGenomeMatrix), <span class="dt">col =</span> <span class="kw">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;blue&quot;</span>),
      <span class="dt">main =</span> <span class="st">&quot;Presence Absence&quot;</span>)</code></pre></div>
<p><img src="Cooley_DECIPHER_files/figure-html/unnamed-chunk-3-1.png" width="672" /></p>
<p>PanGenomeMatrix represents every homolog set, and every singleton gene in our genomes.</p>
<p>This matrix is constructed from the object <code>Homologs</code> which is a list of matrices. Each matrix represents of a set of predicted homologous pairs. Some of the sets are small, say a single pair. Some are large, encompassing genes in every genome in our set. However, the one thing this matrix is missing is singleton genes from each genome. Genes that have no predicted homologs by our method.</p>
<p>And we can create a dendrogram from this matrix.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">PanGenome &lt;-<span class="st"> </span><span class="kw">dist</span>(PanGenomeMatrix,
                  <span class="dt">method =</span> <span class="st">&quot;binary&quot;</span>)
PanDend &lt;-<span class="st"> </span><span class="kw">IdClusters</span>(<span class="dt">myDistMatrix =</span> PanGenome,
                      <span class="dt">method =</span> <span class="st">&quot;NJ&quot;</span>,
                      <span class="dt">type =</span> <span class="st">&quot;dendrogram&quot;</span>,
                      <span class="dt">showPlot =</span> <span class="ot">TRUE</span>,
                      <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p><img src="Cooley_DECIPHER_files/figure-html/create%20dendrogram%20from%20pan%20genome-1.png" width="672" /></p>
<p>We can, additionally, create a simple tangleogram from these two phylogenetic trees. Allowing a comparison of the core, and pan genomes. The core is on the left, while the pan is on the right.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">tf1 &lt;-<span class="st"> </span><span class="kw">tempfile</span>()
tf2 &lt;-<span class="st"> </span><span class="kw">tempfile</span>()

<span class="kw">WriteDendrogram</span>(<span class="dt">x =</span>PanDend,
                <span class="dt">file =</span> tf1)

<span class="kw">WriteDendrogram</span>(<span class="dt">x =</span> CoreDend,
                <span class="dt">file =</span> tf2)

<span class="kw">unlink</span>(tf1)
<span class="kw">unlink</span>(tf2)
<span class="kw">layout</span>(<span class="kw">matrix</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>, <span class="dt">nrow =</span> 1L))
p &lt;-<span class="st"> </span><span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">plot</span>(CoreDend, <span class="dt">horiz =</span> <span class="ot">TRUE</span>, <span class="dt">leaflab =</span> <span class="st">&quot;none&quot;</span>)
<span class="kw">par</span>(<span class="dt">mar =</span> <span class="kw">c</span>(<span class="dv">5</span>, <span class="dv">2</span>, <span class="dv">1</span>, <span class="dv">2</span>))
<span class="kw">plot</span>(PanDend, <span class="dt">horiz =</span> <span class="ot">TRUE</span>, <span class="dt">xlim =</span> <span class="kw">c</span>(<span class="dv">0</span>, <span class="kw">attr</span>(PanDend, <span class="st">&quot;height&quot;</span>)), <span class="dt">leaflab =</span> <span class="st">&quot;none&quot;</span>)
C.Ord &lt;-<span class="st"> </span><span class="kw">unlist</span>(CoreDend)
P.Ord &lt;-<span class="st"> </span><span class="kw">unlist</span>(PanDend)
<span class="kw">segments</span>(<span class="op">-</span><span class="fl">0.10</span>, <span class="kw">seq_along</span>(C.Ord), <span class="op">-</span><span class="fl">0.01</span>, <span class="kw">match</span>(C.Ord, P.Ord), <span class="dt">xpd =</span> <span class="ot">NA</span>, <span class="dt">col =</span> <span class="st">&quot;blue&quot;</span>)</code></pre></div>
<p><img src="Cooley_DECIPHER_files/figure-html/simple%20tangleogram-1.png" width="672" /></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw">par</span>(p)</code></pre></div>
<p>The proposition at the beginning of this workshop was that a specific set of genes, that may not normally be included in a core genome, could be included in the core genome, for this set of Archaea. Specifically, genes involved in some of the unusual chemistry that Archaea do with their lipids. If we wanted to really prove this, a significant amount of lab work would also be required. However we can collect the annotations for all of the sets of homologs that have been predicted, and see how much these annotations agree, and if any of these annotations provide evidence for our proposition.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">CoreGenes &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dt">data =</span> <span class="ot">NA_integer_</span>,
                        <span class="dt">ncol =</span> <span class="kw">length</span>(GeneCalls),
                        <span class="dt">nrow =</span> <span class="kw">length</span>(CoreSet))
CoreAnnotations &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="dt">data =</span> <span class="ot">NA_character_</span>,
                              <span class="dt">ncol =</span> <span class="kw">length</span>(GeneCalls),
                              <span class="dt">nrow =</span> <span class="kw">length</span>(CoreSet))
<span class="cf">for</span> (i <span class="cf">in</span> <span class="kw">seq_len</span>(<span class="kw">ncol</span>(CoreGenes))) {
  <span class="cf">for</span> (j <span class="cf">in</span> <span class="kw">seq_len</span>(<span class="kw">nrow</span>(CoreGenes))) {
    CoreGenes[j, i] &lt;-<span class="st"> </span><span class="kw">unique</span>(Homologs[CoreSet[j]][[<span class="dv">1</span>]][, i])[<span class="op">!</span><span class="kw">is.na</span>(<span class="kw">unique</span>(Homologs[CoreSet[j]][[<span class="dv">1</span>]][, i]))]
    CoreAnnotations[j, i] &lt;-<span class="st"> </span>GeneCalls[[i]][CoreGenes[j, i], <span class="st">&quot;Annotation&quot;</span>]
  }
}

CoreAnnotations &lt;-<span class="st"> </span><span class="kw">t</span>(CoreAnnotations)
CoreAnnotations &lt;-<span class="st"> </span><span class="kw">data.frame</span>(CoreAnnotations,
                                  <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p>The annotations for each core set can now be viewed. Some provide near-universal agreement, while others indicate genes of similar family and function, but potentially not an exact match.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">CoreAnnotations[, <span class="kw">c</span>(1L, 3L)]
<span class="co">#&gt;                                X1                                       X3</span>
<span class="co">#&gt; 1        acetoacetyl-CoA thiolase    DNA-directed RNA polymerase subunit H</span>
<span class="co">#&gt; 2 propanoyl-CoA C-acyltransferase                      RNA polymerase Rpb5</span>
<span class="co">#&gt; 3 propanoyl-CoA C-acyltransferase                      RNA polymerase Rpb5</span>
<span class="co">#&gt; 4        acetoacetyl-CoA thiolase    DNA-directed RNA polymerase subunit H</span>
<span class="co">#&gt; 5        acetoacetyl-CoA thiolase    DNA-directed RNA polymerase subunit H</span>
<span class="co">#&gt; 6    acetyl-CoA acetyltransferase DNA-directed RNA polymerase%2C subunit H</span>
<span class="co">#&gt; 7        acetoacetyl-CoA thiolase    DNA-directed RNA polymerase subunit H</span>
<span class="co">#&gt; 8 Propanoyl-CoA C-acyltransferase                      RNA polymerase Rpb5</span></code></pre></div>
<p>There are 60 of these!</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">CoreAnnotations[, 2L]
<span class="co">#&gt; [1] &quot;ribonucleoside-diphosphate reductase&quot;                               </span>
<span class="co">#&gt; [2] &quot;ribonucleoside-diphosphate reductase%2C adenosylcobalamin-dependent&quot;</span>
<span class="co">#&gt; [3] &quot;ribonucleoside-diphosphate reductase%2C adenosylcobalamin-dependent&quot;</span>
<span class="co">#&gt; [4] &quot;Ribonucleoside-diphosphate reductase&quot;                               </span>
<span class="co">#&gt; [5] &quot;Ribonucleoside-diphosphate reductase&quot;                               </span>
<span class="co">#&gt; [6] &quot;ribonucleoside-diphosphate reductase class II&quot;                      </span>
<span class="co">#&gt; [7] &quot;ribonucleoside-diphosphate reductase&quot;                               </span>
<span class="co">#&gt; [8] &quot;ribonucleoside-diphosphate reductase%2C adenosylcobalamin-dependent&quot;</span></code></pre></div>
<p>Now, a little insider baseball is that the unique chemistry we’re interested in is often the chemistry of radicals, especially in biological systems, where the process is tightly controlled by enzymes. A quick perusal of all the sets of annotations provides two sets that support our original hypothesis.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">CoreAnnotations[, <span class="kw">c</span>(17L, 31L)]
<span class="co">#&gt;                            X17                                 X31</span>
<span class="co">#&gt; 1   radical SAM domain protein putative radical SAM domain protein</span>
<span class="co">#&gt; 2          radical SAM protein                 radical SAM protein</span>
<span class="co">#&gt; 3          radical SAM protein                 radical SAM protein</span>
<span class="co">#&gt; 4   Radical SAM domain protein                 Radical SAM protein</span>
<span class="co">#&gt; 5         hypothetical protein                 Radical SAM protein</span>
<span class="co">#&gt; 6 putative Fe-S oxidoreductase                 Fe-S oxidoreductase</span>
<span class="co">#&gt; 7   radical SAM domain protein putative radical SAM domain protein</span>
<span class="co">#&gt; 8   Radical SAM domain protein          Radical SAM domain protein</span></code></pre></div>
<p>This concludes the workshop, as any analysis after this would require work in a wet lab. Hopefully this was a useful showcase for the power of using DECIPHER to analyze sets of genomic data. The package FindHomology is currently located on github, and is still under construction. Upon completion, it will be available on Bioconductor.</p>
<p>Thank you for your time and attention! We hope you learned something, and if you come back to this at a later date and have any questions, concerns, or comments, feel free to <a href="npc19@pitt.edu">email me</a>, or you can <a href="https://twitter.com/npcooley">tweet</a> at me.</p>

</div>
</div>
</div>
</div>
<div class="footnotes">
<hr />
<ol start="18">
<li id="fn18"><p>University of Pittsburgh<a href="working-with-genomic-data-in-r-with-the-decipher-package.html#fnref18">↩</a></p></li>
<li id="fn19"><p><a href="https://journal.r-project.org/archive/2016-1/wright.pdf">Wright, E. S. The R Journal 2016, 8 (1), 352–359.</a><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#fnref19">↩</a></p></li>
<li id="fn20"><p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC432104/">(1) Woese, C. R.; Fox, G. E. Proc. Natl. Acad. Sci. 1977, 74 (11), 5088–5090.</a><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#fnref20">↩</a></p></li>
<li id="fn21"><p><a href="https://commons.wikimedia.org/wiki/File:Grand_prismatic_spring.jpg">Grand Prismatic Spring</a><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#fnref21">↩</a></p></li>
<li id="fn22"><p><a href="https://www.ncbi.nlm.nih.gov/pubmed/28007654">(1) Caforio, A.; Driessen, A. J. M. Biochimica et Biophysica Acta - Molecular and Cell Biology of Lipids. Elsevier November 1, 2017, pp 1325–1339.</a><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#fnref22">↩</a></p></li>
<li id="fn23"><p><a href="https://www.ncbi.nlm.nih.gov/books/NBK179288/">Entrez Direct</a><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#fnref23">↩</a></p></li>
<li id="fn24"><p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2848648/">(1) Hyatt, D.; Chen, G.-L.; LoCascio, P. F.; Land, M. L.; Larimer, F. W.; Hauser, L. J. BMC Bioinformatics 2010, 11 (1), 119.</a><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#fnref24">↩</a></p></li>
<li id="fn25"><p><a href="https://www.ncbi.nlm.nih.gov/pubmed/24642063">(1) Seemann, T. Bioinformatics 2014, 30 (14), 2068–2069.</a><a href="working-with-genomic-data-in-r-with-the-decipher-package.html#fnref25">↩</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="variant-functional-annotation-using-statepaintr-funcivar-motifbreakr.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="functional-enrichment-analysis-of-high-throughput-omics-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": true,
"facebook": false,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"download": ["BioC2018.pdf", "BioC2018.epub"],
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "";
    if (src === "" || src === "true") src = "https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:" && /^https?:/.test(src))
      src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
